
<!doctype html>
<html lang="zh" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="生物信息云">
      
      
        <meta name="author" content="DoubleHelix">
      
      
        <link rel="canonical" href="https://bioinfocloud.io/BioInfo/RNAseq/pipeline_1/">
      
      
        <link rel="prev" href="../../database/scList/">
      
      
        <link rel="next" href="../../TCGA/">
      
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.22">
    
    
      
        <title>Bulk RNAseq上游分析 - 生物信息云</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.84d31ad4.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../mkdocs/css/no-footer.css">
    
      <link rel="stylesheet" href="../../../mkdocs/css/unordered-list-symbols.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="light-blue">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#fastqc" class="md-skip">
          跳转至
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="页眉">
    <a href="../../.." title="生物信息云" class="md-header__button md-logo" aria-label="生物信息云" data-md-component="logo">
      
  <img src="../../../assets/logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            生物信息云
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Bulk RNAseq上游分析
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="light-blue"  aria-label="切换风格"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="切换风格" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.04 8.71V4h-4.7L12 .69 8.71 4H4v4.71L.69 12 4 15.34v4.7h4.71L12 23.35l3.34-3.31h4.7v-4.7L23.35 12zM8.83 7.05c.98 0 1.77.79 1.77 1.78a1.77 1.77 0 0 1-1.77 1.77c-.99 0-1.78-.79-1.78-1.77 0-.99.79-1.78 1.78-1.78M15.22 17c-.98 0-1.77-.8-1.77-1.78a1.77 1.77 0 0 1 1.77-1.77c.98 0 1.78.79 1.78 1.77A1.78 1.78 0 0 1 15.22 17m-6.72.03L7 15.53 15.53 7l1.5 1.5z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="orange"  aria-label="切换风格"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="切换风格" for="__palette_2" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5s-1.65.15-2.39.42zM3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29zm.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14zM20.65 7l-1.77 3.79a7.02 7.02 0 0 0-2.38-4.15zm-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29zM12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="deep-purple"  aria-label="切换风格"  type="radio" name="__palette" id="__palette_2">
    
      <label class="md-header__button md-icon" title="切换风格" for="__palette_3" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="purple" data-md-color-primary="black" data-md-color-accent="pink"  aria-label="切换风格"  type="radio" name="__palette" id="__palette_3">
    
      <label class="md-header__button md-icon" title="切换风格" for="__palette_4" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="light-green" data-md-color-primary="black" data-md-color-accent="light-blue"  aria-label="切换至日间模式"  type="radio" name="__palette" id="__palette_4">
    
      <label class="md-header__button md-icon" title="切换至日间模式" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9 2c-1.05 0-2.05.16-3 .46 4.06 1.27 7 5.04 7 9.54s-2.94 8.27-7 9.54c.95.3 1.95.46 3 .46a10 10 0 0 0 10-10A10 10 0 0 0 9 2"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="搜索" placeholder="搜索" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="查找">
        
        <button type="reset" class="md-search__icon md-icon" title="清空当前内容" aria-label="清空当前内容" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            正在初始化搜索引擎
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://bioinfocloud.github.io/note" title="前往仓库" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    BioInfoCloud
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="标签" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../../Programming/" class="md-tabs__link">
          
  
  
    
  
  生信编程

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../" class="md-tabs__link">
          
  
  
    
  
  生信基础

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../../Programming/R/ggplot2/" class="md-tabs__link">
          
  
  
    
  
  可视化

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../../bigdata/" class="md-tabs__link">
          
  
  
    
  
  大数据

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../../ML/knn/kNN/" class="md-tabs__link">
          
  
  
    
  
  机器学习

        </a>
      </li>
    
  

      
        
  
  
  
    
  
  
    
    
      
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../../scRNAseq/scRNASeq_UP/" class="md-tabs__link">
          
  
  
    
  
  专题

        </a>
      </li>
    
  

    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../../Literature/RNAseq_review1/" class="md-tabs__link">
          
  
  
    
  
  文献

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../database/scList/" class="md-tabs__link">
          
  
  
    
  
  数据库工具

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../../WeChatOfficialAccount/HOME/" class="md-tabs__link">
          
  
  
    
  
  微信

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../../ABOUT/" class="md-tabs__link">
          
  
  
    
  
  关于

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="导航栏" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="生物信息云" class="md-nav__button md-logo" aria-label="生物信息云" data-md-component="logo">
      
  <img src="../../../assets/logo.png" alt="logo">

    </a>
    生物信息云
  </label>
  
    <div class="md-nav__source">
      <a href="https://bioinfocloud.github.io/note" title="前往仓库" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    BioInfoCloud
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../../Programming/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    生信编程
    
  </span>
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_1" id="__nav_1_label" tabindex="0">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            生信编程
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1_2" >
        
          
          <label class="md-nav__link" for="__nav_1_2" id="__nav_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Linux
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_2">
            <span class="md-nav__icon md-icon"></span>
            Linux
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/linux/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Linux讲座
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/commands-concise/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    命令行入门精简版
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/vim/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    vim
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/tmux/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    tmux
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/unix/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    UNIX
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/kernel-compile/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    内核编译
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1_3" >
        
          
          <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Markdown
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_3">
            <span class="md-nav__icon md-icon"></span>
            Markdown
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Markdown/markdown/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Markdown讲座
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/vscode-python/vscode-python/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VSCode及插件安装
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Markdown/grammars/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Markdown语法
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1_4" >
        
          
          <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Git
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_4">
            <span class="md-nav__icon md-icon"></span>
            Git
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Git/git/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Git
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Git/git-demos/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Git例子
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Git/git-teamwork/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Git合作
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Git/windows-install/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Windows安装Git
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1_5" >
        
          
          <label class="md-nav__link" for="__nav_1_5" id="__nav_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    R
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5">
            <span class="md-nav__icon md-icon"></span>
            R
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/R/Grammar/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    R语法
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/R/ggplot2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    可视化
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1_5_3" >
        
          
          <label class="md-nav__link" for="__nav_1_5_3" id="__nav_1_5_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    R包开发
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_3">
            <span class="md-nav__icon md-icon"></span>
            R包开发
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/R/01-software/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    软件安装
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/R/02-DevelopmentManual/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    快速手册
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/R/shiny/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Shiny
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1_6" >
        
          
          <label class="md-nav__link" for="__nav_1_6" id="__nav_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Python
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_6">
            <span class="md-nav__icon md-icon"></span>
            Python
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/python/links/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    语法基础
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/python/numpy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Numpy
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/python/pandas/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pandas
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/python/Matplotlib_visualization/Matplotlib_visualization/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Matplotlib
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Biopython/01_install/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Biopython
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_2" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    生信基础
    
  </span>
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_2" id="__nav_2_label" tabindex="0">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            生信基础
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_2_2" >
        
          
          <label class="md-nav__link" for="__nav_2_2" id="__nav_2_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    生信基础
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2_2">
            <span class="md-nav__icon md-icon"></span>
            生信基础
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/Introduction/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    生信引言
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/01/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    基础应用
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/02_noun/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    生信名词
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/03_sequence/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    测序技术原理
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/04_format/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    数据文件格式
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/biomaRt/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    biomaRt包
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/counts_RPM_RPKM_FPKM_1/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RNASeq表达定量
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/counts_RPM_RPKM_FPKM_2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RNASeq表达定量转换
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/DESeq2_edgeR_limma/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RNASeq差异表达分析原理
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/GO/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    什么是GO
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/KEGG_GO_GSEA_what/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    富集分析理论基础
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/hsa2mus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    人鼠基因比较
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/hsa2mus_2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    同源基因转换
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/IDtrans/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    各种ID转换
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/miRNA_ID/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    miRNA的ID转换
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_2_3" >
        
          
          <label class="md-nav__link" for="__nav_2_3" id="__nav_2_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    软件使用问题
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2_3">
            <span class="md-nav__icon md-icon"></span>
            软件使用问题
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../hisat2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Hisat2
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    可视化
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            可视化
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/R/ggplot2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    可视化
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../visualization/visualizationList/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    公众号文章参考
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../visualization/ScatterPlot/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    散点图
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../visualization/mult_volcano/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    多重火山图
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../visualization/colorScheme01/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    配色方案1
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_4" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../../bigdata/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    大数据
    
  </span>
  

            </a>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            大数据
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    机器学习
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            机器学习
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../ML/knn/kNN/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    1.kNN算法理论
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../ML/knn/knn_py/knn_py/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    2.kNN实践案例
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" checked>
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    专题
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            专题
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6_1" >
        
          
          <label class="md-nav__link" for="__nav_6_1" id="__nav_6_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    单细胞专题
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_6_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6_1">
            <span class="md-nav__icon md-icon"></span>
            单细胞专题
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../scRNAseq/scRNASeq_UP/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    单细胞上游分析
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../scRNAseq/SeuratObj/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    了解Seurat对象
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../scRNAseq/scRNASeq_down/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    单细胞下游分析
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../scRNAseq/GSE130001/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    常规分析案例一
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../scRNAseq/SingleR/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    细胞注释-SingleR
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../scRNAseq/Scissor/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    表型关联分析
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../scRNAseq/CellChat/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    cellChat
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../scRNAseq/scMetabolism/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    scMetabolism
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../database/scList/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    单细胞相关数据库
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6_2" checked>
        
          
          <label class="md-nav__link" for="__nav_6_2" id="__nav_6_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    RNAseq专题
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_6_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_6_2">
            <span class="md-nav__icon md-icon"></span>
            RNAseq专题
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Bulk RNAseq上游分析
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Bulk RNAseq上游分析
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="目录">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      目录
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#fastqc" class="md-nav__link">
    <span class="md-ellipsis">
      一. 使用fastqc质控
    </span>
  </a>
  
    <nav class="md-nav" aria-label="一. 使用fastqc质控">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#fastqc_1" class="md-nav__link">
    <span class="md-ellipsis">
      （一）fastqc介绍
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fastqc_2" class="md-nav__link">
    <span class="md-ellipsis">
      （二）fastqc质控
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#_1" class="md-nav__link">
    <span class="md-ellipsis">
      （三）结果解读
    </span>
  </a>
  
    <nav class="md-nav" aria-label="（三）结果解读">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1basic-statistics" class="md-nav__link">
    <span class="md-ellipsis">
      1.Basic Statistics
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2per-base-sequence-quality" class="md-nav__link">
    <span class="md-ellipsis">
      2.Per base sequence quality
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3per-tile-sequence-quality" class="md-nav__link">
    <span class="md-ellipsis">
      3.Per tile sequence quality
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4per-sequence-quality-scores" class="md-nav__link">
    <span class="md-ellipsis">
      4.Per sequence quality scores
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#5per-base-sequence-content" class="md-nav__link">
    <span class="md-ellipsis">
      5.Per base sequence content
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#6per-sequence-gc-content" class="md-nav__link">
    <span class="md-ellipsis">
      6.Per sequence GC content
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#7per-base-n-content" class="md-nav__link">
    <span class="md-ellipsis">
      7.Per base N content
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#8sequence-length-distribution" class="md-nav__link">
    <span class="md-ellipsis">
      8.Sequence Length Distribution
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#9sequence-duplication-levels" class="md-nav__link">
    <span class="md-ellipsis">
      9.Sequence Duplication Levels
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#10overrepresented-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      10.Overrepresented sequences
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#11adapter-content" class="md-nav__link">
    <span class="md-ellipsis">
      11.Adapter Content
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#trim-galorereadsadaptor" class="md-nav__link">
    <span class="md-ellipsis">
      二.使用trim-galore去除低质量的reads和adaptor
    </span>
  </a>
  
    <nav class="md-nav" aria-label="二.使用trim-galore去除低质量的reads和adaptor">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#trim-galore" class="md-nav__link">
    <span class="md-ellipsis">
      （一）Trim Galore介绍
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#trim-galorereadsadaptor_1" class="md-nav__link">
    <span class="md-ellipsis">
      （二）使用trim-galore去除低质量的reads和adaptor
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#hisat2" class="md-nav__link">
    <span class="md-ellipsis">
      三.使用Hisat2进行序列比对
    </span>
  </a>
  
    <nav class="md-nav" aria-label="三.使用Hisat2进行序列比对">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#index" class="md-nav__link">
    <span class="md-ellipsis">
      （一）index文件下载
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hisat2_1" class="md-nav__link">
    <span class="md-ellipsis">
      （二）hisat2介绍
    </span>
  </a>
  
    <nav class="md-nav" aria-label="（二）hisat2介绍">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1" class="md-nav__link">
    <span class="md-ellipsis">
      1.背景
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2hisat" class="md-nav__link">
    <span class="md-ellipsis">
      2.Hisat设计原则
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3hisat2-2" class="md-nav__link">
    <span class="md-ellipsis">
      3.HiSat2进行比对的参数设置 【引用：2】
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hisat2_2" class="md-nav__link">
    <span class="md-ellipsis">
      （三）使用Hisat2进行序列比对
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#_2" class="md-nav__link">
    <span class="md-ellipsis">
      （四）结果解读
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#samtoolssambam" class="md-nav__link">
    <span class="md-ellipsis">
      四.使用SAMtools将SAM文件转换为BAM文件、排序、建立索引
    </span>
  </a>
  
    <nav class="md-nav" aria-label="四.使用SAMtools将SAM文件转换为BAM文件、排序、建立索引">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#samtools" class="md-nav__link">
    <span class="md-ellipsis">
      （一）SAMtools介绍
    </span>
  </a>
  
    <nav class="md-nav" aria-label="（一）SAMtools介绍">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1samtools-view" class="md-nav__link">
    <span class="md-ellipsis">
      1.samtools view
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2samtools-sort" class="md-nav__link">
    <span class="md-ellipsis">
      2.samtools sort
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4samtools-flagstat" class="md-nav__link">
    <span class="md-ellipsis">
      4.samtools flagstat
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#samtools_1" class="md-nav__link">
    <span class="md-ellipsis">
      （二）SAMtools实战
    </span>
  </a>
  
    <nav class="md-nav" aria-label="（二）SAMtools实战">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1samtools-view_1" class="md-nav__link">
    <span class="md-ellipsis">
      1.格式转换(samtools view)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-samtools-sort" class="md-nav__link">
    <span class="md-ellipsis">
      2 排序(samtools sort)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-samtools-index" class="md-nav__link">
    <span class="md-ellipsis">
      3. 建立索引(samtools index)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4readssamtools-flagstat" class="md-nav__link">
    <span class="md-ellipsis">
      4.查看reads比对情况(samtools flagstat)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rseqcbam" class="md-nav__link">
    <span class="md-ellipsis">
      五.使用RSeQC软件对生成的BAM文件进行质控
    </span>
  </a>
  
    <nav class="md-nav" aria-label="五.使用RSeQC软件对生成的BAM文件进行质控">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#rseqc" class="md-nav__link">
    <span class="md-ellipsis">
      （一）RSeQC介绍
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bam_statpy" class="md-nav__link">
    <span class="md-ellipsis">
      （二）使用bam_stat.py命令查看比对的总体情况
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#read_distributionpy" class="md-nav__link">
    <span class="md-ellipsis">
      （三）使用read_distribution.py命令查看基因组覆盖率
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#stringtie" class="md-nav__link">
    <span class="md-ellipsis">
      六.使用Stringtie对数据进行下游处理
    </span>
  </a>
  
    <nav class="md-nav" aria-label="六.使用Stringtie对数据进行下游处理">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#stringtie_1" class="md-nav__link">
    <span class="md-ellipsis">
      （一）StringTie介绍
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#stringtie_2" class="md-nav__link">
    <span class="md-ellipsis">
      （二）StringTie应用
    </span>
  </a>
  
    <nav class="md-nav" aria-label="（二）StringTie应用">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1_1" class="md-nav__link">
    <span class="md-ellipsis">
      1.单样本转录本组装
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2" class="md-nav__link">
    <span class="md-ellipsis">
      2.多样本转录本整合
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-gffcompare" class="md-nav__link">
    <span class="md-ellipsis">
      3. 使用gffcompare检验数据比对到基因组上的情况(可选)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4" class="md-nav__link">
    <span class="md-ellipsis">
      4.重新组装转录本并估算基因表达丰度
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#5read-count" class="md-nav__link">
    <span class="md-ellipsis">
      5.read count数据输出
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
          
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6_3" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../TCGA/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    TCGA-GEO专题
    
  </span>
  

            </a>
            
          </div>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_6_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6_3">
            <span class="md-nav__icon md-icon"></span>
            TCGA-GEO专题
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6_4" >
        
          
          <label class="md-nav__link" for="__nav_6_4" id="__nav_6_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    药物分析
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_6_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6_4">
            <span class="md-nav__icon md-icon"></span>
            药物分析
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../drugAna/cellminer/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CellMiner数据库
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../drugAna/ADMET/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ADMET计算
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6_5" >
        
          
          <label class="md-nav__link" for="__nav_6_5" id="__nav_6_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    免疫浸润
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_6_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6_5">
            <span class="md-nav__icon md-icon"></span>
            免疫浸润
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../immune/EaSIeR/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    EaSIeR
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6_6" >
        
          
          <label class="md-nav__link" for="__nav_6_6" id="__nav_6_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    数据库工具
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_6_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6_6">
            <span class="md-nav__icon md-icon"></span>
            数据库工具
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../database/CoMutDB/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CoMutDB数据库
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../database/CysModDB/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CysModDB数据库
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    文献
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            文献
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/RNAseq_review1/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RNASeq综述
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/Review/001_immunoediting/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    癌症免疫编辑
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/Review/003_2021_immunotherapy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    癌症免疫治疗
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/Review/002_DTIS/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    药物靶点预测
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/Review/004_2021_ICB/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ICB治疗
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/scRNAseq/2023_NC_HNSCC/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    2023_HNSCC
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/Review/006_Nano/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    纳米材料
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/AI/001_AI/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    人工智能
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_8" >
        
          
          <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    数据库工具
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_8">
            <span class="md-nav__icon md-icon"></span>
            数据库工具
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../database/scList/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    单细胞相关数据库
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../database/database/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    在线数据库
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../toolsandsoftware/tands/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    工具或软件
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_8_4" >
        
          
          <label class="md-nav__link" for="__nav_8_4" id="__nav_8_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    MedBioInfoCloud包
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_8_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_8_4">
            <span class="md-nav__icon md-icon"></span>
            MedBioInfoCloud包
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../package/MedBioInfoCloud/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    安装
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../package/baseFun/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    基础函数
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../package/BulkRNA-Seq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Bulk RNA-Seq
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../package/enrichAnalysis/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    富集分析
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../package/TCGA/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TCGA数据挖掘
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_9" >
        
          
          <label class="md-nav__link" for="__nav_9" id="__nav_9_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    微信
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_9_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_9">
            <span class="md-nav__icon md-icon"></span>
            微信
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../WeChatOfficialAccount/HOME/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    HOME
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../WeChatOfficialAccount/articleIndex/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    文章索引
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_10" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../../ABOUT/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    关于
    
  </span>
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_10" id="__nav_10_label" tabindex="0">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_10_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_10">
            <span class="md-nav__icon md-icon"></span>
            关于
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../ABOUT/announcement/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    工具
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../ABOUT/studentsGroup/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    微信群
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="目录">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      目录
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#fastqc" class="md-nav__link">
    <span class="md-ellipsis">
      一. 使用fastqc质控
    </span>
  </a>
  
    <nav class="md-nav" aria-label="一. 使用fastqc质控">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#fastqc_1" class="md-nav__link">
    <span class="md-ellipsis">
      （一）fastqc介绍
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fastqc_2" class="md-nav__link">
    <span class="md-ellipsis">
      （二）fastqc质控
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#_1" class="md-nav__link">
    <span class="md-ellipsis">
      （三）结果解读
    </span>
  </a>
  
    <nav class="md-nav" aria-label="（三）结果解读">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1basic-statistics" class="md-nav__link">
    <span class="md-ellipsis">
      1.Basic Statistics
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2per-base-sequence-quality" class="md-nav__link">
    <span class="md-ellipsis">
      2.Per base sequence quality
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3per-tile-sequence-quality" class="md-nav__link">
    <span class="md-ellipsis">
      3.Per tile sequence quality
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4per-sequence-quality-scores" class="md-nav__link">
    <span class="md-ellipsis">
      4.Per sequence quality scores
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#5per-base-sequence-content" class="md-nav__link">
    <span class="md-ellipsis">
      5.Per base sequence content
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#6per-sequence-gc-content" class="md-nav__link">
    <span class="md-ellipsis">
      6.Per sequence GC content
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#7per-base-n-content" class="md-nav__link">
    <span class="md-ellipsis">
      7.Per base N content
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#8sequence-length-distribution" class="md-nav__link">
    <span class="md-ellipsis">
      8.Sequence Length Distribution
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#9sequence-duplication-levels" class="md-nav__link">
    <span class="md-ellipsis">
      9.Sequence Duplication Levels
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#10overrepresented-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      10.Overrepresented sequences
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#11adapter-content" class="md-nav__link">
    <span class="md-ellipsis">
      11.Adapter Content
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#trim-galorereadsadaptor" class="md-nav__link">
    <span class="md-ellipsis">
      二.使用trim-galore去除低质量的reads和adaptor
    </span>
  </a>
  
    <nav class="md-nav" aria-label="二.使用trim-galore去除低质量的reads和adaptor">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#trim-galore" class="md-nav__link">
    <span class="md-ellipsis">
      （一）Trim Galore介绍
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#trim-galorereadsadaptor_1" class="md-nav__link">
    <span class="md-ellipsis">
      （二）使用trim-galore去除低质量的reads和adaptor
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#hisat2" class="md-nav__link">
    <span class="md-ellipsis">
      三.使用Hisat2进行序列比对
    </span>
  </a>
  
    <nav class="md-nav" aria-label="三.使用Hisat2进行序列比对">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#index" class="md-nav__link">
    <span class="md-ellipsis">
      （一）index文件下载
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hisat2_1" class="md-nav__link">
    <span class="md-ellipsis">
      （二）hisat2介绍
    </span>
  </a>
  
    <nav class="md-nav" aria-label="（二）hisat2介绍">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1" class="md-nav__link">
    <span class="md-ellipsis">
      1.背景
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2hisat" class="md-nav__link">
    <span class="md-ellipsis">
      2.Hisat设计原则
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3hisat2-2" class="md-nav__link">
    <span class="md-ellipsis">
      3.HiSat2进行比对的参数设置 【引用：2】
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hisat2_2" class="md-nav__link">
    <span class="md-ellipsis">
      （三）使用Hisat2进行序列比对
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#_2" class="md-nav__link">
    <span class="md-ellipsis">
      （四）结果解读
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#samtoolssambam" class="md-nav__link">
    <span class="md-ellipsis">
      四.使用SAMtools将SAM文件转换为BAM文件、排序、建立索引
    </span>
  </a>
  
    <nav class="md-nav" aria-label="四.使用SAMtools将SAM文件转换为BAM文件、排序、建立索引">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#samtools" class="md-nav__link">
    <span class="md-ellipsis">
      （一）SAMtools介绍
    </span>
  </a>
  
    <nav class="md-nav" aria-label="（一）SAMtools介绍">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1samtools-view" class="md-nav__link">
    <span class="md-ellipsis">
      1.samtools view
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2samtools-sort" class="md-nav__link">
    <span class="md-ellipsis">
      2.samtools sort
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4samtools-flagstat" class="md-nav__link">
    <span class="md-ellipsis">
      4.samtools flagstat
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#samtools_1" class="md-nav__link">
    <span class="md-ellipsis">
      （二）SAMtools实战
    </span>
  </a>
  
    <nav class="md-nav" aria-label="（二）SAMtools实战">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1samtools-view_1" class="md-nav__link">
    <span class="md-ellipsis">
      1.格式转换(samtools view)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-samtools-sort" class="md-nav__link">
    <span class="md-ellipsis">
      2 排序(samtools sort)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-samtools-index" class="md-nav__link">
    <span class="md-ellipsis">
      3. 建立索引(samtools index)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4readssamtools-flagstat" class="md-nav__link">
    <span class="md-ellipsis">
      4.查看reads比对情况(samtools flagstat)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rseqcbam" class="md-nav__link">
    <span class="md-ellipsis">
      五.使用RSeQC软件对生成的BAM文件进行质控
    </span>
  </a>
  
    <nav class="md-nav" aria-label="五.使用RSeQC软件对生成的BAM文件进行质控">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#rseqc" class="md-nav__link">
    <span class="md-ellipsis">
      （一）RSeQC介绍
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bam_statpy" class="md-nav__link">
    <span class="md-ellipsis">
      （二）使用bam_stat.py命令查看比对的总体情况
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#read_distributionpy" class="md-nav__link">
    <span class="md-ellipsis">
      （三）使用read_distribution.py命令查看基因组覆盖率
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#stringtie" class="md-nav__link">
    <span class="md-ellipsis">
      六.使用Stringtie对数据进行下游处理
    </span>
  </a>
  
    <nav class="md-nav" aria-label="六.使用Stringtie对数据进行下游处理">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#stringtie_1" class="md-nav__link">
    <span class="md-ellipsis">
      （一）StringTie介绍
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#stringtie_2" class="md-nav__link">
    <span class="md-ellipsis">
      （二）StringTie应用
    </span>
  </a>
  
    <nav class="md-nav" aria-label="（二）StringTie应用">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1_1" class="md-nav__link">
    <span class="md-ellipsis">
      1.单样本转录本组装
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2" class="md-nav__link">
    <span class="md-ellipsis">
      2.多样本转录本整合
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-gffcompare" class="md-nav__link">
    <span class="md-ellipsis">
      3. 使用gffcompare检验数据比对到基因组上的情况(可选)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4" class="md-nav__link">
    <span class="md-ellipsis">
      4.重新组装转录本并估算基因表达丰度
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#5read-count" class="md-nav__link">
    <span class="md-ellipsis">
      5.read count数据输出
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


  <h1>Bulk RNAseq上游分析</h1>

<h2 id="fastqc">一. 使用fastqc质控<a class="headerlink" href="#fastqc" title="Permanent link">&para;</a></h2>
<p>做转录组测序，通常公司是不给分析的，分析也要自己多花钱，当然不同公司收费不一样，有的可能带有简单的分析。之前测序的第一家公司给了简单的分析，后面换了一家测序公司，不给分析。所以我得自己分析啦，在分析的时候顺便写一下教程。分享给大家，要分析转录组数据，首先得知道测序原理【参考文章：<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651272832&amp;idx=1&amp;sn=7d208df1a65e3696c41c666ef78f5881&amp;chksm=841ed2fdb3695bebedbbc733b0bca112421a91647d2038266d0a24b8ce46ee19503a801fdd2b&amp;scene=21#wechat_redirect">illumina、Sanger、第三代和第四代测序技术原理</a>】，还有就是了解生信分析中一些文件格式【参考文章：<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651272899&amp;idx=1&amp;sn=6779b2bd21f3b607a08227d31c7212c6&amp;chksm=841ed2beb3695ba8bee23563c28caa005447b2298785719964732b16cafe3a15d7d4937b95c1&amp;scene=21#wechat_redirect">生信中常见的数据文件格式</a>】，当然，还有其他一些生物背景知识，除此以外，还需要会Linux，这个是一个漫长的学习过程。本文就介绍转录组数据分析的第一步分析：质控，主要就是fastqc这个软件的使用和结果解读。</p>
<h3 id="fastqc_1">（一）fastqc介绍<a class="headerlink" href="#fastqc_1" title="Permanent link">&para;</a></h3>
<p>拿到原始数据后我们首先采用fastqc程序进行质控，看原始数据质量情况，fastqc会生成一个html结果报告，根据图形化界面，我们可以判断下机数据情况是否符合分析要求。我们测序得到的是带有质量值的碱基序列fastq格式。</p>
<p>所以这里你需要先了解<a href="https://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651272899&amp;idx=1&amp;sn=6779b2bd21f3b607a08227d31c7212c6&amp;scene=21#wechat_redirect">生信中常见的数据文件格式 。</a></p>
<p>FastQC的官网：<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">http://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a></p>
<p>FastQC的下载地址：<a href="http://www.bioinformatics.babraham.ac.uk/projects/download.html#fastqc">http://www.bioinformatics.babraham.ac.uk/projects/download.html#fastqc</a></p>
<p>但在Linux中，通常我们是搭建环境安装分析流程相关的软件的，具体参考文章：<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273219&amp;idx=1&amp;sn=44ad6dfa492aa50f8360ab2bb161b28c&amp;chksm=841edc7eb36955685650eaf067fbded8d3b1d427c1f5e52caf2aa5a25fb9a57250ea12a217a4&amp;scene=21#wechat_redirect">Linux系统下Anaconda的安装和使用教程</a>。</p>
<p>将测序的数据上传到自己的服务器或者虚拟机中，我这里上传数据到目录：/data/RNAseq下。总共12个文件，27G。</p>
<p><img alt="" src="001.png" /></p>
<p>首先，我介绍一下fastqc的用法：</p>
<p><div class="highlight"><pre><span></span><code>   fastqc seqfile1 seqfile2 .. seqfileN
   fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam]
          [-c contaminant file] seqfile1 .. seqfileN
</code></pre></div>
重要参数：
<div class="highlight"><pre><span></span><code> -o --outdir 输出目录，需自己创建目录
 –(no)extract 是否解压输出文件，默认是自动解压缩zip文件。加上–noextract不解压文件。
 -f 指定输入文件的类型，支持fastq|bam|sam三种格式的文件，默认自动识别。
 -t --threads选择程序运行的线程数，即同时处理的文件数目。
 -c --contaminants，污染物选项，输入的是一个文件，格式是Name [Tab] Sequence，里面是可能的污染序列，如果有这个选项，FastQC会在计算时候评估污染的情况，并在统计的时候进行分析，一般用不到。
</code></pre></div>
FastQC读取一组序列文件，并从每个文件生成质量控制报告，该报告由许多不同的模块组成，每个模块将帮助识别数据中可能存在的不同类型的问题。
如果命令行上没有指定要处理的文件，那么程序将作为交互式图形应用程序启动。如果在命令行上提供了文件，那么程序将运行而不需要用户交互。在这种模式下，它适合纳入一个标准化的分析管道。
更多参数可以查看帮助文档，我这里翻译了一部分。</p>
<p>-h --help ：打印此帮助文件并退出</p>
<p>-v --version：打印程序的版本并退出</p>
<p>-o --outdir：在指定的输出目录中创建所有输出文件。
请注意，这个目录必须存在，因为程序不会创建它。如果没有设置此选项，则在处理的序列文件所在的目录中创建每个序列文件的输出文件。</p>
<p>--casava：文件来自原始Casava输出。同一采样组中的文件(仅组号不同)将作为一组文件进行分析，而不是单独分析。在标题中设置了筛选器标志的序列将从分析中排除。文件必须与Casava为其指定的名称相同(包括被gzip压缩并以.gz结尾)，否则它们将无法正确分组在一起。</p>
<p>--nano ：文件来自nanopore序列并且是fast5格式。在这种模式下，您可以将目录传递给进程，程序将接收这些目录中的所有fast5文件，并根据所有文件中的序列生成单个输出文件。</p>
<p>--nofilter ：如果运行-casava，那么在执行QC分析时，不要删除由casava标记的不良质量。</p>
<p>--extract ：如果设置，则压缩后的输出文件将在创建后在同一目录中解压。默认情况下，fastqc在非交互模式下运行时将设置此选项。</p>
<p>-j --java：提供要用于启动fastqc的java二进制文件的完整路径。如果没有提供，则假定java在您的路径中。</p>
<p>--noextract：创建输出文件后，不要解压缩它。如果在非交互模式下运行时不希望解压缩输出，则应该设置此选项。  </p>
<p>--min_length：为报告中显示的序列长度设置一个人为的下限。只要您将其设置为大于或等于最长读取长度的值，那么这将是用于创建读取组的序列长度。这对于从读取长度可变的数据集中生成可直接分解的统计信息非常有用。</p>
<p>-f --format：绕过正常序列文件格式检测，并强制程序使用指定的格式。有效的格式是bam、sam、bam_mapped、sam_mapped和fastq。</p>
<p>-t --threads：指定可同时处理的文件数目。每个线程将被分配250MB的内存，所以您不应该运行超过可用内存应付的线程，并且在32位机器上不应该超过6个线程</p>
<p>-k --kmers ：指定要在Kmer内容模块中查找的Kmer的长度。指定的Kmer长度必须在2到10之间。如果未指定，默认长度为7。</p>
<p>-q --quiet：在标准输出上压缩所有进度消息，只报告错误。</p>
<p>-d --dir：选择要用于生成报表映像时写入的临时文件的目录。如果未指定，默认为系统临时目录。</p>
<h3 id="fastqc_2">（二）fastqc质控<a class="headerlink" href="#fastqc_2" title="Permanent link">&para;</a></h3>
<p>我这里将路径已经切换到了数据所在文件路径：data/RNAseq，数据在当前环境目录下，我打算输出结果路径：/root/HGJ_RNAseq/ ，你可以自定义，事先先创建好文件夹。质控只需要一行命令就搞定，注意，星号前面有一个空格，前面是定义的输出路径，后面是分析的文件名称。</p>
<p><div class="highlight"><pre><span></span><code>fastqc -o /root/HGJ_RNAseq/ *.fq.gz
</code></pre></div>
<img alt="" src="002.png" />
然后就开始一个文件一个文件处理，时间长短和文件大小、个数以及计算机配制有关，我这里的数据30分钟左右。
处理结束后，查看输出路径下的文件信息。</p>
<p><div class="highlight"><pre><span></span><code>ll -h /root/HGJ_RNAseq/
</code></pre></div>
<img alt="" src="003.png" />
我们可以看到，一个数据文件处理后会得到2个文件，一个html格式文件和一个zip的压缩文件，zip解压后和html格式文件内容是一样的，只需要下载html格式文件到本地，用浏览器打开查看。</p>
<h3 id="_1">（三）结果解读<a class="headerlink" href="#_1" title="Permanent link">&para;</a></h3>
<p>FastqC有3种结果：绿色代表PASS；黄色代表WARN；红色代表FAIL。当出现黄色时说明需要查看结果。当然，我这里没有黄色的结果。</p>
<h4 id="1basic-statistics"><strong>1.Basic Statistics</strong><a class="headerlink" href="#1basic-statistics" title="Permanent link">&para;</a></h4>
<p>Basic statistics是该fastq一些基本信息，主要有
Filename:文件名
File type: 文件类型
Encoding：测序平台的版本和相应的编码版本号，用于计算Phred反推error P时用
Total Sequences: 输入文本的reads的数量
Sequence length: 测序长度
%GC: GC含量，表示整体序列的GC含量，由于二代测序GC偏好性高，且深度越高，GC含量会越高。
           <img alt="" src="004.png" />             </p>
<h4 id="2per-base-sequence-quality"><strong>2.Per base sequence quality</strong><a class="headerlink" href="#2per-base-sequence-quality" title="Permanent link">&para;</a></h4>
<p>横轴为read长度，纵轴为质量得分，Q = -10*log10（error P）。关于错误率计算，可参考文章【<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651272899&amp;idx=1&amp;sn=6779b2bd21f3b607a08227d31c7212c6&amp;chksm=841ed2beb3695ba8bee23563c28caa005447b2298785719964732b16cafe3a15d7d4937b95c1&amp;scene=21#wechat_redirect">生信中常见的数据文件格式</a>】。柱状表示该位置所有序列的测序质量的统计，柱状是25%~75%区间质量分布，error bar是10%~90%区间质量分布，蓝线表示平均数。一般要求所有位置的10%分位数大于20，即大于最多允许该位置10%的序列低于Q20。当任何碱基质量低于10，或者任何中位数低于25报WARN,需注意；当任何碱基质量低于5或者任何中位数低于20报FAIL。
           <img alt="" src="005.png" />             </p>
<h4 id="3per-tile-sequence-quality"><strong>3.Per tile sequence quality</strong><a class="headerlink" href="#3per-tile-sequence-quality" title="Permanent link">&para;</a></h4>
<p>每个tail测序情况，横轴表示碱基位置，纵轴表示tail的index编号，这个图主要是为了防止在测序过程中某些tail受到不可控因素的影响而出现测序质量偏低，蓝色表示测序质量很高，暖色表示测序质量不高。当某些tail出现暖色，在后续的分析种把该tail测序结果全部去除。我这里的样本测序质量还是不错的。
           <img alt="" src="006.png" />             </p>
<h4 id="4per-sequence-quality-scores"><strong>4.Per sequence quality scores</strong><a class="headerlink" href="#4per-sequence-quality-scores" title="Permanent link">&para;</a></h4>
<p>横轴表示Q值，纵轴表示每个值对应的read数目，当测序结果主要集中在高分中，证明测序质量良好。
           <img alt="" src="007.png" />             </p>
<h4 id="5per-base-sequence-content"><strong>5.Per base sequence content</strong><a class="headerlink" href="#5per-base-sequence-content" title="Permanent link">&para;</a></h4>
<p>统计在序列中的每一个位置，四种不同碱基占总碱基数的比例，检测有无AT、GC分离的现象。横轴为位置，纵轴为百分比。正常情况下四种碱基出现的频率应是接近的，且没有位置差异，因此好的样品中四条线应该是平行且接近的，由于刚开始测序仪状态不稳定，造成前几个碱基有波动。在 reads 开头出现碱基组成偏离往往是我们的建库操作造成的，比如建 GBS 文库时在 reads 开头加了 barcode；barcode 的碱基组成不是均一的，酶切位点的碱基组成是固定不变的，这样会造成明显的碱基组成偏离；在 reads 结尾出现的碱基组成偏离，往往是测序接头的污染造成的。当所有位置的碱基比例一致现出偏差时，即四条线平行且分开，代表文库有偏差，或测序中的系统误差；当部分位置碱基的比例出现偏差时，即四条线在某些位置纷乱交织，则有overrepresented sequence的污染。当任一位置的A/T比例与G/C比例相差超过10%，报"WARN"；当任一位置的A/T比例与G/C比例相差超过20%，报"FAIL"。
           <img alt="" src="008.png" />             </p>
<h4 id="6per-sequence-gc-content"><strong>6.Per sequence GC content</strong><a class="headerlink" href="#6per-sequence-gc-content" title="Permanent link">&para;</a></h4>
<p>横轴表示GC含量，纵轴表示不同GC含量对应的read数，蓝线是理论分布（正态分布，通过从所测数据计算并构建理论分布），红色是实际情况，两个比较接近判为好的。曲线形状的偏差往往是由于文库的污染或是部分reads构成的子集有偏差（overrepresented reads）；形状接近正态分布但偏离理论分布的情况提示我们可能有系统偏差；如果出现两个或多个峰值，表明测序数据里可能有其他来源的DNA序列污染，或者有接头序列的二聚体污染。偏离理论分布的reads超过15%时，报"WARN"；偏离理论分布的reads超过30%时，报"FAIL"。
           <img alt="" src="009.png" />             </p>
<h4 id="7per-base-n-content"><strong>7.Per base N content</strong><a class="headerlink" href="#7per-base-n-content" title="Permanent link">&para;</a></h4>
<p>当出现测序仪不能分辨的碱基时会产生N，横轴为碱基分布，纵轴为N比率，当任一位置N的比率超过5%报WARN，超过20%报FAIL。我这里几乎没有。
           <img alt="" src="010.png" />             </p>
<h4 id="8sequence-length-distribution"><strong>8.Sequence Length Distribution</strong><a class="headerlink" href="#8sequence-length-distribution" title="Permanent link">&para;</a></h4>
<p>理论上每次测序仪测出的read长度是一致的，但是由于建库等因素通常会导致一些小片段，如果报FAIL，表明此次测序过程中产生的数据不可信。
           <img alt="" src="011.png" />             </p>
<h4 id="9sequence-duplication-levels"><strong>9.Sequence Duplication Levels</strong><a class="headerlink" href="#9sequence-duplication-levels" title="Permanent link">&para;</a></h4>
<p>统计序列完全一致的reads的频率，横轴表示重复的次数，纵轴表示重复的reads的数目。一般测序深度越高，越容易产生一定程度的重复序列。           <img alt="" src="012.png" />             </p>
<h4 id="10overrepresented-sequences"><strong>10.Overrepresented sequences</strong><a class="headerlink" href="#10overrepresented-sequences" title="Permanent link">&para;</a></h4>
<p>当有某个序列大量出现时，超过总reads数的0.1%时报WARN，超过1%时报FAIL。
           <img alt="" src="013.png" />             </p>
<h4 id="11adapter-content"><strong>11.Adapter Content</strong><a class="headerlink" href="#11adapter-content" title="Permanent link">&para;</a></h4>
<p>横轴表示碱基位置，纵轴表示百分比。当fastqc分析时没有选择参数-a adapter list时，默认使用图例中的4种通用adapter序列进行统计。若有adapter残留，后续必须去接头。
           <img alt="" src="014.png" />           <br />
以上就是一个完整的fastqC结果报告的简单说明，更多信息可参考：<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/">http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/</a>
结果解读部分来自文章：<a href="https://blog.csdn.net/gateswell/article/details/78858579">https://blog.csdn.net/gateswell/article/details/78858579</a>  ，对此感谢原作者。</p>
<h2 id="trim-galorereadsadaptor">二.使用trim-galore去除低质量的reads和adaptor<a class="headerlink" href="#trim-galorereadsadaptor" title="Permanent link">&para;</a></h2>
<p>我前面已经介绍了转录组分析中利用fastqc这个软件来查看测序质量【文章：<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273374&amp;idx=1&amp;sn=dfac9e82a09fbb18cd8b2c8179afcad8&amp;chksm=841edce3b36955f5ef17ae96b03c11b9d0052cd3490f1d6b91e3600a5d2eafa77ad79d4a40ff&amp;scene=21#wechat_redirect">转录组分析 | fastqc进行质控与结果解读</a>】，通过分析结果报告，我测序的数据还是可以的，但不管怎样，还是要清除一些不好的reads。这里我用trim-galore去除低质量的reads和adaptor。</p>
<h3 id="trim-galore">（一）Trim Galore介绍<a class="headerlink" href="#trim-galore" title="Permanent link">&para;</a></h3>
<p>Trim Galore是对FastQC和Cutadapt的包装。适用于所有高通量测序，包括RRBS(Reduced Representation Bisulfite-Seq ), Illumina、Nextera 和smallRNA测序平台的双端和单端数据。主要功能包括两步：
首先去除低质量碱基，然后去除3' 末端的adapter, 如果没有指定具体的adapter，程序会自动检测前1million的序列，然后对比前12-13bp的序列是否符合以下类型的adapter:</p>
<p>● Illumina:   AGATCGGAAGAGC</p>
<p>● Small RNA:  TGGAATTCTCGG</p>
<p>● Nextera:    CTGTCTCTTATA</p>
<hr />
<p>当然，Trim Galore是可以自动检测adapter。
接下来我们看软件参数：
--quality：设定Phred quality score阈值，默认为20。我后面分析改成25，稍微严格一些。
--phred33：：选择-phred33或者-phred64，表示测序平台使用的Phred quality score。具体怎么选择，看你用什么测序平台，这个在上一篇文章中的报告中就有【<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273374&amp;idx=1&amp;sn=dfac9e82a09fbb18cd8b2c8179afcad8&amp;chksm=841edce3b36955f5ef17ae96b03c11b9d0052cd3490f1d6b91e3600a5d2eafa77ad79d4a40ff&amp;scene=21#wechat_redirect">转录组分析 | fastqc进行质控与结果解读</a>】。
<img alt="" src="015.png" /></p>
<p>具体是-phred33还是-phred64我在文章【<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651272899&amp;idx=1&amp;sn=6779b2bd21f3b607a08227d31c7212c6&amp;chksm=841ed2beb3695ba8bee23563c28caa005447b2298785719964732b16cafe3a15d7d4937b95c1&amp;scene=21#wechat_redirect">生信中常见的数据文件格式</a>】中有提到。</p>
<p>重要参数如下：</p>
<p>--adapter：输入adapter序列。也可以不输入，Trim Galore!会自动寻找可能性最高的平台对应的adapter。自动搜选的平台三个，也直接显式输入这三种平台，即--illumina、--nextera和--small_rna。</p>
<p>--stringency：设定可以忍受的前后adapter重叠的碱基数，默认为1（非常苛刻)。可以适度放宽，因为后一个adapter几乎不可能被测序仪读到。</p>
<p>--length：设定输出reads长度阈值，小于设定值会被抛弃。</p>
<p>--paired：对于双端测序结果，一对reads中，如果有一个被剔除，那么另一个会被同样抛弃，而不管是否达到标准。</p>
<p>--retain_unpaired：对于双端测序结果，一对reads中，如果一个read达到标准，但是对应的另一个要被抛弃，达到标准的read会被单独保存为一个文件。</p>
<p>--gzip和--dont_gzip：清洗后的数据zip打包或者不打包。</p>
<p>--output_dir：输入目录。需要提前建立目录，否则运行会报错。</p>
<p>-- trim-n : 移除read一端的reads</p>
<h3 id="trim-galorereadsadaptor_1">（二）使用trim-galore去除低质量的reads和adaptor<a class="headerlink" href="#trim-galorereadsadaptor_1" title="Permanent link">&para;</a></h3>
<p>首先，创建保存输出数据的文件夹。
<div class="highlight"><pre><span></span><code>mkdir cleandata mkdir ./cleandata/trim_galoredata
</code></pre></div>
然后接下来我们就可以处理数据了。下面命令是处理单个样本的案例，可以不先运行，只需要知道是什么意思。
<div class="highlight"><pre><span></span><code>trim_galore -q 25 --phred33 --stringency 3 --length 36  --paired CK-4_1.fq.gz CK-4_2.fq.gz --gzip -o ./cleandata/trim_galoredata/
</code></pre></div>
参数解释前面已经介绍，这里提一下，我的测序是双端测序，我们前面12个文件对应6个样本，都是成对的。CK-4_1.fq.gz CK-4_2.fq.gz是一对，处理数据的时候就是一起处理。
<img alt="" src="016.png" />  </p>
<p>但是我们有6个样本的数据，就需要写6个命令，好像有点不高效，所以我们可以写一个脚本，执行一次就行。
在当前路径下创建一个脚本文件trim_galore_batch.sh</p>
<p><div class="highlight"><pre><span></span><code>vim trim_galore_batch.sh
</code></pre></div>
 输入下面内容：
<div class="highlight"><pre><span></span><code>#!/bin/bash
# This is for trimming a batch data
for i in CK-4 CK-7 CK-8 HGJ-10 HGJ-6 HGJ-9
do
        trim_galore -q 25 --phred33 --length 36 --stringency 3 --paired /data/RNAseq/${i}_1.fq.gz /data/RNAseq/${i}_2.fq.gz --gzip -o /data/RNAseq/cleandata/trim_galoredata
done
</code></pre></div>
其实就是一个for循环：CK-4 CK-7 CK-8 HGJ-10 HGJ-6 HGJ-9是文件名称的前部分，也就是样本名称，测序数据文件命名都是有规律的，认真看一下，就知道上面内容是什么意思。
         <img alt="" src="017.png" /> 保存脚本，然后执行脚本。</p>
<p><div class="highlight"><pre><span></span><code>bash trim_galore_batch.sh
</code></pre></div>
  脚本没有问题的话，会看到类似下面的信息
<div class="highlight"><pre><span></span><code>(RNAseq1) [root@localhost RNAseq]# bash trim_galore_batch.sh
Multicore support not enabled. Proceeding with single-core trimming.
Path to Cutadapt set as: &#39;cutadapt&#39; (default)
Cutadapt seems to be working fine (tested command &#39;cutadapt --version&#39;)
Cutadapt version: 1.18
single-core operation.
Output will be written into the directory: /data/RNAseq/cleandata/trim_galoredata/
AUTO-DETECTING ADAPTER TYPE
===========================
Attempting to auto-detect adapter type from the first 1 million sequences of the first file (&gt;&gt; /data/RNAseq/CK-4_1.fq.gz &lt;&lt;)

Found perfect matches for the following adapter sequences:
Adapter type    Count   Sequence        Sequences analysed      Percentage
Illumina        28671   AGATCGGAAGAGC   1000000 2.87
smallRNA        5       TGGAATTCTCGG    1000000 0.00
Nextera 4       CTGTCTCTTATA    1000000 0.00
Using Illumina adapter for trimming (count: 28671). Second best hit was smallRNA (count: 5)

Writing report to &#39;/data/RNAseq/cleandata/trim_galoredata/CK-4_1.fq.gz_trimming_report.txt&#39;

SUMMARISING RUN PARAMETERS
==========================
Input filename: /data/RNAseq/CK-4_1.fq.gz
Trimming mode: paired-end
Trim Galore version: 0.6.6
Cutadapt version: 1.18
Number of cores used for trimming: 1
Quality Phred score cutoff: 25
Quality encoding type selected: ASCII+33
Adapter sequence: &#39;AGATCGGAAGAGC&#39; (Illumina TruSeq, Sanger iPCR; auto-detected)
Maximum trimming error rate: 0.1 (default)
Minimum required adapter overlap (stringency): 3 bp
Minimum required sequence length for both reads before a sequence pair gets removed: 36 bp
Output file(s) will be GZIP compressed

Cutadapt seems to be reasonably up-to-date. Setting -j 1
Writing final adapter and quality trimmed output to CK-4_1_trimmed.fq.gz


  &gt;&gt;&gt; Now performing quality (cutoff &#39;-q 25&#39;) and adapter trimming in a single pass for the adapter sequence: &#39;AGATCGGAAGAGC&#39; from file /data/RNAseq/CK-4_1.fq.gz &lt;&lt;&lt;
</code></pre></div></p>
<p><img alt="" src="018.png" />
这个时间很长很长。一个文件1个小时左右。慢慢等..................
运行结束后查看文件：</p>
<p><div class="highlight"><pre><span></span><code>ll -h /data/RNAseq/cleandata/trim_galoredata
</code></pre></div>
<img alt="" src="019.png" /></p>
<p>fq.gz格式文件是处理后得到的数据，如果还记得的话，前面我们的数据是27G，现在质控后只有22G的数据。txt格式文件是样品处理的结果报告，也包括软件运行的参数信息。下面是其中一个的结果。</p>
<p><div class="highlight"><pre><span></span><code>SUMMARISING<span class="w"> </span>RUN<span class="w"> </span><span class="nv">PARAMETERS</span>
<span class="o">==========================</span>
Input<span class="w"> </span>filename:<span class="w"> </span>/data/RNAseq/CK-4_1.fq.gz
Trimming<span class="w"> </span>mode:<span class="w"> </span>paired-end
Trim<span class="w"> </span>Galore<span class="w"> </span>version:<span class="w"> </span><span class="m">0</span>.6.6
Cutadapt<span class="w"> </span>version:<span class="w"> </span><span class="m">1</span>.18
Number<span class="w"> </span>of<span class="w"> </span>cores<span class="w"> </span>used<span class="w"> </span><span class="k">for</span><span class="w"> </span>trimming:<span class="w"> </span><span class="m">1</span>
Quality<span class="w"> </span>Phred<span class="w"> </span>score<span class="w"> </span>cutoff:<span class="w"> </span><span class="m">25</span>
Quality<span class="w"> </span>encoding<span class="w"> </span><span class="nb">type</span><span class="w"> </span>selected:<span class="w"> </span>ASCII+33
Using<span class="w"> </span>Illumina<span class="w"> </span>adapter<span class="w"> </span><span class="k">for</span><span class="w"> </span>trimming<span class="w"> </span><span class="o">(</span>count:<span class="w"> </span><span class="m">28671</span><span class="o">)</span>.<span class="w"> </span>Second<span class="w"> </span>best<span class="w"> </span>hit<span class="w"> </span>was<span class="w"> </span>smallRNA<span class="w"> </span><span class="o">(</span>count:<span class="w"> </span><span class="m">5</span><span class="o">)</span>
Adapter<span class="w"> </span>sequence:<span class="w"> </span><span class="s1">&#39;AGATCGGAAGAGC&#39;</span><span class="w"> </span><span class="o">(</span>Illumina<span class="w"> </span>TruSeq,<span class="w"> </span>Sanger<span class="w"> </span>iPCR<span class="p">;</span><span class="w"> </span>auto-detected<span class="o">)</span>
Maximum<span class="w"> </span>trimming<span class="w"> </span>error<span class="w"> </span>rate:<span class="w"> </span><span class="m">0</span>.1<span class="w"> </span><span class="o">(</span>default<span class="o">)</span>
Minimum<span class="w"> </span>required<span class="w"> </span>adapter<span class="w"> </span>overlap<span class="w"> </span><span class="o">(</span>stringency<span class="o">)</span>:<span class="w"> </span><span class="m">3</span><span class="w"> </span>bp
Minimum<span class="w"> </span>required<span class="w"> </span>sequence<span class="w"> </span>length<span class="w"> </span><span class="k">for</span><span class="w"> </span>both<span class="w"> </span>reads<span class="w"> </span>before<span class="w"> </span>a<span class="w"> </span>sequence<span class="w"> </span>pair<span class="w"> </span>gets<span class="w"> </span>removed:<span class="w"> </span><span class="m">36</span><span class="w"> </span>bp
Output<span class="w"> </span>file<span class="w"> </span>will<span class="w"> </span>be<span class="w"> </span>GZIP<span class="w"> </span>compressed


This<span class="w"> </span>is<span class="w"> </span>cutadapt<span class="w"> </span><span class="m">1</span>.18<span class="w"> </span>with<span class="w"> </span>Python<span class="w"> </span><span class="m">2</span>.7.15
Command<span class="w"> </span>line<span class="w"> </span>parameters:<span class="w"> </span>-j<span class="w"> </span><span class="m">1</span><span class="w"> </span>-e<span class="w"> </span><span class="m">0</span>.1<span class="w"> </span>-q<span class="w"> </span><span class="m">25</span><span class="w"> </span>-O<span class="w"> </span><span class="m">3</span><span class="w"> </span>-a<span class="w"> </span>AGATCGGAAGAGC<span class="w"> </span>/data/RNAseq/CK-4_1.fq.gz
Processing<span class="w"> </span>reads<span class="w"> </span>on<span class="w"> </span><span class="m">1</span><span class="w"> </span>core<span class="w"> </span><span class="k">in</span><span class="w"> </span>single-end<span class="w"> </span>mode<span class="w"> </span>...
Finished<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1496</span>.55<span class="w"> </span>s<span class="w"> </span><span class="o">(</span><span class="m">62</span><span class="w"> </span>us/read<span class="p">;</span><span class="w"> </span><span class="m">0</span>.96<span class="w"> </span>M<span class="w"> </span>reads/minute<span class="o">)</span>.

<span class="o">===</span><span class="w"> </span><span class="nv">Summary</span><span class="w"> </span><span class="o">===</span>

Total<span class="w"> </span>reads<span class="w"> </span>processed:<span class="w">              </span><span class="m">23</span>,985,482
Reads<span class="w"> </span>with<span class="w"> </span>adapters:<span class="w">                 </span><span class="m">1</span>,508,012<span class="w"> </span><span class="o">(</span><span class="m">6</span>.3%<span class="o">)</span>
Reads<span class="w"> </span>written<span class="w"> </span><span class="o">(</span>passing<span class="w"> </span>filters<span class="o">)</span>:<span class="w">    </span><span class="m">23</span>,985,482<span class="w"> </span><span class="o">(</span><span class="m">100</span>.0%<span class="o">)</span>

Total<span class="w"> </span>basepairs<span class="w"> </span>processed:<span class="w"> </span><span class="m">3</span>,597,822,300<span class="w"> </span>bp
Quality-trimmed:<span class="w">               </span><span class="m">6</span>,269,587<span class="w"> </span>bp<span class="w"> </span><span class="o">(</span><span class="m">0</span>.2%<span class="o">)</span>
Total<span class="w"> </span>written<span class="w"> </span><span class="o">(</span>filtered<span class="o">)</span>:<span class="w">  </span><span class="m">3</span>,567,424,870<span class="w"> </span>bp<span class="w"> </span><span class="o">(</span><span class="m">99</span>.2%<span class="o">)</span>

<span class="o">===</span><span class="w"> </span>Adapter<span class="w"> </span><span class="nv">1</span><span class="w"> </span><span class="o">===</span>

Sequence:<span class="w"> </span>AGATCGGAAGAGC<span class="p">;</span><span class="w"> </span>Type:<span class="w"> </span>regular<span class="w"> </span><span class="m">3</span><span class="err">&#39;</span><span class="p">;</span><span class="w"> </span>Length:<span class="w"> </span><span class="m">13</span><span class="p">;</span><span class="w"> </span>Trimmed:<span class="w"> </span><span class="m">1508012</span><span class="w"> </span>times.

No.<span class="w"> </span>of<span class="w"> </span>allowed<span class="w"> </span>errors:
<span class="m">0</span>-9<span class="w"> </span>bp:<span class="w"> </span><span class="m">0</span><span class="p">;</span><span class="w"> </span><span class="m">10</span>-13<span class="w"> </span>bp:<span class="w"> </span><span class="m">1</span>

Bases<span class="w"> </span>preceding<span class="w"> </span>removed<span class="w"> </span>adapters:
<span class="w">  </span>A:<span class="w"> </span><span class="m">19</span>.9%
<span class="w">  </span>C:<span class="w"> </span><span class="m">34</span>.7%
<span class="w">  </span>G:<span class="w"> </span><span class="m">33</span>.0%
<span class="w">  </span>T:<span class="w"> </span><span class="m">12</span>.4%
<span class="w">  </span>none/other:<span class="w"> </span><span class="m">0</span>.0%

Overview<span class="w"> </span>of<span class="w"> </span>removed<span class="w"> </span>sequences
length<span class="w">  </span>count<span class="w">  </span>expect<span class="w">  </span>max.err<span class="w">  </span>error<span class="w"> </span>counts
<span class="m">3</span><span class="w">  </span><span class="m">490193</span><span class="w">  </span><span class="m">374773</span>.2<span class="w">  </span><span class="m">0</span><span class="w">  </span><span class="m">490193</span>
<span class="m">4</span><span class="w">  </span><span class="m">129883</span><span class="w">  </span><span class="m">93693</span>.3<span class="w">  </span><span class="m">0</span><span class="w">  </span><span class="m">129883</span>
<span class="m">5</span><span class="w">  </span><span class="m">55548</span><span class="w">  </span><span class="m">23423</span>.3<span class="w">  </span><span class="m">0</span><span class="w">  </span><span class="m">55548</span>
<span class="m">6</span><span class="w">  </span><span class="m">30967</span><span class="w">  </span><span class="m">5855</span>.8<span class="w">  </span><span class="m">0</span><span class="w">  </span><span class="m">30967</span>
<span class="m">7</span><span class="w">  </span><span class="m">27693</span><span class="w">  </span><span class="m">1464</span>.0<span class="w">  </span><span class="m">0</span><span class="w">  </span><span class="m">27693</span>
<span class="m">8</span><span class="w">  </span><span class="m">27513</span><span class="w">  </span><span class="m">366</span>.0<span class="w">  </span><span class="m">0</span><span class="w">  </span><span class="m">27513</span>
<span class="m">9</span><span class="w">  </span><span class="m">25573</span><span class="w">  </span><span class="m">91</span>.5<span class="w">  </span><span class="m">0</span><span class="w">  </span><span class="m">24770</span><span class="w"> </span><span class="m">803</span>
<span class="m">10</span><span class="w">  </span><span class="m">24873</span><span class="w">  </span><span class="m">22</span>.9<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">23690</span><span class="w"> </span><span class="m">1183</span>
<span class="m">11</span><span class="w">  </span><span class="m">25827</span><span class="w">  </span><span class="m">5</span>.7<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">24471</span><span class="w"> </span><span class="m">1356</span>
<span class="m">12</span><span class="w">  </span><span class="m">25970</span><span class="w">  </span><span class="m">1</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">24553</span><span class="w"> </span><span class="m">1417</span>
<span class="m">13</span><span class="w">  </span><span class="m">24143</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">23149</span><span class="w"> </span><span class="m">994</span>
<span class="m">14</span><span class="w">  </span><span class="m">25530</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">24384</span><span class="w"> </span><span class="m">1146</span>
<span class="m">15</span><span class="w">  </span><span class="m">24349</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">23344</span><span class="w"> </span><span class="m">1005</span>
<span class="m">16</span><span class="w">  </span><span class="m">24383</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">23251</span><span class="w"> </span><span class="m">1132</span>
<span class="m">17</span><span class="w">  </span><span class="m">24256</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">22978</span><span class="w"> </span><span class="m">1278</span>
<span class="m">18</span><span class="w">  </span><span class="m">21920</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">21005</span><span class="w"> </span><span class="m">915</span>
<span class="m">19</span><span class="w">  </span><span class="m">19694</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">18969</span><span class="w"> </span><span class="m">725</span>
<span class="m">20</span><span class="w">  </span><span class="m">20484</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">19673</span><span class="w"> </span><span class="m">811</span>
<span class="m">21</span><span class="w">  </span><span class="m">19751</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">18973</span><span class="w"> </span><span class="m">778</span>
<span class="m">22</span><span class="w">  </span><span class="m">19049</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">18345</span><span class="w"> </span><span class="m">704</span>
<span class="m">23</span><span class="w">  </span><span class="m">19704</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">18845</span><span class="w"> </span><span class="m">859</span>
<span class="m">24</span><span class="w">  </span><span class="m">20170</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">19256</span><span class="w"> </span><span class="m">914</span>
<span class="m">25</span><span class="w">  </span><span class="m">19510</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">18680</span><span class="w"> </span><span class="m">830</span>
<span class="m">26</span><span class="w">  </span><span class="m">18750</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">18000</span><span class="w"> </span><span class="m">750</span>
<span class="m">27</span><span class="w">  </span><span class="m">18256</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">17503</span><span class="w"> </span><span class="m">753</span>
<span class="m">28</span><span class="w">  </span><span class="m">16883</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">16261</span><span class="w"> </span><span class="m">622</span>
<span class="m">29</span><span class="w">  </span><span class="m">15923</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">15224</span><span class="w"> </span><span class="m">699</span>
<span class="m">30</span><span class="w">  </span><span class="m">15198</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">14587</span><span class="w"> </span><span class="m">611</span>
<span class="m">31</span><span class="w">  </span><span class="m">13894</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">13376</span><span class="w"> </span><span class="m">518</span>
<span class="m">32</span><span class="w">  </span><span class="m">14121</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">13607</span><span class="w"> </span><span class="m">514</span>
<span class="m">33</span><span class="w">  </span><span class="m">13453</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">12847</span><span class="w"> </span><span class="m">606</span>
<span class="m">34</span><span class="w">  </span><span class="m">14484</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">13845</span><span class="w"> </span><span class="m">639</span>
<span class="m">35</span><span class="w">  </span><span class="m">14362</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">13630</span><span class="w"> </span><span class="m">732</span>
<span class="m">36</span><span class="w">  </span><span class="m">13907</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">13328</span><span class="w"> </span><span class="m">579</span>
<span class="m">37</span><span class="w">  </span><span class="m">15268</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">14744</span><span class="w"> </span><span class="m">524</span>
<span class="m">38</span><span class="w">  </span><span class="m">8918</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">8453</span><span class="w"> </span><span class="m">465</span>
<span class="m">39</span><span class="w">  </span><span class="m">10640</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">10184</span><span class="w"> </span><span class="m">456</span>
<span class="m">40</span><span class="w">  </span><span class="m">10771</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">10404</span><span class="w"> </span><span class="m">367</span>
<span class="m">41</span><span class="w">  </span><span class="m">7766</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">7388</span><span class="w"> </span><span class="m">378</span>
<span class="m">42</span><span class="w">  </span><span class="m">8181</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">7900</span><span class="w"> </span><span class="m">281</span>
<span class="m">43</span><span class="w">  </span><span class="m">9762</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">9364</span><span class="w"> </span><span class="m">398</span>
<span class="m">44</span><span class="w">  </span><span class="m">8479</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">8090</span><span class="w"> </span><span class="m">389</span>
<span class="m">45</span><span class="w">  </span><span class="m">14727</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">14310</span><span class="w"> </span><span class="m">417</span>
<span class="m">46</span><span class="w">  </span><span class="m">3156</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">2941</span><span class="w"> </span><span class="m">215</span>
<span class="m">47</span><span class="w">  </span><span class="m">5179</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">4934</span><span class="w"> </span><span class="m">245</span>
<span class="m">48</span><span class="w">  </span><span class="m">7148</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">6891</span><span class="w"> </span><span class="m">257</span>
<span class="m">49</span><span class="w">  </span><span class="m">5975</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">5749</span><span class="w"> </span><span class="m">226</span>
<span class="m">50</span><span class="w">  </span><span class="m">4647</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">4454</span><span class="w"> </span><span class="m">193</span>
<span class="m">51</span><span class="w">  </span><span class="m">5267</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">5063</span><span class="w"> </span><span class="m">204</span>
<span class="m">52</span><span class="w">  </span><span class="m">4253</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">4097</span><span class="w"> </span><span class="m">156</span>
<span class="m">53</span><span class="w">  </span><span class="m">3958</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">3805</span><span class="w"> </span><span class="m">153</span>
<span class="m">54</span><span class="w">  </span><span class="m">4345</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">4159</span><span class="w"> </span><span class="m">186</span>
<span class="m">55</span><span class="w">  </span><span class="m">4843</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">4663</span><span class="w"> </span><span class="m">180</span>
<span class="m">56</span><span class="w">  </span><span class="m">4152</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">3991</span><span class="w"> </span><span class="m">161</span>
<span class="m">57</span><span class="w">  </span><span class="m">4613</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">4460</span><span class="w"> </span><span class="m">153</span>
<span class="m">58</span><span class="w">  </span><span class="m">3645</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">3474</span><span class="w"> </span><span class="m">171</span>
<span class="m">59</span><span class="w">  </span><span class="m">2935</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">2810</span><span class="w"> </span><span class="m">125</span>
<span class="m">60</span><span class="w">  </span><span class="m">2315</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">2232</span><span class="w"> </span><span class="m">83</span>
<span class="m">61</span><span class="w">  </span><span class="m">1453</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">1366</span><span class="w"> </span><span class="m">87</span>
<span class="m">62</span><span class="w">  </span><span class="m">2271</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">2180</span><span class="w"> </span><span class="m">91</span>
<span class="m">63</span><span class="w">  </span><span class="m">2382</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">2295</span><span class="w"> </span><span class="m">87</span>
<span class="m">64</span><span class="w">  </span><span class="m">1702</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">1646</span><span class="w"> </span><span class="m">56</span>
<span class="m">65</span><span class="w">  </span><span class="m">1394</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">1276</span><span class="w"> </span><span class="m">118</span>
<span class="m">66</span><span class="w">  </span><span class="m">2765</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">2667</span><span class="w"> </span><span class="m">98</span>
<span class="m">67</span><span class="w">  </span><span class="m">1689</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">1595</span><span class="w"> </span><span class="m">94</span>
<span class="m">68</span><span class="w">  </span><span class="m">1880</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">1789</span><span class="w"> </span><span class="m">91</span>
<span class="m">69</span><span class="w">  </span><span class="m">867</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">819</span><span class="w"> </span><span class="m">48</span>
<span class="m">70</span><span class="w">  </span><span class="m">553</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">503</span><span class="w"> </span><span class="m">50</span>
<span class="m">71</span><span class="w">  </span><span class="m">85</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">41</span><span class="w"> </span><span class="m">44</span>
<span class="m">72</span><span class="w">  </span><span class="m">66</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">38</span><span class="w"> </span><span class="m">28</span>
<span class="m">73</span><span class="w">  </span><span class="m">271</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">225</span><span class="w"> </span><span class="m">46</span>
<span class="m">74</span><span class="w">  </span><span class="m">514</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">468</span><span class="w"> </span><span class="m">46</span>
<span class="m">75</span><span class="w">  </span><span class="m">687</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">645</span><span class="w"> </span><span class="m">42</span>
<span class="m">76</span><span class="w">  </span><span class="m">779</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">725</span><span class="w"> </span><span class="m">54</span>
<span class="m">77</span><span class="w">  </span><span class="m">761</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">709</span><span class="w"> </span><span class="m">52</span>
<span class="m">78</span><span class="w">  </span><span class="m">802</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">738</span><span class="w"> </span><span class="m">64</span>
<span class="m">79</span><span class="w">  </span><span class="m">641</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">608</span><span class="w"> </span><span class="m">33</span>
<span class="m">80</span><span class="w">  </span><span class="m">537</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">499</span><span class="w"> </span><span class="m">38</span>
<span class="m">81</span><span class="w">  </span><span class="m">488</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">445</span><span class="w"> </span><span class="m">43</span>
<span class="m">82</span><span class="w">  </span><span class="m">406</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">341</span><span class="w"> </span><span class="m">65</span>
<span class="m">83</span><span class="w">  </span><span class="m">323</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">295</span><span class="w"> </span><span class="m">28</span>
<span class="m">84</span><span class="w">  </span><span class="m">395</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">356</span><span class="w"> </span><span class="m">39</span>
<span class="m">85</span><span class="w">  </span><span class="m">400</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">371</span><span class="w"> </span><span class="m">29</span>
<span class="m">86</span><span class="w">  </span><span class="m">398</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">356</span><span class="w"> </span><span class="m">42</span>
<span class="m">87</span><span class="w">  </span><span class="m">419</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">364</span><span class="w"> </span><span class="m">55</span>
<span class="m">88</span><span class="w">  </span><span class="m">330</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">295</span><span class="w"> </span><span class="m">35</span>
<span class="m">89</span><span class="w">  </span><span class="m">335</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">300</span><span class="w"> </span><span class="m">35</span>
<span class="m">90</span><span class="w">  </span><span class="m">270</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">224</span><span class="w"> </span><span class="m">46</span>
<span class="m">91</span><span class="w">  </span><span class="m">210</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">175</span><span class="w"> </span><span class="m">35</span>
<span class="m">92</span><span class="w">  </span><span class="m">213</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">168</span><span class="w"> </span><span class="m">45</span>
<span class="m">93</span><span class="w">  </span><span class="m">201</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">169</span><span class="w"> </span><span class="m">32</span>
<span class="m">94</span><span class="w">  </span><span class="m">160</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">121</span><span class="w"> </span><span class="m">39</span>
<span class="m">95</span><span class="w">  </span><span class="m">197</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">147</span><span class="w"> </span><span class="m">50</span>
<span class="m">96</span><span class="w">  </span><span class="m">167</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">126</span><span class="w"> </span><span class="m">41</span>
<span class="m">97</span><span class="w">  </span><span class="m">193</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">167</span><span class="w"> </span><span class="m">26</span>
<span class="m">98</span><span class="w">  </span><span class="m">189</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">149</span><span class="w"> </span><span class="m">40</span>
<span class="m">99</span><span class="w">  </span><span class="m">192</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">151</span><span class="w"> </span><span class="m">41</span>
<span class="m">100</span><span class="w">  </span><span class="m">182</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">130</span><span class="w"> </span><span class="m">52</span>
<span class="m">101</span><span class="w">  </span><span class="m">124</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">94</span><span class="w"> </span><span class="m">30</span>
<span class="m">102</span><span class="w">  </span><span class="m">116</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">75</span><span class="w"> </span><span class="m">41</span>
<span class="m">103</span><span class="w">  </span><span class="m">96</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">66</span><span class="w"> </span><span class="m">30</span>
<span class="m">104</span><span class="w">  </span><span class="m">96</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">62</span><span class="w"> </span><span class="m">34</span>
<span class="m">105</span><span class="w">  </span><span class="m">104</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">64</span><span class="w"> </span><span class="m">40</span>
<span class="m">106</span><span class="w">  </span><span class="m">85</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">55</span><span class="w"> </span><span class="m">30</span>
<span class="m">107</span><span class="w">  </span><span class="m">92</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">73</span><span class="w"> </span><span class="m">19</span>
<span class="m">108</span><span class="w">  </span><span class="m">121</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">80</span><span class="w"> </span><span class="m">41</span>
<span class="m">109</span><span class="w">  </span><span class="m">104</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">64</span><span class="w"> </span><span class="m">40</span>
<span class="m">110</span><span class="w">  </span><span class="m">85</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">60</span><span class="w"> </span><span class="m">25</span>
<span class="m">111</span><span class="w">  </span><span class="m">83</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">53</span><span class="w"> </span><span class="m">30</span>
<span class="m">112</span><span class="w">  </span><span class="m">71</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">37</span><span class="w"> </span><span class="m">34</span>
<span class="m">113</span><span class="w">  </span><span class="m">58</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">39</span><span class="w"> </span><span class="m">19</span>
<span class="m">114</span><span class="w">  </span><span class="m">88</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">34</span><span class="w"> </span><span class="m">54</span>
<span class="m">115</span><span class="w">  </span><span class="m">65</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">36</span><span class="w"> </span><span class="m">29</span>
<span class="m">116</span><span class="w">  </span><span class="m">70</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">37</span><span class="w"> </span><span class="m">33</span>
<span class="m">117</span><span class="w">  </span><span class="m">69</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">41</span><span class="w"> </span><span class="m">28</span>
<span class="m">118</span><span class="w">  </span><span class="m">75</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">38</span><span class="w"> </span><span class="m">37</span>
<span class="m">119</span><span class="w">  </span><span class="m">47</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">20</span><span class="w"> </span><span class="m">27</span>
<span class="m">120</span><span class="w">  </span><span class="m">69</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">36</span><span class="w"> </span><span class="m">33</span>
<span class="m">121</span><span class="w">  </span><span class="m">65</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">29</span><span class="w"> </span><span class="m">36</span>
<span class="m">122</span><span class="w">  </span><span class="m">58</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">22</span><span class="w"> </span><span class="m">36</span>
<span class="m">123</span><span class="w">  </span><span class="m">41</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">15</span><span class="w"> </span><span class="m">26</span>
<span class="m">124</span><span class="w">  </span><span class="m">52</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">15</span><span class="w"> </span><span class="m">37</span>
<span class="m">125</span><span class="w">  </span><span class="m">52</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">17</span><span class="w"> </span><span class="m">35</span>
<span class="m">126</span><span class="w">  </span><span class="m">33</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">7</span><span class="w"> </span><span class="m">26</span>
<span class="m">127</span><span class="w">  </span><span class="m">46</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">11</span><span class="w"> </span><span class="m">35</span>
<span class="m">128</span><span class="w">  </span><span class="m">61</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">9</span><span class="w"> </span><span class="m">52</span>
<span class="m">129</span><span class="w">  </span><span class="m">47</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">17</span><span class="w"> </span><span class="m">30</span>
<span class="m">130</span><span class="w">  </span><span class="m">57</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">11</span><span class="w"> </span><span class="m">46</span>
<span class="m">131</span><span class="w">  </span><span class="m">40</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">10</span><span class="w"> </span><span class="m">30</span>
<span class="m">132</span><span class="w">  </span><span class="m">29</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">8</span><span class="w"> </span><span class="m">21</span>
<span class="m">133</span><span class="w">  </span><span class="m">28</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">4</span><span class="w"> </span><span class="m">24</span>
<span class="m">134</span><span class="w">  </span><span class="m">36</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">4</span><span class="w"> </span><span class="m">32</span>
<span class="m">135</span><span class="w">  </span><span class="m">41</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">3</span><span class="w"> </span><span class="m">38</span>
<span class="m">136</span><span class="w">  </span><span class="m">17</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">1</span><span class="w"> </span><span class="m">16</span>
<span class="m">137</span><span class="w">  </span><span class="m">44</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">3</span><span class="w"> </span><span class="m">41</span>
<span class="m">138</span><span class="w">  </span><span class="m">25</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">3</span><span class="w"> </span><span class="m">22</span>
<span class="m">139</span><span class="w">  </span><span class="m">28</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">1</span><span class="w"> </span><span class="m">27</span>
<span class="m">140</span><span class="w">  </span><span class="m">15</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">1</span><span class="w"> </span><span class="m">14</span>
<span class="m">141</span><span class="w">  </span><span class="m">31</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">0</span><span class="w"> </span><span class="m">31</span>
<span class="m">142</span><span class="w">  </span><span class="m">24</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">0</span><span class="w"> </span><span class="m">24</span>
<span class="m">143</span><span class="w">  </span><span class="m">19</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">1</span><span class="w"> </span><span class="m">18</span>
<span class="m">144</span><span class="w">  </span><span class="m">4</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">1</span><span class="w"> </span><span class="m">3</span>
<span class="m">145</span><span class="w">  </span><span class="m">12</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">0</span><span class="w"> </span><span class="m">12</span>
<span class="m">146</span><span class="w">  </span><span class="m">16</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">2</span><span class="w"> </span><span class="m">14</span>
<span class="m">147</span><span class="w">  </span><span class="m">69</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">1</span><span class="w"> </span><span class="m">68</span>
<span class="m">148</span><span class="w">  </span><span class="m">22</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">0</span><span class="w"> </span><span class="m">22</span>
<span class="m">149</span><span class="w">  </span><span class="m">14</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">0</span><span class="w"> </span><span class="m">14</span>
<span class="m">150</span><span class="w">  </span><span class="m">344</span><span class="w">  </span><span class="m">0</span>.4<span class="w">  </span><span class="m">1</span><span class="w">  </span><span class="m">4</span><span class="w"> </span><span class="m">340</span>


RUN<span class="w"> </span>STATISTICS<span class="w"> </span>FOR<span class="w"> </span>INPUT<span class="w"> </span>FILE:<span class="w"> </span>/data/RNAseq/CK-4_1.fq.gz
<span class="o">=============================================</span>
<span class="m">23985482</span><span class="w"> </span>sequences<span class="w"> </span>processed<span class="w"> </span><span class="k">in</span><span class="w"> </span>total
</code></pre></div>
这一部分就介绍到这里，本公众号文章目录可查看文章：<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273306&amp;idx=1&amp;sn=d92056ee1fde1ff12b1506e02f8eb56f&amp;chksm=841edc27b36955310bac7e1b4e5e4ab7e26e669c192926dbee3efc4f6d185cf3509a8fc87696&amp;scene=21#wechat_redirect">公众号文章目录</a>，快速浏览相应文章。</p>
<h2 id="hisat2">三.使用Hisat2进行序列比对<a class="headerlink" href="#hisat2" title="Permanent link">&para;</a></h2>
<p>我前面已经对数据进行了质控：</p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273374&amp;idx=1&amp;sn=dfac9e82a09fbb18cd8b2c8179afcad8&amp;chksm=841edce3b36955f5ef17ae96b03c11b9d0052cd3490f1d6b91e3600a5d2eafa77ad79d4a40ff&amp;scene=21#wechat_redirect"><strong>转录组分析 | fastqc进行质控与结果解读</strong></a></p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273401&amp;idx=1&amp;sn=ff9e3478309d805950c40d0ebae74296&amp;chksm=841edcc4b36955d2a5bf8c85429114b5b83ec4e5afe869378b42ec4f659d4953d5f6c8e9e379&amp;scene=21#wechat_redirect"><strong>转录组分析 | 使用trim-galore去除低质量的reads和adaptor</strong></a></p>
<p>接下来我们进行序列比对，利用的软件是Hisat2。</p>
<h3 id="index">（一）index文件下载<a class="headerlink" href="#index" title="Permanent link">&para;</a></h3>
<p>index文件直接去官网下载
(<a href="http://daehwankimlab.github.io/hisat2/download/">http://daehwankimlab.github.io/hisat2/download/</a>)，我测序的组织来自小鼠，所以我这里下载的是小鼠的。</p>
<p><img alt="" src="020.png" /></p>
<p>下载完后上传到Linux服务器，解压后就可以直接用了。
我上传到了： /data/mouse_genome/  ，就是mm10_genome.tar.gz这个文件。
<img alt="" src="021.png" />
解压文件，解压过程中会在当前文件夹下创建mm10文件，解压后的文件就在mm10文件夹中。</p>
<p><div class="highlight"><pre><span></span><code>tar -zxvf /data/mouse_genome/mm10_genome_tar.gz
</code></pre></div>
<img alt="" src="022.png" /></p>
<p>查看解压后的文件。</p>
<p><div class="highlight"><pre><span></span><code>ll -h ./mm10
</code></pre></div>
<img alt="" src="023.png" /></p>
<p>总共8个ht2格式文件，一个sh格式文件。</p>
<h3 id="hisat2_1">（二）hisat2介绍<a class="headerlink" href="#hisat2_1" title="Permanent link">&para;</a></h3>
<p>Hisat是一种高效的RNA-seq实验比对工具。它使用了基于BWT和Ferragina-manzini (Fm) index 两种算法的索引框架。使用了两类索引去比对，一类是全基因组范围的FM索引来锚定每一个比对，另一类是大量的局部索引对这些比对做快速的扩展。比对原理可阅读文献原文：HISAT: a fast spliced aligner with low memory requirements.</p>
<h4 id="1">1.背景<a class="headerlink" href="#1" title="Permanent link">&para;</a></h4>
<p>自2008年起，RNAseq已经成为研究基因表达、转录本结构、长链非编码RNA确定以及融合转录本的重要手段。随着测序深度的加深和read读长的延长，给比对工作带来很多困难。目前的工具如Tophat2和GSNAP等在对单个转录组实验的比对中耗时几天，而新型的STAR虽然很快，但是会吃掉大量的内存空间，如基于人类基因组的比对需要消耗28G左右，而hisat2是4.3G，小鼠3.8G左右。</p>
<h4 id="2hisat">2.Hisat设计原则<a class="headerlink" href="#2hisat" title="Permanent link">&para;</a></h4>
<p><strong>优化了索引建立策略</strong></p>
<p>hisat应用了基于bowtie2的方法去处理很多低水平的用于构建和查询FM索引的操作。但是与其它比对器不同的是，该软件应用了两类不同的索引类型：代表全基因组的全局FM索引和大量的局部小索引，每个索引代表64000bp。以人类基因组为例，创建了48000个局部索引，每一个覆盖1024bp，最终可以覆盖这个3 billion 的碱基的基因组。这种存在交叉（overlap）的边界可以轻松的比对那些跨区域的read（可变剪切体）。尽管有很多索引，但是hisat会把他们使用合适方法压缩，最终只占4gb左右的内存。</p>
<p><strong>采用了新的比对策略</strong></p>
<p>RNA-seq产生的reads可能跨长度比较大的内含子，哺乳动物中甚至最长能达到1MB，同时外显子比较短，read也比较短，会有很多read（模拟数据中大概34%）跨两个外显子的情况。为了更好的比对，将跨外显子的reads分成了三类：1）长锚定read，至少有16bp在两个外显子的每一个上 2）中间锚定read，有8-15bp在一个外显子上 3）短锚定read，只有1-7bp在一个外显子上。所以总的reads可以被划分为五类：1）不跨外显子的read 2）长锚定read 3）中间锚定read 4）短锚定read 5）跨两个外显子以上的read。在模拟的数据中，有25%左右的read是长锚定read，这种read在大多数情况下可以被唯一的定位到人的基因组上。5%为中间锚定read，对于这类，很多依赖于全局索引的算法就很难执行下去（需要比对很多次），而hisat，可以先将read中的长片段实现唯一比对，之后再使用局部索引对剩下的小片段进行比对（局部索引可以实现快速检索）。4.2%为短锚定read，因为这些序列特别短，因此只能通过在hisat比对其它read时发现的剪切位点或者用户自己提供的剪切位点来辅助比对。最后还有3%的是跨多个外显子的read，比对策略在hisat的online method中有介绍，文章中没有详解。比对过程中，中间锚定read、短锚定read、跨多个外显子read的比对占总比对时长的30%-60%，而且比对错误率很高！<strong>【引用：1】</strong></p>
<hr />
<p>官方手册：<a href="https://daehwankimlab.github.io/hisat2/manual/">https://daehwankimlab.github.io/hisat2/manual/</a></p>
<h4 id="3hisat2-2">3.HiSat2进行比对的参数设置 【引用：2】<a class="headerlink" href="#3hisat2-2" title="Permanent link">&para;</a></h4>
<p><div class="highlight"><pre><span></span><code>HISAT2 version 2.1.0 by Daehwan Kim (infphilo@gmail.com, www.ccb.jhu.edu/people/infphilo)
Usage: 
  hisat2 [options]* -x &lt;ht2-idx&gt; {-1 &lt;m1&gt; -2 &lt;m2&gt; | -U &lt;r&gt;} [-S &lt;sam&gt;]

  &lt;ht2-idx&gt;  Index filename prefix (minus trailing .X.ht2).
             索引文件的前缀
  &lt;m1&gt;       Files with #1 mates, paired with files in &lt;m2&gt;.
             Could be gzip&#39;ed (extension: .gz) or bzip2&#39;ed (extension: .bz2).
             Paired-end测序的第一个文件，可以是压缩格式的（`.gz`或者`.bz2`）
  &lt;m2&gt;       Files with #2 mates, paired with files in &lt;m1&gt;.
             Could be gzip&#39;ed (extension: .gz) or bzip2&#39;ed (extension: .bz2).
             Paired-end测序的第二个文件
  &lt;r&gt;        Files with unpaired reads.
             Could be gzip&#39;ed (extension: .gz) or bzip2&#39;ed (extension: .bz2).
             single-end测序的文件
  &lt;sam&gt;      File for SAM output (default: stdout)
             输出比对的结果（`.sam`）

  &lt;m1&gt;, &lt;m2&gt;, &lt;r&gt; can be comma-separated lists (no whitespace) and can be
  specified many times.  E.g. &#39;-U file1.fq,file2.fq -U file3.fq&#39;.
  可以是多个序列文件用逗号分隔

Options (defaults in parentheses):

 Input:
  -q                 query input files are FASTQ .fq/.fastq (default)
                     输入检索序列为FASTQG格式
  --qseq             query input files are in Illumina&#39;s qseq format
                     输入检索序列是Illumina qseq格式
  -f                 query input files are (multi-)FASTA .fa/.mfa
                     输入文件是多个FASTA文件
  -r                 query input files are raw one-sequence-per-line
                     输入检索序列是每行一个原始序列
  -c                 &lt;m1&gt;, &lt;m2&gt;, &lt;r&gt; are sequences themselves, not files
                     直接输入序列，而不是文件
  -s/--skip &lt;int&gt;    skip the first &lt;int&gt; reads/pairs in the input (none)
                     忽略输入序列的前`&lt;int&gt;`个
  -u/--upto &lt;int&gt;    stop after first &lt;int&gt; reads/pairs (no limit)
                     只分析输入序列的前`&lt;int&gt;`个
  -5/--trim5 &lt;int&gt;   trim &lt;int&gt; bases from 5&#39;/left end of reads (0)
                     剪去reads长度为`&lt;int&gt;`的5&#39;端序列
  -3/--trim3 &lt;int&gt;   trim &lt;int&gt; bases from 3&#39;/right end of reads (0)
                     剪去reads长度为`&lt;int&gt;`的3&#39;端序列
  --phred33          qualities are Phred+33 (default)
                     序列质量打分为Phred+33
  --phred64          qualities are Phred+64
                     序列质量打分为Phred+64
  --int-quals        qualities encoded as space-delimited integers
                     序列质量打分为用空格分隔的整数

 Alignment:
  --n-ceil &lt;func&gt;    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)
                     允许在比对结果中出现的最多非ACTG的计算函数
  --ignore-quals     treat all quality values as 30 on Phred scale (off)
                     将所有质量分值设为30
  --nofw             do not align forward (original) version of read (off)
                     不比对前向的read
  --norc             do not align reverse-complement version of read (off)
                     不比对后向的read

 Spliced Alignment:
  --pen-cansplice &lt;int&gt;              penalty for a canonical splice site (0)
                                     对已知剪接位点的罚分
  --pen-noncansplice &lt;int&gt;           penalty for a non-canonical splice site (12)
                                     对新的剪接位点的罚分
  --pen-canintronlen &lt;func&gt;          penalty for long introns (G,-8,1) with canonical splice sites
                                     对已知剪接位点中长内含子的罚分
  --pen-noncanintronlen &lt;func&gt;       penalty for long introns (G,-8,1) with noncanonical splice sites
                                     对未知剪接位点中长内含子的罚分
  --min-intronlen &lt;int&gt;              minimum intron length (20)
                                     最小允许的内含子长度
  --max-intronlen &lt;int&gt;              maximum intron length (500000)
                                     最大允许的内含子长度
  --known-splicesite-infile &lt;path&gt;   provide a list of known splice sites
                                     提供一个已知的剪接位点列表
  --novel-splicesite-outfile &lt;path&gt;  report a list of splice sites
                                     报告新的剪接位点
  --novel-splicesite-infile &lt;path&gt;   provide a list of novel splice sites
                                     提供新的剪接位点
  --no-temp-splicesite               disable the use of splice sites found
                                     不使用发现的新剪接位点
  --no-spliced-alignment             disable spliced alignment
                                     不允许剪接比对
  --rna-strandness &lt;string&gt;          specify strand-specific information (unstranded)
                                     指定链特异性的信息
  --tmo                              reports only those alignments within known transcriptome
                                     报告只包含已知转录组信息的比对结果
  --dta                              reports alignments tailored for transcript assemblers
                                     结果适合用于stringtie进行转录本的拼接
  --dta-cufflinks                    reports alignments tailored specifically for cufflinks
                                     结果适合于cufflinks的应用
  --avoid-pseudogene                 tries to avoid aligning reads to pseudogenes (experimental option)
                                     避免与假基因的比对
  --no-templatelen-adjustment        disables template length adjustment for RNA-seq reads
                                     不允许对reads的模板长度进行调整

 Scoring:
  --mp &lt;int&gt;,&lt;int&gt;   max and min penalties for mismatch; lower qual = lower penalty &lt;6,2&gt;
                     不匹配的最大与最小罚分值，低质量对应低罚分
  --sp &lt;int&gt;,&lt;int&gt;   max and min penalties for soft-clipping; lower qual = lower penalty &lt;2,1&gt;
                     soft-clipping的最大与最小罚分
  --no-softclip      no soft-clipping
                     不考虑Soft-clipping
  --np &lt;int&gt;         penalty for non-A/C/G/Ts in read/ref (1)
                     对read或参考序列中出现的非ATCG的罚分
  --rdg &lt;int&gt;,&lt;int&gt;  read gap open, extend penalties (5,3)
                     对read的gap-open, gap-extension的罚分
  --rfg &lt;int&gt;,&lt;int&gt;  reference gap open, extend penalties (5,3)
                     对参考序列的gap-open,gap-extension的罚分
  --score-min &lt;func&gt; min acceptable alignment score w/r/t read length
                     (L,0.0,-0.2)
                     相对read长度的最小接受比对分值

 Reporting:
  -k &lt;int&gt; (default: 5) report up to &lt;int&gt; alns per read
                     对每个read只报告`&lt;int&gt;`个比对结果

 Paired-end:
  -I/--minins &lt;int&gt;  minimum fragment length (0), only valid with --no-spliced-alignment
                     paired-end的最小片段长度
  -X/--maxins &lt;int&gt;  maximum fragment length (500), only valid with --no-spliced-alignment
                     最大片段长度
  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)
                     `--fr==fw/rev`，`--rf==rev/fw`, `--ff==fw/fw`
  --no-mixed         suppress unpaired alignments for paired reads
                     不输出paired reads的不能匹配的比对结果
  --no-discordant    suppress discordant alignments for paired reads
                     不输出paired-reads的不统一的比对结果

 Output:
  -t/--time          print wall-clock time taken by search phases
                     输出搜索阶段所花费的wall-clock时间
  --un &lt;path&gt;           write unpaired reads that didn&#39;t align to &lt;path&gt;
                        输出没有与`&lt;path&gt;`比对的unpaired reads
  --al &lt;path&gt;           write unpaired reads that aligned at least once to &lt;path&gt;
                        输出至少与`&lt;path&gt;`有一次比对的unpaired reads
  --un-conc &lt;path&gt;      write pairs that didn&#39;t align concordantly to &lt;path&gt;
  --al-conc &lt;path&gt;      write pairs that aligned concordantly at least once to &lt;path&gt;
  (Note: for --un, --al, --un-conc, or --al-conc, add &#39;-gz&#39; to the option name, e.g.
  --un-gz &lt;path&gt;, to gzip compress output, or add &#39;-bz2&#39; to bzip2 compress output.)
                    将输出结果进行`gz`或者`bz2`压缩
  --summary-file     print alignment summary to this file.
                     将比对的总结输出到文件中
  --new-summary      print alignment summary in a new style, which is more machine-friendly.
                     将比对总结以新的形式输出
  --quiet            print nothing to stderr except serious errors
                     除了严重错误以外，不将任何结果输出到stderr
  --met-file &lt;path&gt;  send metrics to file at &lt;path&gt; (off)
                     将度量值i输出到文件
  --met-stderr       send metrics to stderr (off)
                     将度量值输出到stderr
  --met &lt;int&gt;        report internal counters &amp; metrics every &lt;int&gt; secs (1)
                     每隔`&lt;int&gt;`秒输出当前的内部计数器和度量值
  --no-head          supppress header lines, i.e. lines starting with @
                     不输出header行，也就是所有以`@`开始的行
  --no-sq            supppress @SQ header lines
                     不输出`@SQ`行
  --rg-id &lt;text&gt;     set read group id, reflected in @RG line and RG:Z: opt field
                     设置`@RG`
  --rg &lt;text&gt;        add &lt;text&gt; (&quot;lab:value&quot;) to @RG line of SAM header.
                     Note: @RG line only printed when --rg-id is set.
                     将`&lt;text&gt;`加到SAM文件的`@RG`行
  --omit-sec-seq     put &#39;*&#39; in SEQ and QUAL fields for secondary alignments.
                     将&quot;*&quot;置于SEQ和QUAL两列中，进行二次比对

 Performance:
  -o/--offrate &lt;int&gt; override offrate of index; must be &gt;= index&#39;s offrate
                     重置索引的`offrate`，该值必须大于原有的`offrate`
  -p/--threads &lt;int&gt; number of alignment threads to launch (1)
                     比对所用的线程数
  --reorder          force SAM output order to match order of input reads
                     强制要求SAM输出顺序比对与输入reads一致
  --mm               use memory-mapped I/O for index; many &#39;hisat2&#39;s can share
                     使用内存映射的I/O作为索引

 Other:
  --qc-filter        filter out reads that are bad according to QSEQ filter
                     根据QSEQ filter过滤输入的reads
  --seed &lt;int&gt;       seed for random number generator (0)
                     设置RNG seeds的值
  --non-deterministic seed rand. gen. arbitrarily instead of using read attributes
                     不根据reads的属性产生随机数
  --remove-chrname   remove &#39;chr&#39; from reference names in alignment
                     比对结果中的参考序列名称中去除`chr`
  --add-chrname      add &#39;chr&#39; to reference names in alignment
                     在比对结果的参考序列名称中加入`chr` 
  --version          print version information and quit
                     输出版本信息
  -h/--help          print this usage message
                     输出帮助信息
</code></pre></div>
hisat2可以自己建立index文件，如果没有现成的index的话，那就得自己去建立了，这个就比较麻烦而且耗内存和时间，下面是建立index的一些参数介绍。因为我们的index文件是直接下载的，所以我后面不需要建立index文件，关于这部分，后面有机会介绍。
<div class="highlight"><pre><span></span><code>HISAT2 version 2.1.0 by Daehwan Kim (infphilo@gmail.com, http://www.ccb.jhu.edu/people/infphilo)
Usage: hisat2-build [options]* &lt;reference_in&gt; &lt;ht2_index_base&gt;
    reference_in            comma-separated list of files with ref sequences
                            以逗号分隔的参考序列文件的列表
    hisat2_index_base       write ht2 data to files with this dir/basename
                            输出的索引文件的`basename`
Options:
    -c                      reference sequences given on cmd line (as
                            &lt;reference_in&gt;)
                            输入的参考序列
    --large-index           force generated index to be &#39;large&#39;, even if ref
                            has fewer than 4 billion nucleotides
                            强制要求产生的索引为`large`
    -a/--noauto             disable automatic -p/--bmax/--dcv memory-fitting
                            关闭-p/--bmax/--dcv等自动内存匹配
    -p                      number of threads
                            线程数
    --bmax &lt;int&gt;            max bucket sz for blockwise suffix-array builder
                            逐块后缀数组建立采用的最大bucket size
    --bmaxdivn &lt;int&gt;        max bucket sz as divisor of ref len (default: 4)
    --dcv &lt;int&gt;             diff-cover period for blockwise (default: 1024)
    --nodc                  disable diff-cover (algorithm becomes quadratic)
    -r/--noref              don&#39;t build .3/.4.ht2 (packed reference) portion
    -3/--justref            just build .3/.4.ht2 (packed reference) portion
    -o/--offrate &lt;int&gt;      SA is sampled every 2^offRate BWT chars (default: 5)
    -t/--ftabchars &lt;int&gt;    # of chars consumed in initial lookup (default: 10)
    --localoffrate &lt;int&gt;    SA (local) is sampled every 2^offRate BWT chars (default: 3)
    --localftabchars &lt;int&gt;  # of chars consumed in initial lookup in a local index (default: 6)
    --snp &lt;path&gt;            SNP file name
    --haplotype &lt;path&gt;      haplotype file name
    --ss &lt;path&gt;             Splice site file name
    --exon &lt;path&gt;           Exon file name
    --seed &lt;int&gt;            seed for random number generator
    -q/--quiet              verbose output (for debugging)
    -h/--help               print detailed description of tool and its options
    --usage                 print this usage message
    --version               print version information and quit
</code></pre></div></p>
<h3 id="hisat2_2">（三）使用Hisat2进行序列比对<a class="headerlink" href="#hisat2_2" title="Permanent link">&para;</a></h3>
<p>创建输出数据的文件夹
<div class="highlight"><pre><span></span><code>mkdir cleandata/hisat2_mm10data
</code></pre></div>
因为比对这一过程很耗内存，所以样本多话，计算机内存不够大，需要分批比对，我就先介绍一个样本比对的命令：
<div class="highlight"><pre><span></span><code>hisat2 --dta -t -x /data/RNAseq/mm10/genome -1 /data/RNAseq/cleandata/trim_galoredata/CK-4_1_val_1.fq.gz -2 /data/RNAseq/cleandata/trim_galoredata/CK-4_2_val_2.fq.gz -S cleandata/hisat2_mm10data/CK4.sam 
</code></pre></div>
-1和-2分别表示双端测序的1个文件，后面跟的是文件路径，一定要注意 /data/RNAseq/mm10/genome文件的目录，genome这个不是文件夹，是index文件的前缀，我的mm10文件下并没有这个文件，如果不加genome就会发生如下报错：
<div class="highlight"><pre><span></span><code>## Could not locate a HISAT2 index corresponding to basename &quot;/data/RNAseq/mm10/genome/&quot; ##
</code></pre></div>
另外一定要加上 --dta，后续用Stringtie处理会更容易一些，这是StringTie使用说明里面的一句话：【NOTE: be sure to run HISAT2 with the --dta
option for alignment, or your results will suffer.】后续再介绍。-S后面cleandata/hisat2_mm10data/CK4.sam 是输出文件路径，hisat2_mm10data这个文件夹如果没有建立，需要事先建立。
命令没有问题的话，出现下面提示表示程序正在执行，就去干其他的，等执行结束再往下：</p>
<p><div class="highlight"><pre><span></span><code>Time loading forward index: 00:02:51
Time loading reference: 00:00:16
</code></pre></div>
<img alt="" src="024.png" />
我这里，一个样本43分钟。
计算机内存足够大的话，我们可以像前文【<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273401&amp;idx=1&amp;sn=ff9e3478309d805950c40d0ebae74296&amp;chksm=841edcc4b36955d2a5bf8c85429114b5b83ec4e5afe869378b42ec4f659d4953d5f6c8e9e379&amp;scene=21#wechat_redirect"><strong>转录组分析 | 使用trim-galore去除低质量的reads和adaptor</strong></a>】一样，通过一个脚本一次执行。
创建脚本文件：hisat2_mm10_batch.sh</p>
<p><div class="highlight"><pre><span></span><code>vim hisat2_mm10_batch.sh
</code></pre></div>
输入下面内容：
<div class="highlight"><pre><span></span><code>#!/bin/bash
# This is for hisat2 batch aligne
for i in CK-7 CK-8 HGJ-10 HGJ-6 HGJ-9
do
hisat2 --dta -t -p 8 -x /data/RNAseq/mm10/genome \
-1 /data/RNAseq/cleandata/trim_galoredata/&quot;$i&quot;_1_val_1.fq.gz \
-2 /data/RNAseq/cleandata/trim_galoredata/&quot;$i&quot;_2_val_2.fq.gz \
-S /data/RNAseq/cleandata/hisat2_mm10data/&quot;$i&quot;.sam
</code></pre></div>
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1612932193899-887f402a-43f0-44e5-87d4-5d58ce863101.png#align=left&amp;display=inline&amp;height=224&amp;margin=%5Bobject%20Object%5D&amp;originHeight=224&amp;originWidth=634&amp;size=0&amp;status=done&amp;style=none&amp;width=634" /></p>
<p>我这里由于CK-4这个样本前面单独处理了，批量处理就不需要处理。
保存脚本，然后执行脚本。</p>
<p><div class="highlight"><pre><span></span><code>bash hisat2_mm10_batch.sh
</code></pre></div>
脚本运行时间和计算机配制还有数据量的大小有关。运行结束后我们查看结果：
<div class="highlight"><pre><span></span><code>ll -h ./cleandata/hisat2_mm10data/
</code></pre></div>
原本22G的数据，运行结束后155G，所以自己用本地计算机跑的话，数据多注意一下内存。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1612932194033-bb32bdd1-4b55-4208-99fb-75245895e485.png#align=left&amp;display=inline&amp;height=184&amp;margin=%5Bobject%20Object%5D&amp;originHeight=184&amp;originWidth=807&amp;size=0&amp;status=done&amp;style=none&amp;width=807" /></p>
<h3 id="_2"><strong>（四）结果解读</strong><a class="headerlink" href="#_2" title="Permanent link">&para;</a></h3>
<p>我们比对后得到的是sam文件，关于sam文件是一个什么样的文件，参考文章：<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651272899&amp;idx=1&amp;sn=6779b2bd21f3b607a08227d31c7212c6&amp;chksm=841ed2beb3695ba8bee23563c28caa005447b2298785719964732b16cafe3a15d7d4937b95c1&amp;scene=21#wechat_redirect"><strong>生信中常见的数据文件格式</strong></a>。</p>
<hr />
<p>为了快速查看本公众号文章，可阅读文章：<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273306&amp;idx=1&amp;sn=d92056ee1fde1ff12b1506e02f8eb56f&amp;chksm=841edc27b36955310bac7e1b4e5e4ab7e26e669c192926dbee3efc4f6d185cf3509a8fc87696&amp;scene=21#wechat_redirect"><strong>公众号文章目录</strong></a></p>
<hr />
<p><strong>致谢：本文还参考了以下内容：</strong></p>
<p>【1】<a href="https://www.jianshu.com/p/ce3f4afb9b60">https://www.jianshu.com/p/ce3f4afb9b60</a>。</p>
<p>【2】<a href="https://github.com/ricket-sjtu/bioinformatics/wiki/HiSat2%25E5%25BA%2594%25E7%2594%25A8%25E4%25BB%258B%25E7%25BB%258D">https://github.com/ricket-sjtu/bioinformatics/wiki/HiSat2%E5%BA%94%E7%94%A8%E4%BB%8B%E7%BB%8D</a></p>
<p>对原作者表示感谢！</p>
<h2 id="samtoolssambam">四.使用SAMtools将SAM文件转换为BAM文件、排序、建立索引<a class="headerlink" href="#samtoolssambam" title="Permanent link">&para;</a></h2>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273374&amp;idx=1&amp;sn=dfac9e82a09fbb18cd8b2c8179afcad8&amp;chksm=841edce3b36955f5ef17ae96b03c11b9d0052cd3490f1d6b91e3600a5d2eafa77ad79d4a40ff&amp;scene=21#wechat_redirect"><strong>转录组分析 | fastqc进行质控与结果解读</strong></a></p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273401&amp;idx=1&amp;sn=ff9e3478309d805950c40d0ebae74296&amp;chksm=841edcc4b36955d2a5bf8c85429114b5b83ec4e5afe869378b42ec4f659d4953d5f6c8e9e379&amp;scene=21#wechat_redirect"><strong>转录组分析 | 使用trim-galore去除低质量的reads和adaptor</strong></a></p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273450&amp;idx=1&amp;sn=efab631b255e12cf5ffd259dd53fb2b3&amp;chksm=841edc97b3695581de924856b0e6901c67b87648f85bd491729eaa2ee0592271d84042b9eb76&amp;scene=21#wechat_redirect"><strong>转录组分析 | 使用Hisat2进行序列比对</strong></a></p>
<p>我们前面经过质控，比对得到sam文件，接下来为什么要进行格式转换？为了让计算机好处理。SAM（sequence Alignment/mapping)数据格式是目前高通量测序中存放比对数据的标准格式。bam是sam的二进制格式，减少了sam文件的储存量。这在我之前的文章【<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651272899&amp;idx=1&amp;sn=6779b2bd21f3b607a08227d31c7212c6&amp;chksm=841ed2beb3695ba8bee23563c28caa005447b2298785719964732b16cafe3a15d7d4937b95c1&amp;scene=21#wechat_redirect"><strong>生信中常见的数据文件格式</strong></a>】中也有介绍。
接下来，我们要做的事情就是使用SAMtools将SAM文件转换为BAM文件、排序、建立索引。</p>
<h3 id="samtools">（一）SAMtools介绍<a class="headerlink" href="#samtools" title="Permanent link">&para;</a></h3>
<p>SAMtools是一个用于操作sam和bam文件的工具合集。能够实现二进制查看、格式转换、排序及合并等功能，结合sam格式中的flag、tag等信息，还可以完成比对结果的统计汇总。同时利用linux中的grep、awk等操作命令，还可以大大扩展samtools的使用范围与功能。包含有许多命令。我这里主要介绍几个，重点是samtools view。
samtools更多信息：<strong><a href="http://www.htslib.org/">http://www.htslib.org/</a></strong></p>
<h4 id="1samtools-view"><strong>1.samtools view</strong><a class="headerlink" href="#1samtools-view" title="Permanent link">&para;</a></h4>
<p>samtools view主要用来转换SAM/BAM/CRAM文件。将sam文件与bam文件互换；然后对bam文件进行各种操作，比如数据的排序(sort)和提取(这些操作 是对bam文件进行的，因而当输入为sam文件的时候，不能进行该操作)；最后将排序或提取得到的数据输出为bam或sam（默认的）格式。如果没有指定选项或区域，则将指定的输入对齐文件(SAM、BAM或CRAM格式)中的所有对齐打印到SAM格式的标准输出(没有标头)。可以在输入文件名后指定一个或多个空格分隔的区域规范，以将输出限制为仅覆盖指定区域的那些对齐。使用区域规范需要一个协调排序和索引的输入文件(BAM或CRAM格式)。samtools view的参数很多，-b、-C、-1、-u、-h、-H和-C选项将更改缺省的无header SAM的输出格式，而-o和-u选项将设置输出文件名。-t和-T选项提供了额外的参考数据。当SAM输入不包含@SQ headers时，这两个选项中的一个是必需的，当编写CRAM输出时，-T选项是必需的。-L、-M、-r、-R、-d、-D、-s、-q、-L、-M、-f、-F和-G选项过滤将包含在输出中的对齐，只筛选那些匹配特定条件的对齐。-x和-B选项修改包含在每次对齐中的数据。如果数据文件夹不包含任何索引文件，可以使用-X选项允许用户指定定制的索引文件位置。</p>
<p><div class="highlight"><pre><span></span><code>samtools view [options] &lt;in.bam&gt;|&lt;in.sam&gt;|&lt;in.cram&gt; [region ...]
</code></pre></div>
参数介绍：
<div class="highlight"><pre><span></span><code># samtools view 使用说明：
 -b output BAM
  默认下输出是 SAM 格式文件，该参数设置输出 BAM 格式
 -h print header for the SAM output
  默认下输出的 sam 格式文件不带 header，该参数设定输出sam文件时带 header 信息
 -H print header only (no alignments)
  仅仅输出文件的头
 -S input is SAM
  默认下输入是 BAM 文件，若是输入是 SAM 文件，则最好加该参数，否则有时候会报错。
 -u uncompressed BAM output (force -b)
  该参数的使用需要有-b参数，能节约时间，但是需要更多磁盘空间。
 -c Instead of printing the alignments, only count them and print the 
  total number. All filter options, such as ‘-f’, ‘-F’ and ‘-q’ ,   are taken into account.
  过滤功统计功能
 -c print only the count of matching records
 -L FILE  output alignments overlapping the input BED FILE [null]
 -t FILE  list of reference names and lengths (force -S) [null]
  使用一个list文件来作为header的输入
 -T FILE  reference sequence file (force -S) [null]
  使用序列fasta文件作为header的输入
 -o FILE  output file name [stdout]
 -F INT   filtering flag, 0 for unset [0] 
  Skip alignments with bits present in INT [0]
  数字4代表该序列没有比对到参考序列上
  数字8代表该序列的mate序列没有比对到参考序列上
  过滤功能。如F12过滤只有双端map的
 -q INT   minimum mapping quality [0]
    比对的最低质量值，一般认为20就为unique比对了，可以结合上述-bF参数使用使用提取特定的比对结果
</code></pre></div>
其他更多参数介绍，阅读：<a href="http://www.htslib.org/doc/samtools-view.html">http://www.htslib.org/doc/samtools-view.html</a></p>
<h4 id="2samtools-sort"><strong>2.samtools sort</strong><a class="headerlink" href="#2samtools-sort" title="Permanent link">&para;</a></h4>
<p>samtools sort用来对SAM/BAM/CRAM文件进行排序，按最左坐标排序，或使用-n时按读取名称排序。默认情况下，排序后的输出被写到标准输出，或者在使用-o时写到指定的文件(out.bam)。此命令还将创建临时文件tmpprefixv .%d。当整个对齐数据无法装入内存时(通过-m选项控制)，根据需要使用bam。</p>
<p><div class="highlight"><pre><span></span><code>samtools sort [-l level] [-m maxMem] [-o out.bam] [-O format] [-n] [-T tmpprefix] [-@ threads] [in.sam|in.bam|in.cram]
</code></pre></div>
重要参数：
<div class="highlight"><pre><span></span><code>-l INT 设置输出文件压缩等级。0-9，0是不压缩，9是压缩等级最高。不设置此参数时，使用默认压缩等级；
-m INT 设置每个线程运行时的内存大小，可以使用K，M和G表示内存大小。
-n 设置按照read名称进行排序；
-o FILE 设置最终排序后的输出文件名；
-O FORMAT 设置最终输出的文件格式，可以是bam，sam或者cram，默认为bam；
-T PREFIX 设置临时文件的前缀；
-@ INT 设置排序和压缩是的线程数量，默认是单线程。
</code></pre></div>
<strong>3.samtools index</strong></p>
<p>必须对bam文件进行默认情况下的排序后，才能进行index。否则会报错。建立索引后将产生后缀为.bai的文件，用于快速的随机处理。很多情况下需要有bai文件的存在，特别是显示序列比对情况下。比如samtool的tview命令就需要；gbrowse2显示reads的比对图形的时候也需要。IGV显示比对情况也需要。</p>
<p><div class="highlight"><pre><span></span><code>samtools index [-bc] [-m INT] aln.sam|aln.bam|aln.cram [out.index]
</code></pre></div>
参数：
<div class="highlight"><pre><span></span><code>-b 创建一个BAI索引。当不使用格式选项时，这是当前的默认设置。
-c 创建CSI索引。默认情况下，索引的最小间隔大小为2^14，与BAI格式使用的固定值相同。
-m INT 创建CSI索引，最小间隔大小为2^INT。
-@, --threads INT
    除了主线程[0]之外，要使用的输入/输出压缩线程数。
</code></pre></div>
samtools sort命令按默认染色体位置排序，顺利建立Index，如果前面排序有出入，可能不能正确建立索引。
参考：<a href="http://www.htslib.org/doc/samtools-index.html">http://www.htslib.org/doc/samtools-index.html</a></p>
<h4 id="4samtools-flagstat"><strong>4.samtools flagstat</strong><a class="headerlink" href="#4samtools-flagstat" title="Permanent link">&para;</a></h4>
<p>samtools flagstat用于给出BAM文件的比对结果。</p>
<p><div class="highlight"><pre><span></span><code>samtools flagstat in.sam|in.bam|in.cram
</code></pre></div>
参数：
<div class="highlight"><pre><span></span><code>-@ INT 设置读取文件时要使用的额外线程数。
-O FORMAT 设置输出格式。FORMAT可以设置为&#39;default&#39;， &#39;json&#39;或&#39;tsv&#39;来选择默认的，json或标签分隔值输出格式。如果不使用此选项，将选择默认格式。
</code></pre></div>
参考：<a href="http://www.htslib.org/doc/samtools-flagstat.html">http://www.htslib.org/doc/samtools-flagstat.html</a></p>
<hr />
<p>关于SAMtools工具的用法还很多，我就介绍上面几个，其他更多用法，阅读：<a href="http://www.htslib.org/doc/samtools.html">http://www.htslib.org/doc/samtools.html</a>
参考：<a href="https://blog.csdn.net/sinat_38163598/article/details/72910115">https://blog.csdn.net/sinat_38163598/article/details/72910115</a></p>
<h3 id="samtools_1">（二）SAMtools实战<a class="headerlink" href="#samtools_1" title="Permanent link">&para;</a></h3>
<h4 id="1samtools-view_1"><strong>1.格式转换(samtools view)</strong><a class="headerlink" href="#1samtools-view_1" title="Permanent link">&para;</a></h4>
<p>我们先查看一下上一篇文章【<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273450&amp;idx=1&amp;sn=efab631b255e12cf5ffd259dd53fb2b3&amp;chksm=841edc97b3695581de924856b0e6901c67b87648f85bd491729eaa2ee0592271d84042b9eb76&amp;scene=21#wechat_redirect"><strong>转录组分析 | 使用Hisat2进行序列比对</strong></a>】中产生的sam文件。</p>
<p><div class="highlight"><pre><span></span><code>ll -h ./cleandata/hisat2_mm10data/
</code></pre></div>
<img alt="" src="026.png" /></p>
<p>先建立一个输出数据的文件夹。</p>
<p><div class="highlight"><pre><span></span><code>mkdir ./cleandata/samtools_bam
</code></pre></div>
同样的，我们可以单个转换，也可以写一个脚本批量转换，首先，我先介绍单个转换。
<div class="highlight"><pre><span></span><code>samtools view -S ./cleandata/hisat2_mm10data/CK4.sam -b &gt; ./cleandata/samtools_bam/CK4.bam
</code></pre></div>
这个过程的时间也是挺长的。
批量转换，创建脚本文件
<div class="highlight"><pre><span></span><code>vim samtools_view_batch.sh
</code></pre></div>
输入下面内容：
<div class="highlight"><pre><span></span><code>for i in CK-7 CK-8 HGJ-10 HGJ-6 HGJ-9
do
  samtools view -S ./cleandata/hisat2_mm10data/${i}.sam -b -o ./cleandata/samtools_bam/${i}.bam
done
</code></pre></div>
<img alt="" src="027.png" />
保存，然后执行脚本。</p>
<p><div class="highlight"><pre><span></span><code>bash samtools_view_batch.sh
</code></pre></div>
运行结束后我么你查看一下输入的数据：
<div class="highlight"><pre><span></span><code>ll -h ./cleandata/samtools_bam/
</code></pre></div>
<img alt="" src="028.png" /></p>
<p>我们可以看到，转换后的数据只有32G，原来可是155G。相差5倍左右。</p>
<h4 id="2-samtools-sort"><strong>2 排序(samtools sort)</strong><a class="headerlink" href="#2-samtools-sort" title="Permanent link">&para;</a></h4>
<p>单个排序:</p>
<p><div class="highlight"><pre><span></span><code>samtools sort -l 4 -o ./cleandata/samtools_bam/CK-4_sort.bam ./cleandata/samtools_bam/CK4.bam
</code></pre></div>
批量排序：创建脚本文件
<div class="highlight"><pre><span></span><code>vim samtools_sort.sh
</code></pre></div>
输入下面内容：
<div class="highlight"><pre><span></span><code>#!/bin/bash
# This is for samtools_sort
for i in CK-7 CK-8 HGJ-10 HGJ-6 HGJ-9
do
samtools sort -l 4 -o ./cleandata/samtools_bam/${i}_sort.bam ./cleandata/samtools_bam/${i}.bam
done
</code></pre></div>
<img alt="" src="029.png" />
保存并执行脚本。</p>
<p><div class="highlight"><pre><span></span><code>bash samtools_sort.sh
</code></pre></div>
执行结束后，我们查看一下文件：
<div class="highlight"><pre><span></span><code>ll -h cleandata/samtools_bam/*sort.bam
</code></pre></div>
<img alt="" src="030.png" />
sort后文件几乎小了2倍。</p>
<h4 id="3-samtools-index"><strong>3. 建立索引(samtools index)</strong><a class="headerlink" href="#3-samtools-index" title="Permanent link">&para;</a></h4>
<p>samtools sort命令时，按默认染色体位置排序，顺利建立Index，如果前面排序有出入，可能不能正确建立索引。
这里我就一次建立索引了。</p>
<div class="highlight"><pre><span></span><code>for i in CK-4 CK-7 CK-8 HGJ-10 HGJ-6 HGJ-9;do samtools index ./cleandata/samtools_bam/${i}_sort.bam ./cleandata/samtools_bam/${i}_sort.bam.bai;done
</code></pre></div>
<h4 id="4readssamtools-flagstat"><strong>4.查看reads比对情况(samtools flagstat)</strong><a class="headerlink" href="#4readssamtools-flagstat" title="Permanent link">&para;</a></h4>
<p><div class="highlight"><pre><span></span><code>samtools flagstat cleandata/samtools_bam/CK-4_sort.bam
</code></pre></div>
<img alt="" src="031.png" />
<div class="highlight"><pre><span></span><code>(RNAseq1) [root@localhost RNAseq]# samtools flagstat cleandata/samtools_bam/CK-4_sort.bam
51977802 + 0 in total (QC-passed reads + QC-failed reads)
#总共的reads数
4063860 + 0 secondary
0 + 0 supplementary
0 + 0 duplicates
50969572 + 0 mapped (98.06% : N/A)
#总体上reads的匹配率
47913942 + 0 paired in sequencing
#有多少reads是属于paired reads
23956971 + 0 read1 
#reads1中的reads数
23956971 + 0 read2
#reads2中的reads数
45980036 + 0 properly paired (95.96% : N/A)
#完美匹配的reads数：比对到同一条参考序列，并且两条reads之间的距离符合设置的阈值
46308722 + 0 with itself and mate mapped
#paired reads中两条都比对到参考序列上的reads数
596990 + 0 singletons (1.25% : N/A)
#单独一条匹配到参考序列上的reads数，和上一个相加，则是总的匹配上的reads数。
232008 + 0 with mate mapped to a different chr
#paired reads中两条分别比对到两条不同的参考序列的reads数
210169 + 0 with mate mapped to a different chr (mapQ&gt;=5)
#同上一个，只是其中比对质量&gt;=5的reads的数量
</code></pre></div>
参考：</p>
<p>【1】<a href="https://blog.csdn.net/sinat_38163598/article/details/72910115">https://blog.csdn.net/sinat_38163598/article/details/72910115</a></p>
<p>【2】<a href="http://www.htslib.org/doc/samtools.html">http://www.htslib.org/doc/samtools.html</a></p>
<p>这一部分就介绍到这里，本公众号文章目录可查看文章：<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273306&amp;idx=1&amp;sn=d92056ee1fde1ff12b1506e02f8eb56f&amp;chksm=841edc27b36955310bac7e1b4e5e4ab7e26e669c192926dbee3efc4f6d185cf3509a8fc87696&amp;scene=21#wechat_redirect"><strong>公众号文章目录</strong></a>，快速浏览相应文章。</p>
<h2 id="rseqcbam">五.使用RSeQC软件对生成的BAM文件进行质控<a class="headerlink" href="#rseqcbam" title="Permanent link">&para;</a></h2>
<p>接前面文章：</p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273374&amp;idx=1&amp;sn=dfac9e82a09fbb18cd8b2c8179afcad8&amp;chksm=841edce3b36955f5ef17ae96b03c11b9d0052cd3490f1d6b91e3600a5d2eafa77ad79d4a40ff&amp;scene=21#wechat_redirect">转录组分析 | fastqc进行质控与结果解读</a></p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273401&amp;idx=1&amp;sn=ff9e3478309d805950c40d0ebae74296&amp;chksm=841edcc4b36955d2a5bf8c85429114b5b83ec4e5afe869378b42ec4f659d4953d5f6c8e9e379&amp;scene=21#wechat_redirect">转录组分析 | 使用trim-galore去除低质量的reads和adaptor</a></p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273450&amp;idx=1&amp;sn=efab631b255e12cf5ffd259dd53fb2b3&amp;chksm=841edc97b3695581de924856b0e6901c67b87648f85bd491729eaa2ee0592271d84042b9eb76&amp;scene=21#wechat_redirect">转录组分析 | 使用Hisat2进行序列比对</a></p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273488&amp;idx=1&amp;sn=f5ed53f20164b19720d31f0a2cfcd268&amp;chksm=841edd6db369547bc04ae15a2a8a7661fefa1193cabbe05d7e2f9e2fa986211603537cf8fbd0&amp;scene=21#wechat_redirect">转录组分析 | 使用SAMtools将SAM文件转换为BAM文件、排序、建立索引</a></p>
<p>本文接着介绍：使用RSeQC软件对生成的BAM文件进行质控。也就是看看前面数据的质量怎么样。</p>
<p>首先介绍一下RSeQC这个软件。</p>
<h3 id="rseqc">（一）RSeQC介绍<a class="headerlink" href="#rseqc" title="Permanent link">&para;</a></h3>
<p>RSeQC是发表于2012年的一个RNA-Seq质控工具，属于python包。它提供了一系列有用的小工具能够评估高通量测序尤其是RNA-seq数据，比如一些基本模块，检查序列质量, 核酸组分偏性, PCR偏性, GC含量偏性,还有RNA-seq特异性模块: 评估测序饱和度， 映射读数分布， 覆盖均匀性， 链特异性， 转录水平RNA完整性等。该软件的使用命令非常多，但很多功能并不是用来诊断转录组测序的，所以不在我们的考虑范围内。你可以参考官方教程文档：
<a href="https://pythonhosted.org/RSeQC/#bam2fq-py">https://pythonhosted.org/RSeQC/#bam2fq-py</a>
RSeQC接受4种文件格式:</p>
<ul>
<li>
<p>BED 格式: Tab 分割, 12列的表示基因模型的纯文本文件</p>
</li>
<li>
<p>SAM 或BAM 格式: 用来存储reads 比对结果信息.</p>
</li>
<li>
<p>染色体大小文件: 只有两列的纯文本文</p>
</li>
<li>
<p>Fasta文件的参考基因组</p>
</li>
</ul>
<h3 id="bam_statpy">（二）使用bam_stat.py命令查看比对的总体情况<a class="headerlink" href="#bam_statpy" title="Permanent link">&para;</a></h3>
<p>用 bam_stat.py来统计总比对记录, PCR重复数, Non Primary Hits表示多匹配位点, 不匹配的reads数, 比对到+链的reads, 比对到-链的reads, 有剪切位点的reads等。
<img alt="" src="032.png" /></p>
<p>我们查看前面产生的文件CK-4_sort.bam【文章:<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273488&amp;idx=1&amp;sn=f5ed53f20164b19720d31f0a2cfcd268&amp;chksm=841edd6db369547bc04ae15a2a8a7661fefa1193cabbe05d7e2f9e2fa986211603537cf8fbd0&amp;scene=21#wechat_redirect">转录组分析 | 使用SAMtools将SAM文件转换为BAM文件、排序、建立索引</a>】。</p>
<p><div class="highlight"><pre><span></span><code>bam_stat.py -i cleandata/samtools_bam/CK-4_sort.bam
</code></pre></div>
<img alt="" src="033.png" /></p>
<h3 id="read_distributionpy"><strong>（三）使用read_distribution.py命令查看基因组覆盖率</strong><a class="headerlink" href="#read_distributionpy" title="Permanent link">&para;</a></h3>
<p>该命令需要输入两个文件， -i为BAM或SAM文件；-r为参考的bed文件。
<img alt="" src="034.png" /></p>
<p>bed文件参考文章：<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651272899&amp;idx=1&amp;sn=6779b2bd21f3b607a08227d31c7212c6&amp;chksm=841ed2beb3695ba8bee23563c28caa005447b2298785719964732b16cafe3a15d7d4937b95c1&amp;scene=21#wechat_redirect">生信中常见的数据文件格式</a>。</p>
<p>bed文件下载地址RSeQC官网：<a href="https://sourceforge.net/projects/rseqc/files/BED/">https://sourceforge.net/projects/rseqc/files/BED/</a>，</p>
<p>我这里下载的是小鼠的：<a href="https://sourceforge.net/projects/rseqc/files/BED/Mouse_Mus_musculus/">https://sourceforge.net/projects/rseqc/files/BED/Mouse_Mus_musculus/</a>
<img alt="" src="035.png" />
下载后解压上传到Linux服务器中，我的bed文件在 /data/mouse_RefSeq/  这个目录下。</p>
<p><div class="highlight"><pre><span></span><code>ll -h /data/mouse_RefSeq/
</code></pre></div>
<img alt="" src="036.png" />
<div class="highlight"><pre><span></span><code>read_distribution.py -i cleandata/samtools_bam/CK-4_sort.bam -r /data/mouse_RefSeq/mm10_RefSeq.bed
</code></pre></div>
<img alt="" src="037.png" /></p>
<p>组学分析流程的每一个步骤都很重要，实验设计，质控，read比对，表达定量，可视化，差异表达，识别可变剪切，功能注释，融合基因检测，eQTL定位等都需要考虑。我们后面介绍：使用Stringtie对数据进行下游处理。
为了快速查看本公众号文章，可阅读文章：<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273306&amp;idx=1&amp;sn=d92056ee1fde1ff12b1506e02f8eb56f&amp;chksm=841edc27b36955310bac7e1b4e5e4ab7e26e669c192926dbee3efc4f6d185cf3509a8fc87696&amp;scene=21#wechat_redirect">公众号文章目录</a></p>
<hr />
<p>参考：<a href="https://mp.weixin.qq.com/s/KpQwwDm_aIvIbxXHiaNrLg">https://mp.weixin.qq.com/s/KpQwwDm_aIvIbxXHiaNrLg</a></p>
<h2 id="stringtie">六.使用Stringtie对数据进行下游处理<a class="headerlink" href="#stringtie" title="Permanent link">&para;</a></h2>
<p>接前文：</p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273374&amp;idx=1&amp;sn=dfac9e82a09fbb18cd8b2c8179afcad8&amp;chksm=841edce3b36955f5ef17ae96b03c11b9d0052cd3490f1d6b91e3600a5d2eafa77ad79d4a40ff&amp;scene=21#wechat_redirect">转录组分析 | fastqc进行质控与结果解读</a></p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273401&amp;idx=1&amp;sn=ff9e3478309d805950c40d0ebae74296&amp;chksm=841edcc4b36955d2a5bf8c85429114b5b83ec4e5afe869378b42ec4f659d4953d5f6c8e9e379&amp;scene=21#wechat_redirect">转录组分析 | 使用trim-galore去除低质量的reads和adaptor</a></p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273450&amp;idx=1&amp;sn=efab631b255e12cf5ffd259dd53fb2b3&amp;chksm=841edc97b3695581de924856b0e6901c67b87648f85bd491729eaa2ee0592271d84042b9eb76&amp;scene=21#wechat_redirect">转录组分析 | 使用Hisat2进行序列比对</a></p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273488&amp;idx=1&amp;sn=f5ed53f20164b19720d31f0a2cfcd268&amp;chksm=841edd6db369547bc04ae15a2a8a7661fefa1193cabbe05d7e2f9e2fa986211603537cf8fbd0&amp;scene=21#wechat_redirect">转录组分析 | 使用SAMtools将SAM文件转换为BAM文件、排序、建立索引</a></p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273531&amp;idx=1&amp;sn=8a9b12fe030f6c0f8bb1d6c8abbf622d&amp;chksm=841edd46b36954501a57801f8149301d1bbb7b2d62a6f0d73b8a4795071e2aace94e02c2f392&amp;scene=21#wechat_redirect">转录组分析 | 使用RSeQC软件对生成的BAM文件进行质控</a></p>
<p>我们接下来使用Stringtie对数据进行下游处理。</p>
<h3 id="stringtie_1">（一）StringTie介绍<a class="headerlink" href="#stringtie_1" title="Permanent link">&para;</a></h3>
<p>StringTie 是用于 RNA-seq 的转录本组装和定量软件，StringTie 可以看做是cufflinks软件的升级版本，其功能和Cufflinks是一样的，包括下面两个主要功能：转录本组装和定量；相比Cuffinks, 其运行速度更快。</p>
<p>该软件的官网：<a href="https://ccb.jhu.edu/software/stringtie/index.shtml">https://ccb.jhu.edu/software/stringtie/index.shtml</a>。</p>
<p>1、Stringtie通过使用genome指导的组装的方法与从头组装的概念结合的新方法来改善转录组组装。</p>
<p>2、Stringtie的输入不仅可以是经过比对的结果，也可以是Stringtie通过从头组装read所得到的contig，当这两种输入都用到的时候，我们下面称之为stringtie+SR。</p>
<p>3、对于很多使用参考基因组辅助组装的方法，组装的的策略都是先对read进行cluter，然后建立一个graph model来推测每个基因所有可能的isoform，最终通过不同的graph的解析方法得到对转录本的组装结果。</p>
<p>4、有名的cufflinks用的是overlap graph，该模型中nodes代表fragment，如果两个fragment存在overlap并存在兼容的剪切模式，则对应的node连接起来。其解析方法为一种保守的算法，可以产生能够解释所有read的最少的转录本，尽管这种方法很吸引人，但是它没有考虑到转录本的丰度并且对于某些isoform来说该方法没有办法组装！</p>
<p>5、stringtie采用了组装转录本和估计表达量同步进行的方法，这不同于cufflinks的先组装后定量的策略。</p>
<p>6、首先，stringtie将read聚成cluster，然后采用了splice graph，其中node代表外显子或外显子的一部分，path将graph中可能 的剪切现象都连起来，最终对每个转录本通过创建一个网络流的方法，利用最大流算法（maximum flow algorithm）估计转录本的表达量。</p>
<p>7、最大流的问题是最优理论中的经典问题，但是目前还没有应用到转录本定量中。</p>
<p>8、与其它组装软件相比，stringtie具有很高的准确性和新型isoform的发现能力，其优势在于使用了网络流算法，同时stringtie也支持将read从头组装成更长的片段，这进一步提高了其组装的正确性。</p>
<p>9、其另一个优势在于它的最优化策略，它平衡了每次组装中每条转录本的覆盖度，这样可以对组装算法产生一定的限制，因为在组装基因组时，覆盖度是很重要的一个参数因为它需要被用来限制算法，否则组装器可能将重复的片段错误地堆叠到一起，相似地转录组装也是如此，在isoform中的每一个外显子需要有相似的覆盖度，如果忽略这个参数可能会产生一些保守但是错误的转录本，其中含有大量剪切位点的基因组装起来尤其困难。
常用的参数及描述：</p>
<p><img alt="" src="038.png" /></p>
<div class="highlight"><pre><span></span><code>-o [&lt;path/&gt;]&lt;out.gtf&gt;  #设置StringTie组装转录本的输出GTF文件的路径和文件名。此处可指定完整路径，在这种情况下，将根据需要创建目录。默认情况下，StringTie将GTF写入标准输出。
-p &lt;int&gt;               #指定组装转录本的线程数（CPU）。默认值是1
-G &lt;ref_ann.gff&gt;       #使用参考注释基因文件指导组装过程，格式GTF/GFF3。输出文件中既包含已知表达的转录本，也包含新的转录本。选项-B，-b，-e，-C需要此选项（详情如下）
-l &lt;label&gt;             #将&lt;label&gt;设置为输出转录本名称的前缀。默认：STRG
-A &lt;gene_abund.tab&gt;    #输出基因丰度的文件（制表符分隔格式）
-C &lt;cov_refs.gtf&gt;      #输出所有转录本对应的reads覆盖度的文件，此处的转录本是指参考注释基因文件中提供的转录本。(需要参数 -G).
-B                     #应用该选项，则会输出Ballgown输入表文件（* .ctab），其中包含用-G选项给出的参考转录本的覆盖率数据。（有关这些文件的说明，请参阅Ballgown文档。）如果选项-o 给出输出转录文件的完整路径，则* .ctab文件与输出GTF文件在相同的目录下。
-b &lt;path&gt;              #指定 *.ctab 文件的输出路径, 而非由-o选项指定的目录。注意: 建议在使用-B/-b选项中同时使用-e选项，除非StringTie GTF输出文件中仍需要新的转录本,-B和-b选一个使用就行。
-e                     #限制reads比对的处理，仅估计和输出与用-G选项给出的参考转录本匹配的组装转录本。使用该选项，则会跳过处理与参考转录本不匹配的组装转录本，这将大大的提升了处理速度。
--merge                #转录本合并模式。在合并模式下，StringTie将所有样品的GTF/GFF文件列表作为输入，并将这些转录本合并/组装成非冗余的转录本集合。这种模式被用于新的差异分析流程中，用以生成一个跨多个RNA-Seq样品的全局的、统一的转录本。如果提供了-G选项（参考注释基因组文件），则StringTie将从输入的GTF文件中将参考转录本组装到transfrags中。(个人理解：transfrags可能指的是拼接成更大的转录本片段，tanscript fragments)。在此模式下可以使用以下附加选项：
-G &lt;guide_gff&gt;         #参考注释基因组文件(GTF/GFF3)
-o &lt;out_gtf&gt;           #指定输出合并的GTF文件的路径和名称 (默认值：标准输出)
-m &lt;min_len&gt;           #合并文件中，指定允许最小输入转录本的长度 (默认值: 50)
-c &lt;min_cov&gt;           #合并文件中，指定允许最低输入转录本的覆盖度(默认值: 0)
-F &lt;min_fpkm&gt;          #合并文件中，指定允许最低输入转录本的FPKM值 (默认值: 0)
-T &lt;min_tpm&gt;           #合并文件中，指定允许最低输入转录本的TPM值 (默认值: 0)
-f &lt;min_iso&gt;           #minimum isoform fraction (默认值: 0.01)
-i                     #合并后，保留含retained introns的转录本 (默认值: 除非有强有力的证据，否则不予保留)
-l &lt;label&gt;             #输出转录本的名称前缀 (默认值: MSTRG)
</code></pre></div>
<h3 id="stringtie_2">（二）StringTie应用<a class="headerlink" href="#stringtie_2" title="Permanent link">&para;</a></h3>
<h4 id="1_1"><strong>1.单样本转录本组装</strong><a class="headerlink" href="#1_1" title="Permanent link">&para;</a></h4>
<p>输入文档是BAM格式的比对结果文档，该文档必需经过排序。这些文档可以是来源于Tophat比对的结果文档，也可以是hisat2的结果文档经过转换和排序的文档(使用samtools)。我们前面使用的就是hisat2比对后用samtools排序后的bam文件。除此以外，我们还需要gtf注释文件。关于gtf注释文件格式参考文章：<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651272899&amp;idx=1&amp;sn=6779b2bd21f3b607a08227d31c7212c6&amp;chksm=841ed2beb3695ba8bee23563c28caa005447b2298785719964732b16cafe3a15d7d4937b95c1&amp;scene=21#wechat_redirect">生信中常见的数据文件格式</a>。
gtf注释文件可以去genecode下载你需要的gtf文件，我这里下载的是小鼠的。
genecode官网：<a href="https://www.gencodegenes.org/">https://www.gencodegenes.org/</a>
我的gtf文件在/data/mouse_annotation/ 目录下：</p>
<p><img alt="" src="039.png" /></p>
<p>单样本组装：</p>
<p><div class="highlight"><pre><span></span><code>stringtie -p 8 -G /data/mouse_annotation/gencode.mouse.annotation.gtf -o cleandata/stringtiedata/CK-4.gtf cleandata/samtools_bam/CK-4_sort.bam
</code></pre></div>
批量处理：
<div class="highlight"><pre><span></span><code>for i in CK-7 CK-8 HGJ-10 HGJ-6 HGJ-9
do stringtie -p 8 \
   -G /data/mouse_annotation/gencode.mouse.annotation.gtf \
   -o cleandata/stringtiedata/${i}.gtf cleandata/samtools_bam/${i}_sort.bam
done
</code></pre></div>
查看输出结果：
<div class="highlight"><pre><span></span><code>ll -h cleandata/stringtiedata/
</code></pre></div>
<img alt="" src="040.png" /></p>
<h4 id="2"><strong>2.多样本转录本整合</strong><a class="headerlink" href="#2" title="Permanent link">&para;</a></h4>
<p>stringtie --merge [options] gtf.list :转录组merge模式，在该模式下，Stringtie可以利用输入的一个gtf list并将他们中的转录本进行非冗余的整理。可以在处理多个RNA-seq样本的时候，由于转录组存在时空特异性，可以将每个样本各自的转录组进行非冗余的整合，如果-G提供了参考gtf文件，可以将其一起整合到一个文件中,最终输出成一个完整的gtf文件。
将文件名的完整路径输入到一个文件中：</p>
<p><div class="highlight"><pre><span></span><code>find /data/RNAseq/cleandata/stringtiedata/ -name *.gtf &gt; /data/RNAseq/cleandata/stringtiedata/merglist.txt
</code></pre></div>
查看一下merglist.txt文件中的内容：
<div class="highlight"><pre><span></span><code>cat /data/RNAseq/cleandata/stringtiedata/merglist.txt
</code></pre></div>
<img alt="" src="041.png" />
<div class="highlight"><pre><span></span><code>stringtie --merge -p 8 -G /data/mouse_annotation/gencode.mouse.annotation.gtf -o cleandata/stringtiedata/stringtie_merged.gtf cleandata/stringtiedata/merglist.txt
</code></pre></div>
<img alt="" src="042.png" /></p>
<h4 id="3-gffcompare"><strong>3. 使用gffcompare检验数据比对到基因组上的情况(可选)</strong><a class="headerlink" href="#3-gffcompare" title="Permanent link">&para;</a></h4>
<p>程序gffcompare可用于比较、合并、注释和估计一个或多个GFF文件(“查询”文件)的准确性。</p>
<p><div class="highlight"><pre><span></span><code>gffcompare [options]* {-i &lt;input_gtf_list&gt; | &lt;input1.gtf&gt; [&lt;input2.gtf&gt; .. &lt;inputN.gtf&gt;]}
</code></pre></div>
参数：
<div class="highlight"><pre><span></span><code>-i  provide a text file with a list of (query) GTF files to process instead of expecting them as command-line arguments (useful when a large number of GTF files should be processed).
-h/-help  Prints the help message and exits
-v/--version  Prints the version number and exits
-o &lt;outprefix&gt;  All output files created by gffcompare will have this prefix (e.g. .loci, .tracking, etc.). If this option is not provided the default output prefix being used is: “gffcmp”
-r  An optional “reference” annotation GFF file. Each sample is matched against this file, and sample isoforms are tagged as overlapping, matching, or novel where appropriate. See the .refmap and .tmap output file descriptions below.
-R  If -r was specified, this option causes gffcompare to ignore reference transcripts that are not overlapped by any transcript in one of input1.gtf,…,inputN.gtf. Useful for ignoring annotated transcripts that are not present in your RNA-Seq samples and thus adjusting the “sensitivity” calculation in the accuracy report written in the file.
-Q  If -r was specified, this option causes gffcompare to ignore input transcripts that are not overlapped by any transcript in the reference. Useful for adjusting the “precision” calculation in the accuracy report written in the file.
-M  discard (ignore) single-exon transfrags and reference transcripts (i.e. consider only multi-exon transcripts)
-N  discard (ignore) single-exon reference transcripts; single-exon transfrags are still considered, but they will never find an exact match
-D  discard &quot;duplicate&quot; (redundant) query transfrags (i.e. those with the same intron chain) within a single sample (and thus disable &quot;annotation&quot; mode)
-s &lt;genome_path&gt;  path to genome sequences (optional); this will enable the &quot;repeat&quot; (&#39;r&#39;) classcode assessment; &lt;genome_path&gt; should be a full path to a multi-FASTA file, preferrably indexed with samtools faidx; repeats must be soft-masked (lower case) in the genomic sequence
-e &lt;dist&gt;  Maximum distance (range) allowed from free ends of terminal exons of reference transcripts when assessing exon accuracy. By default, this is 100.
-d &lt;dist&gt;  Maximum distance (range) for grouping transcript start sites, by default 100.
-p &lt;tprefix&gt;  The name prefix to use for consensus/combined transcripts in the &lt;outprefix&gt;.combined.gtf file (default: &#39;TCONS&#39;)
-C  Discard the “contained” transfrags from the .combined.gtf output. By default, without this option, gffcompare writes in that file isoforms that were found to be fully contained/covered (with the same compatible intron structure) by other transfrags in the same locus, with the attribute “contained_in” showing the first container transfrag found. (Note: this behavior is the opposite of Cuffcompare&#39;s -C option).
-A  Like -C but will not discard intron-redundant transfrags if they start on a different 5&#39; exon (keep alternate transcript start sites)
-X  Like -C but also discard contained transfrags if transfrag ends stick out within the container&#39;s introns
-K  For -C/-A/-X, do NOT discard any redundant transfrag matching a reference
-T  Do not generate .tmap and .refmap files for each input file
-V  Gffcompare is a little more verbose about what it&#39;s doing, printing messages to stderr, and it will also show warning messages about any inconsistencies or potential issues found while reading the given GFF file(s).
--debug  Enables debug mode, which enables -V and generates additional files: &lt;outprefix&gt;.Q_discarded.lst, &lt;outprefix&gt;.missed_introns.gtf and &lt;outprefix&gt;.R_missed.lst
</code></pre></div>
参考：<a href="http://ccb.jhu.edu/software/stringtie/gffcompare.shtml">http://ccb.jhu.edu/software/stringtie/gffcompare.shtml</a>
我们将前面我们整合的转录本文件检验数据比对到基因组上的情况。</p>
<p><div class="highlight"><pre><span></span><code>gffcompare -r /data/mouse_annotation/gencode.mouse.annotation.gtf -G cleandata/stringtiedata/stringtie_merged.gtf
</code></pre></div>
会在输入的文件所在文件夹下产生几个文件：
<img alt="" src="043.png" />
其中gffcompare.annotated.gtf存储的是StringTie组装的转录本与注释文件内的转录本的差别信息，通过class_code来表示：</p>
<p><div class="highlight"><pre><span></span><code>head cleandata/stringtiedata/gffcompare.annotated.gtf |grep  class_code | cut -d &quot;;&quot; -f 5-8
</code></pre></div>
<img alt="" src="044.png" />
<div class="highlight"><pre><span></span><code>(RNAseq1) [root@localhost RNAseq]# head cleandata/stringtiedata/gffcompare.annotated.gtf |grep  class_code | cut -d &quot;;&quot; -f 5-8
 ref_gene_id &quot;ENSMUSG00000116111.1&quot;; cmp_ref &quot;ENSMUST00000229069.1&quot;; class_code &quot;=&quot;; tss_id &quot;TSS1&quot;
 ref_gene_id &quot;ENSMUSG00000115879.1&quot;; cmp_ref &quot;ENSMUST00000229227.1&quot;; class_code &quot;=&quot;; tss_id &quot;TSS2&quot;
 ref_gene_id &quot;ENSMUSG00000116214.1&quot;; contained_in &quot;ENSMUST00000229087.1&quot;; cmp_ref &quot;ENSMUST00000229989.1&quot;; class_code &quot;=&quot;
</code></pre></div>
对照表如下：
<img alt="" src="045.png" />
gffcmp.stats文件存储比对结果的准确性和预测率。</p>
<p><div class="highlight"><pre><span></span><code>cat cleandata/stringtiedata/gffcompare.stats
</code></pre></div>
<img alt="" src="046.png" />
<div class="highlight"><pre><span></span><code># gffcompare v0.11.2 | Command line was:
#gffcompare -r /data/mouse_annotation/gencode.mouse.annotation.gtf -G cleandata/stringtiedata/stringtie_merged.gtf -o cleandata/stringtiedata/gffcompare
#

#= Summary for dataset: cleandata/stringtiedata/stringtie_merged.gtf 
#     Query mRNAs :  155156 in   54744 loci  (127176 multi-exon transcripts)
#            (20889 multi-transcript loci, ~2.8 transcripts per locus)
# Reference mRNAs :  143543 in   54276 loci  (116302 multi-exon)
# Super-loci w/ reference transcripts:    53940
#-----------------| Sensitivity | Precision  |
        Base level:   100.0     |    97.3    |
        Exon level:    94.0     |    96.1    |
      Intron level:   100.0     |    97.7    |
Intron chain level:   100.0     |    91.4    |
  Transcript level:    99.5     |    92.1    |
       Locus level:    99.7     |    98.3    |

     Matching intron chains:  116302
       Matching transcripts:  142855
              Matching loci:   54117

          Missed exons:       0/451928  (  0.0%)
           Novel exons:    3037/433435  (  0.7%)
        Missed introns:       6/288208  (  0.0%)
         Novel introns:    2186/295137  (  0.7%)
           Missed loci:       0/54276  (  0.0%)
            Novel loci:     804/54744  (  1.5%)

 Total union super-loci across all input datasets: 54744 
155156 out of 155156 consensus transcripts written in cleandata/stringtiedata/gffcompare.annotated.gtf (0 discarded as redundant)
</code></pre></div></p>
<h4 id="4"><strong>4.重新组装转录本并估算基因表达丰度</strong><a class="headerlink" href="#4" title="Permanent link">&para;</a></h4>
<p>利用merge得到的gtf重新对各个样本做定量，并创建ballgown可读取文件。</p>
<p><div class="highlight"><pre><span></span><code>mkdir cleandata/ballgown
for i in CK-4 CK-7 CK-8 HGJ-10 HGJ-6 HGJ-9; do stringtie -e -B -p 8 -G cleandata/stringtiedata/stringtie_merged.gtf  -o cleandata/ballgown/${i}/${i}.gtf cleandata/samtools_bam/${i}_sort.bam; done
</code></pre></div>
查看ballgown目录下生成的文件:</p>
<p><img alt="" src="047.png" /></p>
<p>每一个样本对应一个文件夹。
查看其中一个：</p>
<p><img alt="" src="048.png" /></p>
<h4 id="5read-count"><strong>5.read count数据输出</strong><a class="headerlink" href="#5read-count" title="Permanent link">&para;</a></h4>
<p>这里需要prepDE.py这个脚本。</p>
<p><div class="highlight"><pre><span></span><code>prepDE.py -i cleandata/ballgown/
</code></pre></div>
会在当前文件夹下产生2个csv文件。</p>
<p><img alt="" src="049.png" />
分别查看一下：</p>
<p><img alt="" src="050.png" /></p>
<p>基础的转录组分析大概流程就是这样，当然，分析流程很多，我这里只是其中一个分析流程，可以参考文献，这里面涉及的软件也要不断应用，才知道那里出问题怎么解决，后续的分析算是下游分析，比如利用edgR包或DESeq2进行差异表达分析，这里得到的gene_count_matrix.csv数据就可以用于后续的下游分析了，比如差异表达分析，可参考TCGA序列的文章【<a href="https://mp.weixin.qq.com/mp/appmsgalbum?action=getalbum&amp;album_id=1338047035672526848&amp;__biz=MzA4NDAzODkzMA==#wechat_redirect"><strong>TCGA</strong></a>】，本公众号文章目录可通过文章浏览：<a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651273306&amp;idx=1&amp;sn=d92056ee1fde1ff12b1506e02f8eb56f&amp;chksm=841edc27b36955310bac7e1b4e5e4ab7e26e669c192926dbee3efc4f6d185cf3509a8fc87696&amp;scene=21#wechat_redirect"><strong>公众号文章目录</strong></a>。</p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  回到页面顶部
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2022 - 2025 DoubleHelix
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
    <a href="https://space.bilibili.com/434318322" target="_blank" rel="noopener" title="Bilibili | BioInfoCloud" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M488.6 104.1c16.7 18.1 24.4 39.7 23.3 65.7v202.4c-.4 26.4-9.2 48.1-26.5 65.1-17.2 17-39.1 25.9-65.5 26.7H92c-26.4-.8-48.2-9.8-65.3-27.2S.7 396.5 0 368.2V169.8c.8-26 9.7-47.6 26.7-65.7C43.8 87.8 65.5 78.8 92 78h29.4L96 52.2c-5.7-5.7-8.6-13-8.6-21.8S90.3 14.3 96 8.6 109 0 117.9 0s16.1 2.9 21.9 8.6L213.1 78h88l74.5-69.4C381.7 2.9 389.2 0 398 0s16.1 2.9 21.9 8.6c5.7 5.7 8.6 13 8.6 21.8s-2.9 16.1-8.6 21.8L394.6 78h29.3c26.4.8 48 9.8 64.7 26.1m-38.8 69.7c-.4-9.6-3.7-17.4-10.7-23.5-5.2-6.1-14-9.4-22.7-9.8H96c-9.6.4-17.4 3.7-23.6 9.8-6.1 6.1-9.4 13.9-9.8 23.5v194.4c0 9.2 3.3 17 9.8 23.5s14.4 9.8 23.6 9.8h320.4c9.2 0 17-3.3 23.3-9.8s9.7-14.3 10.1-23.5zm-264.3 42.7c6.3 6.3 9.7 14.1 10.1 23.2V273c-.4 9.2-3.7 16.9-9.8 23.2-6.2 6.3-14 9.5-23.6 9.5s-17.5-3.2-23.6-9.5-9.4-14-9.8-23.2v-33.3c.4-9.1 3.8-16.9 10.1-23.2s13.2-9.6 23.3-10c9.2.4 17 3.7 23.3 10m191.5 0c6.3 6.3 9.7 14.1 10.1 23.2V273c-.4 9.2-3.7 16.9-9.8 23.2s-14 9.5-23.6 9.5-17.4-3.2-23.6-9.5c-7-6.3-9.4-14-9.7-23.2v-33.3c.3-9.1 3.7-16.9 10-23.2s14.1-9.6 23.3-10c9.2.4 17 3.7 23.3 10"/></svg>
    </a>
  
    
    
    
    
    <a href="https://github.com/BioInfoCloud" target="_blank" rel="noopener" title="GitHub | BioInfoCloud" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../../..", "features": ["navigation.tabs", "navigation.top", "navigation.indexes", "navigation.expand", "search.suggest", "search.highlight"], "search": "../../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "\u5df2\u590d\u5236", "clipboard.copy": "\u590d\u5236", "search.result.more.one": "\u5728\u8be5\u9875\u4e0a\u8fd8\u6709 1 \u4e2a\u7b26\u5408\u6761\u4ef6\u7684\u7ed3\u679c", "search.result.more.other": "\u5728\u8be5\u9875\u4e0a\u8fd8\u6709 # \u4e2a\u7b26\u5408\u6761\u4ef6\u7684\u7ed3\u679c", "search.result.none": "\u6ca1\u6709\u627e\u5230\u7b26\u5408\u6761\u4ef6\u7684\u7ed3\u679c", "search.result.one": "\u627e\u5230 1 \u4e2a\u7b26\u5408\u6761\u4ef6\u7684\u7ed3\u679c", "search.result.other": "# \u4e2a\u7b26\u5408\u6761\u4ef6\u7684\u7ed3\u679c", "search.result.placeholder": "\u952e\u5165\u4ee5\u5f00\u59cb\u641c\u7d22", "search.result.term.missing": "\u7f3a\u5c11", "select.version": "\u9009\u62e9\u5f53\u524d\u7248\u672c"}, "version": null}</script>
    
    
      <script src="../../../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
        <script src="../../../mkdocs/javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://unpkg.com/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>