
<!doctype html>
<html lang="zh" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="生物信息云笔记">
      
      
        <meta name="author" content="DoubleHelix">
      
      
        <link rel="canonical" href="https://bioinfocloud.io/BioInfo/scRNAseq/scRNASeq_down/">
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-8.5.10">
    
    
      
        <title>单细胞下游分析 - 生物信息云笔记</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.472b142f.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.08040f6c.min.css">
        
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../mkdocs/css/no-footer.css">
    
      <link rel="stylesheet" href="../../../mkdocs/css/unordered-list-symbols.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="light-blue">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#1" class="md-skip">
          跳转至
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="页眉">
    <a href="../../.." title="生物信息云笔记" class="md-header__button md-logo" aria-label="生物信息云笔记" data-md-component="logo">
      
  <img src="../../../assets/logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            生物信息云笔记
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              单细胞下游分析
            
          </span>
        </div>
      </div>
    </div>
    
      <form class="md-header__option" data-md-component="palette">
        
          
          
          
          <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="light-blue"  aria-label="切换风格"  type="radio" name="__palette" id="__palette_1">
          
            <label class="md-header__button md-icon" title="切换风格" for="__palette_5" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.04 8.71V4h-4.7L12 .69 8.71 4H4v4.71L.69 12 4 15.34v4.7h4.71L12 23.35l3.34-3.31h4.7v-4.7L23.35 12l-3.31-3.29M8.83 7.05c.98 0 1.77.79 1.77 1.78a1.77 1.77 0 0 1-1.77 1.77c-.99 0-1.78-.79-1.78-1.77 0-.99.79-1.78 1.78-1.78M15.22 17c-.98 0-1.77-.8-1.77-1.78a1.77 1.77 0 0 1 1.77-1.77c.98 0 1.78.79 1.78 1.77A1.78 1.78 0 0 1 15.22 17m-6.72.03L7 15.53 15.53 7l1.5 1.5-8.53 8.53Z"/></svg>
            </label>
          
        
          
          
          
          <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="orange"  aria-label="切换风格"  type="radio" name="__palette" id="__palette_2">
          
            <label class="md-header__button md-icon" title="切换风格" for="__palette_1" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5c-.84 0-1.65.15-2.39.42L12 2M3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29L3.34 7m.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14L3.36 17M20.65 7l-1.77 3.79a7.023 7.023 0 0 0-2.38-4.15l4.15.36m-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29L20.64 17M12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44L12 22Z"/></svg>
            </label>
          
        
          
          
          
          <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="deep-purple"  aria-label="切换风格"  type="radio" name="__palette" id="__palette_3">
          
            <label class="md-header__button md-icon" title="切换风格" for="__palette_2" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
            </label>
          
        
          
          
          
          <input class="md-option" data-md-color-media="" data-md-color-scheme="purple" data-md-color-primary="black" data-md-color-accent="pink"  aria-label="切换风格"  type="radio" name="__palette" id="__palette_4">
          
            <label class="md-header__button md-icon" title="切换风格" for="__palette_3" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
            </label>
          
        
          
          
          
          <input class="md-option" data-md-color-media="" data-md-color-scheme="light-green" data-md-color-primary="black" data-md-color-accent="light-blue"  aria-label="切换至日间模式"  type="radio" name="__palette" id="__palette_5">
          
            <label class="md-header__button md-icon" title="切换至日间模式" for="__palette_4" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9 2c-1.05 0-2.05.16-3 .46 4.06 1.27 7 5.04 7 9.54 0 4.5-2.94 8.27-7 9.54.95.3 1.95.46 3 .46a10 10 0 0 0 10-10A10 10 0 0 0 9 2Z"/></svg>
            </label>
          
        
      </form>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="搜索" placeholder="搜索" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="查找">
        
        <button type="reset" class="md-search__icon md-icon" title="清空当前内容" aria-label="清空当前内容" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            正在初始化搜索引擎
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://bioinfocloud.github.io/note" title="前往仓库" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    BioInfoCloud
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="标签" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../Programming/" class="md-tabs__link">
        编程
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../" class="md-tabs__link">
        生信基础
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../Biopython/01_install/" class="md-tabs__link">
        Biopython
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../bigdata/" class="md-tabs__link">
        大数据
      </a>
    </li>
  

      
        
  
  
    
  


  
  
  
    

  
  
  
    <li class="md-tabs__item">
      <a href="../scRNASeq_UP/" class="md-tabs__link md-tabs__link--active">
        专题
      </a>
    </li>
  

  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../Literature/RNAseq_review1/" class="md-tabs__link">
        文献
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../ABOUT/" class="md-tabs__link">
        关于
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="导航栏" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="生物信息云笔记" class="md-nav__button md-logo" aria-label="生物信息云笔记" data-md-component="logo">
      
  <img src="../../../assets/logo.png" alt="logo">

    </a>
    生物信息云笔记
  </label>
  
    <div class="md-nav__source">
      <a href="https://bioinfocloud.github.io/note" title="前往仓库" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    BioInfoCloud
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_1" type="checkbox" id="__nav_1" checked>
      
      
      
        
          
            
          
        
          
        
          
        
          
        
          
        
          
        
      
      
        
        
        <div class="md-nav__link md-nav__link--index ">
          <a href="../../../Programming/">编程</a>
          
            <label for="__nav_1">
              <span class="md-nav__icon md-icon"></span>
            </label>
          
        </div>
      
      <nav class="md-nav" aria-label="编程" data-md-level="1">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          编程
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_1_2" type="checkbox" id="__nav_1_2" checked>
      
      
      
        
          
        
          
        
          
        
          
        
          
        
          
        
      
      
        <label class="md-nav__link" for="__nav_1_2">
          Linux
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Linux" data-md-level="2">
        <label class="md-nav__title" for="__nav_1_2">
          <span class="md-nav__icon md-icon"></span>
          Linux
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/linux/" class="md-nav__link">
        Linux讲座
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/commands-concise/" class="md-nav__link">
        命令行入门精简版
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/vim/" class="md-nav__link">
        vim
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/tmux/" class="md-nav__link">
        tmux
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/unix/" class="md-nav__link">
        UNIX
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/kernel-compile/" class="md-nav__link">
        内核编译
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_1_3" type="checkbox" id="__nav_1_3" checked>
      
      
      
        
          
        
          
        
          
        
      
      
        <label class="md-nav__link" for="__nav_1_3">
          Markdown
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Markdown" data-md-level="2">
        <label class="md-nav__title" for="__nav_1_3">
          <span class="md-nav__icon md-icon"></span>
          Markdown
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Markdown/markdown/" class="md-nav__link">
        Markdown讲座
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/vscode-python/vscode-python/" class="md-nav__link">
        VSCode及插件安装
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Markdown/grammars/" class="md-nav__link">
        Markdown语法
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_1_4" type="checkbox" id="__nav_1_4" checked>
      
      
      
        
          
        
          
        
          
        
          
        
      
      
        <label class="md-nav__link" for="__nav_1_4">
          Git
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Git" data-md-level="2">
        <label class="md-nav__title" for="__nav_1_4">
          <span class="md-nav__icon md-icon"></span>
          Git
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Git/git/" class="md-nav__link">
        Git
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Git/git-demos/" class="md-nav__link">
        Git例子
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Git/git-teamwork/" class="md-nav__link">
        Git合作
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Git/windows-install/" class="md-nav__link">
        Windows安装Git
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_1_5" type="checkbox" id="__nav_1_5" checked>
      
      
      
        
          
        
          
        
          
        
      
      
        <label class="md-nav__link" for="__nav_1_5">
          R
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="R" data-md-level="2">
        <label class="md-nav__title" for="__nav_1_5">
          <span class="md-nav__icon md-icon"></span>
          R
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/R/Grammar/" class="md-nav__link">
        R语法
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/R/ggplot2/" class="md-nav__link">
        可视化
      </a>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_1_5_3" type="checkbox" id="__nav_1_5_3" checked>
      
      
      
        
          
        
          
        
      
      
        <label class="md-nav__link" for="__nav_1_5_3">
          R包开发
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="R包开发" data-md-level="3">
        <label class="md-nav__title" for="__nav_1_5_3">
          <span class="md-nav__icon md-icon"></span>
          R包开发
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/R/01-software/" class="md-nav__link">
        软件安装
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/R/02-DevelopmentManual/" class="md-nav__link">
        快速手册
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_1_6" type="checkbox" id="__nav_1_6" checked>
      
      
      
        
          
        
      
      
        <label class="md-nav__link" for="__nav_1_6">
          Python
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Python" data-md-level="2">
        <label class="md-nav__title" for="__nav_1_6">
          <span class="md-nav__icon md-icon"></span>
          Python
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/python/links/" class="md-nav__link">
        语法基础
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_2" type="checkbox" id="__nav_2" checked>
      
      
      
        
          
            
          
        
          
        
      
      
        
        
        <div class="md-nav__link md-nav__link--index ">
          <a href="../../">生信基础</a>
          
            <label for="__nav_2">
              <span class="md-nav__icon md-icon"></span>
            </label>
          
        </div>
      
      <nav class="md-nav" aria-label="生信基础" data-md-level="1">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          生信基础
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_2_2" type="checkbox" id="__nav_2_2" checked>
      
      
      
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
      
      
        <label class="md-nav__link" for="__nav_2_2">
          生信基础
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="生信基础" data-md-level="2">
        <label class="md-nav__title" for="__nav_2_2">
          <span class="md-nav__icon md-icon"></span>
          生信基础
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/Introduction/" class="md-nav__link">
        生信引言
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/01/" class="md-nav__link">
        基础应用
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/02_noun/" class="md-nav__link">
        生信名词
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/03_sequence/" class="md-nav__link">
        测序技术原理
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/04_format/" class="md-nav__link">
        数据文件格式
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/biomaRt/" class="md-nav__link">
        biomaRt包
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/counts_RPM_RPKM_FPKM_1/" class="md-nav__link">
        RNASeq表达定量
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/counts_RPM_RPKM_FPKM_2/" class="md-nav__link">
        RNASeq表达定量转换
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/DESeq2_edgeR_limma/" class="md-nav__link">
        RNASeq差异表达分析原理
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/GO/" class="md-nav__link">
        什么是GO
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/KEGG_GO_GSEA_what/" class="md-nav__link">
        富集分析理论基础
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/hsa2mus/" class="md-nav__link">
        人鼠基因比较
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/hsa2mus_2/" class="md-nav__link">
        同源基因转换
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/IDtrans/" class="md-nav__link">
        各种ID转换
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/miRNA_ID/" class="md-nav__link">
        miRNA的ID转换
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" checked>
      
      
      
        
          
        
      
      
        <label class="md-nav__link" for="__nav_3">
          Biopython
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Biopython" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Biopython
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../Biopython/01_install/" class="md-nav__link">
        安装
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" checked>
      
      
      
        
          
            
          
        
      
      
        
        
        <div class="md-nav__link md-nav__link--index ">
          <a href="../../../bigdata/">大数据</a>
          
        </div>
      
      <nav class="md-nav" aria-label="大数据" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          大数据
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" checked>
      
      
      
        
          
        
          
        
          
        
          
        
          
        
      
      
        <label class="md-nav__link" for="__nav_5">
          专题
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="专题" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          专题
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5_1" type="checkbox" id="__nav_5_1" checked>
      
      
      
        
          
        
          
        
          
        
          
        
          
        
      
      
        <label class="md-nav__link" for="__nav_5_1">
          单细胞专题
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="单细胞专题" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_1">
          <span class="md-nav__icon md-icon"></span>
          单细胞专题
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../scRNASeq_UP/" class="md-nav__link">
        单细胞上游分析
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../SeuratObj/" class="md-nav__link">
        了解Seurat对象
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          单细胞下游分析
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        单细胞下游分析
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="目录">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      目录
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1" class="md-nav__link">
    1.数据读入
  </a>
  
    <nav class="md-nav" aria-label="1.数据读入">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1csvseurat" class="md-nav__link">
    (1).读入csv文件的表达矩阵构建Seurat对象
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#210x3" class="md-nav__link">
    (2).读入10X标准输出的3个文件和融合多个样本数据
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3h5" class="md-nav__link">
    (3).读入h5格式文件
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-h5ad" class="md-nav__link">
    (4). h5ad格式
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2" class="md-nav__link">
    2.数据质控
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3" class="md-nav__link">
    3.数据标准化
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4" class="md-nav__link">
    4.细胞周期分析
  </a>
  
    <nav class="md-nav" aria-label="4.细胞周期分析">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1seurat" class="md-nav__link">
    (1).Seurat包
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2scran" class="md-nav__link">
    (2).scran包
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5doublets" class="md-nav__link">
    5.检测doublets
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6" class="md-nav__link">
    6.单细胞多样本整合
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7" class="md-nav__link">
    7.聚类
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#8" class="md-nav__link">
    8.细胞类型注释
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../GSE130001/" class="md-nav__link">
        常规分析案例一
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../SingleR/" class="md-nav__link">
        细胞注释-SingleR
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_5_2" type="checkbox" id="__nav_5_2" checked>
      
      
      
        
          
        
      
      
        <label class="md-nav__link" for="__nav_5_2">
          RNAseq专题
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="RNAseq专题" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_2">
          <span class="md-nav__icon md-icon"></span>
          RNAseq专题
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../RNAseq/pipeline_1/" class="md-nav__link">
        Bulk RNAseq上游分析
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_5_3" type="checkbox" id="__nav_5_3" checked>
      
      
      
        
          
            
          
        
      
      
        
        
        <div class="md-nav__link md-nav__link--index ">
          <a href="../../TCGA/">TCGA-GEO专题</a>
          
        </div>
      
      <nav class="md-nav" aria-label="TCGA-GEO专题" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_3">
          <span class="md-nav__icon md-icon"></span>
          TCGA-GEO专题
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_5_4" type="checkbox" id="__nav_5_4" checked>
      
      
      
        
          
        
      
      
        <label class="md-nav__link" for="__nav_5_4">
          药物分析
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="药物分析" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_4">
          <span class="md-nav__icon md-icon"></span>
          药物分析
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../drugAna/cellminer/" class="md-nav__link">
        CellMiner数据库
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_5_5" type="checkbox" id="__nav_5_5" checked>
      
      
      
        
          
        
          
        
      
      
        <label class="md-nav__link" for="__nav_5_5">
          数据库工具
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="数据库工具" data-md-level="2">
        <label class="md-nav__title" for="__nav_5_5">
          <span class="md-nav__icon md-icon"></span>
          数据库工具
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../database/CoMutDB/" class="md-nav__link">
        CoMutDB数据库
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../database/CysModDB/" class="md-nav__link">
        CysModDB数据库
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" checked>
      
      
      
        
          
        
          
        
      
      
        <label class="md-nav__link" for="__nav_6">
          文献
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="文献" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          文献
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/RNAseq_review1/" class="md-nav__link">
        RNASeq综述-01
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/RNAseq_review1/" class="md-nav__link">
        RNASeq综述-02
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" data-md-toggle="__nav_7" type="checkbox" id="__nav_7" checked>
      
      
      
        
          
            
          
        
          
        
      
      
        
        
        <div class="md-nav__link md-nav__link--index ">
          <a href="../../../ABOUT/">关于</a>
          
            <label for="__nav_7">
              <span class="md-nav__icon md-icon"></span>
            </label>
          
        </div>
      
      <nav class="md-nav" aria-label="关于" data-md-level="1">
        <label class="md-nav__title" for="__nav_7">
          <span class="md-nav__icon md-icon"></span>
          关于
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../ABOUT/studentsGroup/" class="md-nav__link">
        学生群
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="目录">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      目录
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1" class="md-nav__link">
    1.数据读入
  </a>
  
    <nav class="md-nav" aria-label="1.数据读入">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1csvseurat" class="md-nav__link">
    (1).读入csv文件的表达矩阵构建Seurat对象
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#210x3" class="md-nav__link">
    (2).读入10X标准输出的3个文件和融合多个样本数据
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3h5" class="md-nav__link">
    (3).读入h5格式文件
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-h5ad" class="md-nav__link">
    (4). h5ad格式
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2" class="md-nav__link">
    2.数据质控
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3" class="md-nav__link">
    3.数据标准化
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4" class="md-nav__link">
    4.细胞周期分析
  </a>
  
    <nav class="md-nav" aria-label="4.细胞周期分析">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1seurat" class="md-nav__link">
    (1).Seurat包
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2scran" class="md-nav__link">
    (2).scran包
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5doublets" class="md-nav__link">
    5.检测doublets
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6" class="md-nav__link">
    6.单细胞多样本整合
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7" class="md-nav__link">
    7.聚类
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#8" class="md-nav__link">
    8.细胞类型注释
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  
  



  <a href="https://bioinfocloud.github.io/note/edit/main/docs/BioInfo/scRNAseq/scRNASeq_down.md" title="编辑此页" class="md-content__button md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg>
  </a>


  <h1>单细胞下游分析</h1>

<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651277403&amp;idx=1&amp;sn=1deef9220392274b9eb7018e764f56a5&amp;chksm=841eac26b3692530adc7a2d611ea00f4fb64eaeb32619cb78ed82553f47e713a1aa038b41f02#rd">单细胞专题 | 1.单细胞测序（10×genomics技术）的原理</a></p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651277453&amp;idx=1&amp;sn=cda31840d53f64763d84b6e3fde01b2c&amp;chksm=841eacf0b36925e6c453dbdb042cf1a8073f5919868a48e2f661bfda3e53660fe18d92f9969a#rd">单细胞专题 | 2.如何开始单细胞RNASeq数据分析</a></p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651277619&amp;idx=1&amp;sn=59c73d767aed61f01a56a0165aed1061&amp;chksm=841ead4eb36924587ee97e8d109a3cc1d2a6c6cc9d87c729e75d220cae1f63838cce8503d506#rd">单细胞专题 | 3.单细胞转录组的上游分析-从BCL到FASTQ</a></p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651277662&amp;idx=1&amp;sn=1f7abfda069a921b948f2cb00a0b4979&amp;chksm=841ead23b369243565a7f88d378302b509aefff9647b4f684a47269d6c2d374b3b37cf4696f1#rd">单细胞专题 | 4.单细胞转录组的上游分析-从SRA到FASTQ</a></p>
<p><a href="http://mp.weixin.qq.com/s?__biz=MzA4NDAzODkzMA==&amp;mid=2651277697&amp;idx=1&amp;sn=059a6f22487e94648d38d45453dff6b6&amp;chksm=841eadfcb36924ea64b3c0fc6e616a8313bf3773977c74db94544032556ab432640554397d85#rd">单细胞专题 | 5.单细胞转录组的上游分析-从FASTQ到count矩阵</a></p>
<h2 id="1">1.数据读入<a class="headerlink" href="#1" title="Permanent link">&para;</a></h2>
<p>Cell Ranger生成的主要表格文件主要包括3个文件，分别是barcodes, features, matrix三个表格文件，其中barcodes文件存储的是细胞的信息，features存储的是基因的名称信息，matrix存储的是表达量的信息。还有一种数据是作者在GEO数据库直接提供表达矩阵（csv或txt）</p>
<h3 id="1csvseurat">(1).读入csv文件的表达矩阵构建Seurat对象<a class="headerlink" href="#1csvseurat" title="Permanent link">&para;</a></h3>
<p>Seurat需要的输入信息为表达量矩阵，矩阵行为基因，列为细胞。使用Seurat提供的Read10X函数可以很方便的将10x结果读入到R矩阵中。使用CreateSeuratObject生成Seurat对象，后续分析都是在该对象上进行操作。</p>
<div class="highlight"><pre><span></span><code><span class="nf">rm</span><span class="p">(</span><span class="n">list</span><span class="o">=</span><span class="nf">ls</span><span class="p">())</span>
<span class="nf">options</span><span class="p">(</span><span class="n">stringsAsFactors</span> <span class="o">=</span> <span class="bp">F</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">Seurat</span><span class="p">)</span>
<span class="c1"># Load data </span>
<span class="n">dir</span><span class="o">=</span><span class="s">&#39;./&#39;</span>
<span class="nf">Sys.time</span><span class="p">()</span>
<span class="n">raw.data</span> <span class="o">&lt;-</span> <span class="nf">read.csv</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="n">dir</span><span class="p">,</span><span class="s">&quot;Data_input/csv_files/S01_datafinal.csv&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span> <span class="n">header</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="m">1</span><span class="p">)</span>
<span class="nf">Sys.time</span><span class="p">()</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">raw.data</span><span class="p">)</span>
<span class="n">raw.data</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">]</span>
<span class="nf">head</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">raw.data</span><span class="p">))</span> 
<span class="c1"># Load metadata </span>
<span class="n">metadata</span> <span class="o">&lt;-</span> <span class="nf">read.csv</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="n">dir</span><span class="p">,</span><span class="s">&quot;Data_input/csv_files/S01_metacells.csv&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span> <span class="n">row.names</span><span class="o">=</span><span class="m">1</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="bp">T</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">metadata</span><span class="p">)</span> 

<span class="c1"># Find ERCC&#39;s, compute the percent ERCC, and drop them from the raw data.</span>
<span class="n">erccs</span> <span class="o">&lt;-</span> <span class="nf">grep</span><span class="p">(</span><span class="n">pattern</span> <span class="o">=</span> <span class="s">&quot;^ERCC-&quot;</span><span class="p">,</span> <span class="n">x</span> <span class="o">=</span> <span class="nf">rownames</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">raw.data</span><span class="p">),</span> <span class="n">value</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="n">percent.ercc</span> <span class="o">&lt;-</span> <span class="n">Matrix</span><span class="o">::</span><span class="nf">colSums</span><span class="p">(</span><span class="n">raw.data</span><span class="p">[</span><span class="n">erccs</span><span class="p">,</span> <span class="p">])</span><span class="o">/</span><span class="n">Matrix</span><span class="o">::</span><span class="nf">colSums</span><span class="p">(</span><span class="n">raw.data</span><span class="p">)</span>
<span class="nf">fivenum</span><span class="p">(</span><span class="n">percent.ercc</span><span class="p">)</span>
<span class="n">ercc.index</span> <span class="o">&lt;-</span> <span class="nf">grep</span><span class="p">(</span><span class="n">pattern</span> <span class="o">=</span> <span class="s">&quot;^ERCC-&quot;</span><span class="p">,</span> <span class="n">x</span> <span class="o">=</span> <span class="nf">rownames</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">raw.data</span><span class="p">),</span> <span class="n">value</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="n">raw.data</span> <span class="o">&lt;-</span> <span class="n">raw.data</span><span class="p">[</span><span class="o">-</span><span class="n">ercc.index</span><span class="p">,]</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">raw.data</span><span class="p">)</span> 
</code></pre></div>
<p>有了表达矩阵，直接使用 CreateSeuratObject 函数即可，然后慢慢添加这个表达矩阵的一些其它外部属性，全部代码如下：</p>
<div class="highlight"><pre><span></span><code># Create the Seurat object with all the data (unfiltered)
main_tiss &lt;- CreateSeuratObject(counts = raw.data)
# add rownames to metadta 
row.names(metadata) &lt;- metadata$cell_id
# add metadata to Seurat object 
main_tiss &lt;- AddMetaData(object = main_tiss, metadata = metadata)
main_tiss &lt;- AddMetaData(object = main_tiss, percent.ercc, col.name = &quot;percent.ercc&quot;)
# Head to check
head(main_tiss@meta.data)

# Calculate percent ribosomal genes and add to metadata
ribo.genes &lt;- grep(pattern = &quot;^RP[SL][[:digit:]]&quot;, x = rownames(x = main_tiss@assays$RNA@data), value = TRUE)
percent.ribo &lt;- Matrix::colSums(main_tiss@assays$RNA@counts[ribo.genes, ])/Matrix::colSums(main_tiss@assays$RNA@data)
fivenum(percent.ribo)
main_tiss &lt;- AddMetaData(object = main_tiss, metadata = percent.ribo, col.name = &quot;percent.ribo&quot;)
main_tiss

# Filter cells so that remaining cells have nGenes &gt;= 500 and nReads &gt;= 50000
main_tiss_filtered &lt;- subset(x=main_tiss, subset = nCount_RNA &gt; 50000 &amp; nFeature_RNA &gt; 500)
main_tiss_filtered
</code></pre></div>
<p>我们得到了main_tiss_filtered这个变量，是一个Seurat对象，就可以follow我们的教程后后续分析流程啦。参考代码仍然是：</p>
<ul>
<li><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247495635&amp;idx=1&amp;sn=b80a060573dfc266fe9d4c7773c88c7f&amp;scene=21#wechat_redirect">祖传的单个10x样本的seurat标准代码</a></li>
<li><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247495646&amp;idx=1&amp;sn=b24b77c21e99f11793ae8166be9ad02f&amp;scene=21#wechat_redirect">祖传的单个10x样本的seurat标准代码（人和鼠需要区别对待）</a></li>
<li><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247495874&amp;idx=1&amp;sn=1189c08b96ce1ea2c70964db93f480d6&amp;scene=21#wechat_redirect">seurat标准流程实例之2个10x样本的项目（GSE135927数据集）</a></li>
</ul>
<h3 id="210x3">(2).读入10X标准输出的3个文件和融合多个样本数据<a class="headerlink" href="#210x3" title="Permanent link">&para;</a></h3>
<p>这三个文件指的是：barcodes.tsv, features.tsv, matrix.mtx。这个情况就比较好处理了，barcodes.tsv就是 cell id，features.tsv就是 gene id，matrix.mtx就是计数 counts 矩阵。</p>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_001.jpg" /></p>
<p>例如:</p>
<div class="highlight"><pre><span></span><code><span class="c1">############### 10X标准输出的3个文件 ##############</span>
<span class="c1"># 列出当前目录下所有开头是GSM的文件</span>
<span class="n">fs</span> <span class="o">&lt;-</span> <span class="nf">list.files</span><span class="p">(</span><span class="s">&#39;./data/GSE106273_RAW/&#39;</span><span class="p">,</span><span class="s">&#39;^GSM&#39;</span><span class="p">)</span>

<span class="c1"># 然后获取四个样本信息</span>
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>
<span class="n">samples</span> <span class="o">&lt;-</span> <span class="nf">str_split</span><span class="p">(</span><span class="n">fs</span><span class="p">,</span><span class="s">&#39;_&#39;</span><span class="p">,</span> <span class="n">simplify</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)[,</span><span class="m">1</span><span class="p">]</span>

<span class="c1"># 设置一个循环，对每个样本信息做同样的事：</span>
<span class="c1">#（1）找到包含这个样本的文件(用grepl)</span>
<span class="c1"># (2)设置对应的目录名（str_split+paste）然后创建目录（用dir.create）</span>
<span class="c1"># (3)将文件放到对应目录(采用的是file.rename)并重命名文件</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;data/GSE106273_RAW/&quot;</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">R.utils</span><span class="p">)</span>

<span class="nf">lapply</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">samples</span><span class="p">),</span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
  <span class="n">y</span> <span class="o">&lt;-</span> <span class="n">fs</span><span class="p">[</span><span class="nf">grepl</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">fs</span><span class="p">)]</span>
  <span class="n">folder</span> <span class="o">&lt;-</span> <span class="nf">paste</span><span class="p">(</span><span class="nf">str_split</span><span class="p">(</span><span class="n">y</span><span class="p">[</span><span class="m">1</span><span class="p">],</span><span class="s">&#39;_&#39;</span><span class="p">,</span><span class="n">simplify</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)[,</span><span class="m">2</span><span class="o">:</span><span class="m">3</span><span class="p">],</span>
               <span class="n">collapse</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">)</span>
  <span class="c1"># 创建文件夹</span>
  <span class="nf">dir.create</span><span class="p">(</span><span class="n">folder</span><span class="p">,</span> <span class="n">recursive</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span>
  <span class="c1"># 文件重命名：</span>
  <span class="nf">file.rename</span><span class="p">(</span><span class="n">y</span><span class="p">[</span><span class="m">1</span><span class="p">],</span><span class="nf">file.path</span><span class="p">(</span><span class="n">folder</span><span class="p">,</span><span class="s">&quot;barcodes.tsv.gz&quot;</span><span class="p">))</span>
  <span class="nf">file.rename</span><span class="p">(</span><span class="n">y</span><span class="p">[</span><span class="m">2</span><span class="p">],</span><span class="nf">file.path</span><span class="p">(</span><span class="n">folder</span><span class="p">,</span><span class="s">&quot;genes.tsv.gz&quot;</span><span class="p">))</span>
  <span class="nf">file.rename</span><span class="p">(</span><span class="n">y</span><span class="p">[</span><span class="m">3</span><span class="p">],</span><span class="nf">file.path</span><span class="p">(</span><span class="n">folder</span><span class="p">,</span><span class="s">&quot;matrix.mtx.gz&quot;</span><span class="p">))</span>
  <span class="c1"># 解压，其实不解压也行</span>
  <span class="nf">sapply</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="n">folder</span><span class="p">,</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;/barcodes.tsv.gz&quot;</span><span class="p">,</span><span class="s">&quot;/genes.tsv.gz&quot;</span><span class="p">,</span><span class="s">&quot;/matrix.mtx.gz&quot;</span><span class="p">)),</span> 
         <span class="n">gunzip</span><span class="p">)</span>
<span class="p">})</span>
<span class="c1"># 批量读取成10X对象:</span>
<span class="c1"># 因为Read10X函数需要对目录进行操作，所以先把目录名提取出来</span>
<span class="n">folders</span><span class="o">=</span><span class="nf">list.files</span><span class="p">(</span><span class="s">&#39;./&#39;</span><span class="p">,</span><span class="n">pattern</span><span class="o">=</span><span class="s">&#39;[12]$&#39;</span><span class="p">)</span>
<span class="n">folders</span>
<span class="c1"># [1] &quot;G1&quot;  &quot;G2&quot;  &quot;L1&quot;  &quot;L2&quot;  &quot;NP1&quot; &quot;NP2&quot; &quot;PI1&quot; &quot;PI2&quot;</span>

<span class="c1"># 然后使用lapply进行循环(之前出过一期apply系列函数教程，可以查阅一下，</span>
<span class="c1"># lapply是对列表或向量进行循环，而apply是对数据框或矩阵操作)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">Seurat</span><span class="p">)</span>
<span class="n">sceList</span> <span class="o">&lt;-</span> <span class="nf">lapply</span><span class="p">(</span><span class="n">folders</span><span class="p">,</span><span class="nf">function</span><span class="p">(</span><span class="n">folder</span><span class="p">){</span> 
  <span class="nf">CreateSeuratObject</span><span class="p">(</span><span class="n">counts</span> <span class="o">=</span> <span class="nf">Read10X</span><span class="p">(</span><span class="n">folder</span><span class="p">),</span> 
                     <span class="n">project</span> <span class="o">=</span> <span class="n">folder</span> <span class="p">)</span>
<span class="p">})</span>

<span class="c1"># 此时的sceList是一个包含8个10X对象的集合，下一步需要将其合并</span>
<span class="n">sceList</span>
<span class="c1"># 合并：</span>
<span class="n">sce_big</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">sceList</span><span class="p">[[</span><span class="m">1</span><span class="p">]],</span> 
                 <span class="n">y</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="n">sceList</span><span class="p">[[</span><span class="m">2</span><span class="p">]],</span><span class="n">sceList</span><span class="p">[[</span><span class="m">3</span><span class="p">]],</span><span class="n">sceList</span><span class="p">[[</span><span class="m">4</span><span class="p">]],</span>
                       <span class="n">sceList</span><span class="p">[[</span><span class="m">5</span><span class="p">]],</span><span class="n">sceList</span><span class="p">[[</span><span class="m">6</span><span class="p">]],</span>
                       <span class="n">sceList</span><span class="p">[[</span><span class="m">7</span><span class="p">]],</span><span class="n">sceList</span><span class="p">[[</span><span class="m">8</span><span class="p">]]),</span> 
                 <span class="n">add.cell.ids</span> <span class="o">=</span> <span class="n">folders</span><span class="p">,</span> 
                 <span class="n">project</span> <span class="o">=</span> <span class="s">&quot;mouse8&quot;</span><span class="p">)</span>

<span class="nf">table</span><span class="p">(</span><span class="n">sce_big</span><span class="o">$</span><span class="n">orig.ident</span><span class="p">)</span>
<span class="c1"># G_1  G_2  L_1  L_2 NP_1 NP_2 PI_1 PI_2 </span>
<span class="c1"># 2915 3106 5906 3697 2249 2127 1500 4306 </span>

<span class="c1"># 保存：</span>
<span class="nf">save</span><span class="p">(</span><span class="n">sce_big</span><span class="p">,</span><span class="n">file</span> <span class="o">=</span> <span class="s">&#39;sce_big.Rdata&#39;</span><span class="p">)</span> <span class="c1"># 保存的数据</span>
</code></pre></div>
<p>在如：</p>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_002.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="c1">###### step1:导入数据 ###### </span>
<span class="nf">rm</span><span class="p">(</span><span class="n">list</span><span class="o">=</span><span class="nf">ls</span><span class="p">())</span>
<span class="nf">options</span><span class="p">(</span><span class="n">stringsAsFactors</span> <span class="o">=</span> <span class="bp">F</span><span class="p">)</span> 
<span class="nf">library</span><span class="p">(</span><span class="n">Seurat</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">clustree</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">cowplot</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>

<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="n">samples</span><span class="o">=</span><span class="nf">list.files</span><span class="p">(</span><span class="s">&#39;GSE129139_RAW//&#39;</span><span class="p">)</span>
<span class="n">samples</span>
<span class="n">sceList</span> <span class="o">=</span> <span class="nf">lapply</span><span class="p">(</span><span class="n">samples</span><span class="p">,</span><span class="nf">function</span><span class="p">(</span><span class="n">pro</span><span class="p">){</span> 
  <span class="c1">#pro=samples[1]</span>
  <span class="n">folder</span><span class="o">=</span><span class="nf">file.path</span><span class="p">(</span><span class="s">&#39;GSE129139_RAW&#39;</span><span class="p">,</span><span class="n">pro</span> <span class="p">)</span> 
  <span class="nf">print</span><span class="p">(</span><span class="n">pro</span><span class="p">)</span>
  <span class="nf">print</span><span class="p">(</span><span class="n">folder</span><span class="p">)</span>
  <span class="nf">print</span><span class="p">(</span><span class="nf">list.files</span><span class="p">(</span><span class="n">folder</span><span class="p">))</span>
  <span class="nf">print</span><span class="p">(</span> <span class="nf">Sys.time</span><span class="p">()</span> <span class="p">)</span>
  <span class="n">sce</span><span class="o">=</span><span class="nf">CreateSeuratObject</span><span class="p">(</span><span class="n">counts</span> <span class="o">=</span> <span class="nf">Read10X</span><span class="p">(</span><span class="n">folder</span><span class="p">),</span> 
                         <span class="n">project</span> <span class="o">=</span> <span class="n">pro</span> <span class="p">)</span>
  <span class="nf">print</span><span class="p">(</span> <span class="nf">Sys.time</span><span class="p">()</span> <span class="p">)</span>
  <span class="nf">return</span><span class="p">(</span><span class="n">sce</span><span class="p">)</span>
<span class="p">})</span>
<span class="n">sceList</span>
<span class="n">samples</span> 
<span class="n">sce.all</span> <span class="o">=</span> <span class="nf">merge</span><span class="p">(</span><span class="n">sceList</span><span class="p">[[</span><span class="m">1</span><span class="p">]],</span> <span class="n">y</span> <span class="o">=</span> <span class="n">sceList</span><span class="p">[</span><span class="m">-1</span><span class="p">],</span> 
                <span class="n">add.cell.ids</span> <span class="o">=</span> <span class="n">samples</span><span class="p">,</span>
                <span class="n">merge.data</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="nf">as.data.frame</span><span class="p">(</span><span class="n">sce.all</span><span class="o">@</span><span class="n">assays</span><span class="o">$</span><span class="n">RNA</span><span class="o">@</span><span class="n">counts</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">,</span> <span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">])</span>
<span class="nf">head</span><span class="p">(</span><span class="n">sce.all</span><span class="o">@</span><span class="n">meta.data</span><span class="p">,</span> <span class="m">10</span><span class="p">)</span>
<span class="nf">table</span><span class="p">(</span><span class="n">sce.all</span><span class="o">@</span><span class="n">meta.data</span><span class="o">$</span><span class="n">orig.ident</span><span class="p">)</span> 
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span> 
<span class="nf">table</span><span class="p">(</span><span class="n">sce.all</span><span class="o">$</span><span class="n">orig.ident</span><span class="p">)</span>
</code></pre></div>
<h3 id="3h5">(3).读入h5格式文件<a class="headerlink" href="#3h5" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="c1">############### h5格式文件 ##############</span>
<span class="c1"># 读取单个</span>
<span class="n">sce</span> <span class="o">&lt;-</span> <span class="nf">Read10X_h5</span><span class="p">(</span><span class="n">filename</span> <span class="o">=</span> <span class="s">&quot;GSM4107899_LH16.3814_raw_gene_bc_matrices_h5.h5&quot;</span><span class="p">)</span>
<span class="n">sce</span> <span class="o">&lt;-</span> <span class="nf">CreateSeuratObject</span><span class="p">(</span><span class="n">counts</span> <span class="o">=</span> <span class="n">sce</span><span class="p">)</span>

<span class="c1"># 批量读取并合并：</span>
<span class="nf">library</span><span class="p">(</span><span class="n">hdf5r</span><span class="p">)</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;../GSE138433_RAW/&quot;</span><span class="p">)</span>
<span class="n">files</span> <span class="o">&lt;-</span> <span class="nf">list.files</span><span class="p">(</span><span class="s">&#39;./&#39;</span><span class="p">)</span>
<span class="n">files</span>
<span class="n">names</span> <span class="o">&lt;-</span> <span class="nf">str_split</span><span class="p">(</span><span class="n">files</span><span class="p">,</span><span class="s">&#39;_&#39;</span><span class="p">,</span> <span class="n">simplify</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)[,</span><span class="m">2</span><span class="p">]</span>

<span class="c1"># 批量读取：</span>
<span class="n">sceList</span> <span class="o">&lt;-</span> <span class="nf">lapply</span><span class="p">(</span><span class="n">files</span><span class="p">,</span> <span class="nf">function</span><span class="p">(</span><span class="n">files</span><span class="p">){</span> 
  <span class="nf">CreateSeuratObject</span><span class="p">(</span><span class="n">counts</span> <span class="o">=</span> <span class="nf">Read10X_h5</span><span class="p">(</span><span class="n">files</span><span class="p">),</span> 
                     <span class="n">project</span> <span class="o">=</span> <span class="n">names</span><span class="p">)</span>
<span class="p">})</span>

<span class="c1"># 合并：</span>
<span class="n">sce_big</span> <span class="o">&lt;-</span> <span class="nf">merge</span><span class="p">(</span><span class="n">sceList</span><span class="p">[[</span><span class="m">1</span><span class="p">]],</span> 
                 <span class="n">y</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="n">sceList</span><span class="p">[[</span><span class="m">2</span><span class="p">]],</span><span class="n">sceList</span><span class="p">[[</span><span class="m">3</span><span class="p">]],</span><span class="n">sceList</span><span class="p">[[</span><span class="m">4</span><span class="p">]],</span>
                       <span class="n">sceList</span><span class="p">[[</span><span class="m">5</span><span class="p">]],</span><span class="n">sceList</span><span class="p">[[</span><span class="m">6</span><span class="p">]]),</span> 
                 <span class="n">add.cell.ids</span> <span class="o">=</span> <span class="n">names</span><span class="p">)</span>

<span class="nf">table</span><span class="p">(</span><span class="n">sce_big</span><span class="o">$</span><span class="n">orig.ident</span><span class="p">)</span>
<span class="c1"># LH16.3814 LH17.3222 LH17.3554  LH17.364  LH17.530  LH18.277 </span>
<span class="c1"># 737280    737280    737280    737280    737280    737280 </span>

<span class="c1"># 保存：</span>
<span class="nf">save</span><span class="p">(</span><span class="n">sce_big</span><span class="p">,</span><span class="n">file</span> <span class="o">=</span> <span class="s">&#39;sce_big.Rdata&#39;</span><span class="p">)</span> <span class="c1"># 保存的数据</span>
</code></pre></div>
<h3 id="4-h5ad">(4). h5ad格式<a class="headerlink" href="#4-h5ad" title="Permanent link">&para;</a></h3>
<p>需要安装SeuratDisk包，先将后h5ad格式转换为h5seurat格式，再使用LoadH5Seurat()函数读取Seurat对象。</p>
<div class="highlight"><pre><span></span><code><span class="c1">############### h5ad格式文件 ##############</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;../GSE153643_RAW/&quot;</span><span class="p">)</span>
<span class="c1"># remotes::install_github(&quot;mojaveazure/seurat-disk&quot;)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">SeuratDisk</span><span class="p">)</span>

<span class="c1"># 先将后`h5ad`格式转换为`h5seurat`格式:</span>
<span class="nf">Convert</span><span class="p">(</span><span class="s">&quot;GSM4648565_liver_raw_counts.h5ad&quot;</span><span class="p">,</span> <span class="s">&quot;h5seurat&quot;</span><span class="p">,</span>
        <span class="n">overwrite</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">assay</span> <span class="o">=</span> <span class="s">&quot;RNA&quot;</span><span class="p">)</span>

<span class="c1"># 再使用`LoadH5Seurat()`函数读取Seurat对象。</span>
<span class="n">sce</span> <span class="o">&lt;-</span> <span class="nf">LoadH5Seurat</span><span class="p">(</span><span class="s">&quot;GSM4648565_liver_raw_counts.h5seurat&quot;</span><span class="p">)</span>
</code></pre></div>
<h2 id="2">2.数据质控<a class="headerlink" href="#2" title="Permanent link">&para;</a></h2>
<p>可参考：<a href="https://mp.weixin.qq.com/s/g_tAhFQfr2IhFyHxzcFRTg">https://mp.weixin.qq.com/s/g_tAhFQfr2IhFyHxzcFRTg</a>
质控的目的是去除掉低质量的数据，包括破损或死亡的细胞、没捕获到细胞的empty droplet和捕获到2个以上细胞的doublets。一般低质量的细胞或者empty droplet通常含有很少的基因，而doublets容易测到更多的基因。另一方面，低质量或者死亡细胞会测到更多的线粒体基因表达的RNA。
使用PercentageFeatureSet函数评估每个细胞中的线粒体表达比例：</p>
<div class="highlight"><pre><span></span><code><span class="c1">###### step2:QC质控 ######</span>
<span class="nf">dir.create</span><span class="p">(</span><span class="s">&quot;./1-QC&quot;</span><span class="p">)</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;./1-QC&quot;</span><span class="p">)</span>
<span class="c1"># sce.all=readRDS(&quot;../sce.all_raw.rds&quot;)</span>
<span class="c1">#计算线粒体基因比例</span>
<span class="c1"># 人和鼠的基因名字稍微不一样</span>


<span class="n">mito_genes</span><span class="o">=</span><span class="nf">rownames</span><span class="p">(</span><span class="n">sce.all</span><span class="p">)[</span><span class="nf">grep</span><span class="p">(</span><span class="s">&quot;^mt-&quot;</span><span class="p">,</span> <span class="nf">rownames</span><span class="p">(</span><span class="n">sce.all</span><span class="p">))]</span> 
<span class="n">mito_genes</span> <span class="c1">#13个线粒体基因</span>
<span class="n">sce.all</span><span class="o">=</span><span class="nf">PercentageFeatureSet</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="s">&quot;^mt-&quot;</span><span class="p">,</span> <span class="n">col.name</span> <span class="o">=</span> <span class="s">&quot;percent_mito&quot;</span><span class="p">)</span>
<span class="c1">#计算核糖体基因比例</span>
<span class="n">ribo_genes</span><span class="o">=</span><span class="nf">rownames</span><span class="p">(</span><span class="n">sce.all</span><span class="p">)[</span><span class="nf">grep</span><span class="p">(</span><span class="s">&quot;^Rp[sl]&quot;</span><span class="p">,</span> <span class="nf">rownames</span><span class="p">(</span><span class="n">sce.all</span><span class="p">))]</span>
<span class="n">ribo_genes</span>
<span class="n">sce.all</span><span class="o">=</span><span class="nf">PercentageFeatureSet</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="s">&quot;^Rp[sl]&quot;</span><span class="p">,</span> <span class="n">col.name</span> <span class="o">=</span> <span class="s">&quot;percent_ribo&quot;</span><span class="p">)</span>
<span class="c1">#计算红血细胞基因比例</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">sce.all</span><span class="p">)[</span><span class="nf">grep</span><span class="p">(</span><span class="s">&quot;^Hb[^(p)]&quot;</span><span class="p">,</span> <span class="nf">rownames</span><span class="p">(</span><span class="n">sce.all</span><span class="p">))]</span>
<span class="n">sce.all</span><span class="o">=</span><span class="nf">PercentageFeatureSet</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="s">&quot;^Hb[^(p)]&quot;</span><span class="p">,</span> <span class="n">col.name</span> <span class="o">=</span> <span class="s">&quot;percent_hb&quot;</span><span class="p">)</span>

<span class="c1">#可视化细胞的上述比例情况</span>
<span class="n">feats</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;nFeature_RNA&quot;</span><span class="p">,</span> <span class="s">&quot;nCount_RNA&quot;</span><span class="p">,</span> <span class="s">&quot;percent_mito&quot;</span><span class="p">,</span> <span class="s">&quot;percent_ribo&quot;</span><span class="p">,</span> <span class="s">&quot;percent_hb&quot;</span><span class="p">)</span>
<span class="n">feats</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;nFeature_RNA&quot;</span><span class="p">,</span> <span class="s">&quot;nCount_RNA&quot;</span><span class="p">)</span>
<span class="n">p1</span><span class="o">=</span><span class="nf">VlnPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;orig.ident&quot;</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="n">feats</span><span class="p">,</span> <span class="n">pt.size</span> <span class="o">=</span> <span class="m">0.01</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span> <span class="o">+</span> 
    <span class="nf">NoLegend</span><span class="p">()</span>
<span class="n">p1</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s">&quot;Vlnplot1.pdf&quot;</span><span class="p">,</span><span class="n">plot</span><span class="o">=</span><span class="n">p1</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s">&quot;Vlnplot1.png&quot;</span><span class="p">,</span><span class="n">plot</span><span class="o">=</span><span class="n">p1</span><span class="p">)</span>
<span class="n">feats</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;percent_mito&quot;</span><span class="p">,</span> <span class="s">&quot;percent_ribo&quot;</span><span class="p">,</span> <span class="s">&quot;percent_hb&quot;</span><span class="p">)</span>
<span class="n">p2</span><span class="o">=</span><span class="nf">VlnPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;orig.ident&quot;</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="n">feats</span><span class="p">,</span> <span class="n">pt.size</span> <span class="o">=</span> <span class="m">0.01</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">3</span><span class="p">,</span> <span class="n">same.y.lims</span><span class="o">=</span><span class="bp">T</span><span class="p">)</span> <span class="o">+</span> 
    <span class="nf">scale_y_continuous</span><span class="p">(</span><span class="n">breaks</span><span class="o">=</span><span class="nf">seq</span><span class="p">(</span><span class="m">0</span><span class="p">,</span> <span class="m">100</span><span class="p">,</span> <span class="m">5</span><span class="p">))</span> <span class="o">+</span>
    <span class="nf">NoLegend</span><span class="p">()</span>
<span class="n">p2</span>  
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s">&quot;Vlnplot2.pdf&quot;</span><span class="p">,</span><span class="n">plot</span><span class="o">=</span><span class="n">p2</span><span class="p">)</span>

<span class="n">p3</span><span class="o">=</span><span class="nf">FeatureScatter</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="s">&quot;nCount_RNA&quot;</span><span class="p">,</span> <span class="s">&quot;nFeature_RNA&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;orig.ident&quot;</span><span class="p">,</span> <span class="n">pt.size</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s">&quot;Scatterplot.pdf&quot;</span><span class="p">,</span><span class="n">plot</span><span class="o">=</span><span class="n">p3</span><span class="p">)</span>
<span class="c1">#根据上述指标，过滤低质量细胞/基因</span>
<span class="c1">#过滤指标1:最少表达基因数的细胞&amp;最少表达细胞数的基因</span>
<span class="n">selected_c</span> <span class="o">&lt;-</span> <span class="nf">WhichCells</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">expression</span> <span class="o">=</span> <span class="n">nFeature_RNA</span> <span class="o">&gt;</span> <span class="m">300</span><span class="p">)</span>
<span class="n">selected_f</span> <span class="o">&lt;-</span> <span class="nf">rownames</span><span class="p">(</span><span class="n">sce.all</span><span class="p">)[</span><span class="n">Matrix</span><span class="o">::</span><span class="nf">rowSums</span><span class="p">(</span><span class="n">sce.all</span><span class="o">@</span><span class="n">assays</span><span class="o">$</span><span class="n">RNA</span><span class="o">@</span><span class="n">counts</span> <span class="o">&gt;</span> <span class="m">0</span> <span class="p">)</span> <span class="o">&gt;</span> <span class="m">3</span><span class="p">]</span>

<span class="n">sce.all.filt</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="n">selected_f</span><span class="p">,</span> <span class="n">cells</span> <span class="o">=</span> <span class="n">selected_c</span><span class="p">)</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">sce.all</span><span class="p">)</span> 
<span class="nf">dim</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">)</span> 
<span class="nf">table</span><span class="p">(</span><span class="n">sce.all</span><span class="o">@</span><span class="n">meta.data</span><span class="o">$</span><span class="n">orig.ident</span><span class="p">)</span> 
<span class="nf">table</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="o">@</span><span class="n">meta.data</span><span class="o">$</span><span class="n">orig.ident</span><span class="p">)</span> 
<span class="c1">#  可以看到，主要是过滤了基因，其次才是细胞</span>

<span class="c1"># par(mar = c(4, 8, 2, 1))</span>
<span class="n">C</span><span class="o">=</span><span class="n">sce.all.filt</span><span class="o">@</span><span class="n">assays</span><span class="o">$</span><span class="n">RNA</span><span class="o">@</span><span class="n">counts</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">C</span><span class="p">)</span>
<span class="n">C</span><span class="o">=</span><span class="n">Matrix</span><span class="o">::</span><span class="nf">t</span><span class="p">(</span><span class="n">Matrix</span><span class="o">::</span><span class="nf">t</span><span class="p">(</span><span class="n">C</span><span class="p">)</span><span class="o">/</span><span class="n">Matrix</span><span class="o">::</span><span class="nf">colSums</span><span class="p">(</span><span class="n">C</span><span class="p">))</span> <span class="o">*</span> <span class="m">100</span>
<span class="c1"># 这里的C 这个矩阵，有一点大，可以考虑随抽样</span>
<span class="n">C</span><span class="o">=</span><span class="n">C</span><span class="p">[,</span><span class="nf">sample</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">C</span><span class="p">),</span><span class="m">1000</span><span class="p">)]</span>
<span class="n">most_expressed</span> <span class="o">&lt;-</span> <span class="nf">order</span><span class="p">(</span><span class="nf">apply</span><span class="p">(</span><span class="n">C</span><span class="p">,</span> <span class="m">1</span><span class="p">,</span> <span class="n">median</span><span class="p">),</span> <span class="n">decreasing</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)[</span><span class="m">50</span><span class="o">:</span><span class="m">1</span><span class="p">]</span>
<span class="nf">pdf</span><span class="p">(</span><span class="s">&quot;TOP50_most_expressed_gene.pdf&quot;</span><span class="p">,</span><span class="n">width</span><span class="o">=</span><span class="m">14</span><span class="p">)</span>
<span class="nf">boxplot</span><span class="p">(</span><span class="nf">as.matrix</span><span class="p">(</span><span class="n">Matrix</span><span class="o">::</span><span class="nf">t</span><span class="p">(</span><span class="n">C</span><span class="p">[</span><span class="n">most_expressed</span><span class="p">,</span> <span class="p">])),</span>
        <span class="n">cex</span> <span class="o">=</span> <span class="m">0.1</span><span class="p">,</span> <span class="n">las</span> <span class="o">=</span> <span class="m">1</span><span class="p">,</span> 
        <span class="n">xlab</span> <span class="o">=</span> <span class="s">&quot;% total count per cell&quot;</span><span class="p">,</span> 
    <span class="n">col</span> <span class="o">=</span> <span class="p">(</span><span class="n">scales</span><span class="o">::</span><span class="nf">hue_pal</span><span class="p">())(</span><span class="m">50</span><span class="p">)[</span><span class="m">50</span><span class="o">:</span><span class="m">1</span><span class="p">],</span> 
    <span class="n">horizontal</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="nf">dev.off</span><span class="p">()</span>
<span class="nf">rm</span><span class="p">(</span><span class="n">C</span><span class="p">)</span>

<span class="c1">#过滤指标2:线粒体/核糖体基因比例(根据上面的violin图)</span>
<span class="n">selected_mito</span> <span class="o">&lt;-</span> <span class="nf">WhichCells</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">,</span> <span class="n">expression</span> <span class="o">=</span> <span class="n">percent_mito</span> <span class="o">&lt;</span> <span class="m">15</span><span class="p">)</span>
<span class="n">selected_ribo</span> <span class="o">&lt;-</span> <span class="nf">WhichCells</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">,</span> <span class="n">expression</span> <span class="o">=</span> <span class="n">percent_ribo</span> <span class="o">&gt;</span> <span class="m">3</span><span class="p">)</span>
<span class="n">selected_hb</span> <span class="o">&lt;-</span> <span class="nf">WhichCells</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">,</span> <span class="n">expression</span> <span class="o">=</span> <span class="n">percent_hb</span> <span class="o">&lt;</span> <span class="m">0.1</span><span class="p">)</span>
<span class="nf">length</span><span class="p">(</span><span class="n">selected_hb</span><span class="p">)</span>
<span class="nf">length</span><span class="p">(</span><span class="n">selected_ribo</span><span class="p">)</span>
<span class="nf">length</span><span class="p">(</span><span class="n">selected_mito</span><span class="p">)</span>


<span class="n">sce.all.filt</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">,</span> <span class="n">cells</span> <span class="o">=</span> <span class="n">selected_mito</span><span class="p">)</span>
<span class="n">sce.all.filt</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">,</span> <span class="n">cells</span> <span class="o">=</span> <span class="n">selected_ribo</span><span class="p">)</span>
<span class="n">sce.all.filt</span> <span class="o">&lt;-</span> <span class="nf">subset</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">,</span> <span class="n">cells</span> <span class="o">=</span> <span class="n">selected_hb</span><span class="p">)</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">)</span>

<span class="nf">table</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="o">$</span><span class="n">orig.ident</span><span class="p">)</span> 
<span class="c1">#可视化过滤后的情况</span>
<span class="n">feats</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;nFeature_RNA&quot;</span><span class="p">,</span> <span class="s">&quot;nCount_RNA&quot;</span><span class="p">)</span>
<span class="n">p1_filtered</span><span class="o">=</span><span class="nf">VlnPlot</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;orig.ident&quot;</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="n">feats</span><span class="p">,</span> <span class="n">pt.size</span> <span class="o">=</span> <span class="m">0.1</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">)</span> <span class="o">+</span> 
    <span class="nf">NoLegend</span><span class="p">()</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s">&quot;Vlnplot1_filtered.pdf&quot;</span><span class="p">,</span><span class="n">plot</span><span class="o">=</span><span class="n">p1_filtered</span><span class="p">)</span>

<span class="n">feats</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;percent_mito&quot;</span><span class="p">,</span> <span class="s">&quot;percent_ribo&quot;</span><span class="p">,</span> <span class="s">&quot;percent_hb&quot;</span><span class="p">)</span>
<span class="n">p2_filtered</span><span class="o">=</span><span class="nf">VlnPlot</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;orig.ident&quot;</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="n">feats</span><span class="p">,</span> <span class="n">pt.size</span> <span class="o">=</span> <span class="m">0.1</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="m">3</span><span class="p">)</span> <span class="o">+</span> 
    <span class="nf">NoLegend</span><span class="p">()</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s">&quot;Vlnplot2_filtered.pdf&quot;</span><span class="p">,</span><span class="n">plot</span><span class="o">=</span><span class="n">p2_filtered</span><span class="p">)</span>

<span class="nf">dim</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">)</span>  
</code></pre></div>
<p>pattern = "^MT-"表示线粒体基因的匹配模式，人的线粒体基因以MT-为前缀，小鼠的以Mt-为前缀。
Seurat对象在生成时，已经将每个细胞表达的基因数和reads数记录在meta.data的nFeature_RNA和nCount_RNA中。我们使用VlnPlot函数将nFeature_RNA、nCount_RNA和percent.mt三个指标进行可视化。</p>
<h2 id="3">3.数据标准化<a class="headerlink" href="#3" title="Permanent link">&para;</a></h2>
<p>基因的表达矩阵需要经过标准化后才能进行后续的分析。使用NormalizeData函数进行标准化，normalization.method：标准化方法，可选项，</p>
<ul>
<li>logNormalize： log(每个细胞的特征counts/细胞count总数*scale.factor)</li>
<li>CLR：中心化对数标准化</li>
<li>RC：相对计数，计算公式同logNormalize，不取log，计算CPM表达量，设置scale.factor=1e6</li>
</ul>
<p>可选择LogNormalize：</p>
<div class="highlight"><pre><span></span><code><span class="n">sce.all.filt</span> <span class="o">=</span> <span class="nf">NormalizeData</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">)</span>
</code></pre></div>
<h2 id="4">4.细胞周期分析<a class="headerlink" href="#4" title="Permanent link">&para;</a></h2>
<p>A cell cycle is a series of events that takes place in a cell as it grows and divides.即描述细胞生长、分裂整个过程中细胞变化过程。最重要的两个特点就是DNA复制、分裂成两个一样的子细胞。
如下图，一般分成4个阶段</p>
<ul>
<li>G1(gap1)：Cell increases in size(Cellular contents duplicated)</li>
<li>S(synthesis) ：DNA replication, each of the 46 chromosomes (23 pairs) is replicated by the cell</li>
<li>G2(gap2)：Cell grows more，organelles and proteins develop in preparation for cell division，为分裂做准备</li>
<li>M(mitosis)：'Old' cell partitions the two copies of the genetic material into the two daughter cells. And the cell cycle can begin again.</li>
</ul>
<p><img alt="" src="../Figure/scRNASeq_Down_003.jpg" />
在分析单细胞数据时，同一类型的细胞往往来自于不同的细胞周期阶段，这可能对下游聚类分析，细胞类型注释产生混淆；由于细胞周期也是通过cell cycle related protein 调控，即每个阶段有显著的marker基因；通过分析细胞周期有关基因的表达情况，可以对细胞所处周期阶段进行注释；
在单细胞周期分析时，通常只考虑三个阶段：G1、S、G2M。(即把G2和M当做一个phase)
主要学习两种来自scran与Seurat包鉴定细胞周期的方法介绍与演示。</p>
<h3 id="1seurat">(1).Seurat包<a class="headerlink" href="#1seurat" title="Permanent link">&para;</a></h3>
<p>Seurat包CellCycleScoring较scran包cyclone函数最主要的区别是直接根据每个cycle，一组marker基因表达值判断。</p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">Seurat</span><span class="p">)</span>
<span class="nf">str</span><span class="p">(</span><span class="n">cc.genes</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_004.jpg" />
如上是Seurat包提供的人的细胞中分别与S期、G2M期直接相关的marker基因。
CellCycleScoring即根据此，对每个细胞的S期、G2M期可能性进行打分；具体如何计算的，暂时在Seurat官方文档中没有提及。在satijalab的github(<a href="https://github.com/satijalab/seurat/issues/728">https://github.com/satijalab/seurat/issues/728)</a>)中，作者这样回复类似的提问：
As we say in the vignette, the scores are computed using an algorithm developed by Itay Tirosh, when he was a postdoc in the Regev Lab (Science et al., 2016). The gene sets are also taken from his work.
You can read the methods section of <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4944528/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4944528/</a> to see how the scoring works, but essentially the method scores cells based on the expression of each gene in a signature set - after controlling for the expected expression of genes with similar abundance. 
结合一些中文教程（https://www.jianshu.com/p/e4a5b5c67de1)的介绍，认为就是根据每个细胞的S期(或者G2/M期)基因集是否显著高表达，对应的score就是表示在该细胞中，S期(或者G2/M期)基因集高表达的程度(如果是负数，就认为不属于该phase)。
<img alt="" src="E:\PersonalNotes\note\docs\BioInfo\scRNAseq\Figure/scRNASeq_Down_005.jpg" />
区别于scran包的另外重要的一点就是Seurat包仅提供了人类细胞有关的cell cycle related gene，没有小鼠的。对此，作者这样回复：<code>Thanks. We don't provide different capitalizations because this gene list was developed on a human dataset, and we don't want to create ambiguity by suggesting its created from a mouse reference dataset. **In practice however, we've found it works quite well for mouse also, and recommend the solution above.**</code>
简言之，作者认为可以将对应人的cc.gene转换为鼠对应的基因名，当做后者的cell cycle related gene（因为鼠和人类基因的高度相似性)。提到的solution就是采用biomaRt包转换一下。这在我之前的教程中有介绍。</p>
<div class="highlight"><pre><span></span><code><span class="n">convertHumanGeneList</span> <span class="o">&lt;-</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
  <span class="nf">require</span><span class="p">(</span><span class="s">&quot;biomaRt&quot;</span><span class="p">)</span>
  <span class="n">human</span> <span class="o">=</span> <span class="nf">useMart</span><span class="p">(</span><span class="s">&quot;ensembl&quot;</span><span class="p">,</span> <span class="n">dataset</span> <span class="o">=</span> <span class="s">&quot;hsapiens_gene_ensembl&quot;</span><span class="p">)</span>
  <span class="n">mouse</span> <span class="o">=</span> <span class="nf">useMart</span><span class="p">(</span><span class="s">&quot;ensembl&quot;</span><span class="p">,</span> <span class="n">dataset</span> <span class="o">=</span> <span class="s">&quot;mmusculus_gene_ensembl&quot;</span><span class="p">)</span>
  <span class="n">genesV2</span> <span class="o">=</span> <span class="nf">getLDS</span><span class="p">(</span><span class="n">attributes</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;hgnc_symbol&quot;</span><span class="p">),</span> <span class="n">filters</span> <span class="o">=</span> <span class="s">&quot;hgnc_symbol&quot;</span><span class="p">,</span> <span class="n">values</span> <span class="o">=</span> <span class="n">x</span> <span class="p">,</span> <span class="n">mart</span> <span class="o">=</span> <span class="n">human</span><span class="p">,</span> <span class="n">attributesL</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;mgi_symbol&quot;</span><span class="p">),</span> <span class="n">martL</span> <span class="o">=</span> <span class="n">mouse</span><span class="p">,</span> <span class="n">uniqueRows</span><span class="o">=</span><span class="bp">T</span><span class="p">)</span>
  <span class="n">humanx</span> <span class="o">&lt;-</span> <span class="nf">unique</span><span class="p">(</span><span class="n">genesV2</span><span class="p">[,</span> <span class="m">2</span><span class="p">])</span>
  <span class="c1"># Print the first 6 genes found to the screen</span>
  <span class="nf">print</span><span class="p">(</span><span class="nf">head</span><span class="p">(</span><span class="n">humanx</span><span class="p">))</span>
  <span class="nf">return</span><span class="p">(</span><span class="n">humanx</span><span class="p">)</span>
<span class="p">}</span>
<span class="n">m.s.genes</span> <span class="o">&lt;-</span> <span class="nf">convertHumanGeneList</span><span class="p">(</span><span class="n">cc.genes</span><span class="o">$</span><span class="n">s.genes</span><span class="p">)</span>
<span class="n">m.g2m.genes</span> <span class="o">&lt;-</span> <span class="nf">convertHumanGeneList</span><span class="p">(</span><span class="n">cc.genes</span><span class="o">$</span><span class="n">g2m.genes</span><span class="p">)</span>
</code></pre></div>
<p>但是biomaRt包网络不是很稳定，有人也直接提供了转换后的结果（<a href="https://github.com/satijalab/seurat/issues/462">https://github.com/satijalab/seurat/issues/462</a>），直接下载导入到R里即可，下面演示的代码就是用的该结果。</p>
<div class="highlight"><pre><span></span><code><span class="n">sce.all.filt</span> <span class="o">=</span> <span class="nf">NormalizeData</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">)</span><span class="c1">##数据标准化</span>
<span class="n">mouse.cc.gene</span><span class="o">=</span><span class="nf">readRDS</span><span class="p">(</span><span class="s">&quot;H:/MedBioInfoCloud/analysis/base_files/GeneSet/mouse_cell_cycle_genes/mouse_cell_cycle_genes/mouse_cell_cycle_genes.rds&quot;</span><span class="p">)</span>

<span class="n">sce.all.filt</span><span class="o">=</span><span class="nf">CellCycleScoring</span><span class="p">(</span><span class="n">object</span> <span class="o">=</span> <span class="n">sce.all.filt</span><span class="p">,</span> 
                              <span class="n">g2m.features</span> <span class="o">=</span> <span class="n">mouse.cc.gene</span><span class="o">$</span><span class="n">g2m.genes</span><span class="p">,</span> 
                              <span class="n">s.features</span> <span class="o">=</span> <span class="n">mouse.cc.gene</span><span class="o">$</span><span class="n">s.genes</span><span class="p">)</span>
<span class="n">p4</span><span class="o">=</span><span class="nf">VlnPlot</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;S.Score&quot;</span><span class="p">,</span> <span class="s">&quot;G2M.Score&quot;</span><span class="p">),</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;orig.ident&quot;</span><span class="p">,</span> 
           <span class="n">ncol</span> <span class="o">=</span> <span class="m">2</span><span class="p">,</span> <span class="n">pt.size</span> <span class="o">=</span> <span class="m">0.1</span><span class="p">)</span>



<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s">&quot;Vlnplot4_cycle.pdf&quot;</span><span class="p">,</span><span class="n">plot</span><span class="o">=</span><span class="n">p4</span><span class="p">)</span>

<span class="n">sce.all.filt</span><span class="o">@</span><span class="n">meta.data</span>  <span class="o">%&gt;%</span> <span class="nf">ggplot</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">S.Score</span><span class="p">,</span><span class="n">G2M.Score</span><span class="p">))</span><span class="o">+</span><span class="nf">geom_point</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="n">Phase</span><span class="p">))</span><span class="o">+</span>
  <span class="nf">theme_minimal</span><span class="p">()</span>

<span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s">&quot;cycle_details.pdf&quot;</span> <span class="p">)</span>
<span class="c1"># S.Score较高的为S期，G2M.Score较高的为G2M期，都比较低的为G1期</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="o">@</span><span class="n">meta.data</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="o">@</span><span class="n">meta.data</span><span class="p">[,</span><span class="m">7</span><span class="o">:</span><span class="m">9</span><span class="p">])</span>
<span class="nf">plot</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="o">$</span><span class="n">S.Score</span><span class="p">,</span><span class="n">sce.all.filt</span><span class="o">$</span><span class="n">G2M.Score</span><span class="p">,</span>
     <span class="n">col</span><span class="o">=</span><span class="nf">factor</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="o">$</span><span class="n">Phase</span><span class="p">),</span>
     <span class="n">main</span><span class="o">=</span><span class="s">&quot;CellCycleScoring&quot;</span><span class="p">)</span>
<span class="nf">legend</span><span class="p">(</span><span class="s">&quot;topleft&quot;</span><span class="p">,</span><span class="n">inset</span><span class="o">=</span><span class="n">.</span><span class="m">05</span><span class="p">,</span>
       <span class="n">title</span> <span class="o">=</span> <span class="s">&quot;cell cycle&quot;</span><span class="p">,</span>  
       <span class="nf">c</span><span class="p">(</span><span class="s">&quot;G1&quot;</span><span class="p">,</span><span class="s">&quot;S&quot;</span><span class="p">,</span><span class="s">&quot;G2M&quot;</span><span class="p">),</span> <span class="n">pch</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">),</span><span class="n">col</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="s">&quot;green&quot;</span><span class="p">,</span><span class="s">&quot;red&quot;</span><span class="p">))</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_006.jpg" /></p>
<h3 id="2scran">(2).scran包<a class="headerlink" href="#2scran" title="Permanent link">&para;</a></h3>
<p><a href="http://bioconductor.org/packages/release/bioc/manuals/scran/man/scran.pdf">http://bioconductor.org/packages/release/bioc/manuals/scran/man/scran.pdf</a>
scran包cyclone函数是利用‘marker基因对’表达来对细胞所在周期阶段进行预测的方法Scialdone (2015)
“maker基因对”由作者根据训练集细胞（已注释了cell cycle）的基因表达特征产生，我们可以直接使用。对于每一细胞周期阶段（人/鼠）都有一组“maker基因对”集合。</p>
<div class="highlight"><pre><span></span><code><span class="c1"># BiocManager::install(&quot;scran&quot;)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">scran</span><span class="p">)</span>
<span class="n">mm.pairs</span> <span class="o">&lt;-</span> <span class="nf">readRDS</span><span class="p">(</span><span class="nf">system.file</span><span class="p">(</span><span class="s">&quot;exdata&quot;</span><span class="p">,</span> <span class="s">&quot;mouse_cycle_markers.rds&quot;</span><span class="p">,</span> 
                                <span class="n">package</span><span class="o">=</span><span class="s">&quot;scran&quot;</span><span class="p">))</span>

<span class="nf">str</span><span class="p">(</span><span class="n">mm.pairs</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">mm.pairs</span><span class="o">$</span><span class="n">G1</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_007.jpg" />
如上图，具体来说，即比较某个细胞的ENSMUSG00000000001基因表达值是否大于ENSMUSG00000001785基因表达值。
如果对于所有G1期marker基因对，某个细胞的“first”列基因表达量大于对应的“second”基因的情况越多，则越有把握认为该细胞就是处于G1期（因为越符合训练集特征)。
而cyclone则是通过计算score，即对于某个细胞，符合上述比较关系的marker基因对数占全部marker基因对数的比值。
这里默认提供marker基因对是ensemble格式，如果表达数据提供的是其它类类型的基因ID，比如：SYMBOL，那么我们需要转化一下ID。</p>
<div class="highlight"><pre><span></span><code><span class="c1">###基因转换</span>
<span class="nf">library</span><span class="p">(</span><span class="n">clusterProfiler</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">org.Mm.eg.db</span><span class="p">)</span>
<span class="c1"># x &lt;- names(mm.pairs)[1]</span>
<span class="n">trs</span> <span class="o">&lt;-</span> <span class="nf">lapply</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">mm.pairs</span><span class="p">),</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
  <span class="n">df</span> <span class="o">&lt;-</span> <span class="n">mm.pairs</span><span class="p">[[</span><span class="n">x</span><span class="p">]]</span>
  <span class="n">first</span> <span class="o">&lt;-</span> <span class="nf">bitr</span><span class="p">(</span><span class="n">mm.pairs</span><span class="p">[[</span><span class="n">x</span><span class="p">]][,</span><span class="m">1</span><span class="p">],</span>
                   <span class="n">fromType</span><span class="o">=</span> <span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span> <span class="n">toType</span><span class="o">=</span><span class="s">&quot;SYMBOL&quot;</span><span class="p">,</span>
                   <span class="n">OrgDb</span><span class="o">=</span><span class="s">&quot;org.Mm.eg.db&quot;</span><span class="p">)</span>

  <span class="n">second</span> <span class="o">&lt;-</span> <span class="nf">bitr</span><span class="p">(</span><span class="n">mm.pairs</span><span class="p">[[</span><span class="n">x</span><span class="p">]][,</span><span class="m">2</span><span class="p">],</span>
                    <span class="n">fromType</span><span class="o">=</span> <span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span> <span class="n">toType</span><span class="o">=</span><span class="s">&quot;SYMBOL&quot;</span><span class="p">,</span>
                    <span class="n">OrgDb</span><span class="o">=</span><span class="s">&quot;org.Mm.eg.db&quot;</span><span class="p">)</span>
  <span class="n">df</span> <span class="o">&lt;-</span> <span class="n">df</span><span class="p">[</span><span class="n">first</span><span class="o">$</span><span class="n">ENSEMBL</span> <span class="o">%in%</span> <span class="n">df</span><span class="o">$</span><span class="n">first</span><span class="p">,]</span>
  <span class="n">df</span> <span class="o">&lt;-</span> <span class="n">df</span><span class="p">[</span><span class="n">second</span><span class="o">$</span><span class="n">ENSEMBL</span> <span class="o">%in%</span> <span class="n">df</span><span class="o">$</span><span class="n">second</span><span class="p">,]</span>

  <span class="n">df</span><span class="o">$</span><span class="n">first</span> <span class="o">&lt;-</span>  <span class="nf">lapply</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">first</span><span class="p">,</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
    <span class="n">first</span><span class="p">[</span><span class="n">first</span><span class="o">$</span><span class="n">ENSEMBL</span> <span class="o">==</span> <span class="n">x</span><span class="p">,</span><span class="m">2</span><span class="p">][</span><span class="m">1</span><span class="p">]</span> 
  <span class="p">})</span>  <span class="o">%&gt;%</span> <span class="nf">unlist</span><span class="p">()</span>

  <span class="n">df</span><span class="o">$</span><span class="n">second</span> <span class="o">&lt;-</span>  <span class="nf">lapply</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">second</span><span class="p">,</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
    <span class="n">second</span><span class="p">[</span><span class="n">second</span><span class="o">$</span><span class="n">ENSEMBL</span> <span class="o">==</span> <span class="n">x</span><span class="p">,</span><span class="m">2</span><span class="p">][</span><span class="m">1</span><span class="p">]</span> 
  <span class="p">})</span>  <span class="o">%&gt;%</span> <span class="nf">unlist</span><span class="p">()</span>

  <span class="nf">return</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="p">})</span>
<span class="nf">names</span><span class="p">(</span><span class="n">trs</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">names</span><span class="p">(</span><span class="n">mm.pairs</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1"># BiocManager::install(&quot;scran&quot;)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">scran</span><span class="p">)</span>
<span class="n">hs.pairs</span> <span class="o">&lt;-</span> <span class="nf">readRDS</span><span class="p">(</span><span class="nf">system.file</span><span class="p">(</span><span class="s">&quot;exdata&quot;</span><span class="p">,</span> <span class="s">&quot;human_cycle_markers.rds&quot;</span><span class="p">,</span> 
                                <span class="n">package</span><span class="o">=</span><span class="s">&quot;scran&quot;</span><span class="p">))</span>

<span class="nf">str</span><span class="p">(</span><span class="n">hs.pairs</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">hs.pairs</span><span class="o">$</span><span class="n">G1</span><span class="p">)</span>

<span class="c1">###基因转换</span>
<span class="nf">library</span><span class="p">(</span><span class="n">clusterProfiler</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">org.Hs.eg.db</span><span class="p">)</span>
<span class="c1"># x &lt;- names(hs.pairs)[1]</span>
<span class="n">htrs</span> <span class="o">&lt;-</span> <span class="nf">lapply</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">hs.pairs</span><span class="p">),</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
  <span class="n">df</span> <span class="o">&lt;-</span> <span class="n">hs.pairs</span><span class="p">[[</span><span class="n">x</span><span class="p">]]</span>
  <span class="n">first</span> <span class="o">&lt;-</span> <span class="nf">bitr</span><span class="p">(</span><span class="n">hs.pairs</span><span class="p">[[</span><span class="n">x</span><span class="p">]][,</span><span class="m">1</span><span class="p">],</span>
                <span class="n">fromType</span><span class="o">=</span> <span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span> <span class="n">toType</span><span class="o">=</span><span class="s">&quot;SYMBOL&quot;</span><span class="p">,</span>
                <span class="n">OrgDb</span><span class="o">=</span><span class="s">&quot;org.Hs.eg.db&quot;</span><span class="p">)</span>

  <span class="n">second</span> <span class="o">&lt;-</span> <span class="nf">bitr</span><span class="p">(</span><span class="n">hs.pairs</span><span class="p">[[</span><span class="n">x</span><span class="p">]][,</span><span class="m">2</span><span class="p">],</span>
                 <span class="n">fromType</span><span class="o">=</span> <span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span> <span class="n">toType</span><span class="o">=</span><span class="s">&quot;SYMBOL&quot;</span><span class="p">,</span>
                 <span class="n">OrgDb</span><span class="o">=</span><span class="s">&quot;org.Hs.eg.db&quot;</span><span class="p">)</span>
  <span class="n">df</span> <span class="o">&lt;-</span> <span class="n">df</span><span class="p">[</span><span class="n">first</span><span class="o">$</span><span class="n">ENSEMBL</span> <span class="o">%in%</span> <span class="n">df</span><span class="o">$</span><span class="n">first</span><span class="p">,]</span>
  <span class="n">df</span> <span class="o">&lt;-</span> <span class="n">df</span><span class="p">[</span><span class="n">second</span><span class="o">$</span><span class="n">ENSEMBL</span> <span class="o">%in%</span> <span class="n">df</span><span class="o">$</span><span class="n">second</span><span class="p">,]</span>

  <span class="n">df</span><span class="o">$</span><span class="n">first</span> <span class="o">&lt;-</span>  <span class="nf">lapply</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">first</span><span class="p">,</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
    <span class="n">first</span><span class="p">[</span><span class="n">first</span><span class="o">$</span><span class="n">ENSEMBL</span> <span class="o">==</span> <span class="n">x</span><span class="p">,</span><span class="m">2</span><span class="p">][</span><span class="m">1</span><span class="p">]</span> 
  <span class="p">})</span>  <span class="o">%&gt;%</span> <span class="nf">unlist</span><span class="p">()</span>

  <span class="n">df</span><span class="o">$</span><span class="n">second</span> <span class="o">&lt;-</span>  <span class="nf">lapply</span><span class="p">(</span><span class="n">df</span><span class="o">$</span><span class="n">second</span><span class="p">,</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
    <span class="n">second</span><span class="p">[</span><span class="n">second</span><span class="o">$</span><span class="n">ENSEMBL</span> <span class="o">==</span> <span class="n">x</span><span class="p">,</span><span class="m">2</span><span class="p">][</span><span class="m">1</span><span class="p">]</span> 
  <span class="p">})</span>  <span class="o">%&gt;%</span> <span class="nf">unlist</span><span class="p">()</span>

  <span class="nf">return</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="p">})</span>
<span class="nf">names</span><span class="p">(</span><span class="n">htrs</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">names</span><span class="p">(</span><span class="n">hs.pairs</span><span class="p">)</span>
</code></pre></div>
<p>下面是计算三个周期阶段的scores：</p>
<div class="highlight"><pre><span></span><code><span class="n">assigned</span> <span class="o">&lt;-</span> <span class="nf">cyclone</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="o">@</span><span class="n">assays</span><span class="p">[[</span><span class="s">&quot;RNA&quot;</span><span class="p">]]</span><span class="o">@</span><span class="n">data</span><span class="p">,</span> <span class="n">pairs</span><span class="o">=</span><span class="n">trs</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">assigned</span><span class="o">$</span><span class="n">scores</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_008.jpg" />
根据每一个细胞对于三个周期阶段的scores，可进行判断；具体规则为 <strong>G1或G2M评分高于0.5的细胞分别被分配到G1期或G2M期</strong>。若G1 or G2M phases均小于0.5，则可判断为S期(虽然可以直接看S期的score)；若G1 or G2M phases均大于0.5，则 the higher score is used for assignment。</p>
<h2 id="5doublets">5.检测doublets<a class="headerlink" href="#5doublets" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="c1">#检测doublets </span>
<span class="n">sce.all.filt</span> <span class="o">=</span> <span class="nf">FindVariableFeatures</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">)</span>
<span class="n">sce.all.filt</span> <span class="o">=</span> <span class="nf">ScaleData</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">,</span> 
                         <span class="n">vars.to.regress</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;nFeature_RNA&quot;</span><span class="p">,</span> <span class="s">&quot;percent_mito&quot;</span><span class="p">))</span>
<span class="n">sce.all.filt</span> <span class="o">=</span> <span class="nf">RunPCA</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">,</span> <span class="n">npcs</span> <span class="o">=</span> <span class="m">20</span><span class="p">)</span>
<span class="n">sce.all.filt</span> <span class="o">=</span> <span class="nf">RunTSNE</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">,</span> <span class="n">npcs</span> <span class="o">=</span> <span class="m">20</span><span class="p">)</span>
<span class="n">sce.all.filt</span> <span class="o">=</span> <span class="nf">RunUMAP</span><span class="p">(</span><span class="n">sce.all.filt</span><span class="p">,</span> <span class="n">dims</span> <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">)</span>
<span class="c1"># define the expected number of doublet cellscells.</span>
</code></pre></div>
<h2 id="6">6.单细胞多样本整合<a class="headerlink" href="#6" title="Permanent link">&para;</a></h2>
<p>CCA和RPCA整合方法都是Seurat包内置并推荐使用的。区别在于：
CCA 方法非常适合在细胞类型保守的情况下识别锚点，但在不同实验处理下基因表达存在非常大的差异，</p>
<ul>
<li>简而言之，CCA适合多个样本具有大致相似的细胞类型分布，也适合跨物种的多样本整合。</li>
<li>值得注意的是，CCA方法可能会导致整合过度，即可能会消除多样本间的生物学差异；</li>
</ul>
<p>而RPCA方法的整合运行速度更快，整合的力度较CCA弱一些（更保守），</p>
<ul>
<li>适合多个样本有不同的细胞类型分布情况；</li>
<li>建议多样本数据均是来自同一个平台的，例如要整合的目的样本均来自10X测序平台；</li>
<li>如大样本量的单细胞数据集整合。</li>
</ul>
<div class="highlight"><pre><span></span><code><span class="c1">###### step3:合并2个数据集###### </span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&#39;../&#39;</span><span class="p">)</span>
<span class="nf">dir.create</span><span class="p">(</span><span class="s">&quot;2-int&quot;</span><span class="p">)</span>
<span class="nf">getwd</span><span class="p">()</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;2-int&quot;</span><span class="p">)</span>
<span class="c1"># sce.all=readRDS(&quot;../1-QC/sce.all_qc.rds&quot;)</span>
<span class="n">sce.all</span><span class="o">=</span><span class="n">sce.all.filt</span>
<span class="n">sce.all</span>
<span class="c1">#拆分为*个seurat子对象</span>
<span class="n">sce.all.list</span> <span class="o">&lt;-</span> <span class="nf">SplitObject</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">split.by</span> <span class="o">=</span> <span class="s">&quot;orig.ident&quot;</span><span class="p">)</span>
<span class="n">sce.all.list</span>

<span class="nf">for </span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">sce.all.list</span><span class="p">))</span> <span class="p">{</span>
    <span class="nf">print</span><span class="p">(</span><span class="n">i</span><span class="p">)</span>
    <span class="n">sce.all.list</span><span class="p">[[</span><span class="n">i</span><span class="p">]]</span> <span class="o">&lt;-</span> <span class="nf">NormalizeData</span><span class="p">(</span><span class="n">sce.all.list</span><span class="p">[[</span><span class="n">i</span><span class="p">]],</span> <span class="n">verbose</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
    <span class="n">sce.all.list</span><span class="p">[[</span><span class="n">i</span><span class="p">]]</span> <span class="o">&lt;-</span> <span class="nf">FindVariableFeatures</span><span class="p">(</span><span class="n">sce.all.list</span><span class="p">[[</span><span class="n">i</span><span class="p">]],</span> <span class="n">selection.method</span> <span class="o">=</span> <span class="s">&quot;vst&quot;</span><span class="p">,</span> 
        <span class="n">nfeatures</span> <span class="o">=</span> <span class="m">2000</span><span class="p">,</span> <span class="n">verbose</span> <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
<span class="p">}</span>
</code></pre></div>
<p>FindIntegrationAnchors函数中的reduction参数进行设置：
Dimensional reduction to perform when finding anchors. Can be one of:</p>
<ul>
<li>cca: Canonical correlation analysis</li>
<li>rpca: Reciprocal PCA</li>
<li>rlsi: Reciprocal LSI</li>
</ul>
<div class="highlight"><pre><span></span><code><span class="c1">#CCA</span>
<span class="n">alldata.anchors</span> <span class="o">&lt;-</span> <span class="nf">FindIntegrationAnchors</span><span class="p">(</span><span class="n">object.list</span> <span class="o">=</span> <span class="n">sce.all.list</span><span class="p">,</span> <span class="n">dims</span> <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">,</span> 
    <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;cca&quot;</span><span class="p">)</span>
<span class="c1">#Perform dataset integration using a pre-computed AnchorSet.</span>
<span class="n">sce.all.int</span> <span class="o">&lt;-</span> <span class="nf">IntegrateData</span><span class="p">(</span><span class="n">anchorset</span> <span class="o">=</span> <span class="n">alldata.anchors</span><span class="p">,</span> <span class="n">dims</span> <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">,</span> <span class="n">new.assay.name</span> <span class="o">=</span> <span class="s">&quot;CCA&quot;</span><span class="p">)</span>
<span class="nf">names</span><span class="p">(</span><span class="n">sce.all.int</span><span class="o">@</span><span class="n">assays</span><span class="p">)</span>
<span class="c1">#[1] &quot;RNA&quot; &quot;CCA&quot;</span>
<span class="n">sce.all.int</span><span class="o">@</span><span class="n">active.assay</span>
<span class="c1">#[1] &quot;CCA&quot;</span>
<span class="n">sce.all.int</span><span class="o">=</span><span class="nf">ScaleData</span><span class="p">(</span><span class="n">sce.all.int</span><span class="p">)</span>
<span class="n">sce.all.int</span><span class="o">=</span><span class="nf">RunPCA</span><span class="p">(</span><span class="n">sce.all.int</span><span class="p">,</span> <span class="n">npcs</span> <span class="o">=</span> <span class="m">30</span><span class="p">)</span>
<span class="n">sce.all.int</span><span class="o">=</span><span class="nf">RunTSNE</span><span class="p">(</span><span class="n">sce.all.int</span><span class="p">,</span> <span class="n">dims</span> <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">)</span>
<span class="n">sce.all.int</span><span class="o">=</span><span class="nf">RunUMAP</span><span class="p">(</span><span class="n">sce.all.int</span><span class="p">,</span> <span class="n">dims</span> <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">)</span>
<span class="nf">names</span><span class="p">(</span><span class="n">sce.all.int</span><span class="o">@</span><span class="n">reductions</span><span class="p">)</span>
<span class="nf">names</span><span class="p">(</span><span class="n">sce.all</span><span class="o">@</span><span class="n">reductions</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">sce.all</span><span class="o">@</span><span class="n">meta.data</span><span class="p">)</span> 

<span class="n">p1.compare</span><span class="o">=</span><span class="nf">plot_grid</span><span class="p">(</span><span class="n">ncol</span> <span class="o">=</span> <span class="m">3</span><span class="p">,</span>
  <span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;pca&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;orig.ident&quot;</span><span class="p">)</span><span class="o">+</span><span class="nf">NoAxes</span><span class="p">()</span><span class="o">+</span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;PCA raw_data&quot;</span><span class="p">),</span>
  <span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;tsne&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;orig.ident&quot;</span><span class="p">)</span><span class="o">+</span><span class="nf">NoAxes</span><span class="p">()</span><span class="o">+</span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;tSNE raw_data&quot;</span><span class="p">),</span>
  <span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;umap&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;orig.ident&quot;</span><span class="p">)</span><span class="o">+</span><span class="nf">NoAxes</span><span class="p">()</span><span class="o">+</span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;UMAP raw_data&quot;</span><span class="p">),</span>

  <span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all.int</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;pca&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;orig.ident&quot;</span><span class="p">)</span><span class="o">+</span><span class="nf">NoAxes</span><span class="p">()</span><span class="o">+</span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;PCA integrated&quot;</span><span class="p">),</span>
  <span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all.int</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;tsne&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;orig.ident&quot;</span><span class="p">)</span><span class="o">+</span><span class="nf">NoAxes</span><span class="p">()</span><span class="o">+</span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;tSNE integrated&quot;</span><span class="p">),</span>
  <span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all.int</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;umap&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;orig.ident&quot;</span><span class="p">)</span><span class="o">+</span><span class="nf">NoAxes</span><span class="p">()</span><span class="o">+</span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;UMAP integrated&quot;</span><span class="p">)</span>
<span class="p">)</span>
<span class="n">p1.compare</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">plot</span><span class="o">=</span><span class="n">p1.compare</span><span class="p">,</span><span class="n">filename</span><span class="o">=</span><span class="s">&quot;Before&amp;After_int.pdf&quot;</span><span class="p">)</span> 

<span class="nf">colnames</span><span class="p">(</span><span class="n">sce.all</span><span class="o">@</span><span class="n">meta.data</span><span class="p">)</span>
</code></pre></div>
<p>整合前后：
<img alt="image.png" src="../Figure/scRNASeq_Down_009.jpg" /></p>
<h2 id="7">7.聚类<a class="headerlink" href="#7" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="c1">###### cluster######</span>
<span class="n">sce.all.int</span>
<span class="n">sce.all</span>
<span class="n">sce.all</span><span class="o">=</span><span class="n">sce.all.int</span>
<span class="n">sce.all</span><span class="o">@</span><span class="n">active.assay</span>
<span class="n">sce.all</span><span class="o">=</span><span class="nf">FindNeighbors</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">dims</span> <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">30</span><span class="p">,</span> <span class="n">k.param</span> <span class="o">=</span> <span class="m">60</span><span class="p">,</span> <span class="n">prune.SNN</span> <span class="o">=</span> <span class="m">1</span><span class="o">/</span><span class="m">15</span><span class="p">)</span>
<span class="c1">#设置不同的分辨率，观察分群效果(选择哪一个？)</span>
<span class="nf">for </span><span class="p">(</span><span class="n">res</span> <span class="n">in</span> <span class="nf">c</span><span class="p">(</span><span class="m">0.01</span><span class="p">,</span> <span class="m">0.05</span><span class="p">,</span> <span class="m">0.1</span><span class="p">,</span> <span class="m">0.2</span><span class="p">,</span> <span class="m">0.3</span><span class="p">,</span> <span class="m">0.5</span><span class="p">,</span><span class="m">0.8</span><span class="p">,</span><span class="m">1</span><span class="p">))</span> <span class="p">{</span>
    <span class="n">sce.all</span><span class="o">=</span><span class="nf">FindClusters</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">graph.name</span> <span class="o">=</span> <span class="s">&quot;CCA_snn&quot;</span><span class="p">,</span> <span class="n">resolution</span> <span class="o">=</span> <span class="n">res</span><span class="p">,</span> <span class="n">algorithm</span> <span class="o">=</span> <span class="m">1</span><span class="p">)</span>
<span class="p">}</span>
<span class="nf">apply</span><span class="p">(</span><span class="n">sce.all</span><span class="o">@</span><span class="n">meta.data</span><span class="p">[,</span><span class="nf">grep</span><span class="p">(</span><span class="s">&quot;CCA_snn_res&quot;</span><span class="p">,</span><span class="nf">colnames</span><span class="p">(</span><span class="n">sce.all</span><span class="o">@</span><span class="n">meta.data</span><span class="p">))],</span><span class="m">2</span><span class="p">,</span><span class="n">table</span><span class="p">)</span>


<span class="n">p1_dim</span><span class="o">=</span><span class="nf">plot_grid</span><span class="p">(</span><span class="n">ncol</span> <span class="o">=</span> <span class="m">3</span><span class="p">,</span> <span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;umap&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;CCA_snn_res.0.01&quot;</span><span class="p">)</span> <span class="o">+</span> 
    <span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;louvain_0.01&quot;</span><span class="p">),</span> <span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;umap&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;CCA_snn_res.0.1&quot;</span><span class="p">)</span> <span class="o">+</span> 
    <span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;louvain_0.1&quot;</span><span class="p">),</span> <span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;umap&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;CCA_snn_res.0.2&quot;</span><span class="p">)</span> <span class="o">+</span> 
    <span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;louvain_0.2&quot;</span><span class="p">))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">plot</span><span class="o">=</span><span class="n">p1_dim</span><span class="p">,</span> <span class="n">filename</span><span class="o">=</span><span class="s">&quot;Dimplot_diff_resolution_low.pdf&quot;</span><span class="p">,</span><span class="n">width</span> <span class="o">=</span> <span class="m">14</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_010.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="n">p1_dim</span><span class="o">=</span><span class="nf">plot_grid</span><span class="p">(</span><span class="n">ncol</span> <span class="o">=</span> <span class="m">3</span><span class="p">,</span> <span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;umap&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;CCA_snn_res.0.8&quot;</span><span class="p">)</span> <span class="o">+</span> 
                   <span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;louvain_0.8&quot;</span><span class="p">),</span> <span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;umap&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;CCA_snn_res.1&quot;</span><span class="p">)</span> <span class="o">+</span> 
                   <span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;louvain_1&quot;</span><span class="p">),</span> <span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;umap&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;CCA_snn_res.0.3&quot;</span><span class="p">)</span> <span class="o">+</span> 
                   <span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;louvain_0.3&quot;</span><span class="p">))</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">plot</span><span class="o">=</span><span class="n">p1_dim</span><span class="p">,</span> <span class="n">filename</span><span class="o">=</span><span class="s">&quot;Dimplot_diff_resolution_high.pdf&quot;</span><span class="p">,</span><span class="n">width</span> <span class="o">=</span> <span class="m">18</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_011.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="n">p2_tree</span><span class="o">=</span><span class="nf">clustree</span><span class="p">(</span><span class="n">sce.all</span><span class="o">@</span><span class="n">meta.data</span><span class="p">,</span> <span class="n">prefix</span> <span class="o">=</span> <span class="s">&quot;CCA_snn_res.&quot;</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">plot</span><span class="o">=</span><span class="n">p2_tree</span><span class="p">,</span> <span class="n">filename</span><span class="o">=</span><span class="s">&quot;Tree_diff_resolution.pdf&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_012.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="c1">#接下来分析，按照分辨率为0.8进行 </span>
<span class="n">sel.clust</span> <span class="o">=</span> <span class="s">&quot;CCA_snn_res.0.8&quot;</span>
<span class="n">sce.all</span> <span class="o">&lt;-</span> <span class="nf">SetIdent</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">value</span> <span class="o">=</span> <span class="n">sel.clust</span><span class="p">)</span>
<span class="nf">table</span><span class="p">(</span><span class="n">sce.all</span><span class="o">@</span><span class="n">active.ident</span><span class="p">)</span> 
<span class="nf">saveRDS</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="s">&quot;sce.all_int.rds&quot;</span><span class="p">)</span>
</code></pre></div>
<h2 id="8">8.细胞类型注释<a class="headerlink" href="#8" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="nf">setwd</span><span class="p">(</span><span class="s">&#39;../&#39;</span><span class="p">)</span> 
<span class="nf">dir.create</span><span class="p">(</span><span class="s">&quot;3-cell&quot;</span><span class="p">)</span>

<span class="nf">setwd</span><span class="p">(</span><span class="s">&quot;3-cell&quot;</span><span class="p">)</span> 
<span class="c1">#sce.all=readRDS( &quot;../2-int/sce.all_int.rds&quot;)</span>
<span class="n">sce.all</span>
<span class="n">sce.all</span>
<span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;umap&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;seurat_clusters&quot;</span><span class="p">,</span><span class="n">label</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span> 
<span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;umap&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;CCA_snn_res.0.8&quot;</span><span class="p">,</span><span class="n">label</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span> 
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#39;umap_by_CCA_snn_res.0.8.pdf&#39;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_013.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="c1"># T Cells (CD3D, CD3E, CD8A), </span>
<span class="c1"># B cells (CD19, CD79A, MS4A1 [CD20]), </span>
<span class="c1"># Plasma cells (IGHG1, MZB1, SDC1, CD79A), </span>
<span class="c1"># Monocytes and macrophages (CD68, CD163, CD14),</span>
<span class="c1"># NK Cells (FGFBP2, FCG3RA, CX3CR1),  </span>
<span class="c1"># Photoreceptor cells (RCVRN), </span>
<span class="c1"># Fibroblasts (FGF7, MME), </span>
<span class="c1"># Endothelial cells (PECAM1, VWF). </span>
<span class="c1"># epi or tumor (EPCAM, KRT19, PROM1, ALDH1A1, CD24).</span>
<span class="c1"># immune (CD45+,PTPRC), epithelial/cancer (EpCAM+,EPCAM), </span>
<span class="c1"># stromal (CD10+,MME,fibo or CD31+,PECAM1,endo) </span>

<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span> 
<span class="n">genes_to_check</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#39;PTPRC&#39;</span><span class="p">,</span> <span class="s">&#39;CD3D&#39;</span><span class="p">,</span> <span class="s">&#39;CD3E&#39;</span><span class="p">,</span> <span class="s">&#39;CD4&#39;</span><span class="p">,</span><span class="s">&#39;CD8A&#39;</span><span class="p">,</span><span class="s">&#39;CD19&#39;</span><span class="p">,</span> <span class="s">&#39;CD79A&#39;</span><span class="p">,</span> <span class="s">&#39;MS4A1&#39;</span> <span class="p">,</span>
                   <span class="s">&#39;IGHG1&#39;</span><span class="p">,</span> <span class="s">&#39;MZB1&#39;</span><span class="p">,</span> <span class="s">&#39;SDC1&#39;</span><span class="p">,</span>
                   <span class="s">&#39;CD68&#39;</span><span class="p">,</span> <span class="s">&#39;CD163&#39;</span><span class="p">,</span> <span class="s">&#39;CD14&#39;</span><span class="p">,</span> 
                   <span class="s">&#39;TPSAB1&#39;</span> <span class="p">,</span> <span class="s">&#39;TPSB2&#39;</span><span class="p">,</span>  <span class="c1"># mast cells,</span>
                   <span class="s">&#39;MKI67&#39;</span><span class="p">,</span><span class="s">&#39;TOP2A&#39;</span><span class="p">,</span><span class="s">&#39;KLRC1&#39;</span><span class="p">,</span>
                   <span class="s">&#39;RCVRN&#39;</span><span class="p">,</span><span class="s">&#39;FPR1&#39;</span> <span class="p">,</span> <span class="s">&#39;ITGAM&#39;</span> <span class="p">,</span>
                   <span class="s">&#39;FGF7&#39;</span><span class="p">,</span><span class="s">&#39;MME&#39;</span><span class="p">,</span> <span class="s">&#39;ACTA2&#39;</span><span class="p">,</span>
                   <span class="s">&#39;PECAM1&#39;</span><span class="p">,</span> <span class="s">&#39;VWF&#39;</span><span class="p">,</span>    
                   <span class="s">&#39;KLRB1&#39;</span><span class="p">,</span><span class="s">&#39;NCR1&#39;</span><span class="p">,</span> <span class="c1"># NK </span>
                   <span class="s">&#39;EPCAM&#39;</span> <span class="p">,</span> <span class="s">&#39;KRT19&#39;</span><span class="p">,</span> <span class="s">&#39;PROM1&#39;</span><span class="p">,</span> <span class="s">&#39;ALDH1A1&#39;</span><span class="p">,</span>
                   <span class="s">&#39;MKI67&#39;</span> <span class="p">,</span><span class="s">&#39;TOP2A&#39;</span> <span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>  
<span class="n">genes_to_check</span> <span class="o">=</span> <span class="nf">str_to_title</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">genes_to_check</span><span class="p">))</span>
<span class="n">genes_to_check</span>
<span class="n">p_all_markers</span> <span class="o">&lt;-</span> <span class="nf">DotPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="n">genes_to_check</span><span class="p">,</span>
             <span class="n">assay</span><span class="o">=</span><span class="s">&#39;RNA&#39;</span>  <span class="p">)</span>  <span class="o">+</span> <span class="nf">coord_flip</span><span class="p">()</span>

<span class="n">p_all_markers</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">plot</span><span class="o">=</span><span class="n">p_all_markers</span><span class="p">,</span> <span class="n">filename</span><span class="o">=</span><span class="s">&quot;check_all_marker_by_seurat_cluster.pdf&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_014.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="n">genes_to_check</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#39;PTPRC&#39;</span><span class="p">,</span> <span class="s">&#39;CD3D&#39;</span><span class="p">,</span> <span class="s">&#39;CD3E&#39;</span><span class="p">,</span> <span class="s">&#39;CD4&#39;</span><span class="p">,</span><span class="s">&#39;CD8A&#39;</span><span class="p">,</span>
                   <span class="s">&#39;CCR7&#39;</span><span class="p">,</span> <span class="s">&#39;SELL&#39;</span> <span class="p">,</span> <span class="s">&#39;TCF7&#39;</span><span class="p">,</span><span class="s">&#39;CXCR6&#39;</span> <span class="p">,</span> <span class="s">&#39;ITGA1&#39;</span><span class="p">,</span>
                   <span class="s">&#39;FOXP3&#39;</span><span class="p">,</span> <span class="s">&#39;IL2RA&#39;</span><span class="p">,</span>  <span class="s">&#39;CTLA4&#39;</span><span class="p">,</span><span class="s">&#39;GZMB&#39;</span><span class="p">,</span> <span class="s">&#39;GZMK&#39;</span><span class="p">,</span><span class="s">&#39;CCL5&#39;</span><span class="p">,</span>
                   <span class="s">&#39;IFNG&#39;</span><span class="p">,</span> <span class="s">&#39;CCL4&#39;</span><span class="p">,</span> <span class="s">&#39;CCL3&#39;</span> <span class="p">,</span>
                   <span class="s">&#39;PRF1&#39;</span> <span class="p">,</span> <span class="s">&#39;NKG7&#39;</span><span class="p">)</span> 
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>  
<span class="n">genes_to_check</span><span class="o">=</span><span class="nf">str_to_title</span><span class="p">(</span><span class="n">genes_to_check</span><span class="p">)</span>
<span class="n">genes_to_check</span>
<span class="n">p</span> <span class="o">&lt;-</span> <span class="nf">DotPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="n">genes_to_check</span><span class="p">,</span>
             <span class="n">assay</span><span class="o">=</span><span class="s">&#39;RNA&#39;</span>  <span class="p">)</span>  <span class="o">+</span> <span class="nf">coord_flip</span><span class="p">()</span>

<span class="n">p</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">plot</span><span class="o">=</span><span class="n">p</span><span class="p">,</span> <span class="n">filename</span><span class="o">=</span><span class="s">&quot;check_Tcells_marker_by_seurat_cluster.pdf&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_015.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="c1"># mast cells, TPSAB1 and TPSB2 </span>
<span class="c1"># B cell,  CD79A  and MS4A1 (CD20) </span>
<span class="c1"># naive B cells, such as MS4A1 (CD20), CD19, CD22, TCL1A, and CD83, </span>
<span class="c1"># plasma B cells, such as CD38, TNFRSF17 (BCMA), and IGHG1/IGHG4</span>
<span class="n">genes_to_check</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#39;CD3D&#39;</span><span class="p">,</span><span class="s">&#39;MS4A1&#39;</span><span class="p">,</span><span class="s">&#39;CD79A&#39;</span><span class="p">,</span>
                   <span class="s">&#39;CD19&#39;</span><span class="p">,</span> <span class="s">&#39;CD22&#39;</span><span class="p">,</span> <span class="s">&#39;TCL1A&#39;</span><span class="p">,</span>  <span class="s">&#39;CD83&#39;</span><span class="p">,</span> <span class="c1">#  naive B cells</span>
                   <span class="s">&#39;CD38&#39;</span><span class="p">,</span><span class="s">&#39;TNFRSF17&#39;</span><span class="p">,</span><span class="s">&#39;IGHG1&#39;</span><span class="p">,</span><span class="s">&#39;IGHG4&#39;</span><span class="p">,</span> <span class="c1"># plasma B cells,</span>
                   <span class="s">&#39;TPSAB1&#39;</span> <span class="p">,</span> <span class="s">&#39;TPSB2&#39;</span><span class="p">,</span>  <span class="c1"># mast cells,</span>
                   <span class="s">&#39;PTPRC&#39;</span> <span class="p">)</span> 
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>  
<span class="n">genes_to_check</span><span class="o">=</span><span class="nf">str_to_title</span><span class="p">(</span><span class="n">genes_to_check</span><span class="p">)</span>
<span class="n">genes_to_check</span>
<span class="n">p</span> <span class="o">&lt;-</span> <span class="nf">DotPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="n">genes_to_check</span><span class="p">,</span>
             <span class="n">assay</span><span class="o">=</span><span class="s">&#39;RNA&#39;</span>  <span class="p">)</span>  <span class="o">+</span> <span class="nf">coord_flip</span><span class="p">()</span>

<span class="n">p</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">plot</span><span class="o">=</span><span class="n">p</span><span class="p">,</span> <span class="n">filename</span><span class="o">=</span><span class="s">&quot;check_Bcells_marker_by_seurat_cluster.pdf&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_016.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="n">genes_to_check</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#39;CD68&#39;</span><span class="p">,</span> <span class="s">&#39;CD163&#39;</span><span class="p">,</span> <span class="s">&#39;CD14&#39;</span><span class="p">,</span>  <span class="s">&#39;CD86&#39;</span><span class="p">,</span><span class="s">&#39;C1QA&#39;</span><span class="p">,</span>  <span class="s">&#39;C1QB&#39;</span><span class="p">,</span>  <span class="c1"># mac</span>
                   <span class="s">&#39;S100A9&#39;</span><span class="p">,</span> <span class="s">&#39;S100A8&#39;</span><span class="p">,</span> <span class="s">&#39;MMP19&#39;</span><span class="p">,</span><span class="c1"># monocyte</span>
                   <span class="s">&#39;LAMP3&#39;</span><span class="p">,</span> <span class="s">&#39;IDO1&#39;</span><span class="p">,</span><span class="s">&#39;IDO2&#39;</span><span class="p">,</span><span class="c1">## DC3 </span>
                   <span class="s">&#39;MRC1&#39;</span><span class="p">,</span><span class="s">&#39;MSR1&#39;</span><span class="p">,</span><span class="s">&#39;ITGAE&#39;</span><span class="p">,</span><span class="s">&#39;ITGAM&#39;</span><span class="p">,</span><span class="s">&#39;ITGAX&#39;</span><span class="p">,</span><span class="s">&#39;SIGLEC7&#39;</span><span class="p">,</span> 
                   <span class="s">&#39;CD1E&#39;</span><span class="p">,</span><span class="s">&#39;CD1C&#39;</span><span class="p">,</span> <span class="c1"># DC2</span>
                   <span class="s">&#39;XCR1&#39;</span><span class="p">,</span><span class="s">&#39;CLEC9A&#39;</span><span class="p">,</span><span class="s">&#39;FCER1A&#39;</span><span class="p">,</span><span class="c1"># DC1</span>
                   <span class="s">&#39;GZMB&#39;</span><span class="p">,</span><span class="s">&#39;TCF4&#39;</span><span class="p">,</span><span class="s">&#39;IRF7&#39;</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>  
<span class="n">genes_to_check</span><span class="o">=</span><span class="nf">str_to_title</span><span class="p">(</span><span class="n">genes_to_check</span><span class="p">)</span>
<span class="n">genes_to_check</span>
<span class="n">p</span> <span class="o">&lt;-</span> <span class="nf">DotPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">unique</span><span class="p">(</span><span class="n">genes_to_check</span><span class="p">),</span>
             <span class="n">assay</span><span class="o">=</span><span class="s">&#39;RNA&#39;</span>  <span class="p">)</span>  <span class="o">+</span> <span class="nf">coord_flip</span><span class="p">()</span>

<span class="n">p</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">plot</span><span class="o">=</span><span class="n">p</span><span class="p">,</span> <span class="n">filename</span><span class="o">=</span><span class="s">&quot;check_Myeloid_marker_by_seurat_cluster.pdf&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_017.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="c1"># epi or tumor (EPCAM, KRT19, PROM1, ALDH1A1, CD24).</span>
<span class="c1"># - alveolar type I cell (AT1; AGER+)</span>
<span class="c1"># - alveolar type II cell (AT2; SFTPA1)</span>
<span class="c1"># - secretory club cell (Club; SCGB1A1+)</span>
<span class="c1"># - basal airway epithelial cells (Basal; KRT17+)</span>
<span class="c1"># - ciliated airway epithelial cells (Ciliated; TPPP3+) </span>

<span class="n">genes_to_check</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span>  <span class="s">&#39;EPCAM&#39;</span> <span class="p">,</span> <span class="s">&#39;KRT19&#39;</span><span class="p">,</span> <span class="s">&#39;PROM1&#39;</span><span class="p">,</span> <span class="s">&#39;ALDH1A1&#39;</span> <span class="p">,</span>
                     <span class="s">&#39;AGER&#39;</span><span class="p">,</span><span class="s">&#39;SFTPA1&#39;</span><span class="p">,</span><span class="s">&#39;SCGB1A1&#39;</span><span class="p">,</span><span class="s">&#39;KRT17&#39;</span><span class="p">,</span><span class="s">&#39;TPPP3&#39;</span><span class="p">,</span>
                     <span class="s">&#39;KRT4&#39;</span><span class="p">,</span><span class="s">&#39;KRT14&#39;</span><span class="p">,</span><span class="s">&#39;KRT8&#39;</span><span class="p">,</span><span class="s">&#39;KRT18&#39;</span><span class="p">,</span>
                     <span class="s">&#39;CD3D&#39;</span><span class="p">,</span><span class="s">&#39;PTPRC&#39;</span> <span class="p">)</span> 
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>  
<span class="n">genes_to_check</span><span class="o">=</span><span class="nf">str_to_title</span><span class="p">(</span><span class="n">genes_to_check</span><span class="p">)</span>
<span class="n">genes_to_check</span>
<span class="n">p</span> <span class="o">&lt;-</span> <span class="nf">DotPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">unique</span><span class="p">(</span><span class="n">genes_to_check</span><span class="p">),</span>
             <span class="n">assay</span><span class="o">=</span><span class="s">&#39;RNA&#39;</span>  <span class="p">)</span>  <span class="o">+</span> <span class="nf">coord_flip</span><span class="p">()</span>

<span class="n">p</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">plot</span><span class="o">=</span><span class="n">p</span><span class="p">,</span> <span class="n">filename</span><span class="o">=</span><span class="s">&quot;check_epi_marker_by_seurat_cluster.pdf&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_018.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="n">genes_to_check</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#39;TEK&#39;</span><span class="p">,</span><span class="s">&quot;PTPRC&quot;</span><span class="p">,</span><span class="s">&quot;EPCAM&quot;</span><span class="p">,</span><span class="s">&quot;PDPN&quot;</span><span class="p">,</span><span class="s">&quot;PECAM1&quot;</span><span class="p">,</span><span class="s">&#39;PDGFRB&#39;</span><span class="p">,</span>
                   <span class="s">&#39;CSPG4&#39;</span><span class="p">,</span><span class="s">&#39;GJB2&#39;</span><span class="p">,</span> <span class="s">&#39;RGS5&#39;</span><span class="p">,</span><span class="s">&#39;ITGA7&#39;</span><span class="p">,</span>
                   <span class="s">&#39;ACTA2&#39;</span><span class="p">,</span><span class="s">&#39;RBP1&#39;</span><span class="p">,</span><span class="s">&#39;CD36&#39;</span><span class="p">,</span> <span class="s">&#39;ADGRE5&#39;</span><span class="p">,</span><span class="s">&#39;COL11A1&#39;</span><span class="p">,</span><span class="s">&#39;FGF7&#39;</span><span class="p">,</span> <span class="s">&#39;MME&#39;</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>  
<span class="n">genes_to_check</span><span class="o">=</span><span class="nf">str_to_title</span><span class="p">(</span><span class="n">genes_to_check</span><span class="p">)</span>
<span class="n">genes_to_check</span>
<span class="n">p</span> <span class="o">&lt;-</span> <span class="nf">DotPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">unique</span><span class="p">(</span><span class="n">genes_to_check</span><span class="p">),</span>
             <span class="n">assay</span><span class="o">=</span><span class="s">&#39;RNA&#39;</span>  <span class="p">)</span>  <span class="o">+</span> <span class="nf">coord_flip</span><span class="p">()</span>

<span class="n">p</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">plot</span><span class="o">=</span><span class="n">p</span><span class="p">,</span> <span class="n">filename</span><span class="o">=</span><span class="s">&quot;check_stromal_marker_by_seurat_cluster.pdf&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_019.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="n">p_all_markers</span>
<span class="n">p_umap</span><span class="o">=</span><span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;umap&quot;</span><span class="p">,</span>  <span class="n">label</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span> 
<span class="nf">library</span><span class="p">(</span><span class="n">patchwork</span><span class="p">)</span>
<span class="n">p_all_markers</span><span class="o">+</span><span class="n">p_umap</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#39;markers_umap.pdf&#39;</span><span class="p">,</span><span class="n">width</span> <span class="o">=</span> <span class="m">22</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_020.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="n">sce</span><span class="o">=</span><span class="n">sce.all</span>
<span class="nf">table</span><span class="p">(</span><span class="nf">Idents</span><span class="p">(</span><span class="n">sce</span><span class="p">))</span>   
<span class="n">sce.markers</span> <span class="o">&lt;-</span> <span class="nf">FindAllMarkers</span><span class="p">(</span><span class="n">object</span> <span class="o">=</span> <span class="n">sce</span><span class="p">,</span> <span class="n">only.pos</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">min.pct</span> <span class="o">=</span> <span class="m">0.25</span><span class="p">,</span> 
                              <span class="n">thresh.use</span> <span class="o">=</span> <span class="m">0.25</span><span class="p">)</span>
<span class="n">DT</span><span class="o">::</span><span class="nf">datatable</span><span class="p">(</span><span class="n">sce.markers</span><span class="p">)</span>
<span class="n">pro</span><span class="o">=</span><span class="s">&#39;basic_seurat&#39;</span>
<span class="nf">write.csv</span><span class="p">(</span><span class="n">sce.markers</span><span class="p">,</span><span class="n">file</span><span class="o">=</span><span class="nf">paste0</span><span class="p">(</span><span class="n">pro</span><span class="p">,</span><span class="s">&#39;_sce.markers.csv&#39;</span><span class="p">))</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span> 
<span class="n">top10</span> <span class="o">&lt;-</span> <span class="n">sce.markers</span> <span class="o">%&gt;%</span> <span class="nf">group_by</span><span class="p">(</span><span class="n">cluster</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">top_n</span><span class="p">(</span><span class="m">10</span><span class="p">,</span> <span class="n">avg_log2FC</span><span class="p">)</span>
<span class="nf">DoHeatmap</span><span class="p">(</span><span class="n">sce</span><span class="p">,</span><span class="n">top10</span><span class="o">$</span><span class="n">gene</span><span class="p">,</span><span class="n">size</span><span class="o">=</span><span class="m">3</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_021.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="n">p</span> <span class="o">&lt;-</span> <span class="nf">DotPlot</span><span class="p">(</span><span class="n">sce</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">unique</span><span class="p">(</span><span class="n">top10</span><span class="o">$</span><span class="n">gene</span><span class="p">),</span>
             <span class="n">assay</span><span class="o">=</span><span class="s">&#39;RNA&#39;</span>  <span class="p">)</span>  <span class="o">+</span> <span class="nf">coord_flip</span><span class="p">()</span>

<span class="n">p</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="n">pro</span><span class="p">,</span><span class="s">&#39;DotPlot_check_top10_markers_by_clusters.pdf&#39;</span><span class="p">),</span><span class="n">height</span> <span class="o">=</span> <span class="m">11</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_022.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span> 
<span class="n">top3</span> <span class="o">&lt;-</span> <span class="n">sce.markers</span> <span class="o">%&gt;%</span> <span class="nf">group_by</span><span class="p">(</span><span class="n">cluster</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">top_n</span><span class="p">(</span><span class="m">3</span><span class="p">,</span> <span class="n">avg_log2FC</span><span class="p">)</span>
<span class="nf">DoHeatmap</span><span class="p">(</span><span class="n">sce</span><span class="p">,</span><span class="n">top3</span><span class="o">$</span><span class="n">gene</span><span class="p">,</span><span class="n">size</span><span class="o">=</span><span class="m">3</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="n">pro</span><span class="p">,</span><span class="s">&#39;DoHeatmap_check_top3_markers_by_clusters.pdf&#39;</span><span class="p">))</span>
<span class="n">p</span> <span class="o">&lt;-</span> <span class="nf">DotPlot</span><span class="p">(</span><span class="n">sce</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">unique</span><span class="p">(</span><span class="n">top3</span><span class="o">$</span><span class="n">gene</span><span class="p">),</span>
             <span class="n">assay</span><span class="o">=</span><span class="s">&#39;RNA&#39;</span>  <span class="p">)</span>  <span class="o">+</span> <span class="nf">coord_flip</span><span class="p">()</span>

<span class="n">p</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="n">pro</span><span class="p">,</span><span class="s">&#39;DotPlot_check_top3_markers_by_clusters.pdf&#39;</span><span class="p">),</span><span class="n">height</span> <span class="o">=</span> <span class="m">11</span><span class="p">)</span>
<span class="nf">save</span><span class="p">(</span><span class="n">sce.markers</span><span class="p">,</span><span class="n">file</span> <span class="o">=</span> <span class="nf">paste0</span><span class="p">(</span><span class="n">pro</span><span class="p">,</span><span class="s">&#39;sce.markers.Rdata&#39;</span><span class="p">))</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_023.jpg" /></p>
<hr />
<div class="highlight"><pre><span></span><code><span class="n">sce.all</span><span class="o">=</span><span class="nf">readRDS</span><span class="p">(</span> <span class="s">&quot;../2-int/sce.all_int.rds&quot;</span><span class="p">)</span>

<span class="c1"># T Cells (CD3D, CD3E, CD8A), </span>
<span class="c1"># B cells (CD19, CD79A, MS4A1 [CD20]), </span>
<span class="c1"># Plasma cells (IGHG1, MZB1, SDC1, CD79A), </span>
<span class="c1"># Monocytes and macrophages (CD68, CD163, CD14),</span>
<span class="c1"># NK Cells (FGFBP2, FCG3RA, CX3CR1),  </span>
<span class="c1"># Photoreceptor cells (RCVRN), </span>
<span class="c1"># Fibroblasts (FGF7, MME), </span>
<span class="c1"># Endothelial cells (PECAM1, VWF). </span>
<span class="c1"># epi or tumor (EPCAM, KRT19, PROM1, ALDH1A1, CD24).</span>
<span class="c1">#   immune (CD45+,PTPRC), epithelial/cancer (EpCAM+,EPCAM), </span>
<span class="c1"># stromal (CD10+,MME,fibo or CD31+,PECAM1,endo) </span>


<span class="c1"># TH17 cells (C-12, CCR6+ and RORC)</span>
<span class="c1"># type 1 helper T cells (TH1; C-17, CXCR3+ and IFNG and TBX21)</span>
<span class="c1"># heterogeneous continuum of intermediate TH1/TH17 states (C-13, C-16 and C-19) </span>
<span class="c1"># with varying degrees of CXCR3, CCR6, CCR5 and CD161 surface protein expression and RORC and TBX21 expression</span>
<span class="c1"># CD161+ subset of type 2 helper T (TH2) cells (C-14),</span>
<span class="c1"># described as pathogenic, with higher expression of allergy-associated HPGDS and IL17RB</span>
<span class="c1"># naive (LEF1, SELL, TCF7),</span>
<span class="c1"># effector (IFNG), </span>
<span class="c1"># cytotoxicity (GZMB, PRF1), </span>
<span class="c1"># early and general exhaustion (PDCD1, CTLA4, ENTPD1 ) .</span>
<span class="c1"># antigen presentation (CD74, HLA-DRB1/5, HLA-DQA2)</span>
<span class="c1"># FCGR3A (FCGR3A+ Monocyte), KLRF1 (NK), FCER1A (DC), and PF4 (MP/platelets).</span>

<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span> 
<span class="n">genes_to_check</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#39;PTPRC&#39;</span><span class="p">,</span> <span class="s">&#39;CD3D&#39;</span><span class="p">,</span> <span class="s">&#39;CD3E&#39;</span><span class="p">,</span> <span class="s">&#39;CD4&#39;</span><span class="p">,</span><span class="s">&#39;CD8A&#39;</span><span class="p">,</span>
                   <span class="s">&#39;LEF1&#39;</span><span class="p">,</span> <span class="s">&#39;SELL&#39;</span> <span class="p">,</span> <span class="s">&#39;TCF7&#39;</span><span class="p">,</span> <span class="c1"># naive </span>
                   <span class="s">&#39;FOXP3&#39;</span><span class="p">,</span> 
                   <span class="s">&#39;CCR6&#39;</span><span class="p">,</span> <span class="s">&#39;RORC&#39;</span> <span class="p">,</span>  <span class="c1"># 几乎不表达 TH17 cells </span>
                   <span class="s">&#39;TBX21&#39;</span><span class="p">,</span> <span class="s">&#39;CXCR3&#39;</span><span class="p">,</span> <span class="s">&#39;IFNG&#39;</span><span class="p">,</span> <span class="c1"># 微弱表达  type 1 helper T cells </span>
                   <span class="s">&#39;CCR3&#39;</span><span class="p">,</span><span class="s">&#39;CCR4&#39;</span><span class="p">,</span>
                   <span class="s">&#39;PDCD1&#39;</span><span class="p">,</span>  <span class="s">&#39;CTLA4&#39;</span><span class="p">,</span><span class="s">&#39;ENTPD1&#39;</span><span class="p">,</span> <span class="c1"># early and general exhaustion </span>
                   <span class="s">&#39;GZMB&#39;</span><span class="p">,</span> <span class="s">&#39;GZMK&#39;</span><span class="p">,</span><span class="s">&#39;PRF1&#39;</span><span class="p">,</span> <span class="c1"># cytotoxicity</span>
                   <span class="s">&#39;IFNG&#39;</span><span class="p">,</span>   <span class="s">&#39;CCL3&#39;</span> <span class="p">,</span><span class="s">&#39;CXCR6&#39;</span> <span class="p">,</span> <span class="s">&#39;ITGA1&#39;</span><span class="p">,</span>  <span class="c1"># effector</span>
                   <span class="s">&#39;NKG7&#39;</span><span class="p">,</span><span class="s">&#39;KLRF1&#39;</span><span class="p">,</span><span class="s">&#39;MKI67&#39;</span><span class="p">,</span><span class="s">&#39;PF4&#39;</span><span class="p">,</span><span class="s">&#39;FCER1A&#39;</span><span class="p">,</span><span class="s">&#39;FCGR3A&#39;</span><span class="p">)</span>  

<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>  
<span class="n">genes_to_check</span><span class="o">=</span><span class="nf">str_to_title</span><span class="p">(</span><span class="n">genes_to_check</span><span class="p">)</span>
<span class="n">genes_to_check</span> 
<span class="n">p_all_markers</span> <span class="o">&lt;-</span> <span class="nf">DotPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">unique</span><span class="p">(</span><span class="n">genes_to_check</span><span class="p">),</span>
                         <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;CCA_snn_res.0.8&quot;</span><span class="p">,</span><span class="n">assay</span><span class="o">=</span><span class="s">&#39;RNA&#39;</span>  <span class="p">)</span>  <span class="o">+</span> <span class="nf">coord_flip</span><span class="p">()</span>

<span class="n">p_all_markers</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="n">plot</span><span class="o">=</span><span class="n">p_all_markers</span><span class="p">,</span> <span class="n">filename</span><span class="o">=</span><span class="s">&quot;check_all_marker_by_CCA_snn_res.0.8.pdf&quot;</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">sce.all</span><span class="o">@</span><span class="n">meta.data</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_024.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="n">p_umap</span><span class="o">=</span><span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;umap&quot;</span><span class="p">,</span>
               <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;CCA_snn_res.0.8&quot;</span><span class="p">,</span><span class="n">label</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span> 
<span class="n">p_umap</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#39;umap_by_CCA_snn_res.0.8.pdf&#39;</span><span class="p">,</span><span class="n">width</span> <span class="o">=</span> <span class="m">15</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_026.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">patchwork</span><span class="p">)</span>
<span class="n">p_all_markers</span><span class="o">+</span><span class="n">p_umap</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#39;umap_and_all_markers_by_CCA_snn_res.0.8.pdf&#39;</span><span class="p">,</span><span class="n">width</span> <span class="o">=</span> <span class="m">15</span><span class="p">)</span>

<span class="c1">#  CXCR6 (TRM marker), IFIT3 (IFN response marker),</span>
<span class="c1">#  NME1 (activation marker), PRF1 (cytotoxic marker), </span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_027.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="n">celltype</span><span class="o">=</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">ClusterID</span><span class="o">=</span><span class="m">0</span><span class="o">:</span><span class="m">8</span><span class="p">,</span>
                    <span class="n">celltype</span><span class="o">=</span><span class="s">&#39;cytotoxic&#39;</span><span class="p">)</span> 
<span class="n">celltype</span><span class="p">[</span><span class="n">celltype</span><span class="o">$</span><span class="n">ClusterID</span> <span class="o">%in%</span> <span class="nf">c</span><span class="p">(</span> <span class="m">2</span><span class="p">,</span><span class="m">6</span> <span class="p">),</span><span class="m">2</span><span class="p">]</span><span class="o">=</span><span class="s">&#39;naive&#39;</span>  
<span class="nf">head</span><span class="p">(</span><span class="n">celltype</span><span class="p">)</span>
<span class="n">celltype</span> 
<span class="nf">table</span><span class="p">(</span><span class="n">celltype</span><span class="o">$</span><span class="n">celltype</span><span class="p">)</span>
<span class="n">sce.all</span><span class="o">@</span><span class="n">meta.data</span><span class="o">$</span><span class="n">celltype</span> <span class="o">=</span> <span class="s">&quot;NA&quot;</span>
<span class="nf">for</span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">1</span><span class="o">:</span><span class="nf">nrow</span><span class="p">(</span><span class="n">celltype</span><span class="p">)){</span>
  <span class="n">sce.all</span><span class="o">@</span><span class="n">meta.data</span><span class="p">[</span><span class="nf">which</span><span class="p">(</span><span class="n">sce.all</span><span class="o">@</span><span class="n">meta.data</span><span class="o">$</span><span class="n">CCA_snn_res.0.8</span> <span class="o">==</span> <span class="n">celltype</span><span class="o">$</span><span class="n">ClusterID</span><span class="p">[</span><span class="n">i</span><span class="p">]),</span><span class="s">&#39;celltype&#39;</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">celltype</span><span class="o">$</span><span class="n">celltype</span><span class="p">[</span><span class="n">i</span><span class="p">]}</span>
<span class="nf">table</span><span class="p">(</span><span class="n">sce.all</span><span class="o">@</span><span class="n">meta.data</span><span class="o">$</span><span class="n">celltype</span><span class="p">)</span>


<span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;umap&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;celltype&quot;</span><span class="p">,</span><span class="n">label</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span>  
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#39;umap_by_celltype.pdf&#39;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_028.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="n">tab.1</span><span class="o">=</span><span class="nf">table</span><span class="p">(</span><span class="n">sce.all</span><span class="o">@</span><span class="n">meta.data</span><span class="o">$</span><span class="n">celltype</span><span class="p">,</span><span class="n">sce.all</span><span class="o">@</span><span class="n">meta.data</span><span class="o">$</span><span class="n">CCA_snn_res.0.8</span><span class="p">)</span> 
<span class="nf">library</span><span class="p">(</span><span class="n">gplots</span><span class="p">)</span>
<span class="n">tab.1</span>
<span class="n">pro</span><span class="o">=</span><span class="s">&#39;cluster&#39;</span>
<span class="nf">pdf</span><span class="p">(</span><span class="n">file</span> <span class="o">=</span> <span class="nf">paste0</span><span class="p">(</span><span class="n">pro</span><span class="p">,</span><span class="s">&#39;  celltype VS  res.0.8.pdf&#39;</span><span class="p">),</span><span class="n">width</span> <span class="o">=</span> <span class="m">12</span><span class="p">)</span>
<span class="nf">balloonplot</span><span class="p">(</span><span class="n">tab.1</span><span class="p">,</span> <span class="n">main</span> <span class="o">=</span><span class="s">&quot; celltype VS  res.0.8 &quot;</span><span class="p">,</span> <span class="n">xlab</span> <span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">,</span> <span class="n">ylab</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">,</span>
            <span class="n">label</span> <span class="o">=</span> <span class="bp">T</span><span class="p">,</span> <span class="n">show.margins</span> <span class="o">=</span> <span class="bp">F</span><span class="p">)</span>
<span class="nf">dev.off</span><span class="p">()</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_029.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">patchwork</span><span class="p">)</span>
<span class="n">th</span><span class="o">=</span><span class="nf">theme</span><span class="p">(</span><span class="n">axis.text.x</span> <span class="o">=</span> <span class="nf">element_text</span><span class="p">(</span><span class="n">angle</span> <span class="o">=</span> <span class="m">45</span><span class="p">,</span> 
                                    <span class="n">vjust</span> <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> <span class="n">hjust</span><span class="o">=</span><span class="m">0.5</span><span class="p">))</span>
<span class="n">p_all_markers</span><span class="o">=</span><span class="nf">DotPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">unique</span><span class="p">(</span><span class="n">genes_to_check</span><span class="p">),</span>
                      <span class="n">assay</span><span class="o">=</span><span class="s">&#39;RNA&#39;</span> <span class="p">,</span><span class="n">group.by</span> <span class="o">=</span> <span class="s">&#39;celltype&#39;</span> <span class="p">)</span>  <span class="o">+</span> <span class="nf">coord_flip</span><span class="p">()</span><span class="o">+</span> <span class="n">th</span>
<span class="n">p_umap</span><span class="o">=</span><span class="nf">DimPlot</span><span class="p">(</span><span class="n">sce.all</span><span class="p">,</span> <span class="n">reduction</span> <span class="o">=</span> <span class="s">&quot;umap&quot;</span><span class="p">,</span> <span class="n">group.by</span> <span class="o">=</span> <span class="s">&quot;celltype&quot;</span><span class="p">,</span><span class="n">label</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span> 
<span class="n">p_all_markers</span><span class="o">+</span><span class="n">p_umap</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&#39;markers_umap_by_celltype.pdf&#39;</span><span class="p">,</span><span class="n">width</span> <span class="o">=</span> <span class="m">13</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_030.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="n">sce</span><span class="o">=</span><span class="n">sce.all</span> 
<span class="nf">Idents</span><span class="p">(</span><span class="n">sce</span><span class="p">)</span><span class="o">=</span><span class="n">sce</span><span class="o">$</span><span class="n">celltype</span>
<span class="nf">table</span><span class="p">(</span><span class="nf">Idents</span><span class="p">(</span><span class="n">sce</span><span class="p">))</span>  
<span class="n">sce.markers</span> <span class="o">&lt;-</span> <span class="nf">FindAllMarkers</span><span class="p">(</span><span class="n">object</span> <span class="o">=</span> <span class="n">sce</span><span class="p">,</span> <span class="n">only.pos</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> <span class="n">min.pct</span> <span class="o">=</span> <span class="m">0.25</span><span class="p">,</span> 
                              <span class="n">thresh.use</span> <span class="o">=</span> <span class="m">0.25</span><span class="p">)</span>
<span class="n">DT</span><span class="o">::</span><span class="nf">datatable</span><span class="p">(</span><span class="n">sce.markers</span><span class="p">)</span>
<span class="n">pro</span><span class="o">=</span><span class="s">&#39;celltype_deg&#39;</span>
<span class="nf">write.csv</span><span class="p">(</span><span class="n">sce.markers</span><span class="p">,</span><span class="n">file</span><span class="o">=</span><span class="nf">paste0</span><span class="p">(</span><span class="n">pro</span><span class="p">,</span><span class="s">&#39;_sce.markers.csv&#39;</span><span class="p">))</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span> 
<span class="n">top10</span> <span class="o">&lt;-</span> <span class="n">sce.markers</span> <span class="o">%&gt;%</span> <span class="nf">group_by</span><span class="p">(</span><span class="n">cluster</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">top_n</span><span class="p">(</span><span class="m">10</span><span class="p">,</span> <span class="n">avg_log2FC</span><span class="p">)</span>
<span class="nf">DoHeatmap</span><span class="p">(</span><span class="n">sce</span><span class="p">,</span><span class="n">top10</span><span class="o">$</span><span class="n">gene</span><span class="p">,</span><span class="n">size</span><span class="o">=</span><span class="m">3</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_031.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="n">p</span> <span class="o">&lt;-</span> <span class="nf">DotPlot</span><span class="p">(</span><span class="n">sce</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">unique</span><span class="p">(</span><span class="n">top10</span><span class="o">$</span><span class="n">gene</span><span class="p">),</span>
             <span class="n">assay</span><span class="o">=</span><span class="s">&#39;RNA&#39;</span>  <span class="p">)</span>  <span class="o">+</span> <span class="nf">coord_flip</span><span class="p">()</span>

<span class="n">p</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="n">pro</span><span class="p">,</span><span class="s">&#39;DotPlot_check_top10_markers_by_clusters.pdf&#39;</span><span class="p">),</span><span class="n">height</span> <span class="o">=</span> <span class="m">11</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_032.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span> 
<span class="n">top3</span> <span class="o">&lt;-</span> <span class="n">sce.markers</span> <span class="o">%&gt;%</span> <span class="nf">group_by</span><span class="p">(</span><span class="n">cluster</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">top_n</span><span class="p">(</span><span class="m">3</span><span class="p">,</span> <span class="n">avg_log2FC</span><span class="p">)</span>
<span class="nf">DoHeatmap</span><span class="p">(</span><span class="n">sce</span><span class="p">,</span><span class="n">top3</span><span class="o">$</span><span class="n">gene</span><span class="p">,</span><span class="n">size</span><span class="o">=</span><span class="m">3</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="n">pro</span><span class="p">,</span><span class="s">&#39;DoHeatmap_check_top3_markers_by_clusters.pdf&#39;</span><span class="p">))</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_033.jpg" /></p>
<div class="highlight"><pre><span></span><code><span class="n">p</span> <span class="o">&lt;-</span> <span class="nf">DotPlot</span><span class="p">(</span><span class="n">sce</span><span class="p">,</span> <span class="n">features</span> <span class="o">=</span> <span class="nf">unique</span><span class="p">(</span><span class="n">top3</span><span class="o">$</span><span class="n">gene</span><span class="p">),</span>
             <span class="n">assay</span><span class="o">=</span><span class="s">&#39;RNA&#39;</span>  <span class="p">)</span>  <span class="o">+</span> <span class="nf">coord_flip</span><span class="p">()</span>

<span class="n">p</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="n">pro</span><span class="p">,</span><span class="s">&#39;DotPlot_check_top3_markers_by_clusters.pdf&#39;</span><span class="p">),</span><span class="n">height</span> <span class="o">=</span> <span class="m">11</span><span class="p">)</span>
<span class="nf">save</span><span class="p">(</span><span class="n">sce.markers</span><span class="p">,</span><span class="n">file</span> <span class="o">=</span> <span class="nf">paste0</span><span class="p">(</span><span class="n">pro</span><span class="p">,</span><span class="s">&#39;sce.markers.Rdata&#39;</span><span class="p">))</span>
</code></pre></div>
<p><img alt="image.png" src="../Figure/scRNASeq_Down_034.jpg" /></p>





                
              </article>
            </div>
          
          
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            回到页面顶部
          </a>
        
      </main>
      
        <footer class="md-footer">
  
    
    <nav class="md-footer__inner md-grid" aria-label="页脚" >
      
        
        <a href="../SeuratObj/" class="md-footer__link md-footer__link--prev" aria-label="上一页: 了解Seurat对象" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                上一页
              </span>
              了解Seurat对象
            </div>
          </div>
        </a>
      
      
        
        <a href="../GSE130001/" class="md-footer__link md-footer__link--next" aria-label="下一页: 常规分析案例一" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                下一页
              </span>
              常规分析案例一
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2022 DoubleHelix
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    <a href="https://space.bilibili.com/434318322" target="_blank" rel="noopener" title="Bilibili | BioInfoCloud" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M488.6 104.1c16.7 18.1 24.4 39.7 23.3 65.7v202.4c-.4 26.4-9.2 48.1-26.5 65.1-17.2 17-39.1 25.9-65.5 26.7H92.02c-26.45-.8-48.21-9.8-65.28-27.2C9.682 419.4.767 396.5 0 368.2V169.8c.767-26 9.682-47.6 26.74-65.7C43.81 87.75 65.57 78.77 92.02 78h29.38L96.05 52.19c-5.75-5.73-8.63-13-8.63-21.79 0-8.8 2.88-16.06 8.63-21.797C101.8 2.868 109.1 0 117.9 0s16.1 2.868 21.9 8.603L213.1 78h88l74.5-69.397C381.7 2.868 389.2 0 398 0c8.8 0 16.1 2.868 21.9 8.603 5.7 5.737 8.6 12.997 8.6 21.797 0 8.79-2.9 16.06-8.6 21.79L394.6 78h29.3c26.4.77 48 9.75 64.7 26.1zm-38.8 69.7c-.4-9.6-3.7-17.4-10.7-23.5-5.2-6.1-14-9.4-22.7-9.8H96.05c-9.59.4-17.45 3.7-23.58 9.8-6.14 6.1-9.4 13.9-9.78 23.5v194.4c0 9.2 3.26 17 9.78 23.5s14.38 9.8 23.58 9.8H416.4c9.2 0 17-3.3 23.3-9.8 6.3-6.5 9.7-14.3 10.1-23.5V173.8zm-264.3 42.7c6.3 6.3 9.7 14.1 10.1 23.2V273c-.4 9.2-3.7 16.9-9.8 23.2-6.2 6.3-14 9.5-23.6 9.5-9.6 0-17.5-3.2-23.6-9.5-6.1-6.3-9.4-14-9.8-23.2v-33.3c.4-9.1 3.8-16.9 10.1-23.2 6.3-6.3 13.2-9.6 23.3-10 9.2.4 17 3.7 23.3 10zm191.5 0c6.3 6.3 9.7 14.1 10.1 23.2V273c-.4 9.2-3.7 16.9-9.8 23.2-6.1 6.3-14 9.5-23.6 9.5-9.6 0-17.4-3.2-23.6-9.5-7-6.3-9.4-14-9.7-23.2v-33.3c.3-9.1 3.7-16.9 10-23.2 6.3-6.3 14.1-9.6 23.3-10 9.2.4 17 3.7 23.3 10z"/></svg>
    </a>
  
    
    
    <a href="https://github.com/BioInfoCloud" target="_blank" rel="noopener" title="GitHub | BioInfoCloud" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../../..", "features": ["navigation.tabs", "navigation.top", "navigation.indexes", "navigation.expand", "search.suggest", "search.highlight"], "search": "../../../assets/javascripts/workers/search.16e2a7d4.min.js", "translations": {"clipboard.copied": "\u5df2\u590d\u5236", "clipboard.copy": "\u590d\u5236", "search.config.lang": "ja", "search.config.pipeline": "trimmer, stemmer", "search.config.separator": "[\\s\\-\uff0c\u3002]+", "search.placeholder": "\u641c\u7d22", "search.result.more.one": "\u5728\u8be5\u9875\u4e0a\u8fd8\u6709 1 \u4e2a\u7b26\u5408\u6761\u4ef6\u7684\u7ed3\u679c", "search.result.more.other": "\u5728\u8be5\u9875\u4e0a\u8fd8\u6709 # \u4e2a\u7b26\u5408\u6761\u4ef6\u7684\u7ed3\u679c", "search.result.none": "\u6ca1\u6709\u627e\u5230\u7b26\u5408\u6761\u4ef6\u7684\u7ed3\u679c", "search.result.one": "\u627e\u5230 1 \u4e2a\u7b26\u5408\u6761\u4ef6\u7684\u7ed3\u679c", "search.result.other": "# \u4e2a\u7b26\u5408\u6761\u4ef6\u7684\u7ed3\u679c", "search.result.placeholder": "\u952e\u5165\u4ee5\u5f00\u59cb\u641c\u7d22", "search.result.term.missing": "\u7f3a\u5c11", "select.version.title": "\u9009\u62e9\u5f53\u524d\u7248\u672c"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.d6c3db9e.min.js"></script>
      
        <script src="../../../mkdocs/javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://unpkg.com/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>