
<!doctype html>
<html lang="zh" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="生物信息云">
      
      
        <meta name="author" content="DoubleHelix">
      
      
        <link rel="canonical" href="https://bioinfocloud.io/BioInfo/chip/chip/">
      
      
      
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.22">
    
    
      
        <title>一. 基因芯片的基本原理 - 生物信息云</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.84d31ad4.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../mkdocs/css/no-footer.css">
    
      <link rel="stylesheet" href="../../../mkdocs/css/unordered-list-symbols.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="light-blue">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#_1" class="md-skip">
          跳转至
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="页眉">
    <a href="../../.." title="生物信息云" class="md-header__button md-logo" aria-label="生物信息云" data-md-component="logo">
      
  <img src="../../../assets/logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            生物信息云
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              一. 基因芯片的基本原理
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="light-blue"  aria-label="切换风格"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="切换风格" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.04 8.71V4h-4.7L12 .69 8.71 4H4v4.71L.69 12 4 15.34v4.7h4.71L12 23.35l3.34-3.31h4.7v-4.7L23.35 12zM8.83 7.05c.98 0 1.77.79 1.77 1.78a1.77 1.77 0 0 1-1.77 1.77c-.99 0-1.78-.79-1.78-1.77 0-.99.79-1.78 1.78-1.78M15.22 17c-.98 0-1.77-.8-1.77-1.78a1.77 1.77 0 0 1 1.77-1.77c.98 0 1.78.79 1.78 1.77A1.78 1.78 0 0 1 15.22 17m-6.72.03L7 15.53 15.53 7l1.5 1.5z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="orange"  aria-label="切换风格"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="切换风格" for="__palette_2" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5s-1.65.15-2.39.42zM3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29zm.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14zM20.65 7l-1.77 3.79a7.02 7.02 0 0 0-2.38-4.15zm-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29zM12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="deep-purple"  aria-label="切换风格"  type="radio" name="__palette" id="__palette_2">
    
      <label class="md-header__button md-icon" title="切换风格" for="__palette_3" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="purple" data-md-color-primary="black" data-md-color-accent="pink"  aria-label="切换风格"  type="radio" name="__palette" id="__palette_3">
    
      <label class="md-header__button md-icon" title="切换风格" for="__palette_4" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="light-green" data-md-color-primary="black" data-md-color-accent="light-blue"  aria-label="切换至日间模式"  type="radio" name="__palette" id="__palette_4">
    
      <label class="md-header__button md-icon" title="切换至日间模式" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9 2c-1.05 0-2.05.16-3 .46 4.06 1.27 7 5.04 7 9.54s-2.94 8.27-7 9.54c.95.3 1.95.46 3 .46a10 10 0 0 0 10-10A10 10 0 0 0 9 2"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="搜索" placeholder="搜索" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="查找">
        
        <button type="reset" class="md-search__icon md-icon" title="清空当前内容" aria-label="清空当前内容" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            正在初始化搜索引擎
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://bioinfocloud.github.io/note" title="前往仓库" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    BioInfoCloud
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="标签" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../../Programming/" class="md-tabs__link">
          
  
  
    
  
  生信编程

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../" class="md-tabs__link">
          
  
  
    
  
  生信基础

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../../Programming/R/ggplot2/" class="md-tabs__link">
          
  
  
    
  
  可视化

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../../bigdata/" class="md-tabs__link">
          
  
  
    
  
  大数据

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../../ML/knn/kNN/" class="md-tabs__link">
          
  
  
    
  
  机器学习

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../scRNAseq/scRNASeq_UP/" class="md-tabs__link">
          
  
  
    
  
  专题

        </a>
      </li>
    
  

    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../../Literature/RNAseq_review1/" class="md-tabs__link">
          
  
  
    
  
  文献

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../database/scList/" class="md-tabs__link">
          
  
  
    
  
  数据库工具

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../../WeChatOfficialAccount/HOME/" class="md-tabs__link">
          
  
  
    
  
  微信

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../../ABOUT/" class="md-tabs__link">
          
  
  
    
  
  关于

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="导航栏" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="生物信息云" class="md-nav__button md-logo" aria-label="生物信息云" data-md-component="logo">
      
  <img src="../../../assets/logo.png" alt="logo">

    </a>
    生物信息云
  </label>
  
    <div class="md-nav__source">
      <a href="https://bioinfocloud.github.io/note" title="前往仓库" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    BioInfoCloud
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../../Programming/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    生信编程
    
  </span>
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_1" id="__nav_1_label" tabindex="0">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            生信编程
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1_2" >
        
          
          <label class="md-nav__link" for="__nav_1_2" id="__nav_1_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Linux
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_2">
            <span class="md-nav__icon md-icon"></span>
            Linux
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/linux/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Linux讲座
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/commands-concise/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    命令行入门精简版
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/vim/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    vim
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/tmux/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    tmux
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/unix/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    UNIX
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Linux/kernel-compile/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    内核编译
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1_3" >
        
          
          <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Markdown
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_3">
            <span class="md-nav__icon md-icon"></span>
            Markdown
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Markdown/markdown/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Markdown讲座
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/vscode-python/vscode-python/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VSCode及插件安装
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Markdown/grammars/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Markdown语法
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1_4" >
        
          
          <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Git
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_4">
            <span class="md-nav__icon md-icon"></span>
            Git
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Git/git/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Git
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Git/git-demos/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Git例子
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Git/git-teamwork/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Git合作
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/Git/windows-install/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Windows安装Git
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1_5" >
        
          
          <label class="md-nav__link" for="__nav_1_5" id="__nav_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    R
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5">
            <span class="md-nav__icon md-icon"></span>
            R
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/R/Grammar/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    R语法
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/R/ggplot2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    可视化
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1_5_3" >
        
          
          <label class="md-nav__link" for="__nav_1_5_3" id="__nav_1_5_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    R包开发
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_1_5_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5_3">
            <span class="md-nav__icon md-icon"></span>
            R包开发
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/R/01-software/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    软件安装
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/R/02-DevelopmentManual/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    快速手册
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/R/shiny/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Shiny
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1_6" >
        
          
          <label class="md-nav__link" for="__nav_1_6" id="__nav_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Python
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_6">
            <span class="md-nav__icon md-icon"></span>
            Python
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/python/links/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    语法基础
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/python/numpy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Numpy
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/python/pandas/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pandas
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/python/Matplotlib_visualization/Matplotlib_visualization/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Matplotlib
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Biopython/01_install/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Biopython
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_2" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    生信基础
    
  </span>
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_2" id="__nav_2_label" tabindex="0">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            生信基础
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_2_2" >
        
          
          <label class="md-nav__link" for="__nav_2_2" id="__nav_2_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    生信基础
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2_2">
            <span class="md-nav__icon md-icon"></span>
            生信基础
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/Introduction/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    生信引言
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/01/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    基础应用
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/02_noun/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    生信名词
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/03_sequence/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    测序技术原理
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/04_format/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    数据文件格式
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/biomaRt/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    biomaRt包
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/counts_RPM_RPKM_FPKM_1/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RNASeq表达定量
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/counts_RPM_RPKM_FPKM_2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RNASeq表达定量转换
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/DESeq2_edgeR_limma/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RNASeq差异表达分析原理
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/GO/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    什么是GO
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/KEGG_GO_GSEA_what/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    富集分析理论基础
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/hsa2mus/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    人鼠基因比较
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/hsa2mus_2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    同源基因转换
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/IDtrans/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    各种ID转换
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../BioInfoBasics/miRNA_ID/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    miRNA的ID转换
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_2_3" >
        
          
          <label class="md-nav__link" for="__nav_2_3" id="__nav_2_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    软件使用问题
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2_3">
            <span class="md-nav__icon md-icon"></span>
            软件使用问题
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../RNAseq/hisat2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Hisat2
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    可视化
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            可视化
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Programming/R/ggplot2/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    可视化
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../visualization/visualizationList/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    公众号文章参考
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../visualization/ScatterPlot/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    散点图
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../visualization/mult_volcano/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    多重火山图
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../visualization/colorScheme01/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    配色方案1
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_4" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../../bigdata/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    大数据
    
  </span>
  

            </a>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            大数据
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    机器学习
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            机器学习
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../ML/knn/kNN/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    1.kNN算法理论
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../ML/knn/knn_py/knn_py/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    2.kNN实践案例
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    专题
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            专题
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6_1" >
        
          
          <label class="md-nav__link" for="__nav_6_1" id="__nav_6_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    单细胞专题
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_6_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6_1">
            <span class="md-nav__icon md-icon"></span>
            单细胞专题
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../scRNAseq/scRNASeq_UP/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    单细胞上游分析
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../scRNAseq/SeuratObj/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    了解Seurat对象
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../scRNAseq/scRNASeq_down/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    单细胞下游分析
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../scRNAseq/GSE130001/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    常规分析案例一
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../scRNAseq/SingleR/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    细胞注释-SingleR
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../scRNAseq/Scissor/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    表型关联分析
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../scRNAseq/CellChat/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    cellChat
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../scRNAseq/scMetabolism/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    scMetabolism
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../database/scList/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    单细胞相关数据库
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6_2" >
        
          
          <label class="md-nav__link" for="__nav_6_2" id="__nav_6_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    RNAseq专题
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_6_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6_2">
            <span class="md-nav__icon md-icon"></span>
            RNAseq专题
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../RNAseq/pipeline_1/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Bulk RNAseq上游分析
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
          
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6_3" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../TCGA/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    TCGA-GEO专题
    
  </span>
  

            </a>
            
          </div>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_6_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6_3">
            <span class="md-nav__icon md-icon"></span>
            TCGA-GEO专题
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6_4" >
        
          
          <label class="md-nav__link" for="__nav_6_4" id="__nav_6_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    药物分析
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_6_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6_4">
            <span class="md-nav__icon md-icon"></span>
            药物分析
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../drugAna/cellminer/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CellMiner数据库
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../drugAna/ADMET/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ADMET计算
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6_5" >
        
          
          <label class="md-nav__link" for="__nav_6_5" id="__nav_6_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    免疫浸润
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_6_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6_5">
            <span class="md-nav__icon md-icon"></span>
            免疫浸润
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../immune/EaSIeR/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    EaSIeR
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6_6" >
        
          
          <label class="md-nav__link" for="__nav_6_6" id="__nav_6_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    数据库工具
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_6_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6_6">
            <span class="md-nav__icon md-icon"></span>
            数据库工具
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../database/CoMutDB/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CoMutDB数据库
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../database/CysModDB/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CysModDB数据库
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    文献
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            文献
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/RNAseq_review1/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RNASeq综述
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/Review/001_immunoediting/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    癌症免疫编辑
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/Review/003_2021_immunotherapy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    癌症免疫治疗
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/Review/002_DTIS/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    药物靶点预测
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/Review/004_2021_ICB/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    ICB治疗
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/scRNAseq/2023_NC_HNSCC/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    2023_HNSCC
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/Review/006_Nano/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    纳米材料
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Literature/AI/001_AI/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    人工智能
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_8" >
        
          
          <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    数据库工具
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_8">
            <span class="md-nav__icon md-icon"></span>
            数据库工具
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../database/scList/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    单细胞相关数据库
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../database/database/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    在线数据库
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../toolsandsoftware/tands/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    工具或软件
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_8_4" >
        
          
          <label class="md-nav__link" for="__nav_8_4" id="__nav_8_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    MedBioInfoCloud包
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_8_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_8_4">
            <span class="md-nav__icon md-icon"></span>
            MedBioInfoCloud包
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../package/MedBioInfoCloud/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    安装
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../package/baseFun/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    基础函数
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../package/BulkRNA-Seq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Bulk RNA-Seq
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../package/enrichAnalysis/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    富集分析
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../package/TCGA/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TCGA数据挖掘
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_9" >
        
          
          <label class="md-nav__link" for="__nav_9" id="__nav_9_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    微信
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_9_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_9">
            <span class="md-nav__icon md-icon"></span>
            微信
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../WeChatOfficialAccount/HOME/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    HOME
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../WeChatOfficialAccount/articleIndex/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    文章索引
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_10" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../../ABOUT/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    关于
    
  </span>
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_10" id="__nav_10_label" tabindex="0">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_10_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_10">
            <span class="md-nav__icon md-icon"></span>
            关于
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../ABOUT/announcement/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    工具
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../ABOUT/studentsGroup/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    微信群
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="目录">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      目录
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1-affymetrix" class="md-nav__link">
    <span class="md-ellipsis">
      1. Affymetrix生物芯片介绍
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-illumina" class="md-nav__link">
    <span class="md-ellipsis">
      2. illumina的原理
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-agilent" class="md-nav__link">
    <span class="md-ellipsis">
      3. Agilent 生物芯片原理
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="_1">一. 基因芯片的基本原理<a class="headerlink" href="#_1" title="Permanent link">&para;</a></h1>
<p>基因芯片(genechip)（又称DNA芯片、生物芯片）的原型是80年代中期提出的。基因芯片的测序原理是杂交测序方法，即通过与一组已知序列的核酸探针杂交进行核酸序列测定的方法，在一块基片表面固定了序列已知的靶核苷酸的探针。当溶液中带有荧光标记的核酸序列TATGCAATCTAG，与基因芯片上对应位置的核酸探针产生互补匹配时，通过确定荧光强度最强的探针位置，获得一组序列完全互补的探针序列。据此可重组出靶核酸的序列。</p>
<h2 id="1-affymetrix">1. Affymetrix生物芯片介绍<a class="headerlink" href="#1-affymetrix" title="Permanent link">&para;</a></h2>
<p>美国Affymetrix公司是目前全球基因芯片行业的领头羊，以其专利的寡聚核苷酸原位光刻合成技术，年产各类寡聚核苷酸基因芯片达到几十万张，占据了表达谱基因芯片科研市场的一半以上，经过了将近十年的研究和开发，已经在国际上赢得了很高的盛誉，同时也成为为数极少的已经盈利的生物芯片公司。Affymetrix公司的基因芯片为寡核苷酸芯片（Oligo芯片），这种类型的芯片具有极高的特异性和灵敏度，重复性好，假阳性率非常低，是目前世界上最先进的基因芯片。Affymetrix所利用的原位光刻专利技术可使一张芯片上合成多达500,000 个寡核苷酸。该系统可以分析样品中DNA或者RNA序列的相对含量。
Affymetrix公司率先开发的寡聚核苷酸原位光刻专利技术，是生产高密度寡核苷酸基因芯片的核心关键技术。采用的技术原理是在合成碱基单体的5＇羟基末端连上一个光敏保护基。首先使支持物羟基化，并用光敏保护基团将其保护起来。每次选取适当的蔽光膜(mask)使需要聚合的部位透光，其它部位不透光。这样，光通过蔽光膜照射到支持物上，受光部位的羟基脱保护而活化。因为合成所用的单体分子一端按传统固相合成方法活化，另一端受光敏保护基的保护，所以发生偶联的部位反应后仍旧带有光敏保护基团。因此，每次通过控制蔽光膜的图案(透光与不透光)决定哪些区域应被活化，以及所用单体的种类和反应次序就可以实现在待定位点合成大量预定序列寡聚体的目的。 使用多种蔽光膜能以更少的合成步骤生产出高密度的阵列，在合成循环中探针数目呈指数增长。某一含N个核苷酸的寡聚核苷酸，通过4×N个化学步骤能合成出4N个可能结构。例如：一段8个碱基的寡核苷酸有65，536种排列的可能，通过32个化学步骤，8个小时就能合成65，536个探针。其基本原理如图所示：
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1629791811289-9dc9a6ed-748a-4dbe-ad51-5974ddb28335.png#clientId=ud07b1a42-41f6-4&amp;from=paste&amp;height=358&amp;id=uc3cd0007&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=556&amp;originWidth=1259&amp;originalType=binary&amp;ratio=1&amp;size=658301&amp;status=done&amp;style=none&amp;taskId=u62837631-8a87-4a0e-a325-51d4b60a83b&amp;width=810.5" />
视频讲解：<a href="https://v.qq.com/x/page/j0173fqkgbb.html">https://v.qq.com/x/page/j0173fqkgbb.html</a>
昂飞的表达谱芯片，分成传统的 <strong>「In Vitro Transcription」</strong> 芯片，简称 IVT 芯片和新一代的 <strong>「Whole Transcriptome」</strong> 芯片，简称 WT 芯片。
IVT 芯片用 Oligo dT 引物和 T7 逆转录酶来得到 cDNA 链，所以，它得到的 cDNA 主要是靠近 mRNA 3'位末端的 cDNA。</p>
<p><img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1629793161121-6688acd9-bdc1-405a-9b31-f1b03f37eaf2.png#clientId=ud07b1a42-41f6-4&amp;from=paste&amp;id=u72394b4d&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=553&amp;originWidth=552&amp;originalType=binary&amp;ratio=1&amp;size=208527&amp;status=done&amp;style=none&amp;taskId=ub67b9693-4db6-4f3c-8c41-5af96ad9acc" /></p>
<p>WT 芯片用随机引物和 T7 逆转录酶来得到 cDNA 链，所以其 cDNA 会覆盖转录本上更多区域。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1629793268904-50586415-d2ba-4904-addd-2e8c8b303865.png#clientId=ud07b1a42-41f6-4&amp;from=paste&amp;id=u1d043f9e&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=625&amp;originWidth=551&amp;originalType=binary&amp;ratio=1&amp;size=229871&amp;status=done&amp;style=none&amp;taskId=u9d8dfbe0-17ed-420a-b5db-d22675cba62" />
WT 芯片的好处：</p>
<ul>
<li>可以覆盖转录本上更多的区域，实验结果的代表性就会更强（如图绿框）。</li>
<li>可以针对因为差异剪接所形成的不同转录本分别设计探针，方便得到不同转录本的表达量变化情况（如图红框）。</li>
<li>可以检测到 lncRNA。</li>
</ul>
<p>比较著名的 WT 芯片有 HTA 2.0、Exon 1.0、Gene 2.0/2.1 等。
参考文章：<a href="https://mp.weixin.qq.com/s/OURXVzaQF05ykQEWpXsLGw">https://mp.weixin.qq.com/s/OURXVzaQF05ykQEWpXsLGw</a></p>
<h2 id="2-illumina">2. illumina的原理<a class="headerlink" href="#2-illumina" title="Permanent link">&para;</a></h2>
<p>illumina 方案的测序芯片称为「Infinium」（曾用名「Bead Array」），顾名思义由玻璃基片和微珠组成。
玻璃基片形状大小与普通载玻片相近，其作用主要是为了给测序微珠提供容器。
基片内表面，通过光蚀刻的方法，做出许多排列整齐的微米级小孔，作为容纳测序微珠的空间。两者大小恰好匹配，一个小孔正好能容纳一个微珠。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1629795222311-9e6545a3-61b1-4c29-87fe-4de5efc5812e.png#clientId=ud07b1a42-41f6-4&amp;from=paste&amp;height=365&amp;id=ue6410702&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=546&amp;originWidth=1091&amp;originalType=binary&amp;ratio=1&amp;size=485451&amp;status=done&amp;style=none&amp;taskId=ue5d6aebf-56bc-4eb7-a2dd-1e81e70c646&amp;width=729.5" />
微珠则为芯片的核心部分，每个微珠的表面，都各偶联某种序列的 DNA 片段。每个微珠上，有几十万个片段，而一个珠子上的片段，都是同一种序列。
其序列设计如图所示，由靠近微珠端的 Address 序列和 Probe 序列构成。Address 序列是标识微珠的标签序列，通过碱基的排列组合组成该微珠的 ID。在 illumina 生产芯片的过程当中，是把要做芯片的几十万种微珠，按设定的比例或者客户需求进行混合，撒到玻璃基片上。在一张芯片的一个反应当中，每种珠子平均有 15 颗或以上。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1629795346605-4227e299-ed6a-4c88-aa7d-c1104f861e9f.png#clientId=ud07b1a42-41f6-4&amp;from=paste&amp;height=383&amp;id=u7fea8283&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=541&amp;originWidth=1170&amp;originalType=binary&amp;ratio=1&amp;size=468201&amp;status=done&amp;style=none&amp;taskId=udb6672ad-1f47-4651-89cf-c6947504581&amp;width=829" />
微珠随机地落入基片的小孔当中，再通过检测芯片上每个小孔当中的微珠上的 Address 序列，就可以知道，这个小孔当中是哪种微珠。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1629795726269-7babc3b9-3bb4-4c2f-84c4-9456e4af1790.png#clientId=ud07b1a42-41f6-4&amp;from=paste&amp;height=298&amp;id=u8645112a&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=322&amp;originWidth=783&amp;originalType=binary&amp;ratio=1&amp;size=403066&amp;status=done&amp;style=none&amp;taskId=ue38651d5-61ba-4418-8b60-ce7ea3965ab&amp;width=724.5" />
所以，illumina 公司出厂的每一张芯片，都要跟一个.dmap文件，标注每一张芯片的每一个微孔当中，分别是哪种微珠。自然地，获得芯片相应的地图文件之后，才能解读测序数据。
<strong>Probe 序列</strong>：作为探针序列，其作用是与目标 DNA 互补杂交；且与 Address 序列存在一一对应的关系。</p>
<p>检测样本的制备和 Affymetrix 芯片基本相似，也是要产生带标记的 cRNA，再和探针杂交，这里我们常用的标记有两种，一种是生物素（biotin）标记，再加荧光结合的抗生物素蛋白与生物素结合即可，见于最常用的 Human HT-12 系列，目前为 V4 版本。
另一种是待检测 cRNA 上直接标记荧光，比如专供石蜡包埋（FFPE）标本的 Human Whole Genome DASL HT 芯片就是这种。大家可以看到，标记了荧光的 cRNA 被探针“毛珠”粘住了，表达量越高的 RNA 被粘住的机会也越多，荧光就越强。这些荧光的点阵正是一个个长满探针排列在硅片孔里的微珠。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1629804901311-1b90f721-9752-4c24-b248-a031384e556e.png#clientId=ud07b1a42-41f6-4&amp;from=paste&amp;height=470&amp;id=u70ad8c3b&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=498&amp;originWidth=221&amp;originalType=binary&amp;ratio=1&amp;size=40066&amp;status=done&amp;style=none&amp;taskId=ud748079f-e92c-4eed-b032-163116eda3e&amp;width=208.5" />
上图显示了 WG-6 BeadChip 芯片中典型珠子(顶部)的示意图，其中标记了阳性(红色正方形)和局部背景(黄色正方形)信号的像素。许多包含 50-mer 探针的微珠位于每个 BeadArray 反应池中。图示为了清晰一些，大大降低了微珠的密度。
视频讲解：<a href="https://v.qq.com/x/page/d0171ihr74i.html">https://v.qq.com/x/page/d0171ihr74i.html</a>
参考文章：<a href="https://mp.weixin.qq.com/s/7D0TmxWubcMnnaXOtn4bow">https://mp.weixin.qq.com/s/7D0TmxWubcMnnaXOtn4bow</a></p>
<h2 id="3-agilent">3. Agilent 生物芯片原理<a class="headerlink" href="#3-agilent" title="Permanent link">&para;</a></h2>
<p>Agilent芯片的基片是一个玻璃片。它的大小和一张标准的病理载玻片一样大小。
它的芯片制作过程，是用和喷墨打印一样的技术来进行制作的。喷墨打印机，是在墨盒里面是装了“红、黄、蓝、黑”四种颜色的墨水。而Agilent打印生物芯片的墨盒里面，是用带保护基团的A/C/G/T四种碱基底物，来代替了颜色墨水。分别含有4种碱基底物的小液滴，被按照设计的探针序列，依次、层叠地喷到玻璃板的确定的位置上。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1629813267156-65de3523-458d-4cb3-a643-80351268c35d.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=221&amp;id=uec4635ed&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=331&amp;originWidth=754&amp;originalType=binary&amp;ratio=1&amp;size=424953&amp;status=done&amp;style=none&amp;taskId=u4b984f1b-12fc-4534-90b9-5530c48c3a1&amp;width=504" />
在每一个碱基的延伸过程当中都有 3 个步骤，分别是偶联、氧化、 去除保护基团，即先把一个核苷酸喷到玻璃板上，然后再喷上第二个碱基，让两个碱基之间发生偶联，继而经过氧化，把亚磷酸基团氧化成磷酸基团。然后，把连在第二个碱基 5’ 位羟基上的 DMT 保护基团去掉， 留下一个自由的 5’ 位羟基，再进行下一步的延伸反应。 不断重复这个过程，DNA 链就会不断地延长。 Agilent 的这个 DNA 链合成技术，每一步的合成效率都非常高，可以达到 99%以上。这让 Agilent 可以在芯片上得到很长的 DNA 链， 最长可以达到 300 个碱基的长度。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1629813345659-9265f36d-f3b8-4e93-b0d2-a80e3f89cacc.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=175&amp;id=ufaad8623&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=276&amp;originWidth=954&amp;originalType=binary&amp;ratio=1&amp;size=286801&amp;status=done&amp;style=none&amp;taskId=ucfd977ec-2f6d-4940-8f98-ccf4bbf9e8f&amp;width=605" />
Agilent 表达谱芯片的检测原理也是基于 3’ IVT 原理。首先用 Oligo dT 引物和 T7 逆转录酶来得到第一链的 cDNA， 然后再转录或复制出第二链的 cDNA，得到双链 cDNA，这个双链cDNA 是带有 T7 启动子的。接下来， 如果是单色（或单通道）芯片杂交流程，我们只进行这张图的左边的蓝色部分，使用 T7 逆转录酶，以带有 A、 T、 U 碱基的核苷酸和标有 Cy5 荧光基团 CTP 为底物逆转录出 cRNA 来；如果是双色（或双通道）杂交流程，要加上图示的右半边红色的部分，即对于另一个样本（参照组 reference），就用标有 Cy3 荧光基团 CTP 做为底物之一。这样， Cy3（或 Cy5） 荧光基团就在体外逆转录过程当中，被带入到新合成的 cRNA链当中去了。 经纯化后，将这个制备好的 cRNA 与芯片探针进行杂交，在激光扫描仪下检测每个点的荧光强度，并计算出对应基因的 RNA 表达量来。不同于单色芯片以计算荧光强度作为表达值，双色芯片的表达值则计算的是实验组和参照组的比值。
视频连接：<a href="https://v.qq.com/x/page/h0174dobvcb.html">https://v.qq.com/x/page/h0174dobvcb.html</a>
参考：<a href="https://mp.weixin.qq.com/s/8qUblDoNV3AobGdiwl8Aig">https://mp.weixin.qq.com/s/8qUblDoNV3AobGdiwl8Aig</a>
需要注意的是，对于双色芯片，我们称标记第二种颜色的样本为参照组（reference），它可以是对照组样本（如用同一个患者的癌旁组织作为癌的对照），也可以仅仅是一个基因表达的参考，如使用一种或几种肿瘤细胞系的 RNA 混合物作为参照组。 <strong>GEO 上使用 Agilent 双色芯片的数据集的表达谱，常常给出的是两种基因的表达强度比值，并取 log2；但也有些数据集虽然用的也是这款芯片，但只使用了单色单样本，这时候就直接和普通单色芯片一样分析即可</strong>。</p>
<h1 id="_2">二. 芯片数据库<a class="headerlink" href="#_2" title="Permanent link">&para;</a></h1>
<p>比较大的芯片数据库有美国 NCBI 的 GEO，欧洲 EMBL-EBI 的 ArrayExpress，日本 DDBJ 的 GEA，不过这个 GEA 直接连接到 ArrayExpress，还有美国 NCI 下属的 TCGA，里面也有大量肿瘤芯片数据，不要误以为 TCGA 全是测序数据。其中 GEO 是最大的芯片数据库，其次是 ArrayExpress。这里就介绍GEO这个数据库。</p>
<h2 id="1-geo">1. GEO数据库简介<a class="headerlink" href="#1-geo" title="Permanent link">&para;</a></h2>
<p>GEO，全称 Gene Expression Omnibus（基因表达综合数据库），是由美国国立生物技术信息中心NCBI创建并维护的基因表达数据库。它创建于2000年，收录并整理了全球范围内研究工作者上传的微阵列芯片、二代测序以及其他形式的高通量基因组数据，这些数据包括基于单通道和双通道微阵列的实验，检测mRNA，基因组DNA和蛋白质丰度，以及非阵列技术，如基因表达系列分析（SAGE），质谱蛋白质组学数据和高通量测序数据。也就是说只要是目前已经发表的论文，论文中涉及到的基因表达检测的数据都可以通过这个数据库中找到。关键是这个数据是免费的！随着芯片和测序技术的蓬勃发展和应用， 2000 年美国国立生物技术信息中心, NCBI 创建了 GEO 并维护至今。 2003 年到 2004 年 GEO 数据库逐渐开放供研究者获取和分析， 2001 年到 2006 年， GEO 数据上传的规则，也就是 MIAME（The MinimumInformation About a Microarray Experiment guidelines）逐渐形成并实行，并成为提交芯片数据的全球规范。
GEO 数据库收录的数据类型有：</p>
<ul>
<li>01）基因芯片或高通量测序的基因表达数据，包括 ncRNA</li>
<li>02）基因芯片或高通量测序的 ChIP 数据</li>
<li>03）基因芯片或高通量测序的基因组甲基化数据</li>
<li>04）高通量实时定量 PCR 的数据</li>
<li>05）微阵列比较基因组杂交技术（arrayCGH）的基因组变异数据</li>
<li>06）单核苷酸多态性（SNP）芯片数据</li>
<li>07）基因表达系列分析（SAGE）的数据</li>
<li>08）蛋白芯片数据</li>
<li>09）十几组大规模平行测序 MPSS 数据</li>
<li>10）十几组质谱 MS 数据</li>
<li>11）其他</li>
</ul>
<h2 id="2arrayexpress">2.ArrayExpress 数据库<a class="headerlink" href="#2arrayexpress" title="Permanent link">&para;</a></h2>
<p>NCBI 的基因表达综合数据库 GEO 和欧洲生物信息学研究所(EBI)的 ArrayExpress 是芯片数据的两个主要公共数据库。尽管它们具有不同的设计，但两个数据库都支持由 MIAME 定义的所有数据元素。 因此 GEO 和ArrayExpress 数据库及数据集结构十分相似，可以类比的学习。ArrayExpress 数据库在 2002建立，开始收录芯片数据， 2008 年开始收录高通量测序数据，同样，近几年测序数据的增长量逐渐超过芯片数据。 ArrayExpress 是 GEO 数据的一个良好的补充。
ArrayExpres数据库的主页链接（<a href="https://www.ebi.ac.uk/arrayexpress/">https://www.ebi.ac.uk/arrayexpress/</a>）</p>
<h1 id="geo">三. GEO数据库使用教程<a class="headerlink" href="#geo" title="Permanent link">&para;</a></h1>
<h2 id="1">1.检索<a class="headerlink" href="#1" title="Permanent link">&para;</a></h2>
<p>检索界面，网址：<a href="http://www.ncbi.nlm.nih.gov/geo">http://www.ncbi.nlm.nih.gov/geo</a></p>
<p><img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773180966-58be52c0-3280-4693-9d4c-fdd9ab3c95c9.jpeg#height=601&amp;id=qCHlu&amp;originHeight=601&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
或者通过NCBI首页，All Databases下拉框中选择GEO DataSets，输入关键词即可搜索。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773180955-fbaaf877-28b6-42ce-86db-e432257dbb35.jpeg#height=601&amp;id=f8Q3z&amp;originHeight=601&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
这里以检索肺癌（lung caner）的数据为例。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773180948-15a80dec-629f-43f0-89d9-929d68dab099.jpeg#height=577&amp;id=YsPX8&amp;originHeight=577&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
搜索结果可以通过7来设置每页显示个数，通过8选择排序方式，可以通过左侧的选项对搜索结果进行筛选。Entry type有四种:Datasets，Series，Samples，Platforms。分别对应的是，GEO Dataset (GDS) 数据集的ID号、GEO Series (GSE) 研究的ID号、GEO Sample (GSM) 样本ID号和GEO Platform (GPL) 芯片平台。</p>
<h3 id="_3">平台<a class="headerlink" href="#_3" title="Permanent link">&para;</a></h3>
<p>平台记录描述阵列上的元件列表（例如，cDNA，寡核苷酸探针组，ORF，抗体）或可在该实验中检测和定量的元件列表（例如，SAGE标签，肽）。每个平台记录都分配有唯一且稳定的GEO登录号（GPLxxx）。平台可以引用多个提交者提交的许多样本。</p>
<h3 id="_4">样品<a class="headerlink" href="#_4" title="Permanent link">&para;</a></h3>
<p>样品记录描述了处理单个样品的条件，它经历的操作以及从中得到的每个元素的丰度测量。每个样品记录都分配有唯一且稳定的GEO登录号（GSMxxx）。Sample实体必须仅引用一个Platform，可以包含在多个Series中。</p>
<h3 id="_5">系列<a class="headerlink" href="#_5" title="Permanent link">&para;</a></h3>
<p>系列记录定义了一组被认为是组的一部分的相关样本，样本如何相关，以及它们是否以及如何排序。A系列提供了整个实验的焦点和描述。系列记录还可能包含描述提取数据，摘要结论或分析的表格。每个系列记录都分配有唯一且稳定的GEO登记号（GSExxx）。</p>
<h3 id="_6">数据集<a class="headerlink" href="#_6" title="Permanent link">&para;</a></h3>
<p>GEO DataSet（GDSxxx）是GEO样本数据的精选集合。GDS记录代表了一系列具有生物学和统计学意义的GEO样本，构成了GEO数据显示和分析工具套件的基础。GDS中的样本指的是同一个平台，也就是说，它们共享一组共同的探测元素。假设GDS中每个样本的值测量值以等效方式计算，即背景处理和标准化等考虑因素在整个数据集中是一致的。通过GDS子集提供反映实验设计的信息。
记住大小关系：一个GDS可以有多个GSM，一个GSM可以有多个GSE，至于GPL，一般不接触，我们通常接触的都是GSE系列（一个GSE里面有多个GSM）的数据。
接下来的几个就好理解了，2是指物种类型，点击一下会弹出一个对话框，选择自己要的就可以对搜索结果进行筛选了，也可以直接点击9处进行筛选，一般不是人就是鼠，通常是选择人。
3通过研究类型进行筛选，比如甲基化，单核苷酸突变等进行筛选。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773180974-91664b08-26c0-44de-95fe-bdb37383b195.jpeg#height=845&amp;id=HM5qs&amp;originHeight=845&amp;originWidth=618&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=618" />
4是根据作者进行筛选，个人感觉一般用不到。5是属性名称，表示数据来自于组织还是特定的细胞类型。6是指初版日期。
上面是通过关键词检索，只是为了让大家了解一下检索页面，通常在文章中我们会看到作者都是用GSE<strong><em>*号，我们知道GSE编号，我们也可以直接通过检索GSE编号获取信息，同时也可以通过GDS</em></strong>,和GSM来检索。GSE编号检索比较常用。
GDS编号检索结果页面，以GDS402为例
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773180993-ca39eb59-ce80-4e90-90d9-f7846d9a3be7.jpeg#height=518&amp;id=UsBuE&amp;originHeight=518&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
GSE编号检索结果页面
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773180998-101dd059-1b48-4b03-b5ec-6cf373c3ebc8.jpeg#height=848&amp;id=WgRho&amp;originHeight=848&amp;originWidth=1075&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1075" /><img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181119-b1d12809-a13f-4e35-8a2d-52071a308399.jpeg#height=870&amp;id=UR16M&amp;originHeight=870&amp;originWidth=1073&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1073" />
soft和miniml都是表示该platform的基础信息，比如GPL编号，上传日期等，soft文件的部分内容如下:
<div class="highlight"><pre><span></span><code>^DATABASE = GeoMiame!Database_name = Gene Expression Omnibus (GEO)!Database_institute = NCBI NLM NIH!Database_web_link = http://www.ncbi.nlm.nih.gov/geo!Database_email = geo@ncbi.nlm.nih.gov^PLATFORM = GPL20814!Platform_title = Human miRNA Array miRbase release 19!Platform_geo_accession = GPL20814!Platform_status = Public on Jun 12 2017!Platform_submission_date = Aug 18 2015!Platform_last_update_date = Jun 12 2017
</code></pre></div>
在soft文件中，每种类别的信息以^开头，常见的类别如下所示
DATABASE
PLATFORM
SAMPLE
SERIES</p>
<p>DATABASE代表GEO数据库的基本信息，PLATFORM代表该平台的基本信息，SAMPLE代表用该平台得到的样本信息，SERIES代表使用该平台得到的一组样本。
在每种类别中，!开头代表一种类型的信息，常规格式为key = value, 比如:Platform_geo_accession = GPL17515
对于芯片平台而言，还会提供探针和基因之间的对应关系等信息，在对应的网页上，我们可以看到如下的表格.
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773180973-b5885c18-8a9a-4f3d-97e1-897eba10db39.jpeg#height=332&amp;id=MC3kI&amp;originHeight=332&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
这部分内容在soft文件中也是存在的，示意如下:
<div class="highlight"><pre><span></span><code>!platform_table_beginID    Reporter Name    miRNA_ID    SPOT_ID272    hsa-miR-99b-5p    MIMAT0000689    271    hsa-miR-99b-3p    MIMAT0004678    270    hsa-miR-99a-5p    MIMAT0000097    269    hsa-miR-99a-3p    MIMAT0004511    268    hsa-miR-98-5p    MIMAT0000096...!platform_table_end
</code></pre></div>
采用配对的两个!叹号来表示数据的开始和结束，在中间存放对应的表格数据。
miniml中的内容和soft是一样的，只是用XML格式来存储上述信息，而supplement file则是由提交者自己上传的一些补充文件，没有明确的格式。
GSM编号检索页面
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181100-1a505a42-9890-4cb3-adc5-ef4b67f47487.jpeg#height=696&amp;id=N1lgZ&amp;originHeight=696&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" /><img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773180989-710e8c3d-4d61-4716-946a-9200376bae8a.jpeg#height=597&amp;id=Su6yP&amp;originHeight=597&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" /><img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773180988-9b3fc174-95ab-4548-aa46-5c760884e0d3.jpeg#height=643&amp;id=B3RDq&amp;originHeight=643&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" /><img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181018-bd8ef1ce-7e8f-4eb7-8359-8ad71f2d9512.jpeg#height=273&amp;id=fPWNR&amp;originHeight=273&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
在这个页面，我们可以看见样本在进行试验处理的详细描述，以及每个探针所检测的的信号值。</p>
<p>除此以外，我们还可以通过GEO数据库的仓库浏览器就行特定的检索。
在GEO数据库首页点击Repository Browser
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181002-5bab859e-426e-45b8-9b6e-3fc0924dcc48.jpeg#height=437&amp;id=fYTFy&amp;originHeight=437&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
进入的页面我们可以看见，序列，平台，样本和物种的选项卡，我们可以通过其进行检索。
比如我们通过序列进行检索芯片表达数据或者甲基化数据，就点击相应的选项
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181005-d87f2908-cc37-41d9-81e6-b87e95ce3e54.jpeg#height=540&amp;id=NdXaE&amp;originHeight=540&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
点击Methylation profiling by array进入新的页面，我们就可以看到具有甲基化数据的GSE数据集的详细描述，包括标题，序列类型，样本数，物种等。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181016-02f541ad-9121-4304-8b2f-d4d924c98691.jpeg#height=540&amp;id=MmMv7&amp;originHeight=540&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
我们还可以通过关键词对结果进行筛选。比如你只和癌症有关的数据。输入cancer，结果就只有151个。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181201-e3038a56-595b-43b3-987d-bcad9feb73ad.jpeg#height=543&amp;id=sTTGI&amp;originHeight=543&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
关于GEO数据库检索就介绍到这里，我们接下来就介绍GEO在线的分析工具。</p>
<h2 id="2">2.分析工具<a class="headerlink" href="#2" title="Permanent link">&para;</a></h2>
<p>我们再看看GDS号检索的结果页面
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181042-b4d1a3e4-b6b0-438e-9f56-291fe7ecbafd.jpeg#height=525&amp;id=erJ4N&amp;originHeight=525&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" /></p>
<h3 id="1find-gene">(1).Find gene工具<a class="headerlink" href="#1find-gene" title="Permanent link">&para;</a></h3>
<p>该工具用于直接查找该数据集中该基因的表达谱数据。
我们以检索brca1基因为例
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1610773181177-7d4e05f2-b687-4fb0-ba32-f2e96be0ab4c.png#height=237&amp;id=Hn8hc&amp;originHeight=237&amp;originWidth=639&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=639" /></p>
<p>我们会跳到一个新的检索结果页面
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181028-7b456457-5f88-4fd1-81b1-69e0ebf38d07.jpeg#height=511&amp;id=fv1iZ&amp;originHeight=511&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
往下拉我们会找到一个Brca1的一个表达谱数据集。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181026-b3e7202a-54b9-487c-afb0-fcd688e64c1b.jpeg#height=267&amp;id=t17Hw&amp;originHeight=267&amp;originWidth=1072&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1072" />
点击右侧的图进入详细页面，该页面具有该基因在各个样本中的表达信息，并且样本的分组信息也有。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181122-8a3d3a8f-4865-4de3-ab8c-69af6180659d.jpeg#height=939&amp;id=VhVvN&amp;originHeight=939&amp;originWidth=582&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=582" />
Find genes that are up/down for this condition(s)可以根据选择的实验筛选条件来找到一序列随该筛选条件有较明显表达差异的基因表达谱。假设我们要检索和疾病分期有关的上调或下调的基因。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1610773181060-2f9037f3-dda9-4b68-b829-2a86688ff036.png#height=106&amp;id=yocrz&amp;originHeight=106&amp;originWidth=632&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=632" />
可以看到检索到了79个基因。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181010-85c03d9a-0f6e-4f5d-8cee-293ef5040b8d.jpeg#height=496&amp;id=qf4mj&amp;originHeight=496&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
我们点击第一个基因，同样可以看到该基因的详细信息。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181160-abe9ac68-cc83-4d56-a3e8-c6b718302fc9.jpeg#height=937&amp;id=smIVe&amp;originHeight=937&amp;originWidth=582&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=582" /></p>
<h3 id="2compare-2-sets-of-samples">(2).Compare 2 sets of samples工具<a class="headerlink" href="#2compare-2-sets-of-samples" title="Permanent link">&para;</a></h3>
<p><img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181057-0380bfb7-4cb1-4130-b577-e89393118e81.jpeg#height=241&amp;id=XL2Zi&amp;originHeight=241&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181037-0e29a148-d33b-4947-b428-269fe21e2fae.jpeg#height=242&amp;id=LTnGQ&amp;originHeight=242&amp;originWidth=808&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=808" />
我们点击Step2： Select which Samples to put in Group A and Group B会弹出一个窗口，让我们自己进行分组。我们可以看见A组合B组的GSM号是一样的，需要将他分在哪一组就在哪一组点击一下就行。背景变为深灰色就表示被选中，这里将GSM9920-GSM9925定位A组，其余定位B组。点击OK.
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181061-50ba3299-f571-4f5d-89d6-b6865e654afd.jpeg#height=567&amp;id=BXKdz&amp;originHeight=567&amp;originWidth=760&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=760" />
点击OK后我们会看见Step2下面会出现分组的样本编号。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181071-ed5a3cac-f882-4996-b84d-56dc699ddf47.jpeg#height=272&amp;id=f4cc1&amp;originHeight=272&amp;originWidth=700&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=700" /></p>
<p>点击Query Group A vs. B我们就会获得5608个表达差异的基因。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181068-7e7defb9-d4bb-4ec6-9e2b-88206b5a3fc1.jpeg#height=503&amp;id=im3qE&amp;originHeight=503&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" /></p>
<h3 id="3cluster-heatmaps">(3).Cluster heatmaps工具<a class="headerlink" href="#3cluster-heatmaps" title="Permanent link">&para;</a></h3>
<p><img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181061-bc68b9db-a031-4e9b-9a83-1ee261250c99.jpeg#height=318&amp;id=J42qs&amp;originHeight=318&amp;originWidth=679&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=679" />
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181059-f521c4a7-6168-44ab-8ed0-110431832cd9.jpeg#height=261&amp;id=iYItt&amp;originHeight=261&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
点击Display，或获取一个聚类的热图，鼠标放在热图上谁有一个红色的虚线框，可以通过上下拖动边框调整选择的区域大小。可以点击DOWNLOAD 下载数据。</p>
<p><img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181176-6e29afd8-afb3-46a7-a923-5249f0100eeb.jpeg#height=837&amp;id=BxZ8k&amp;originHeight=837&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
点击Stack up 可以将所选择的区域放大，能够看见基因名称。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181102-3c98575a-eced-4087-8efd-0abcbdbe9ae7.jpeg#height=838&amp;id=QvDxQ&amp;originHeight=838&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
点击Plot value可以看见探针在样本中的曲线图。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181106-db34785e-563b-499b-89f6-ffba422a0b6b.jpeg#height=859&amp;id=tcBWJ&amp;originHeight=859&amp;originWidth=962&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=962" /></p>
<p>此外Cluster heatmaps工具中还有一个Partitional (K-means/K-medians)的聚类。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181098-f21812aa-8f25-4731-a489-ae0fd69cf8f0.jpeg#height=278&amp;id=Vt5pm&amp;originHeight=278&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
然后通过设置高低表达的颜色，设置聚类个数（cluster，2-15），这里选择4，点击Display
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181106-766212a2-21db-4ca2-8bd7-6eae576ab071.jpeg#height=391&amp;id=bg0Fr&amp;originHeight=391&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181184-b71ed816-d9d0-4589-8678-0ff2b8669d1a.jpeg#height=727&amp;id=llzOM&amp;originHeight=727&amp;originWidth=993&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=993" />
而对于By location on chromosome，是将基因定位于染色体上进行分析。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181112-bd93a65c-2c0b-4a2e-9df7-9f4dc3849863.jpeg#height=217&amp;id=lKQr2&amp;originHeight=217&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
结果如下：
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181244-ff18a941-36a8-4fb8-a5c9-8017904302cf.jpeg#height=822&amp;id=nbucW&amp;originHeight=822&amp;originWidth=807&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=807" /></p>
<h3 id="4experiment-design-and-value-distribution">(4).Experiment design and value distribution工具<a class="headerlink" href="#4experiment-design-and-value-distribution" title="Permanent link">&para;</a></h3>
<p>这个工具就是一些箱线图</p>
<p><img alt="" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1610773181135-d8e27770-c2cb-4c71-9a54-1723cfed8b01.png#height=364&amp;id=tlR6k&amp;originHeight=364&amp;originWidth=967&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=967" />
这就是每个样本中所有基因表达值的一个汇总，这是归一化后的箱线图，看起来质量不错。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181170-f77bcab3-df62-4f82-ad24-88fac7a4eb98.jpeg#height=903&amp;id=hN7r4&amp;originHeight=903&amp;originWidth=545&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=545" /></p>
<h3 id="5geo2r">(5).GEO2R<a class="headerlink" href="#5geo2r" title="Permanent link">&para;</a></h3>
<p>GEO2R是一个交互式web工具，它允许用户比较GEO系列中的两组或两组以上的样本，以便识别在不同实验条件下表达不同的基因。结果显示为按重要性排序的基因表。
GEO2R使用Bioconductor项目中的GEOquery和limma R包对原始提交者提供的处理过的数据表执行比较。
与GEO的其他数据集分析工具不同，GEO2R不依赖于精心设计的数据集，而是直接查询原始的系列矩阵数据文件。这使得及时分析更多的地理数据成为可能。然而，重要的是要认识到，无论数据类型和质量如何，这个工具几乎可以访问和分析任何GEO系列。</p>
<p>在GSE检索结果页面就可以看到这个工具，这里以GSE49382为例，
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181116-efc48269-0bc4-47fb-9680-aff3103d4826.jpeg#height=400&amp;id=f16L4&amp;originHeight=400&amp;originWidth=929&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=929" />
点击Analyze with GEO2R进入页面可以看见所有样本的信息列表
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181150-035ba3e8-031a-4cf6-a208-ad74be48d384.jpeg#height=540&amp;id=p4g58&amp;originHeight=540&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" /></p>
<p>通过Define groups将样本进行分组,输入相应的组名，
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1610773181122-c8056737-4ce8-454a-a62a-96036f0a325c.png#height=260&amp;id=nHl79&amp;originHeight=162&amp;originWidth=179&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=287" />
点击组A，会弹出一个对话框，让你选择要归入A组的样本，点击相应的样本即可（按住Crtl多选），我这里随便点。</p>
<p><img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181137-dd368768-f836-4781-9a3e-daffb96d8e7e.jpeg#height=342&amp;id=rWG2X&amp;originHeight=342&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" /></p>
<p>点击Top250会显示前250个基因
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181200-013f342f-1349-42e6-a390-7f7a8a2e5103.jpeg#height=241&amp;id=Cw5r4&amp;originHeight=241&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" />
结果在浏览器中显示为按p值排列的前250个基因的表。p值最小的基因最显著。单击一行显示该基因的基因表达谱图。图中的每个红条表示从原始提交者提供的样例记录的value列中提取的表达式度量。</p>
<p>使用Select columns特性修改表中包含哪些数据和注释列。有关数据列含义的信息在Summary statistics部分中提供。
在Options选项卡中编辑测试参数，然后回到GEO2R选项卡并单击Recalculate来应用编辑。
要查看超过前250个结果，或者如果想保存结果，可以使用save all results按钮下载完整的结果表。下载的文件以制表符分隔，适合在Excel等电子表格应用程序中打开。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181285-a0673523-b4fc-40ee-8f5d-c837174bf1f9.jpeg#height=503&amp;id=xwuDM&amp;originHeight=503&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" /></p>
<p>编辑options和features</p>
<h4 id="1value-distribution">（1）Value distribution<a class="headerlink" href="#1value-distribution" title="Permanent link">&para;</a></h4>
<p>计算和查看所选示例值的分布。值是提交者提供的原始数据，在此基础上执行GEO2R计算。查看分布对于确定选择的样本是否适合进行比较非常重要。通常，以中间值为中心的值表示数据是标准化的和可交叉比较的。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181144-0a1764ff-c74f-419b-9737-8cfe7e122c00.jpeg#height=674&amp;id=k8g0X&amp;originHeight=674&amp;originWidth=742&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=742" /></p>
<h4 id="2options">（2）Options<a class="headerlink" href="#2options" title="Permanent link">&para;</a></h4>
<p><img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181138-cb31aab9-834c-4b83-be8d-fa149197a58b.jpeg#height=446&amp;id=dqsuO&amp;originHeight=446&amp;originWidth=979&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=979" />
Apply adjustment to the P-values</p>
<p>Limma包提供了几个p值调整选项。这些调整，也称为多次测试纠正，试图纠正错误阳性结果的发生。默认选择Benjamini &amp; Hochberg错误发现率方法，因为它是对微阵列数据最常用的调整，并在发现统计上重要的基因和限制假阳性之间提供了良好的平衡。</p>
<p>Apply log transformation to the data</p>
<p>GEO数据库接受各种数据值类型，包括logged和未logged的数据。Limma包需要使用logged数据为了解决这个问题，GEO2R有一个自动检测特性，它检查所选样本的值，并自动执行log2转换。可选择是否自动转换。</p>
<p>Category of Platform annotation to display on results</p>
<p>选择要在结果上显示的注释类别。基因注释来自于相应的平台记录。有两种注释类型:</p>
<p>NCBI生成的注释可用于许多记录。这些注释是通过从平台中提取稳定的序列识别信息，定期查询Entrez基因和UniGene数据库，生成一致的、最新的注释而得到的。默认情况下选择基因符号和基因标题注释。NCBI生成的注释的其他类别包括GO术语和染色体位置信息。
提交者提供的注释可用于所有记录。这些表示提交者提供的原始平台注释。请注意，提交者提供的注释在样式和内容上有很多多样性，而且自提交时起可能就没有更新过。</p>
<h4 id="3profile-graph">（3）Profile graph<a class="headerlink" href="#3profile-graph" title="Permanent link">&para;</a></h4>
<p>通过从平台记录的ID列输入相应的标识符来查看特定的基因表达谱图。此功能不执行任何计算;它只是在样本间显示基因的表达值。要使此功能正常工作，不需要定义示例组。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181141-0ddca2cf-db69-4ede-95f6-cb13146afc53.jpeg#height=303&amp;id=U1ZbS&amp;originHeight=303&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" /></p>
<h4 id="4r-script">（4）R script<a class="headerlink" href="#4r-script" title="Permanent link">&para;</a></h4>
<p>此选项卡打印用于执行计算的R脚本。这些信息可以保存下来，作为计算结果的参考。
<img alt="" src="https://cdn.nlark.com/yuque/0/2021/jpeg/1234840/1610773181240-0ac125a3-8d52-47ad-b136-586907527201.jpeg#height=348&amp;id=yL4vZ&amp;originHeight=348&amp;originWidth=1080&amp;originalType=binary&amp;ratio=1&amp;size=0&amp;status=done&amp;style=none&amp;width=1080" /></p>
<h1 id="arrayexpress">四. ArrayExpress 数据库使用教程<a class="headerlink" href="#arrayexpress" title="Permanent link">&para;</a></h1>
<p>进到主页之后，我们可以先看一下这个数据库的信息，比如它是功能基因组数据，目前有74700个实验的数据，2556953个芯片数据，60.91TB的存档数据，这都是在描述ArrayExpres数据库的数据量的大小。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630052271282-b10c29f9-d2ce-4b3e-ae28-de951a4e22bf.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=392&amp;id=u5f494ca8&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=784&amp;originWidth=1896&amp;originalType=binary&amp;ratio=1&amp;size=189936&amp;status=done&amp;style=none&amp;taskId=u98643f1d-e2a4-476a-b416-ef4229712c6&amp;width=948" />
了解完之后，我们可以在Search栏进行数据的搜索，举个栗子：我想要搜一个关于breast cancer 的研究，那我就输入关键字：breast cancer ，然后点击搜索。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630052517630-43e088d6-154b-4d6e-8ca2-a5c21d956d8c.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=308&amp;id=u1792fbf3&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=616&amp;originWidth=1521&amp;originalType=binary&amp;ratio=1&amp;size=116040&amp;status=done&amp;style=none&amp;taskId=ue0ca34bf-ddf5-494b-b8c5-db94b81362e&amp;width=760.5" />
搜索完之后，我们看一下结果，有2896个实验的研究出来了，其中有一些可能不是你想要的，怎么办呢，咱们可以加一些过滤条件，限定一下，以减少实验研究的数量。"Filter search results"这个就是添加过滤条件的地方，我现在加一些条件，然后点击filter即可过滤。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630052632880-c5e4c70f-1d49-4b14-b0fb-97f4bd55c41a.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=418&amp;id=u42f8b39e&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=535&amp;originWidth=438&amp;originalType=binary&amp;ratio=1&amp;size=32808&amp;status=done&amp;style=none&amp;taskId=u0dab22e6-8a87-4a9d-b621-2f21f2a9a7a&amp;width=342" />
过滤后的结果就少了很多。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630052712983-47a7bb67-5044-4285-80d0-7384c9b87aba.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=281&amp;id=ufdea3d2d&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=561&amp;originWidth=1520&amp;originalType=binary&amp;ratio=1&amp;size=130743&amp;status=done&amp;style=none&amp;taskId=u40debbe3-307e-424b-b222-e69d4ad83f2&amp;width=760" />
我把出来的这个结果对应的每个字段解释一哈：（以E-MTAB-8310为例）
1. Accession:数据库ID，芯片的数据库编号；
2. Title:题目也就是文章的研究方向；
3. Type:实验类型，转录组分析；
4. Organism：物种，研究的物种是人；
5. Assays：芯片样本数，有8个样品；
6. Released:上传日期，2021年3月2号；
7. Processed：处理过的数据（包括：矩阵数据和单样品数据）。如果这里有下载的箭头，可以下载到它们。
8. Raw:原始数据链接，打开后可以下载数据。下载的页面如下。
我们可以点击Accession ID 查看该数据集的详细信息，该信息页面也有下载数据的地址。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630052957911-b5bcfe82-6879-4f3c-b6aa-e24e11b5ed59.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;id=ueb269301&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=721&amp;originWidth=1149&amp;originalType=binary&amp;ratio=1&amp;size=172571&amp;status=done&amp;style=none&amp;taskId=uecf7f473-2d14-4c02-ae34-f547e533985" />
参考：<a href="https://mp.weixin.qq.com/s/UWMcXFjtydTU9W3JTnUrhA">https://mp.weixin.qq.com/s/UWMcXFjtydTU9W3JTnUrhA</a></p>
<h1 id="_7">五. 芯片数据分析<a class="headerlink" href="#_7" title="Permanent link">&para;</a></h1>
<h2 id="1-r">1. 处理芯片数据的R包<a class="headerlink" href="#1-r" title="Permanent link">&para;</a></h2>
<p>芯片原始数据文件包括： ① 芯片图像扫描得到的记录光信号强度的Intensity文件 ② 包含芯片类型、探针排布等芯片具体设计信息的Design文件 ③ 包含探针注释信息、探针序列等信息的Annotation文件 ④ 包含样本分组、实验处理等信息的Targets文件。不同芯片厂商的文件格式不同，下面是三大厂商芯片数据相应文件的格式信息：
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630117712817-09b037f7-2dd7-4391-890a-fd2864b5e6cb.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;id=u199b8592&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=258&amp;originWidth=606&amp;originalType=binary&amp;ratio=1&amp;size=56968&amp;status=done&amp;style=none&amp;taskId=u09bf8361-c216-4b46-a1f8-e6d972ec9a2" />
原始芯片数据的来源： ①  GEO、ArrayExpres等多个存储芯片实验数据的数据库 ② 芯片公司的官方网站。
处理这些数据会用到下面这些R包：
affy/oligo；beadarray/lumi；limma ；AgiMicroRna ；GEOquery ；readr/readxl。</p>
<h3 id="1-affymetrix_1">(1). Affymetrix芯片<a class="headerlink" href="#1-affymetrix_1" title="Permanent link">&para;</a></h3>
<p>Affymetrix芯片原始数据最常用的格式是CEL格式，从读取原始数据到转化为基因表达量矩阵表格,可以使用affy ,oligo等R包进行数据的质量控制,背景矫正,数据标准化,其中标准化方法常用的为rma算法和mas5算法。</p>
<h4 id="1-affy">1). affy包<a class="headerlink" href="#1-affy" title="Permanent link">&para;</a></h4>
<p>R包affy用于读取Affymetrix芯片CEL文件，也可用于读取一些较早期的affy芯片数据，如3'IVT 芯片，常见的U133系列(GPL570、GPL57K GPL96) ，不适合用于读取较新的affy芯片类型，如Exon ST、 GeneSK SNP芯片。  affy读取CEL文件函数是<code>read.affybatch</code>/<code>ReadAffy</code> 读取后的对象类别：AffyBatch</p>
<h4 id="2-oligo">2). oligo包<a class="headerlink" href="#2-oligo" title="Permanent link">&para;</a></h4>
<p>oligo读取Affymetrix芯片CEL文件，也可以读取affymetrix的几乎所有芯片类型的CEL文件，此外，oligo还可读取Roche NimbleGen芯片的.xys原始数据文件 ，oligo读取CEL文件函数是<code>read.celfiles</code>，读取后的对象类别：</p>
<table>
<thead>
<tr>
<th>芯片数据</th>
<th>对象</th>
</tr>
</thead>
<tbody>
<tr>
<td>Expresssion arrays</td>
<td>ExpressionFeatureSet</td>
</tr>
<tr>
<td>Gene ST arrays</td>
<td>GeneFeatureSet</td>
</tr>
<tr>
<td>Exon ST arrays</td>
<td>ExonFeatureSet</td>
</tr>
<tr>
<td>SNP arrays</td>
<td>SnpFeatureSet</td>
</tr>
<tr>
<td>Tiling arrays</td>
<td>TilingFeatureSet</td>
</tr>
</tbody>
</table>
<h3 id="2-illumina_1">(2). Illumina芯片<a class="headerlink" href="#2-illumina_1" title="Permanent link">&para;</a></h3>
<p>Illumina芯片数据分析的4个起点： ① Pixel-level,每个像素点对应一个数据， tif/tiff 格式； ②  Bead-level,每个bead对应一个数据， 类似于affymetrix的单个probe数据， 具体数据包括txt/idat/Iocs /sdf/ xml等多种格式； ③ Summary-level,每个bead type对应 —个数据，类似于affy的probe set数据， txt/CSV 格式； ④ 公共数据库存储的表达矩阵数据，如 GEO/ArrayExpress，部分数据集已经过预处理，属于summary-level。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630119701339-1b6023a9-d14c-4e60-97a7-ca246f162e41.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;id=u428ecf54&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=521&amp;originWidth=839&amp;originalType=binary&amp;ratio=1&amp;size=402592&amp;status=done&amp;style=none&amp;taskId=u7ae9cfc8-96ab-48ba-8a8e-551c63cc725" /></p>
<p>使用iScan/BeadScan扫描图像及处理图像信息时，因软件的具体版 本和设置参数不同，产生多种不同格式的原始数据文件。 ① <code>.txtfiles</code>,包含所有beads坐标数据和已减去背景值的光信号强度值 ② <code>.tiffiles</code>,扫描后得到的图像文件 ③ <code>.Iocs files</code>,包含所有beads的坐标信息 ④ <code>.idatfiles</code>,专有的二进制文件用于储存光信号强度数据，可供 BeadStudio/GenomeStudio读取进行后续分析 ⑤ <code>.xml files</code>,包含仪器的扫描参数设置和提取光强度值数据的具体方法信息 ⑥ <code>.bgx files</code>,包含探针注释信息 ⑦ <code>Metrics.txt</code>,包含扫描芯片的质量参数信息 ⑧ <code>.sdf files</code>,包含样本/芯片的具体构架、布局信息 ⑨ IBS (Illumina Bead Summary) files，<code>.csv</code>格式，summary-level 数据。</p>
<h4 id="1-limma">1). limma包<a class="headerlink" href="#1-limma" title="Permanent link">&para;</a></h4>
<p><code>read.ilmn</code>函数读取summary-level数据 ， <code>read.idat</code>读取idat数据。</p>
<h4 id="2-beadarray">2). beadarray包<a class="headerlink" href="#2-beadarray" title="Permanent link">&para;</a></h4>
<p><code>readlllumina</code>函数最常用，用于读取bead-level数据 ；<code>readTIFF</code>函数读取pixel-level图像数据 ；<code>readldatFiles</code>函数读取idat文件； <code>readLocsFile</code>函数读取.Iocs文件； <code>readBeadSummaryData</code>函数读取summary-level数据。</p>
<h4 id="3-lumi">3). lumi包<a class="headerlink" href="#3-lumi" title="Permanent link">&para;</a></h4>
<p><code>lumiR</code>函数读取summary-level数据 <code>lumiR.batch</code>函数批量读取summary-level数据 </p>
<h4 id="4-illuminaio">4). illuminaio包<a class="headerlink" href="#4-illuminaio" title="Permanent link">&para;</a></h4>
<p><code>readIDAT</code>函数读取idat数据 ；<code>readBGX</code>函数读取.bgx文件。</p>
<h3 id="3-agilent_1">(3). Agilent芯片<a class="headerlink" href="#3-agilent_1" title="Permanent link">&para;</a></h3>
<h4 id="1-limma_1">1). limma<a class="headerlink" href="#1-limma_1" title="Permanent link">&para;</a></h4>
<p>limma包中的<code>read.maimages</code>函数不限于Agilent芯片数据，可用于读取多个平台芯片文件，但不适用于affymetrix及illumina芯片数据 ，可读取单色及双色芯片数据；可读取多种图像分析程序产生的txt数据，如：Agilent Feature Extraction, Arrayvision, BlueFuse, GenePix, ImaGene, QuantArray (Version 3 or later), Stanford Microarray Database (SMD)，SPOT。</p>
<h4 id="2-agimicrorna">2). AgiMicroRna<a class="headerlink" href="#2-agimicrorna" title="Permanent link">&para;</a></h4>
<p><code>readMicroRnaAFE</code> 函数适用于Agilent的miRNA芯片数据。</p>
<h2 id="2_1">2. 数据下载与处理<a class="headerlink" href="#2_1" title="Permanent link">&para;</a></h2>
<p>数据下载方法： 可以通过GEO网站链接、FTP站点直接下载，也可以通过R 包 GEOquery 的 getGEOSuppFiles 函数下载，在Linux系统下，可以使用wgetx axek arial2等软件下载。</p>
<h3 id="1-geoquery">(1). GEOquery包<a class="headerlink" href="#1-geoquery" title="Permanent link">&para;</a></h3>
<h4 id="1-getgeo">1). getGEO()函数<a class="headerlink" href="#1-getgeo" title="Permanent link">&para;</a></h4>
<p>这个函数是GEOquery包中的主要用户级函数。它指导下载(如果没有指定文件名)，并将GEO SOFT格式文件解析为专门设计的R数据结构，使GEO SOFT格式的每个重要部分易于访问。
<div class="highlight"><pre><span></span><code><span class="nf">getGEO</span><span class="p">(</span><span class="n">GEO</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w"> </span><span class="n">filename</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w"> </span><span class="n">destdir</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">tempdir</span><span class="p">(),</span>
<span class="w">  </span><span class="n">GSElimits</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w"> </span><span class="n">GSEMatrix</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">AnnotGPL</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">getGPL</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span>
<span class="w">  </span><span class="n">parseCharacteristics</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="n">Arguments</span>
</code></pre></div>
 参数介绍：</p>
<table>
<thead>
<tr>
<th>GEO</th>
<th>用于下载和解析的表示GEO对象的字符串。(如。,‘GDS505’,‘GSE2’,‘GSM2’,‘GPL96’)</th>
</tr>
</thead>
<tbody>
<tr>
<td>filename</td>
<td>以前下载的GEO SOFT格式文件的文件名或它的gzip文件(在这种情况下，文件名必须以.gz结尾)。可以指定GEO或文件名中的一个，不能同时指定两个。GEO系列矩阵文件也进行了处理。注意，由于解析的是单个文件，所以返回值不是集合列表，而是解析GSE矩阵文件时的单个集合。</td>
</tr>
<tr>
<td>destdir</td>
<td>任何下载的目标目录。默认为当前工作目录。如果要保存文件供以后使用，您可能需要指定一个不同的目录。如果你的连接速度很慢，这样做是个好主意，因为有些GEO文件太大了!</td>
</tr>
<tr>
<td>GSElimits</td>
<td>此参数只能用于从GSE加载GSMs的一个连续子集。它应该被指定为长度为2的向量，指定要加载的开始和结束(包括)GSMs。例如，这对于将大型GSE拆分为更易于管理的部分可能很有用。</td>
</tr>
<tr>
<td>GSEMatrix</td>
<td>一个布尔值，告诉GEOquery是否使用GEO的GSE Series Matrix文件。这些文件的解析比GSE SOFT格式文件的解析快许多数量级。默认为TRUE，意味着不会发生SOFT格式解析;如果出于某种原因需要GSE记录中的其他列，则设置为FALSE。</td>
</tr>
<tr>
<td>AnnotGPL</td>
<td>一个布尔值，默认值为FALSE，用于决定是否使用Annotation GPL信息。这些文件很好用，因为它们包含定期从Entrez Gene映射的最新信息。然而，它们并不适用于所有gpl;一般来说，它们只适用于GDS引用的GPLs</td>
</tr>
<tr>
<td>getGPL</td>
<td>当获取GSEMatrix文件时，是否下载和包含GPL信息的布尔值默认为TRUE。如果你知道你将使用Bioconductor工具注释你的特征数据，而不是依赖于NCBI GEO提供的信息，你可能想把这个设置为FALSE。通过指定FALSE还可以大大减少下载时间。</td>
</tr>
<tr>
<td>parseCharacteristics</td>
<td>一个默认为TRUE的布尔值，用于判断是否解析GSE Matrix文件的特征信息(如果可用)。如果在解析特征时遇到麻烦，请将此设置为FALSE。</td>
</tr>
</tbody>
</table>
<p>案例：
<div class="highlight"><pre><span></span><code><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">requireNamespace</span><span class="p">(</span><span class="s">&quot;BiocManager&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">quietly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span>
<span class="w">    </span><span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;BiocManager&quot;</span><span class="p">)</span>

<span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;GEOquery&quot;</span><span class="p">)</span>
<span class="n">gds</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">getGEO</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="nf">system.file</span><span class="p">(</span><span class="s">&quot;extdata/GDS507.soft.gz&quot;</span><span class="p">,</span><span class="n">package</span><span class="o">=</span><span class="s">&quot;GEOquery&quot;</span><span class="p">))</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="n">gse2553</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">getGEO</span><span class="p">(</span><span class="s">&#39;GSE2553&#39;</span><span class="p">,</span><span class="n">GSEMatrix</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="nf">show</span><span class="p">(</span><span class="n">gse2553</span><span class="p">)</span>
<span class="nf">show</span><span class="p">(</span><span class="nf">pData</span><span class="p">(</span><span class="nf">phenoData</span><span class="p">(</span><span class="n">gse2553</span><span class="p">[[</span><span class="m">1</span><span class="p">]]))[</span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">,</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">6</span><span class="p">,</span><span class="m">8</span><span class="p">)])</span>
</code></pre></div></p>
<h4 id="2-getgeosuppfiles">2). getGEOSuppFiles()函数<a class="headerlink" href="#2-getgeosuppfiles" title="Permanent link">&para;</a></h4>
<p>NCBI GEO允许将补充文件附加到GEO系列(GSE)、GEO平台(GPL)和GEO样本(GSM)。这个函数“知道”如何获得基于GEO加入的这些文件。没有对下载的文件进行解析，因为计算机通常不知道文件格式。
<div class="highlight"><pre><span></span><code><span class="nf">getGEOSuppFiles</span><span class="p">(</span><span class="n">GEO</span><span class="p">,</span><span class="w"> </span><span class="n">makeDirectory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">baseDir</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">getwd</span><span class="p">(),</span>
<span class="w">  </span><span class="n">fetch_files</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">filter_regex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">)</span>
</code></pre></div>
参数介绍：</p>
<table>
<thead>
<tr>
<th>GEO</th>
<th>GEO accession number such as GPL1073 or GSM1137</th>
</tr>
</thead>
<tbody>
<tr>
<td>makeDirectory</td>
<td>应该为下载的文件创建一个“子目录”吗?默认是正确的。如果为FALSE，文件将直接下载到baseDir中。</td>
</tr>
<tr>
<td>baseDir</td>
<td>下载的基本目录。默认是当前工作目录。</td>
</tr>
<tr>
<td>fetch_files</td>
<td>逻辑值。如果为TRUE，则实际下载文件。如果为FALSE，只返回将要下载的文件名。有用的测试和获得一个列表的文件没有实际下载。</td>
</tr>
<tr>
<td>filter_regex</td>
<td>一个字符(1)正则表达式，用于过滤来自GEO的文件名，以限制那些将被下载的文件。这有助于限制，例如，仅下载bed文件。</td>
</tr>
</tbody>
</table>
<h3 id="2-affymetrix">(2). Affymetrix芯片原始数据的下载与处理<a class="headerlink" href="#2-affymetrix" title="Permanent link">&para;</a></h3>
<h4 id="1_1">1). 数据下载<a class="headerlink" href="#1_1" title="Permanent link">&para;</a></h4>
<p>这里以GEO数据库中的数据为例
<div class="highlight"><pre><span></span><code><span class="c1">##下载芯片原始数据</span>
<span class="nf">ifelse</span><span class="p">(</span><span class="o">!</span><span class="nf">dir.exists</span><span class="p">(</span><span class="s">&quot;geodata&quot;</span><span class="p">),</span><span class="nf">dir.create</span><span class="p">(</span><span class="s">&quot;geodata&quot;</span><span class="p">),</span><span class="nf">message</span><span class="p">(</span><span class="s">&quot;文件夹已存在&quot;</span><span class="p">))</span>
<span class="nf">getGEOSuppFiles</span><span class="p">(</span><span class="s">&quot;GSE5823&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">baseDir</span><span class="w"> </span><span class="o">=</span><span class="s">&quot;geodata&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">makeDirectory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span><span class="w">  </span><span class="c1"># 新建一个文件夹</span>
<span class="c1">#getGEOSuppFiles(&quot;GSE5823&quot;, baseDir = &quot;geodata&quot;, makeDirectory = F) # 不会新建对应的文件夹</span>
</code></pre></div>
也可以获取数据地址URL后使用download.file函数下载，但这种下载方式比较慢，不推荐。
<div class="highlight"><pre><span></span><code><span class="c1">##获取下载链接 </span>
<span class="n">data_url</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">getGEOSuppFiles</span><span class="p">(</span><span class="s">&quot;GSE5823&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">fetch_files</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">makeDirectory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
<span class="c1">##是一个数据框</span>
<span class="n">url</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">data_url</span><span class="o">$</span><span class="n">url</span><span class="p">)</span>
<span class="c1">#使用R函数download.file下载</span>
<span class="nf">download.file</span><span class="p">(</span><span class="n">url</span><span class="p">,</span><span class="w"> </span><span class="n">destfile</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE5823.tar&quot;</span><span class="p">)</span>
</code></pre></div>
如果是在Linux系统中，获取下载链接后，使用axel软件下载。
Axel 是 CLI (command-line interface) 下的一个多线程下载工具，通常我都用它取代 wget 下载各类文件，适用于 Linux 及 BSD 等 UNIX 类平台。在Ubuntu 上安装 axel 命令是：<code>sudo apt-get install axel</code>。在CentOS上安装命令是：<code>sudo yum install axel</code>。</p>
<h4 id="2-targets">2). 整理Targets文件<a class="headerlink" href="#2-targets" title="Permanent link">&para;</a></h4>
<p>Targets文件包含样本信息的Tab分割的文本文件，每行对应一个芯片样本，每列对应样本名、样本分组、处理因素等变量信息 等，列名可以自己定义，该文件数据来源芯片原始数据文件，部分信息可能会整合在每个样本的原始数据文件中，但有时不全面，所以通常我们需要从GSE数据主页的实验设计信息和原始文献中去获取，然后自己整理， GSE页面Samples栏目下常有简要分组、处理信息。比如数据集【GSE5823】：
<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE5823">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE5823</a>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630242330859-3f1ad290-cded-4dac-98b2-0b222da8b0af.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;id=u09d68953&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=559&amp;originWidth=579&amp;originalType=binary&amp;ratio=1&amp;size=51905&amp;status=done&amp;style=none&amp;taskId=u52a47fdf-561f-46f1-a6b6-94bd5766db2" />
我们可以将Samples的内容复制到一个txt文本中读入，然后进行整理。下面是我整理GSE5823数据集的代码。
<div class="highlight"><pre><span></span><code><span class="n">sampleInfo</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="s">&quot;geodata/GSE5823/sampleInfo.txt&quot;</span><span class="p">,</span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">sampleInfo</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;sampleName&quot;</span><span class="p">,</span><span class="s">&quot;description&quot;</span><span class="p">)</span>

<span class="n">celFileNames</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="s">&quot;geodata/GSE5823/GSE5823_RAW&quot;</span><span class="p">,</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;.CEL.gz$&quot;</span><span class="p">,</span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
<span class="n">sampleName</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="n">celFileNames</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;[.]&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)[,</span><span class="m">1</span><span class="p">]</span>
<span class="n">celFlieInfo</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">fileName</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">celFileNames</span><span class="p">,</span><span class="n">sampleName</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sampleName</span><span class="p">)</span>
<span class="n">celFlieInfo</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">celFlieInfo</span><span class="p">,</span><span class="n">sampleInfo</span><span class="p">,</span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;sampleName&quot;</span><span class="p">)</span>
<span class="n">celFlieInfo</span><span class="o">$</span><span class="n">cellType</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="nf">str_split</span><span class="p">(</span><span class="n">celFlieInfo</span><span class="o">$</span><span class="n">description</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\\s+&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)[,</span><span class="m">3</span><span class="p">],</span><span class="w"> </span>
<span class="w">                                 </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)[,</span><span class="m">1</span><span class="p">]</span>
<span class="n">celFlieInfo</span><span class="o">$</span><span class="n">group</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">grepl</span><span class="p">(</span><span class="s">&quot;control&quot;</span><span class="p">,</span><span class="n">celFlieInfo</span><span class="o">$</span><span class="n">description</span><span class="p">),</span><span class="s">&quot;control&quot;</span><span class="p">,</span><span class="s">&quot;c-myc KD&quot;</span><span class="p">)</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">celFlieInfo</span><span class="p">,</span>
<span class="w">            </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE5823/Targets.txt&quot;</span><span class="p">,</span>
<span class="w">            </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span>
<span class="w">            </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
<span class="w">            </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
</code></pre></div>
得到的信息如下，说白了，Targets文件就是样本信息，文件名对应的样本名称，分组等信息而已，你可以用Excel整理也是可以的，总之最后在R中读入可以得到类似下面的数据框就行。列名没有特殊要求，内容有一列就文件名称。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630242613469-8967efac-cf9d-4b68-a958-3ba49dd1fd55.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;id=u1b7d1b4c&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=205&amp;originWidth=1029&amp;originalType=binary&amp;ratio=1&amp;size=48652&amp;status=done&amp;style=none&amp;taskId=ub91eda00-69ca-4522-9612-281944b86b1" /></p>
<h4 id="3">3). 数据读入<a class="headerlink" href="#3" title="Permanent link">&para;</a></h4>
<p>根据前面介绍的R包，读入数据的函数很多。affy包中有2个读取CEL文件的函数：<code>read.affybatch()</code>和<code>ReadAffy()</code>。
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">affy</span><span class="p">)</span>
<span class="n">cel_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">affy</span><span class="o">::</span><span class="nf">list.celfiles</span><span class="p">(</span><span class="s">&quot;geodata/GSE5823/GSE5823_RAW&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                                 </span><span class="n">full.name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
<span class="c1">## affy包中有2个读取CEL文件的函数，推荐ReadAffy(),更灵活易用</span>
<span class="n">cel1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.affybatch</span><span class="p">(</span><span class="n">filenames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cel_files</span><span class="p">)</span>
</code></pre></div>
<code>affy::list.celfiles()</code>函数先获取cel文件的完整路径，其实与<code>list.files()</code>和<code>dir()</code>函数通过正则表达式获取一样，就是cel文件的路径而已。
<div class="highlight"><pre><span></span><code><span class="n">cel_files</span>
<span class="nf">list.files</span><span class="p">(</span><span class="n">path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE5823/GSE5823_RAW&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;.CEL.gz$&quot;</span><span class="p">,</span>
<span class="w">           </span><span class="n">full.name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630243276694-f0703b2a-9082-43fd-9a08-572a28a3fb35.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;id=u859e9607&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=582&amp;originWidth=1266&amp;originalType=binary&amp;ratio=1&amp;size=217479&amp;status=done&amp;style=none&amp;taskId=u3467a11d-767e-4826-b34b-946059e1767" />
接着我们接可以通过read.affybatch()函数读取了，这里的参数就是前面的cel文件路径。
<div class="highlight"><pre><span></span><code><span class="n">cel1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.affybatch</span><span class="p">(</span><span class="n">filenames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cel_files</span><span class="p">)</span>
</code></pre></div>
查看一下cel1这个数据的类型：AffyBatch类
<div class="highlight"><pre><span></span><code><span class="o">&gt;</span><span class="w"> </span><span class="nf">class</span><span class="p">(</span><span class="n">cel1</span><span class="p">)</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;AffyBatch&quot;</span>
<span class="nf">attr</span><span class="p">(,</span><span class="s">&quot;package&quot;</span><span class="p">)</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;affy&quot;</span>
</code></pre></div>
这里我们可以看一下里面的信息，AffyBatch对象是一个S4类，你可以通过@符号索引出来看看里面的内容。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630243708251-8dbf46e2-9083-4eb4-a5d2-23d73abe2345.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;id=u2e3211bc&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=272&amp;originWidth=595&amp;originalType=binary&amp;ratio=1&amp;size=27169&amp;status=done&amp;style=none&amp;taskId=u964c77b1-c919-48ac-b317-608dfe157f1" />
比如：<code>head(cel1@assayData[["exprs"]])[,1:3]</code>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630243917141-e16fc96b-40ad-43db-b232-27a30eec166e.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;id=u3d28a40f&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=232&amp;originWidth=725&amp;originalType=binary&amp;ratio=1&amp;size=25100&amp;status=done&amp;style=none&amp;taskId=uaf6f50c4-78b1-43b7-b44f-9dc894e73a9" />
其实这个就是我们的表达数据了，列名数文件名，具体信息可以通过前面整理的Targets文件来识别。
下面是通过ReadAffy()函数来读入，这个函数读入需要Targets文件信息。Targets这个数据框的行名最后是获取矩阵的列名，所以可以把Targets这个数据框的行名设置成对应的样本名称。
<div class="highlight"><pre><span></span><code><span class="n">Targets</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="s">&quot;geodata/GSE5823/Targets.txt&quot;</span><span class="p">,</span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">,</span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">)</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">Targets</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Targets</span><span class="o">$</span><span class="n">sampleName</span>
<span class="n">cel2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ReadAffy</span><span class="p">(</span><span class="n">filenames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Targets</span><span class="o">$</span><span class="n">fileName</span><span class="p">,</span>
<span class="w">                         </span><span class="n">celfile.path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE5823/GSE5823_RAW&quot;</span><span class="p">,</span>
<span class="w">                         </span><span class="n">phenoData</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Targets</span><span class="p">)</span>
<span class="c1"># 表达矩阵 exprs</span>
<span class="nf">head</span><span class="p">(</span><span class="nf">exprs</span><span class="p">(</span><span class="n">cel1</span><span class="p">))[,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">]</span>
<span class="nf">head</span><span class="p">(</span><span class="nf">exprs</span><span class="p">(</span><span class="n">cel2</span><span class="p">))[,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">]</span>
<span class="n">affexp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">exprs</span><span class="p">(</span><span class="n">cel2</span><span class="p">)</span>
</code></pre></div>
得到的cel2对象和cel1是一样的。可以像前面一样通过索引的方式提取矩阵，但我们有现成的轮子，可以直接通过<code>exprs()</code>函数直接提取出来。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630244369522-a7a2ebd2-7c8a-4f4d-a3fe-580386744643.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;id=u7f803ceb&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=430&amp;originWidth=1124&amp;originalType=binary&amp;ratio=1&amp;size=60200&amp;status=done&amp;style=none&amp;taskId=u3e1e78d4-41e8-4f41-8ab2-22e78eaa0b1" />
我们可以看到2个函数提取的数据是一样的。就是列名有点不同，有Targets文件的话，用<code>ReadAffy()</code>函数会好一些。样本数少，<code>read.affybatch()</code>函数可能会更加方便。
得到的数据可以保存为R对象：
<div class="highlight"><pre><span></span><code><span class="n">GSE5823_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cel2</span>
<span class="nf">save</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">,</span><span class="n">file</span><span class="o">=</span><span class="s">&quot;geodata/GSE5823/GSE5823_data.Rdata&quot;</span><span class="p">)</span>
</code></pre></div></p>
<h4 id="4">4). 数据质量评估<a class="headerlink" href="#4" title="Permanent link">&para;</a></h4>
<p>芯片数据的获取经历多个步骤，每个环节都有可能出现问题: </p>
<ul>
<li>① 样本收集：准备细胞、组织、动物样本；样本的储存 </li>
<li>② RNA提取：DNA污染？ RNA降解？ </li>
<li>③ 荧光标记/探针杂交：RNA反转录、标记、杂交、清洗 </li>
<li>④ 图像扫描：信号的检测、转换、数据存储 </li>
<li>⑤ 从图像提取信号值：匹配坐标与探针、提取前景和背景信号值 </li>
</ul>
<p>质量评估是芯片数据分析过程中不可缺少的步骤 ：单个芯片的水平(array level) ；整个实验设计分组的水平(experimental level)；针对芯片原始数据(raw data)的评估；针对标准化处理后的数据(normalized data)的评估。
Affymetrix芯片数据质量评估的R包有： affy, affyPLM, simpleaffy, arrayQualityMetrics
下面我们加载前面处理得到的数据。
<div class="highlight"><pre><span></span><code><span class="c1">####  加载Affymetrix芯片GSE5823数据 ----</span>
<span class="nf">load</span><span class="p">(</span><span class="s">&quot;geodata/GSE5823/GSE5823_data.Rdata&quot;</span><span class="p">)</span><span class="c1">#变量名GSE5823_data</span>
<span class="nf">library</span><span class="p">(</span><span class="n">affy</span><span class="p">)</span>
<span class="n">GSE5823_exp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">exprs</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">)</span><span class="c1">#提取表达矩阵</span>
<span class="nf">head</span><span class="p">(</span><span class="n">GSE5823_exp</span><span class="p">)[,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">]</span>
<span class="n">GSE5823_targets</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">pData</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">)</span><span class="w"> </span><span class="c1">#就是前面Targets的信息，在phenoData中的data</span>
</code></pre></div>
查看数据。
<div class="highlight"><pre><span></span><code><span class="c1">###  查看数据----</span>
<span class="n">affy</span><span class="o">::</span><span class="nf">pm</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">]</span><span class="c1"># perfect matches </span>
<span class="n">affy</span><span class="o">::</span><span class="nf">mm</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">]</span><span class="c1"># mismatches 这是一个背景数据，没有完全匹配的数据</span>
<span class="n">GSE5823_exp</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">]</span>

<span class="n">affy</span><span class="o">::</span><span class="nf">pm</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">)[</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">]</span><span class="c1">#369707这是id号</span>
<span class="n">GSE5823_exp</span><span class="p">[</span><span class="s">&quot;369707&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">]</span><span class="c1">#定位到369707这一行，发现两者是一致的</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630292289110-7ebdacf4-6a15-4b4f-a1d5-9457bb75144d.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;id=u959f1fe4&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=808&amp;originWidth=1254&amp;originalType=binary&amp;ratio=1&amp;size=140418&amp;status=done&amp;style=none&amp;taskId=ufc3f1698-b345-4887-9e6b-4d21af3afa1" />
查看某一探针集的表达情况。
<div class="highlight"><pre><span></span><code><span class="c1">## 查看某个探针集的表达情况 </span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span><span class="c1">#%&gt;% 管道符就是tidyverse里的</span>

<span class="n">GSE5823_pm_1007</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">affy</span><span class="o">::</span><span class="nf">pm</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;1007_s_at&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">as.data.frame</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">rownames_to_column</span><span class="p">(</span><span class="s">&quot;probe_id&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">pivot_longer</span><span class="p">(</span><span class="n">cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">starts_with</span><span class="p">(</span><span class="s">&quot;GSM&quot;</span><span class="p">),</span>
<span class="w">               </span><span class="n">names_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;sampleName&quot;</span><span class="p">,</span>
<span class="w">               </span><span class="n">values_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;pm_value&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">left_join</span><span class="p">(</span><span class="n">GSE5823_targets</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">sampleName</span><span class="p">,</span><span class="w"> </span><span class="n">group</span><span class="p">),</span><span class="w"> </span>
<span class="w">            </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;sampleName&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">probe_id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">probe_id</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;1007_s_at&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">16</span><span class="p">))))</span>

<span class="nf">ggplot</span><span class="p">(</span><span class="n">GSE5823_pm_1007</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">probe_id</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pm_value</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">group</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Probe set 1007_s_at&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Intensity of PM values&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">90</span><span class="p">,</span><span class="w"> </span><span class="n">vjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630292703445-e6e09a40-f9a1-44a5-a841-6798e011a07d.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;id=u6abe21df&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=573&amp;originWidth=642&amp;originalType=binary&amp;ratio=1&amp;size=33267&amp;status=done&amp;style=none&amp;taskId=u73754bfa-dd6a-4eca-85be-7e3673f6c5c" />
我们可以将数据转换为图片信息。
<div class="highlight"><pre><span></span><code><span class="nf">image</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">[,</span><span class="w"> </span><span class="m">1</span><span class="p">],</span><span class="w"> </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GSE5823_targets</span><span class="o">$</span><span class="n">sampleName</span><span class="p">[</span><span class="m">1</span><span class="p">])</span>
<span class="nf">image</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">[,</span><span class="w"> </span><span class="m">2</span><span class="p">],</span><span class="w"> </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GSE5823_targets</span><span class="o">$</span><span class="n">sampleName</span><span class="p">[</span><span class="m">2</span><span class="p">])</span>
<span class="nf">image</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">[,</span><span class="w"> </span><span class="m">6</span><span class="p">],</span><span class="w"> </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GSE5823_targets</span><span class="o">$</span><span class="n">sampleName</span><span class="p">[</span><span class="m">6</span><span class="p">])</span>

<span class="nf">dir.create</span><span class="p">(</span><span class="s">&quot;geodata/GSE5823/GSE5823_plots/&quot;</span><span class="p">)</span>
<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">20</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="nf">tiff</span><span class="p">(</span><span class="n">filename</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;geodata/GSE5823/GSE5823_plots/&quot;</span><span class="p">,</span>
<span class="w">                         </span><span class="n">GSE5823_targets</span><span class="o">$</span><span class="n">sampleName</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="s">&quot;.tif&quot;</span><span class="p">),</span>
<span class="w">       </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span>
<span class="w">       </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span>
<span class="w">       </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;in&quot;</span><span class="p">,</span>
<span class="w">       </span><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">300</span><span class="p">)</span>
<span class="w">  </span><span class="nf">image</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">[,</span><span class="w"> </span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GSE5823_targets</span><span class="o">$</span><span class="n">sampleName</span><span class="p">[</span><span class="n">i</span><span class="p">])</span>
<span class="w">  </span><span class="nf">dev.off</span><span class="p">()</span>
<span class="p">}</span>
</code></pre></div>
可以根据图片的亮点分布是否均匀看数据质量好坏。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630293069696-c1e7e29d-d0c3-416e-a50c-d9d0e384e45b.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=287&amp;id=ua4737cca&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=573&amp;originWidth=642&amp;originalType=binary&amp;ratio=1&amp;size=212110&amp;status=done&amp;style=none&amp;taskId=uded6cba7-5815-4d29-be52-ab88981aa60&amp;width=321" /><img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630293034554-ff0b371f-4337-4195-b735-402dc2ce93e4.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=287&amp;id=u7d08d101&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=573&amp;originWidth=642&amp;originalType=binary&amp;ratio=1&amp;size=214685&amp;status=done&amp;style=none&amp;taskId=u50201048-b433-4580-b1e3-90233d6af1f&amp;width=321" />
我们可以查看芯片的扫描时间。这个日期可作为一种重要的批次效应因素。
<div class="highlight"><pre><span></span><code><span class="n">GSE5823_targets</span><span class="o">$</span><span class="n">batch</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">GSE5823_data</span><span class="o">@</span><span class="n">protocolData</span><span class="o">@</span><span class="n">data</span><span class="o">$</span><span class="n">ScanDate</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">str_sub</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">8</span><span class="p">)</span><span class="c1">#提取时间日期(年月日)，去掉具体详细的时间点。</span>
<span class="n">GSE5823_targets</span><span class="o">$</span><span class="n">batch</span>
<span class="nf">table</span><span class="p">(</span><span class="n">GSE5823_targets</span><span class="o">$</span><span class="n">batch</span><span class="p">)</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630293449236-263a6e0d-4097-43d5-ad63-56493cd04ad3.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=178&amp;id=u3c04f90d&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=282&amp;originWidth=1197&amp;originalType=binary&amp;ratio=1&amp;size=45247&amp;status=done&amp;style=none&amp;taskId=u22dc7b69-073f-44cf-9d72-6f56cf68b56&amp;width=754.5" />
20个样本分了2批，还是有明显的批次效应的，但也不是说同一个数据集中就有问题。需要根据实验信息判断。比如我们这里的数据集中，是不同细胞control组和c-myc敲低组的芯片数据，如果每种细胞类型是通一批次的，不影响，因为我们分析的时候是按同一细胞类型去分析，但如果同一实验设计中，不同批次，那我们需要注意。这个数据会带来批次效应。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630293581885-7940e27c-2a97-4135-bb70-350f55be3568.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=280&amp;id=ude45dab9&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=249&amp;originWidth=666&amp;originalType=binary&amp;ratio=1&amp;size=34386&amp;status=done&amp;style=none&amp;taskId=u2ec3889d-cf01-4a85-85d9-e203df39f62&amp;width=749" />评估RNA的讲解程度：RNA降解从5‘端开始降解，所5'信号值要比3'信号值低一些，所以假如低得多了，斜率就大了，说明降解得很厉害。
<div class="highlight"><pre><span></span><code><span class="c1">#评估RNA的讲解程度</span>
<span class="n">RNAdeg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">AffyRNAdeg</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">)</span>
<span class="nf">summaryAffyRNAdeg</span><span class="p">(</span><span class="n">RNAdeg</span><span class="p">)</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630294076194-6edc5558-c22f-49d6-af00-c89d364a27de.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;id=u1cd36207&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=279&amp;originWidth=1250&amp;originalType=binary&amp;ratio=1&amp;size=46744&amp;status=done&amp;style=none&amp;taskId=u0dc417d2-5dcc-491f-8564-0bd58086a5c" />
可以可视化查看：
<div class="highlight"><pre><span></span><code><span class="n">cols</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rainbow</span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">GSE5823_targets</span><span class="p">))</span>
<span class="nf">plotAffyRNAdeg</span><span class="p">(</span><span class="n">RNAdeg</span><span class="p">,</span><span class="w"> </span><span class="n">cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cols</span><span class="p">)</span>
<span class="nf">legend</span><span class="p">(</span><span class="s">&quot;bottomright&quot;</span><span class="p">,</span>
<span class="w">       </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">,</span>
<span class="w">       </span><span class="n">legend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sampleNames</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">),</span><span class="w"> </span>
<span class="w">       </span><span class="n">lty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span>
<span class="w">       </span><span class="n">lwd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">,</span>
<span class="w">       </span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.8</span><span class="p">,</span>
<span class="w">       </span><span class="n">box.lty</span><span class="o">=</span><span class="m">0</span><span class="p">,</span>
<span class="w">       </span><span class="n">bg</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;transparent&quot;</span><span class="p">,</span>
<span class="w">       </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cols</span><span class="p">)</span>
<span class="nf">box</span><span class="p">()</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630294335470-6d682f0b-50fe-4660-90f0-42b90ee93bb0.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=430&amp;id=uf481d246&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=573&amp;originWidth=516&amp;originalType=binary&amp;ratio=1&amp;size=33992&amp;status=done&amp;style=none&amp;taskId=u977a68bb-9b87-4ed3-82f3-a329ab92776&amp;width=387" />
simpleaffy包的质量评估函数qc()函数可以查看数据的质量。
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">simpleaffy</span><span class="p">)</span>
<span class="n">GSE5823_qc</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">simpleaffy</span><span class="o">::</span><span class="nf">qc</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">)</span>
<span class="nf">plot</span><span class="p">(</span><span class="n">GSE5823_qc</span><span class="p">)</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630294577062-a61d433b-31e9-460e-b9bc-766a51e68700.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;id=ud0c870e2&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=649&amp;originWidth=1044&amp;originalType=binary&amp;ratio=1&amp;size=59791&amp;status=done&amp;style=none&amp;taskId=u7d7878cc-8e17-4a5f-bbcd-eb9dcc74233" /></p>
<p>这里的 P/M/A 分别代表 present/ marginal present/ absent
绘制箱型图，颜色是根据不同细胞类型设定。
<div class="highlight"><pre><span></span><code><span class="c1">#箱线图</span>
<span class="nf">library</span><span class="p">(</span><span class="n">RColorBrewer</span><span class="p">)</span>
<span class="nf">display.brewer.all</span><span class="p">()</span>
<span class="n">col</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Set2&quot;</span><span class="p">)</span>
<span class="n">cols</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="n">col</span><span class="p">[</span><span class="m">1</span><span class="p">],</span><span class="m">4</span><span class="p">),</span><span class="w"> </span><span class="nf">rep</span><span class="p">(</span><span class="n">col</span><span class="p">[</span><span class="m">2</span><span class="p">],</span><span class="m">6</span><span class="p">),</span><span class="nf">rep</span><span class="p">(</span><span class="n">col</span><span class="p">[</span><span class="m">3</span><span class="p">],</span><span class="m">4</span><span class="p">),</span><span class="nf">rep</span><span class="p">(</span><span class="n">col</span><span class="p">[</span><span class="m">4</span><span class="p">],</span><span class="m">6</span><span class="p">))</span>
<span class="nf">boxplot</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">,</span><span class="w"> </span><span class="n">las</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span>
<span class="w">        </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cols</span><span class="w"> </span><span class="p">)</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630298391767-90b64a1b-1322-4c1e-b5a8-b6d5d23613b9.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=520&amp;id=uc0b33c62&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=693&amp;originWidth=701&amp;originalType=binary&amp;ratio=1&amp;size=33125&amp;status=done&amp;style=none&amp;taskId=u47ee4ed9-15dc-42a9-a04b-056d8428a99&amp;width=526" />
我们发现同一实验设计中，数据直接还是有差异的。
绘制密度分布图查看一下表达值的分布情况。
<div class="highlight"><pre><span></span><code><span class="c1">#密度图</span>
<span class="nf">hist</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">,</span><span class="w"> </span><span class="n">lty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cols</span><span class="p">)</span>
<span class="nf">legend</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="m">12</span><span class="p">,</span>
<span class="w">       </span><span class="n">y</span><span class="o">=</span><span class="m">0.85</span><span class="p">,</span>
<span class="w">       </span><span class="n">legend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sampleNames</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">),</span>
<span class="w">       </span><span class="n">lty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">3</span><span class="p">,</span>
<span class="w">       </span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.8</span><span class="p">,</span>
<span class="w">       </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cols</span><span class="p">,</span>
<span class="w">       </span><span class="n">box.col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;transparent&quot;</span><span class="p">,</span>
<span class="w">       </span><span class="n">xpd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="nf">box</span><span class="p">()</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630298356483-d4ce0102-3317-40a5-be48-62d6a00b7d78.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=520&amp;id=u6eb1d1f6&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=693&amp;originWidth=701&amp;originalType=binary&amp;ratio=1&amp;size=52246&amp;status=done&amp;style=none&amp;taskId=u649d4bb9-cf4a-4563-a7f3-9a2ee13c8be&amp;width=526" />
看见表达值的分布有的高有的低，整体来看的话说明是有问题的。
Deseq2包的的PCA函数进行主成分分析查看一下数据的分类情况。我这里的数据集有不同的细胞，所以我按照细胞类型来看。
<div class="highlight"><pre><span></span><code><span class="c1">#参考了Deseq2包来写的PCA函数</span>
<span class="n">PCA_new</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">expr</span><span class="p">,</span><span class="w"> </span><span class="n">ntop</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">500</span><span class="p">,</span><span class="w"> </span><span class="n">group</span><span class="p">,</span><span class="w"> </span><span class="n">show_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">){</span>
<span class="w">  </span><span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="w">  </span><span class="nf">library</span><span class="p">(</span><span class="n">ggrepel</span><span class="p">)</span>
<span class="w">  </span><span class="n">rv</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">genefilter</span><span class="o">::</span><span class="nf">rowVars</span><span class="p">(</span><span class="n">expr</span><span class="p">)</span>
<span class="w">  </span><span class="n">select</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">order</span><span class="p">(</span><span class="n">rv</span><span class="p">,</span><span class="w"> </span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)[</span><span class="nf">seq_len</span><span class="p">(</span><span class="nf">min</span><span class="p">(</span><span class="n">ntop</span><span class="p">,</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">rv</span><span class="p">)))]</span>
<span class="w">  </span><span class="n">pca</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">prcomp</span><span class="p">(</span><span class="nf">t</span><span class="p">(</span><span class="n">expr</span><span class="p">[</span><span class="n">select</span><span class="p">,</span><span class="w"> </span><span class="p">]))</span><span class="c1">#最核心的代码</span>
<span class="w">  </span><span class="n">percentVar</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pca</span><span class="o">$</span><span class="n">sdev</span><span class="o">^</span><span class="m">2</span><span class="o">/</span><span class="nf">sum</span><span class="p">(</span><span class="n">pca</span><span class="o">$</span><span class="n">sdev</span><span class="o">^</span><span class="m">2</span><span class="p">)</span>
<span class="w">  </span><span class="n">d</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">PC1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pca</span><span class="o">$</span><span class="n">x</span><span class="p">[,</span><span class="w"> </span><span class="m">1</span><span class="p">],</span><span class="w"> </span>
<span class="w">                  </span><span class="n">PC2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pca</span><span class="o">$</span><span class="n">x</span><span class="p">[,</span><span class="w"> </span><span class="m">2</span><span class="p">],</span><span class="w"> </span>
<span class="w">                  </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">group</span><span class="p">,</span><span class="w"> </span>
<span class="w">                  </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">expr</span><span class="p">))</span>
<span class="w">  </span><span class="nf">attr</span><span class="p">(</span><span class="n">d</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;percentVar&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">percentVar</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">]</span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">show_name</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">d</span><span class="p">,</span><span class="w"> </span><span class="nf">aes_string</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;PC1&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;PC2&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;group&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">      </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">xlab</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;PC1: &quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">percentVar</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">100</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;% variance&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">      </span><span class="nf">ylab</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;PC2: &quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">percentVar</span><span class="p">[</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">100</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;% variance&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">geom_text_repel</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">name</span><span class="p">),</span>
<span class="w">                      </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">,</span>
<span class="w">                      </span><span class="n">segment.color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;black&quot;</span><span class="p">,</span>
<span class="w">                      </span><span class="n">show.legend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="w"> </span><span class="p">)</span>
<span class="w">  </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">d</span><span class="p">,</span><span class="w"> </span><span class="nf">aes_string</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;PC1&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;PC2&quot;</span><span class="p">,</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;group&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">      </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">      </span><span class="nf">xlab</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;PC1: &quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">percentVar</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">100</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;% variance&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">      </span><span class="nf">ylab</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;PC2: &quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">percentVar</span><span class="p">[</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">100</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;% variance&quot;</span><span class="p">))</span>
<span class="w">  </span><span class="p">}</span>
<span class="p">}</span>

<span class="nf">PCA_new</span><span class="p">(</span><span class="nf">log2</span><span class="p">(</span><span class="n">GSE5823_exp</span><span class="p">),</span><span class="w"> </span>
<span class="w">        </span><span class="nf">nrow</span><span class="p">(</span><span class="n">GSE5823_exp</span><span class="p">),</span><span class="w"> </span>
<span class="w">        </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GSE5823_targets</span><span class="o">$</span><span class="n">cellType</span><span class="p">,</span>
<span class="w">        </span><span class="n">show_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630298786576-f9d5e9a4-cc3c-422b-9748-311daaa4df6b.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=520&amp;id=u67a5bc81&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=693&amp;originWidth=701&amp;originalType=binary&amp;ratio=1&amp;size=40505&amp;status=done&amp;style=none&amp;taskId=u096471b4-abe0-4260-908e-77c91855eb8&amp;width=526" />
但我们做评估的话，这样慢慢看效率是很低下的，有专门的函数生成数据的评估报告。arrayQualityMetrics包就可以用来评估数据，会在设定的文件加下生成很多文件，可以打开index.html网页文件打开查看。
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">arrayQualityMetrics</span><span class="p">)</span>
<span class="nf">dir.create</span><span class="p">(</span><span class="s">&quot;geodata/GSE5823/GSE5823_QC_raw&quot;</span><span class="p">)</span>
<span class="nf">arrayQualityMetrics</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">,</span><span class="w"> </span>
<span class="w">                    </span><span class="n">outdir</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE5823/GSE5823_QC_raw&quot;</span><span class="p">,</span><span class="w"> </span><span class="c1">#通过该文件夹下的index来看结果</span>
<span class="w">                    </span><span class="n">force</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="c1">#如果本身有个文件夹的话就把文件夹刷新了</span>
<span class="w">                    </span><span class="n">intgroup</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;group&quot;</span><span class="p">,</span><span class="c1">#GSE5823_data@phenoData@data[[&quot;group&quot;]]</span>
<span class="w">                    </span><span class="n">do.logtransform</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="nf">dev.off</span><span class="p">()</span>
</code></pre></div>
打开index.html网页文件。网页顶部有报告的目录信息，分为5部分。我们可以慢慢的查看。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630306907380-eba22419-cc4e-4810-9d1e-4e0eec9df50a.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=347&amp;id=uf194758b&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=462&amp;originWidth=480&amp;originalType=binary&amp;ratio=1&amp;size=46482&amp;status=done&amp;style=none&amp;taskId=u71a142b9-16e6-4596-8fb7-2bf24f2455d&amp;width=360" />
可以根据下面这个图来确认数据的好坏，有×的点，代表该数据样本在对应的方法检测下是不合格的数据。如果数据样本多，可以去除不合格的样本，数据样本少的时候没法，但如果一个样本6种检测方法中有3个即以上都不通过，说明数据是有问题的。当然，有的作者也使用。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630307051833-86e497c0-db5f-47c6-ae42-cc997a94cfeb.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=401&amp;id=ufc4e3916&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=534&amp;originWidth=773&amp;originalType=binary&amp;ratio=1&amp;size=96824&amp;status=done&amp;style=none&amp;taskId=u97c0d2f5-3d5b-45f9-995d-cc5761d36d6&amp;width=580" /><img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630307160276-6deac82b-4b22-4404-8ab1-81e7c46197ea.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;id=ue5e72838&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=144&amp;originWidth=557&amp;originalType=binary&amp;ratio=1&amp;size=15795&amp;status=done&amp;style=none&amp;taskId=u714e4835-afe6-45ee-a9f6-0a952de6594" />
这6种方法对应的可视化图片在网页下面可以查看。</p>
<h4 id="5">5). 数据的背景校正<a class="headerlink" href="#5" title="Permanent link">&para;</a></h4>
<p>Affymerix芯片背景校正的主要R包：从CEL文件读取后的数据出发，可使用affy包及oligo包中的相应函数处理。
Affymetrix芯片原始数据的背景校正可以用affy包中的bg.correct()函数。该函数提供5种计算方法，最常用rma方法。可以通过bgcorrect.methods()查看这5种方法。这里不需要了解各种算法，就用rma方法就行了。
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">RColorBrewer</span><span class="p">)</span>
<span class="nf">display.brewer.all</span><span class="p">()</span>
<span class="n">col</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Set2&quot;</span><span class="p">)</span>
<span class="n">cols</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="n">col</span><span class="p">[</span><span class="m">1</span><span class="p">],</span><span class="m">4</span><span class="p">),</span><span class="w"> </span><span class="nf">rep</span><span class="p">(</span><span class="n">col</span><span class="p">[</span><span class="m">2</span><span class="p">],</span><span class="m">6</span><span class="p">),</span><span class="nf">rep</span><span class="p">(</span><span class="n">col</span><span class="p">[</span><span class="m">3</span><span class="p">],</span><span class="m">4</span><span class="p">),</span><span class="nf">rep</span><span class="p">(</span><span class="n">col</span><span class="p">[</span><span class="m">4</span><span class="p">],</span><span class="m">6</span><span class="p">))</span>
<span class="nf">bgcorrect.methods</span><span class="p">()</span>
<span class="n">GSE5823_bgc</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">affy</span><span class="o">::</span><span class="nf">bg.correct</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">,</span><span class="w"> </span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;rma&quot;</span><span class="p">)</span><span class="c1">#最常用rma方法</span>
<span class="c1">#背景矫正后作图</span>
<span class="nf">boxplot</span><span class="p">(</span><span class="n">GSE5823_bgc</span><span class="p">,</span><span class="w"> </span><span class="n">las</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cols</span><span class="p">)</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630308008816-c8c4b2fa-eee4-4e9c-9065-0c7e71be7f87.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=520&amp;id=ud369a5ba&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=693&amp;originWidth=701&amp;originalType=binary&amp;ratio=1&amp;size=35865&amp;status=done&amp;style=none&amp;taskId=u474cd58c-da03-420c-9bb3-ee46c0bb09a&amp;width=526" />
可以比较一下，前面数据质量报告，GSM136093和GSM136094这2个样本在质量评估中6种检测方法是不合格的，这里背景校正后还是和另外4个是有偏离的，可以从前面的质控报告汇中看到，这2个样本的扫描时间和其他的不一样，这可能就是批次效应带来的误差所引起的。所以这2个数据是需要去除的。类似的还有GSM136084这个样本。
再做一个主成分分析看看。
<div class="highlight"><pre><span></span><code><span class="nf">PCA_new</span><span class="p">(</span><span class="nf">log2</span><span class="p">(</span><span class="nf">exprs</span><span class="p">(</span><span class="n">GSE5823_bgc</span><span class="p">)),</span><span class="w"> </span>
<span class="w">        </span><span class="nf">nrow</span><span class="p">(</span><span class="n">GSE5823_bgc</span><span class="p">),</span><span class="w"> </span>
<span class="w">        </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GSE5823_targets</span><span class="o">$</span><span class="n">cellType</span><span class="p">,</span>
<span class="w">        </span><span class="n">show_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>

<span class="nf">PCA_new</span><span class="p">(</span><span class="nf">log2</span><span class="p">(</span><span class="nf">exprs</span><span class="p">(</span><span class="n">GSE5823_bgc</span><span class="p">)),</span><span class="w"> </span>
<span class="w">        </span><span class="nf">nrow</span><span class="p">(</span><span class="n">GSE5823_bgc</span><span class="p">),</span><span class="w"> </span>
<span class="w">        </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GSE5823_targets</span><span class="o">$</span><span class="n">group</span><span class="p">,</span>
<span class="w">        </span><span class="n">show_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
</code></pre></div></p>
<p><img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630308417005-126d49bd-44b5-4e14-9b80-f59cfde9fe50.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=320&amp;id=u894731e0&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=597&amp;originWidth=701&amp;originalType=binary&amp;ratio=1&amp;size=37638&amp;status=done&amp;style=none&amp;taskId=uac92ec33-d88d-4abf-8940-98e62ec6856&amp;width=375.5" /><img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630308481417-d863850e-9ef9-447d-b896-3c305b6d5eaa.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=299&amp;id=ue35fcbe8&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=597&amp;originWidth=701&amp;originalType=binary&amp;ratio=1&amp;size=36797&amp;status=done&amp;style=none&amp;taskId=u2a89afc4-803d-4f23-a565-172e0c4eca2&amp;width=350.5" /></p>
<p>从细胞类型来看的确分为了4个类别，这个和符合，本身就是不给不同的实验设计。但同一实验设计中就不是那么好了，只有MCF-7细胞这组的实验设计能很好的将control组和敲低组区分出来。</p>
<h4 id="6">6). 芯片数据标准化<a class="headerlink" href="#6" title="Permanent link">&para;</a></h4>
<p>通过对芯片数据标准化可以弥补一部分系统偏差对数据的影响 ，使各个样本芯片数据的分布相似，使不同的芯片之间可以进行比较 。
芯片数据标准化方法类别： </p>
<ul>
<li>① 基于所有基因的标准化，需要满足两个假设前提： 一是不同样本中的绝大部分基因表达是稳定不变的 ，二是上调和下调的基因数量是相似的，具有对称性。 </li>
<li>② 基于管家基因的标准化 ：认为管家基因在不同的处理条件下均是恒定的 </li>
<li>③ 基于芯片实验中设置的已知表达水平的对照基因 </li>
</ul>
<p>对于Affymerix芯片来说，rma或者gcrma已经把背景矫正整合进去了，所以没有必要单独做背景矫正。我们先来看rma方法。
<div class="highlight"><pre><span></span><code><span class="n">GSE5823_rma</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">affy</span><span class="o">::</span><span class="nf">rma</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">)</span><span class="c1">#用的是原始cel数据，不是背景矫正后的数据</span>
<span class="nf">boxplot</span><span class="p">(</span><span class="n">GSE5823_rma</span><span class="p">,</span><span class="w"> </span><span class="n">las</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cols</span><span class="p">)</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630309393710-648413fc-5e8e-4bd9-bef3-ae44c664f38e.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=335&amp;id=u0358b0dc&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=446&amp;originWidth=701&amp;originalType=binary&amp;ratio=1&amp;size=21081&amp;status=done&amp;style=none&amp;taskId=ue9de94a5-075a-4b49-9ea9-46b1de9ff6b&amp;width=526" />
<strong>这里需要注意，我这里的数据集里面有4个不同的实验设计，一起校正。我认为，真正分析的时候，将他们区分出来，分别校正，会好一些</strong>。
<div class="highlight"><pre><span></span><code><span class="n">GSE5823_rma_BT474</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">affy</span><span class="o">::</span><span class="nf">rma</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">])</span>
<span class="nf">boxplot</span><span class="p">(</span><span class="n">GSE5823_rma_BT474</span><span class="p">,</span><span class="w"> </span><span class="n">las</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span>
<span class="w">        </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;#66C2A5&quot;</span><span class="p">,</span><span class="s">&quot;#FC8D62&quot;</span><span class="p">),</span><span class="m">2</span><span class="p">))</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630309848183-883b92f9-8b16-42fc-8724-59e8b864aa24.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=388&amp;id=uc711d15a&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=517&amp;originWidth=701&amp;originalType=binary&amp;ratio=1&amp;size=18495&amp;status=done&amp;style=none&amp;taskId=u0a0f017b-0715-4ce4-aa47-36f75281376&amp;width=526" />
芯片数据标准化之后看数据质量是否改善。
<div class="highlight"><pre><span></span><code><span class="nf">arrayQualityMetrics</span><span class="p">(</span><span class="n">GSE5823_rma</span><span class="p">,</span><span class="w"> </span>
<span class="w">                    </span><span class="n">outdir</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;affymetrix/GSE5823/GSE5823_QC_rma&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                    </span><span class="n">force</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span>
<span class="w">                    </span><span class="n">intgroup</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;group&quot;</span><span class="p">,</span>
<span class="w">                    </span><span class="n">do.logtransform</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
</code></pre></div>
和前面一样，就不多说啦。
下面我们使用gcrma方法进行标准化
<div class="highlight"><pre><span></span><code><span class="n">GSE5823_gcrma</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">gcrma</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">)</span>
<span class="n">GSE5823_expr_gcrma</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">exprs</span><span class="p">(</span><span class="n">GSE5823_gcrma</span><span class="p">)</span>
<span class="nf">boxplot</span><span class="p">(</span><span class="n">GSE5823_gcrma</span><span class="p">,</span><span class="w"> </span><span class="n">las</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cols</span><span class="p">)</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630310771990-8bc231fc-0c02-4c3c-9733-e1eacef99fac.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=388&amp;id=u16ef0c1c&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=517&amp;originWidth=701&amp;originalType=binary&amp;ratio=1&amp;size=24436&amp;status=done&amp;style=none&amp;taskId=ub6012afe-651b-4c00-b529-3ee10a019aa&amp;width=526" /></p>
<p>标准化后的数据和未标准化的数据是有很大区别的。
<div class="highlight"><pre><span></span><code><span class="nf">head</span><span class="p">(</span><span class="n">GSE5823_expr_gcrma</span><span class="p">)[,</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">]</span>
<span class="nf">head</span><span class="p">(</span><span class="n">GSE5823_exp</span><span class="p">)[,</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">]</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">GSE5823_expr_gcrma</span><span class="p">)</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">GSE5823_exp</span><span class="p">)</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630311364873-afb7d280-04cf-4756-9faa-8b6d52029f35.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=395&amp;id=u91cfeacc&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=526&amp;originWidth=644&amp;originalType=binary&amp;ratio=1&amp;size=64727&amp;status=done&amp;style=none&amp;taskId=u2c6a8d7c-92f6-46da-a534-7c6f31b33eb&amp;width=483" />
为标准化之前有135万行的数据，任何芯片都没有这么多探针的，我们标准化后就可以看见行名变成了探针名称，行数只有5万多。
我们在看，不同实验设计一起标准化合和分别标准化多多少少是有区别的。所以像本案例这类似的数据集，最好是分别标准化。
<div class="highlight"><pre><span></span><code><span class="n">GSE5823_gcrma_BT474</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">gcrma</span><span class="p">(</span><span class="n">GSE5823_data</span><span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">])</span>
<span class="n">GSE5823_expr_gcrma_BT474</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">exprs</span><span class="p">(</span><span class="n">GSE5823_gcrma_BT474</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">GSE5823_expr_gcrma</span><span class="p">)[,</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">]</span>
<span class="nf">head</span><span class="p">(</span><span class="n">GSE5823_expr_gcrma_BT474</span><span class="p">)[,</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">]</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630311766451-2eb38936-50e8-4023-be43-67c7537dfb74.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=319&amp;id=ue7de8a03&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=425&amp;originWidth=651&amp;originalType=binary&amp;ratio=1&amp;size=62646&amp;status=done&amp;style=none&amp;taskId=ud30765b0-f032-4d60-9c1c-9fdc59a35b8&amp;width=488" />
同样标准化后可以再检测一下数据质量。
<div class="highlight"><pre><span></span><code><span class="nf">arrayQualityMetrics</span><span class="p">(</span><span class="n">GSE5823_gcrma</span><span class="p">,</span><span class="w"> </span>
<span class="w">                    </span><span class="n">outdir</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;affymetrix/GSE5823/GSE5823_QC_gcrma&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                    </span><span class="n">force</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span>
<span class="w">                    </span><span class="n">intgroup</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;group&quot;</span><span class="p">,</span>
<span class="w">                    </span><span class="n">do.logtransform</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
</code></pre></div></p>
<div class="highlight"><pre><span></span><code><span class="n">sampleInfo</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="s">&quot;geodata/GSE5823/sampleInfo.txt&quot;</span><span class="p">,</span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">sampleInfo</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;sampleName&quot;</span><span class="p">,</span><span class="s">&quot;description&quot;</span><span class="p">)</span>

<span class="n">celFileNames</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="s">&quot;geodata/GSE5823/GSE5823_RAW&quot;</span><span class="p">,</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;.CEL.gz$&quot;</span><span class="p">,</span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
<span class="n">sampleName</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="n">celFileNames</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;[.]&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)[,</span><span class="m">1</span><span class="p">]</span>
<span class="n">celFlieInfo</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">fileName</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">celFileNames</span><span class="p">,</span><span class="n">sampleName</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sampleName</span><span class="p">)</span>
<span class="n">celFlieInfo</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">celFlieInfo</span><span class="p">,</span><span class="n">sampleInfo</span><span class="p">,</span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;sampleName&quot;</span><span class="p">)</span>
<span class="n">celFlieInfo</span><span class="o">$</span><span class="n">cellType</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="nf">str_split</span><span class="p">(</span><span class="n">celFlieInfo</span><span class="o">$</span><span class="n">description</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\\s+&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)[,</span><span class="m">3</span><span class="p">],</span><span class="w"> </span>
<span class="w">                                 </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)[,</span><span class="m">1</span><span class="p">]</span>
<span class="n">celFlieInfo</span><span class="o">$</span><span class="n">group</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">grepl</span><span class="p">(</span><span class="s">&quot;control&quot;</span><span class="p">,</span><span class="n">celFlieInfo</span><span class="o">$</span><span class="n">description</span><span class="p">),</span><span class="s">&quot;control&quot;</span><span class="p">,</span><span class="s">&quot;c-myc KD&quot;</span><span class="p">)</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">celFlieInfo</span><span class="p">,</span>
<span class="w">            </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE5823/Targets.txt&quot;</span><span class="p">,</span>
<span class="w">            </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span>
<span class="w">            </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
<span class="w">            </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
</code></pre></div>
<h4 id="7">7). 缺失值处理<a class="headerlink" href="#7" title="Permanent link">&para;</a></h4>
<p>我们标准化后可能有缺失值，缺失值主要来自原始图像信息处理过程 、数据后续处理、转换过程。如果有，那么我们需要处理这些缺失值。补充缺失值的方法有邻近平均值局部法和删除等。
<div class="highlight"><pre><span></span><code><span class="c1">#首先看一下有没有缺失值</span>
<span class="o">&gt;</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="nf">exprs</span><span class="p">(</span><span class="n">GSE5823_gcrma</span><span class="p">)))</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="m">0</span>
</code></pre></div>
在R中，impute包提供的impute.knn()函数来补充缺失值，该函数使用最近邻平均替换缺失值。我们这里模拟一下，缺失值处理。
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">impute</span><span class="p">)</span>
<span class="n">na_test</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">GSE5823_expr_gcrma</span>

<span class="n">row_index</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sample</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">50000</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">]</span>
<span class="n">col_index</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sample</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">20</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">]</span>

<span class="n">na_test</span><span class="p">[</span><span class="n">row_index</span><span class="p">,</span><span class="w"> </span><span class="n">col_index</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NA</span><span class="c1">#造出NA</span>
<span class="nf">sum</span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="n">na_test</span><span class="p">))</span>

<span class="n">na_knn</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">impute.knn</span><span class="p">(</span><span class="n">na_test</span><span class="p">,</span><span class="w"> </span><span class="n">maxp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30000</span><span class="p">)</span><span class="c1">#maxp默认的是1500，根据数据维度大小设置大些</span>
<span class="nf">sum</span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="n">na_test</span><span class="p">))</span>
<span class="c1">#比较一下原始的和补充之后的数值</span>
<span class="n">GSE5823_expr_gcrma</span><span class="p">[</span><span class="n">row_index</span><span class="p">,</span><span class="w"> </span><span class="n">col_index</span><span class="p">]</span>
<span class="n">na_knn</span><span class="o">$</span><span class="n">data</span><span class="p">[</span><span class="n">row_index</span><span class="p">,</span><span class="w"> </span><span class="n">col_index</span><span class="p">]</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630314520914-1aa7a2ac-e880-4791-b51b-c5ebd4450cbb.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=288&amp;id=u85c4b094&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=384&amp;originWidth=803&amp;originalType=binary&amp;ratio=1&amp;size=66919&amp;status=done&amp;style=none&amp;taskId=uf2ee6484-0524-439c-a3f7-9b7306dd70c&amp;width=602" />
我们发现部分探针填补后和原来的缺失值是和大的。你可以多运行几次看看。部分探针没有差别，但部分还是差别很大。如果缺失值不多，缺失值所在的行不是你要的探针(不是你研究的探针/基因)，我个人觉得直接删除该行会好一些。</p>
<h3 id="3-illumina">(3). Illumina芯片原始数据的下载与处理<a class="headerlink" href="#3-illumina" title="Permanent link">&para;</a></h3>
<h4 id="1_2">1). 数据的下载<a class="headerlink" href="#1_2" title="Permanent link">&para;</a></h4>
<p>GEO数据库中，Illumina芯片原始数据下载后解压，哪怕同一芯片，里面的数据也有些不同，甚至有的有缺失，作者并没有上传，比如数据集GSE32873，下载后的原始数据里面只有探针的注释文件。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630458543141-b05915fe-6358-45e6-867e-3d1e0233fac1.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=194&amp;id=ufc584201&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=259&amp;originWidth=764&amp;originalType=binary&amp;ratio=1&amp;size=26895&amp;status=done&amp;style=none&amp;taskId=ue9690a56-c07d-48f7-8063-ad2d97589fe&amp;width=573" />
下面我们下载几个数据集来比较一下。
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">GEOquery</span><span class="p">)</span>
<span class="c1">##下载芯片原始数据</span>
<span class="nf">ifelse</span><span class="p">(</span><span class="o">!</span><span class="nf">dir.exists</span><span class="p">(</span><span class="s">&quot;geodata&quot;</span><span class="p">),</span><span class="nf">dir.create</span><span class="p">(</span><span class="s">&quot;geodata&quot;</span><span class="p">),</span><span class="nf">message</span><span class="p">(</span><span class="s">&quot;文件夹已存在&quot;</span><span class="p">))</span>
<span class="nf">getGEOSuppFiles</span><span class="p">(</span><span class="s">&quot;GSE32873&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">baseDir</span><span class="w"> </span><span class="o">=</span><span class="s">&quot;geodata&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">makeDirectory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span><span class="w"> </span>
<span class="nf">getGEOSuppFiles</span><span class="p">(</span><span class="s">&quot;GSE140882&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">baseDir</span><span class="w"> </span><span class="o">=</span><span class="s">&quot;geodata&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">makeDirectory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span><span class="w"> </span>
<span class="nf">getGEOSuppFiles</span><span class="p">(</span><span class="s">&quot;GSE59183&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">baseDir</span><span class="w"> </span><span class="o">=</span><span class="s">&quot;geodata&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">makeDirectory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span><span class="w"> </span>
</code></pre></div>
下载后得到的原始数据文件GSE<strong><em>_RAW.tar。我们解压到GSE</em></strong>_RAW文件夹下。你需要查看一下里面有没有注释信息的文件，下面是3个不同的数据集，它们使用的是同一芯片。解压后会有相同的GPL文件。我们还可以看到，除了这2个文件，其他的GSM开头的就是样本数据文件，但他们个格式有所不同。前面说过Illumina芯片数据分析的4个起点： ① Pixel-level,每个像素点对应一个数据， tif/tiff 格式； ②  Bead-level,每个bead对应一个数据， 类似于affymetrix的单个probe数据， 具体数据包括txt/idat/Iocs/sdf/xml等多种格式； ③ Summary-level,每个bead type对应 —个数据，类似于affy的probe set数据， txt/CSV 格式； ④ 公共数据库存储的表达矩阵数据。这就看作者上传的是什么数据文件，通常都是 Bead-level的数据文件，所以会有txt/idat/Iocs/sdf/xml这样的数据格式文件，比如数据集GSE59183，而且该数据集还有tif格式的数据文件。但在GSE140882数据集中只有txt格式的Bead-level的数据。GSE32873数据集里面处理GPL文件外，就没有其他的，也就是说没有原始数据，所以该数据集你无法从原始数据开始分析，但作者提供了非标准化的表达矩阵文件。</p>
<p><img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630458879744-f128b553-35ee-47f3-9210-080d7ff388ee.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=116&amp;id=u682398e8&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=155&amp;originWidth=763&amp;originalType=binary&amp;ratio=1&amp;size=16709&amp;status=done&amp;style=none&amp;taskId=u5a4dfe2c-9880-44aa-bc63-e6faa187846&amp;width=572" />
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630458965380-43a49d5b-e5e0-4b4b-a98f-576103c88492.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=230&amp;id=u43b480d6&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=316&amp;originWidth=773&amp;originalType=binary&amp;ratio=1&amp;size=41904&amp;status=done&amp;style=none&amp;taskId=ub1c67995-35e0-447e-bed3-1486f00d783&amp;width=562" />
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630459214930-016146a2-5a97-4948-8744-637a3961016d.png#clientId=ue984c4d8-a874-4&amp;from=paste&amp;height=208&amp;id=uc39df09c&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=298&amp;originWidth=749&amp;originalType=binary&amp;ratio=1&amp;size=40529&amp;status=done&amp;style=none&amp;taskId=uee14995b-8462-40d9-8526-c2051557384&amp;width=524" />
再如GSE108369这个数据集，解压后是还是压缩包需要进一步再解压。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630548085778-3441049a-18c0-48d9-a5e7-836cf68f4c0e.png#clientId=u1315abab-3476-4&amp;from=paste&amp;height=273&amp;id=ua7e7458e&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=364&amp;originWidth=723&amp;originalType=binary&amp;ratio=1&amp;size=54550&amp;status=done&amp;style=none&amp;taskId=uee87cbe7-01ed-4733-91e5-06c2bba3e98&amp;width=542" /></p>
<h4 id="2_2">2). 数据的读入<a class="headerlink" href="#2_2" title="Permanent link">&para;</a></h4>
<p>关于 Illumina芯片原始数据读入使用什么R包在前面有介绍。数据的读入我们首先介绍beadarray包的<code>readIllumina()</code>函数。下面介绍<code>readIllumina()</code>函数。
<div class="highlight"><pre><span></span><code><span class="nf">readIllumina</span><span class="p">(</span><span class="n">dir</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;.&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">useImages</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span>
<span class="w">             </span><span class="n">illuminaAnnotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w"> </span><span class="n">sectionNames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w"> </span><span class="n">metricsFile</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Metrics.txt&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">             </span><span class="n">forceIScan</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">dec</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;.&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sampleSheet</span><span class="o">=</span><span class="s">&quot;sampleSheet.csv&quot;</span><span class="p">,</span><span class="kc">...</span><span class="p">)</span>
</code></pre></div>
dir就是数据所在的路径。useImages指定是否通过图像数据读入，设置为TRUE时，前体是有tif/tiff格式的数据。数据集GSE140882中是没有的，所以处理该数据集时不能通过这种方式。但数据集GSE59183数据很全，是可以的，其实就是如果useImage设置为TRUE，则使用文本文件中提供的珠子中心坐标从TIFF图像中提取强度。对于其他格式文件信息如下：</p>
<ul>
<li>.txt文本文件：列出数组部分中每个解码珠的ID、坐标和强度。强度经过了局部背景校正。如果useImages = FALSE，这些强度将用作分析的起点。</li>
<li>.sdf文件：Illumina的整个芯片或SAM的样本描述文件。这在beadarray中用于确定剖面的物理属性。</li>
<li>.locs文件：数组中所有珠子的位置(即包括所有无法解码的珠子)</li>
</ul>
<p>illuminaAnnotation参数：指定生成数据的Illumina平台的字符串。这是可选的，将有助于自动化对表达式数据的一些分析。目前这个参数的选择是Humanv4, Humanv3, Humanv2, Humanv1, Mousev2, Mousev1, mouse1p1, Ratv1。比如这里的芯片是Illumina HumanHT-12 V4.0 expression beadchip，对应的就是Humanv4。
读入数据集GSE140882的数据。
<div class="highlight"><pre><span></span><code><span class="c1">## 读取目录中的文件, bead-level</span>
<span class="nf">library</span><span class="p">(</span><span class="n">beadarray</span><span class="p">)</span>
<span class="n">path1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;geodata/GSE140882/GSE140882_RAW/&quot;</span>
<span class="n">GSE140882_BLdata</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readIllumina</span><span class="p">(</span><span class="n">dir</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">path1</span><span class="p">)</span>
<span class="c1">##添加注释</span>
<span class="nf">annotation</span><span class="p">(</span><span class="n">GSE140882_BLdata</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w">  </span><span class="nf">suggestAnnotation</span><span class="p">(</span><span class="n">GSE140882_BLdata</span><span class="p">)</span>

<span class="c1">## 查看数据</span>
<span class="nf">head</span><span class="p">(</span><span class="n">GSE140882_BLdata</span><span class="p">[[</span><span class="m">10</span><span class="p">]])</span>
<span class="nf">slotNames</span><span class="p">(</span><span class="n">GSE140882_BLdata</span><span class="p">)</span>
<span class="nf">str</span><span class="p">(</span><span class="n">GSE140882_BLdata</span><span class="p">)</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">GSE140882_BLdata</span><span class="p">)</span>
<span class="c1">## 保存数据</span>
<span class="nf">save</span><span class="p">(</span><span class="n">GSE140882_BLdata</span><span class="p">,</span><span class="w"> </span>
<span class="w">     </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE140882/GSE140882_beadlevel1.Rda&quot;</span><span class="p">)</span>
</code></pre></div>
因为这个数据集只有txt文件，没有其他的格式文件，所以在Targets中相应的变量为NA，如果有，这些数据也会被读入进来。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630463362511-f43966b4-0915-4cb1-9cf8-09af1c562803.png#clientId=u1315abab-3476-4&amp;from=paste&amp;height=293&amp;id=u81d086a6&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=390&amp;originWidth=953&amp;originalType=binary&amp;ratio=1&amp;size=44280&amp;status=done&amp;style=none&amp;taskId=ud7d63a63-ab54-4efa-99f7-c3551e8bd16&amp;width=715" />
取一个值看看，第一列是探针ID，第二列和第三列是坐标值，第四列是信号值，用来指示表达水平。
<div class="highlight"><pre><span></span><code><span class="o">&gt;</span><span class="w"> </span><span class="nf">head</span><span class="p">(</span><span class="n">GSE140882_BLdata</span><span class="p">[[</span><span class="m">10</span><span class="p">]])</span>
<span class="w">     </span><span class="n">ProbeID</span><span class="w">     </span><span class="n">GrnX</span><span class="w">      </span><span class="n">GrnY</span><span class="w"> </span><span class="n">Grn</span>
<span class="p">[</span><span class="m">1</span><span class="p">,]</span><span class="w">   </span><span class="m">10008</span><span class="w"> </span><span class="m">1266.670</span><span class="w"> </span><span class="m">14383.660</span><span class="w"> </span><span class="m">747</span>
<span class="p">[</span><span class="m">2</span><span class="p">,]</span><span class="w">   </span><span class="m">10008</span><span class="w"> </span><span class="m">1816.482</span><span class="w">  </span><span class="m">2781.520</span><span class="w"> </span><span class="m">523</span>
<span class="p">[</span><span class="m">3</span><span class="p">,]</span><span class="w">   </span><span class="m">10008</span><span class="w"> </span><span class="m">1944.487</span><span class="w">  </span><span class="m">3564.491</span><span class="w"> </span><span class="m">342</span>
<span class="p">[</span><span class="m">4</span><span class="p">,]</span><span class="w">   </span><span class="m">10008</span><span class="w"> </span><span class="m">2116.688</span><span class="w">  </span><span class="m">1204.003</span><span class="w"> </span><span class="m">444</span>
<span class="p">[</span><span class="m">5</span><span class="p">,]</span><span class="w">   </span><span class="m">10008</span><span class="w">  </span><span class="m">684.497</span><span class="w">  </span><span class="m">3372.999</span><span class="w"> </span><span class="m">639</span>
<span class="p">[</span><span class="m">6</span><span class="p">,]</span><span class="w">   </span><span class="m">10008</span><span class="w"> </span><span class="m">1445.521</span><span class="w">  </span><span class="m">5622.521</span><span class="w"> </span><span class="m">554</span>
</code></pre></div>
dim函数查看数据维度，总共有27芯片，每个芯片有的Beads数也被列出来。
<div class="highlight"><pre><span></span><code><span class="o">&gt;</span><span class="w"> </span><span class="nf">dim</span><span class="p">(</span><span class="n">GSE140882_BLdata</span><span class="p">)</span>
<span class="w"> </span><span class="n">nArrays</span><span class="w">  </span><span class="n">nBeads1</span><span class="w">  </span><span class="n">nBeads2</span><span class="w">  </span><span class="n">nBeads3</span><span class="w">  </span><span class="n">nBeads4</span><span class="w">  </span><span class="n">nBeads5</span><span class="w">  </span><span class="n">nBeads6</span><span class="w">  </span><span class="n">nBeads7</span><span class="w">  </span><span class="n">nBeads8</span><span class="w">  </span><span class="n">nBeads9</span><span class="w"> </span><span class="n">nBeads10</span><span class="w"> </span>
<span class="w">      </span><span class="m">27</span><span class="w">  </span><span class="m">1049985</span><span class="w">   </span><span class="m">979585</span><span class="w">  </span><span class="m">1032451</span><span class="w">   </span><span class="m">999011</span><span class="w">  </span><span class="m">1020257</span><span class="w">  </span><span class="m">1017478</span><span class="w">   </span><span class="m">987157</span><span class="w">   </span><span class="m">975496</span><span class="w">   </span><span class="m">976775</span><span class="w">   </span><span class="m">976596</span><span class="w"> </span>
<span class="n">nBeads11</span><span class="w"> </span><span class="n">nBeads12</span><span class="w"> </span><span class="n">nBeads13</span><span class="w"> </span><span class="n">nBeads14</span><span class="w"> </span><span class="n">nBeads15</span><span class="w"> </span><span class="n">nBeads16</span><span class="w"> </span><span class="n">nBeads17</span><span class="w"> </span><span class="n">nBeads18</span><span class="w"> </span><span class="n">nBeads19</span><span class="w"> </span><span class="n">nBeads20</span><span class="w"> </span><span class="n">nBeads21</span><span class="w"> </span>
<span class="w">  </span><span class="m">944583</span><span class="w">   </span><span class="m">953363</span><span class="w">  </span><span class="m">1029041</span><span class="w">  </span><span class="m">1043010</span><span class="w">  </span><span class="m">1095933</span><span class="w">  </span><span class="m">1040761</span><span class="w">  </span><span class="m">1086987</span><span class="w">  </span><span class="m">1097331</span><span class="w">  </span><span class="m">1093468</span><span class="w">  </span><span class="m">1117748</span><span class="w">  </span><span class="m">1112479</span><span class="w"> </span>
<span class="n">nBeads22</span><span class="w"> </span><span class="n">nBeads23</span><span class="w"> </span><span class="n">nBeads24</span><span class="w"> </span><span class="n">nBeads25</span><span class="w"> </span><span class="n">nBeads26</span><span class="w"> </span><span class="n">nBeads27</span><span class="w"> </span>
<span class="w"> </span><span class="m">1115702</span><span class="w">  </span><span class="m">1126377</span><span class="w">  </span><span class="m">1120570</span><span class="w">  </span><span class="m">1120508</span><span class="w">  </span><span class="m">1125320</span><span class="w">  </span><span class="m">1122439</span><span class="w"> </span>
</code></pre></div>
接下来，读入GSE59183数据集，通过设置useImages参数来确认是否从TIF图像中读入数据。
<div class="highlight"><pre><span></span><code><span class="c1">## 文件所在目录</span>
<span class="n">path2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE59183/GSE59183_RAW&quot;</span>
<span class="c1">## 读取目录中的文件, bead-level</span>
<span class="nf">library</span><span class="p">(</span><span class="n">beadarray</span><span class="p">)</span>
<span class="n">data_ntiff</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readIllumina</span><span class="p">(</span><span class="n">dir</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">path2</span><span class="p">,</span><span class="w"> </span><span class="n">useImages</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">dec</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;.&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                           </span><span class="n">illuminaAnnotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Humanv4&quot;</span><span class="p">)</span>
<span class="n">data_tiff</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readIllumina</span><span class="p">(</span><span class="n">path2</span><span class="p">,</span><span class="w"> </span><span class="n">useImages</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">,</span><span class="w"> </span><span class="n">dec</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;.&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                          </span><span class="n">illuminaAnnotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Humanv4&quot;</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">data_ntiff</span><span class="p">[[</span><span class="m">1</span><span class="p">]])</span>
<span class="nf">head</span><span class="p">(</span><span class="n">data_tiff</span><span class="p">[[</span><span class="m">1</span><span class="p">]])</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630462718750-4a40813c-8bdc-442b-bca2-29683e96222a.png#clientId=u1315abab-3476-4&amp;from=paste&amp;height=319&amp;id=ua55f318e&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=426&amp;originWidth=739&amp;originalType=binary&amp;ratio=1&amp;size=64599&amp;status=done&amp;style=none&amp;taskId=u64f817c5-3345-4373-8571-bf593b0da90&amp;width=554" />
我们发现，useImages参数是否设置为TRUE，他们的信号强度值(Grn)不一样。GrnF是图片的前景值，GrnB是图片的背景值。前面介绍useImage设置为TRUE，则使用文本文件中提供的珠子中心坐标从TIFF图像中提取强度。前景值是从图像的锐化版本计算出来的。背景值计算为围绕珠心的17x17像素正方形中5个最低像素的中值。所以useImages参数是否设置为TRUE，导致计算的背景值方法不一样，所以Grn的值有区别，但整体的分析的差异应该是不大的。我们的原始数据中大多是没有tif格式的图像文件的，这种你就不用纠结要不是设置为TRUE，如果有，其实设置与否都行，我没有去比较这2者之间最后的差异。但我认为是没有太大差异的（我们验证过）。还要说一下Grn的值其实就是前景值(GrnF)减去背景值(GrnB)。背景值计算的算法参考文献如下：</p>
<ul>
<li>Kuhn K, Baker SC, Chudin E, Lieu MH, Oeser S, Bennett H, Rigault P, Barker D, McDaniel TK, Chee MS. A novel, high-performance random array platform for quantitative gene expression profiling. Genome Research. 2004;14(11):2347-2356</li>
<li>Smith ML, Dunning MJ, Tavare S, Lynch AG. Identification and correction of previously unreported spatial phenomena using raw Illumina BeadArray data. BMC Bioinformatics. 2010;11:208</li>
</ul>
<p>下面介绍一下计算过程。
首先我们需要读入tif文件。用readTIFF()函数读入，该函数一次只能读入一个文件。可以用lapply()函数遍历读入。
<div class="highlight"><pre><span></span><code><span class="c1">### read TIFF </span>
<span class="n">tif_files_path</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="s">&quot;geodata/GSE59183/GSE59183_RAW/&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                        </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;tif$&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
<span class="n">tif_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lapply</span><span class="p">(</span><span class="n">tif_files_path</span><span class="p">,</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">p</span><span class="p">){</span><span class="nf">readTIFF</span><span class="p">(</span><span class="n">p</span><span class="p">)})</span>
</code></pre></div>
我们去第一个图片信息查看一下
<div class="highlight"><pre><span></span><code><span class="nf">head</span><span class="p">(</span><span class="n">tif_data</span><span class="p">[[</span><span class="m">1</span><span class="p">]])[,</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">]</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">tif_data</span><span class="p">[[</span><span class="m">1</span><span class="p">]])</span>
<span class="nf">class</span><span class="p">(</span><span class="n">tif_data</span><span class="p">[[</span><span class="m">1</span><span class="p">]])</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630468251297-33e66b04-c531-4878-9b76-bd2236d9011e.png#clientId=u1315abab-3476-4&amp;from=paste&amp;height=251&amp;id=u1c0bfde4&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=334&amp;originWidth=581&amp;originalType=binary&amp;ratio=1&amp;size=26218&amp;status=done&amp;style=none&amp;taskId=u14b61f93-2a67-4116-8959-1e34f0a1e28&amp;width=436" />
读入的图片数据是一个矩阵。
我们将前面读入的 data_ntiff数据赋值给一个新变量进行操作。getBeadData()函数获取芯片数据。array=1，表示第一个芯片，what是芯片数据的列名。下面是提取第一个芯片的坐标信息。
<div class="highlight"><pre><span></span><code><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data_ntiff</span>
<span class="n">xcoords</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">getBeadData</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="n">array</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">what</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GrnX&quot;</span><span class="p">)</span>
<span class="n">ycoords</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">getBeadData</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="n">array</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">what</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GrnY&quot;</span><span class="p">)</span>
</code></pre></div>
下面我们计算前面读入tif图片矩阵数据的背景值。有2个函数可以计算。medianBackground()函数和illuminaBackground()函数。2个函数计算结果差不多。
<div class="highlight"><pre><span></span><code><span class="n">bg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">medianBackground</span><span class="p">(</span><span class="n">tif_data</span><span class="p">[[</span><span class="m">1</span><span class="p">]],</span><span class="w"> </span><span class="nf">cbind</span><span class="p">(</span><span class="n">xcoords</span><span class="p">,</span><span class="w"> </span><span class="n">ycoords</span><span class="p">))</span>
<span class="n">bg2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">illuminaBackground</span><span class="p">(</span><span class="n">tif_data</span><span class="p">[[</span><span class="m">1</span><span class="p">]],</span><span class="w"> </span><span class="nf">cbind</span><span class="p">(</span><span class="n">xcoords</span><span class="p">,</span><span class="w"> </span><span class="n">ycoords</span><span class="p">))</span>
</code></pre></div>
将计算的背景值添加到data中。what指定添加数据的列名
<div class="highlight"><pre><span></span><code><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">insertBeadData</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="n">array</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">what</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GrnRB&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">bg</span><span class="p">)</span>
<span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">insertBeadData</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="n">array</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">what</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GrnRB2&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">bg2</span><span class="p">)</span>
</code></pre></div>
计算图形前景值，利用illuminaSharpen()函数先过滤一下数据，再用illuminaForeground()函数计算。backgroundCorrectSingleSection函数计算信号值（前景值-背景值=信号值）。
<div class="highlight"><pre><span></span><code><span class="c1"># apply Illumina&#39;s image filtering</span>
<span class="n">Tif</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">illuminaSharpen</span><span class="p">(</span><span class="n">tif_data</span><span class="p">[[</span><span class="m">1</span><span class="p">]])</span>
<span class="n">fg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">illuminaForeground</span><span class="p">(</span><span class="n">Tif</span><span class="p">,</span><span class="w"> </span><span class="nf">cbind</span><span class="p">(</span><span class="n">xcoords</span><span class="p">,</span><span class="w"> </span><span class="n">ycoords</span><span class="p">))</span>
<span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">insertBeadData</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="n">array</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">what</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GrnF&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">fg</span><span class="p">)</span>
<span class="c1">##</span>
<span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">backgroundCorrectSingleSection</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="n">array</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span>
<span class="w">                                       </span><span class="n">fg</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GrnF&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                                       </span><span class="n">bg</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GrnRB&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                                       </span><span class="n">newName</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GrnR&quot;</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">data</span><span class="p">[[</span><span class="m">1</span><span class="p">]])</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630472547814-3b79a841-fcd8-4d5a-9fa5-810cdb1ecf75.png#clientId=u1315abab-3476-4&amp;from=paste&amp;height=172&amp;id=u80ec25a9&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=229&amp;originWidth=912&amp;originalType=binary&amp;ratio=1&amp;size=42311&amp;status=done&amp;style=none&amp;taskId=uff80ddb5-cd3d-4645-9e22-0b9951d1488&amp;width=684" />
 我们可以看到我们自己计算的信号值GrnR与前面直接读入数据Grn是差不多的。
也就是说， 对于illumina数据而言，就不需要矫正背景数据了，就没有了背景矫正这个环节，在useImages参数设置为TRUE时，得到的数据已经进行了背景校正。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630472987572-98c5bd8b-6411-4ff2-ada3-00919f6d4b4d.png#clientId=u1315abab-3476-4&amp;from=paste&amp;height=470&amp;id=u9b7c25cd&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=627&amp;originWidth=917&amp;originalType=binary&amp;ratio=1&amp;size=109306&amp;status=done&amp;style=none&amp;taskId=ud3c1a9e7-17e8-480f-ba98-f5cf742578d&amp;width=688" />
添加注释
<div class="highlight"><pre><span></span><code><span class="nf">suggestAnnotation</span><span class="p">(</span><span class="n">data_ntiff</span><span class="p">,</span><span class="w"> </span><span class="n">verbose</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
<span class="nf">annotation</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">suggestAnnotation</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="n">verbose</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
</code></pre></div>
查看数据结构
<div class="highlight"><pre><span></span><code><span class="nf">slotNames</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
<span class="n">data</span><span class="o">@</span><span class="n">sectionData</span><span class="o">$</span><span class="n">Targets</span><span class="o">$</span><span class="n">greenImage</span>
<span class="n">data</span><span class="o">@</span><span class="n">sectionData</span><span class="o">$</span><span class="n">Targets</span><span class="o">$</span><span class="n">textFile</span>
<span class="nf">head</span><span class="p">(</span><span class="n">data</span><span class="p">[[</span><span class="m">1</span><span class="p">]])</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">data</span><span class="p">[[</span><span class="m">1</span><span class="p">]])</span>
<span class="nf">sectionNames</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
<span class="nf">numBeads</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</code></pre></div>
接着，我们读入idat的数据。使用readIdatFiles()函数读入数据。这里以这个数据集GSE113440为例
<div class="highlight"><pre><span></span><code><span class="n">idat_dir</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;geodata/GSE113440/GSE113440_RAW&quot;</span>
<span class="n">idat_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="n">idat_dir</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;.idat$&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
<span class="c1">## 读取idat files</span>
<span class="nf">library</span><span class="p">(</span><span class="n">beadarray</span><span class="p">)</span>
<span class="n">GSE113440_idat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readIdatFiles</span><span class="p">(</span><span class="n">idatFiles</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">idat_files</span><span class="p">)</span>
</code></pre></div>
可以通过索引的方式提取原始数据的表达矩阵
<div class="highlight"><pre><span></span><code><span class="c1">## 原始表达矩阵数据</span>
<span class="n">GSE113440_expr_idat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">GSE113440_idat</span><span class="o">@</span><span class="n">assayData</span><span class="o">$</span><span class="n">exprs</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">GSE113440_expr_idat</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">GSE113440_expr_idat</span><span class="p">)[,</span><span class="m">1</span><span class="o">:</span><span class="m">3</span><span class="p">]</span>
</code></pre></div>
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630503413021-9d9fb670-c09d-4eba-89e6-29bca305a8c7.png#clientId=u1315abab-3476-4&amp;from=paste&amp;height=214&amp;id=u698eb5fb&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=285&amp;originWidth=1099&amp;originalType=binary&amp;ratio=1&amp;size=48763&amp;status=done&amp;style=none&amp;taskId=u91849fdf-c954-4f0e-809e-68e81b5c079&amp;width=824" />
保存数据用于后续的分析。
<div class="highlight"><pre><span></span><code><span class="nf">save</span><span class="p">(</span><span class="n">GSE113440_idat</span><span class="p">,</span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE113440/GSE113440_idat.Rdata&quot;</span><span class="p">)</span>
</code></pre></div>
下面我们介绍如何使用lumi包读取illumina芯片数据。GSE108369这个数据集解压后需要处理一下，将所有GSM样本的txt文件放在同一目录下，方便后面处理。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630548514727-d8f0c543-0510-48de-893e-322dc84664d1.png#clientId=u1315abab-3476-4&amp;from=paste&amp;height=233&amp;id=u17ef88c2&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=311&amp;originWidth=799&amp;originalType=binary&amp;ratio=1&amp;size=42045&amp;status=done&amp;style=none&amp;taskId=u6c681e78-9fda-4ba0-a5d0-53cdd60f264&amp;width=599" />
虽然都是txt文件，用的芯片也是一样的，但这里面的数据有所区别，数据总共有6列。第一列就是探针，第二列是信号值的log值，第三列是原始的信号值，第5列是该探针的重复数(Beads数)，第四列就是该探针在所有beads中信号值的标准差，第6列就是p值。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630548775173-2d627d24-fc6d-48a6-aa1e-944ff1213d68.png#clientId=u1315abab-3476-4&amp;from=paste&amp;height=193&amp;id=uf68abdb3&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=257&amp;originWidth=859&amp;originalType=binary&amp;ratio=1&amp;size=34979&amp;status=done&amp;style=none&amp;taskId=ud58e953f-c6b1-4f94-b42f-1c0573dce64&amp;width=644" />
先整理一个target文件：
<div class="highlight"><pre><span></span><code><span class="c1">## 整理targets</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="n">files</span>
<span class="n">targets</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">FileName</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">files</span><span class="p">,</span>
<span class="w">                      </span><span class="n">sample_id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_match</span><span class="p">(</span><span class="n">files</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;GSM\\d+&quot;</span><span class="p">)[,</span><span class="w"> </span><span class="m">1</span><span class="p">],</span>
<span class="w">                      </span><span class="n">sample_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_match</span><span class="p">(</span><span class="n">files</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;eum_(.*).txt&quot;</span><span class="p">)[,</span><span class="w"> </span><span class="m">2</span><span class="p">],</span>
<span class="w">                      </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_match</span><span class="p">(</span><span class="n">files</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;eum_(.*)_&quot;</span><span class="p">)[,</span><span class="w"> </span><span class="m">2</span><span class="p">],</span>
<span class="w">                      </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_match</span><span class="p">(</span><span class="n">files</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;GSM\\d+&quot;</span><span class="p">)[,</span><span class="w"> </span><span class="m">1</span><span class="p">],</span>
<span class="w">                      </span><span class="n">stringsAsFactors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
</code></pre></div>
我们去掉第二列的数据，其他列的列名"ProbeID", "AVG_SIGNAL","BEAD_STD",'Avg_NBEADS','DETECTION'都要加上样本ID，如下这种样子。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630551481922-59e3465b-73b2-4cbc-8b78-706f25a3ef0a.png#clientId=u1315abab-3476-4&amp;from=paste&amp;height=90&amp;id=u686f3f69&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=120&amp;originWidth=1184&amp;originalType=binary&amp;ratio=1&amp;size=17333&amp;status=done&amp;style=none&amp;taskId=ud3d2065f-3514-4a0c-a3c9-9cd3efce052&amp;width=888" />我们读入数据然后再重新命名列名后输出就可以啦。
<div class="highlight"><pre><span></span><code><span class="nf">dir.create</span><span class="p">(</span><span class="s">&quot;geodata/GSE108369/GSE108369_RAW_processed&quot;</span><span class="p">)</span>
<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">files</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="c1"># read_tsv读入，dplyr::select选择第1列和第3到第6列</span>
<span class="w">  </span><span class="n">tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">readr</span><span class="o">::</span><span class="nf">read_tsv</span><span class="p">(</span><span class="n">files</span><span class="p">[</span><span class="n">i</span><span class="p">])</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">3</span><span class="o">:</span><span class="m">6</span><span class="p">)</span>

<span class="w">  </span><span class="c1"># 改列名</span>
<span class="w">  </span><span class="nf">colnames</span><span class="p">(</span><span class="n">tmp</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;ProbeID&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;AVG_SIGNAL&quot;</span><span class="p">,</span><span class="s">&quot;BEAD_STD&quot;</span><span class="p">,</span><span class="s">&#39;Avg_NBEADS&#39;</span><span class="p">,</span><span class="s">&#39;DETECTION&#39;</span><span class="p">),</span>
<span class="w">                          </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">targets</span><span class="o">$</span><span class="n">sample_id</span><span class="p">[</span><span class="n">i</span><span class="p">])</span>
<span class="w">  </span><span class="c1"># 写出</span>
<span class="w">  </span><span class="n">readr</span><span class="o">::</span><span class="nf">write_tsv</span><span class="p">(</span><span class="n">tmp</span><span class="p">,</span><span class="w"> </span><span class="n">path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;geodata/GSE108369/GSE108369_RAW_processed/&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">basename</span><span class="p">(</span><span class="n">files</span><span class="p">[</span><span class="n">i</span><span class="p">])))</span>
<span class="w">  </span><span class="c1"># write.table(tmp, file = basename(files[i]), sep = &quot;\t&quot;, col.names = T,row.names = F, quote = F)</span>
<span class="p">}</span>
</code></pre></div>
然后再读入重新命名列名的数据。
<div class="highlight"><pre><span></span><code><span class="c1">## 应用 lumi读取文件 </span>
<span class="nf">library</span><span class="p">(</span><span class="n">lumi</span><span class="p">)</span>
<span class="n">GSE108369_lumi</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lumiR.batch</span><span class="p">(</span><span class="n">fileList</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;geodata/GSE108369/GSE108369_RAW_processed/&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                                                </span><span class="nf">basename</span><span class="p">(</span><span class="n">files</span><span class="p">)),</span><span class="w"> </span>
<span class="w">                              </span><span class="n">detectionTh</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.01</span><span class="p">,</span>
<span class="w">                              </span><span class="n">sampleInfoFile</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">targets</span><span class="p">)</span>

<span class="c1">## exprssion matrix</span>
<span class="n">GSE108369_expr_lumi</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">exprs</span><span class="p">(</span><span class="n">GSE108369_lumi</span><span class="p">)</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">GSE108369_expr_lumi</span><span class="p">)</span>
<span class="nf">sum</span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="n">GSE108369_expr_lumi</span><span class="p">))</span><span class="c1">#判断是否有缺失值，如果有的话就返回1，再计算总和就可以知道有几个缺失值</span>

<span class="c1">## save data </span>
<span class="nf">save</span><span class="p">(</span><span class="n">GSE108369_lumi</span><span class="p">,</span><span class="w"> </span><span class="n">GSE108369_expr_lumi</span><span class="p">,</span><span class="w"> </span>
<span class="w">     </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE108369/GSE108369_lumi.Rdata&quot;</span><span class="p">)</span>
</code></pre></div>
最后介绍一下limma包读入数据，limma包中read.ilmn函数适合读入下面这种类型的数据，下载后的原始数据只在一个文件中。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630554641937-7e98300f-f1a8-4ea9-aa5b-c89b30620fde.png#clientId=u1315abab-3476-4&amp;from=paste&amp;height=163&amp;id=ua10cb8ef&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=217&amp;originWidth=824&amp;originalType=binary&amp;ratio=1&amp;size=35741&amp;status=done&amp;style=none&amp;taskId=ub4c62323-a974-477f-ad67-a4e63bfef45&amp;width=618" />
有一些是非标准化的数据。比如数据集GSE42242。
<img alt="image.png" src="https://cdn.nlark.com/yuque/0/2021/png/1234840/1630569626242-80ba2e41-2ac1-4144-a327-06f8db30cffd.png#clientId=u1315abab-3476-4&amp;from=paste&amp;height=164&amp;id=lCkIa&amp;margin=%5Bobject%20Object%5D&amp;name=image.png&amp;originHeight=252&amp;originWidth=962&amp;originalType=binary&amp;ratio=1&amp;size=46179&amp;status=done&amp;style=none&amp;taskId=u03097219-8e56-402b-9310-2ce1aafdcaf&amp;width=626" />
<div class="highlight"><pre><span></span><code><span class="n">preview</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_tsv</span><span class="p">(</span><span class="s">&quot;geodata/GSE42242/GSE42242_non-normalized.txt&quot;</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">preview</span><span class="p">)</span>

<span class="n">GSE42242_non_norm</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.ilmn</span><span class="p">(</span><span class="n">files</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE42242/GSE42242_non-normalized.txt&quot;</span><span class="p">,</span>
<span class="w">                               </span><span class="n">expr</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SAMPLE&quot;</span><span class="p">,</span><span class="c1">#这里要根据数据实际的样子来改，有时候又变成了sample全</span>
<span class="w">                               </span><span class="n">other.columns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Detection&quot;</span><span class="p">,</span>
<span class="w">                               </span><span class="n">probeid</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ID_REF&quot;</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">GSE42242_non_norm</span><span class="o">$</span><span class="n">E</span><span class="p">)</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">GSE42242_non_norm</span><span class="o">$</span><span class="n">E</span><span class="p">)</span>
<span class="c1">## save data </span>
<span class="nf">save</span><span class="p">(</span><span class="n">GSE42242_non_norm</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE42242/GSE42242_non-normalized.Rdata&quot;</span><span class="p">)</span>
</code></pre></div>
但是有的不规则的，我们要重新改一下列名。比如GSE32873这个数据集。
<div class="highlight"><pre><span></span><code><span class="n">preview</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_tsv</span><span class="p">(</span><span class="s">&quot;geodata/GSE32873/GSE32873_non_normalized.txt&quot;</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">preview</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">preview</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;ID_REF&quot;</span><span class="p">,</span>
<span class="w">                       </span><span class="s">&quot;AllStar_control_1&quot;</span><span class="p">,</span><span class="s">&quot;Detection.Pval&quot;</span><span class="p">,</span>
<span class="w">                       </span><span class="s">&quot;AllStar_control_2&quot;</span><span class="p">,</span><span class="s">&quot;Detection.Pval&quot;</span><span class="p">,</span>
<span class="w">                       </span><span class="s">&quot;AllStar_control_3&quot;</span><span class="p">,</span><span class="s">&quot;Detection.Pval&quot;</span><span class="p">,</span>
<span class="w">                       </span><span class="s">&quot;AllStar_UBE2C_1&quot;</span><span class="p">,</span><span class="s">&quot;Detection.Pval&quot;</span><span class="p">,</span><span class="w">          </span>
<span class="w">                       </span><span class="s">&quot;AllStar_UBE2C_2&quot;</span><span class="p">,</span><span class="s">&quot;Detection.Pval&quot;</span><span class="p">,</span><span class="w">          </span>
<span class="w">                       </span><span class="s">&quot;AllStar_UBE2C_3&quot;</span><span class="p">,</span><span class="s">&quot;Detection.Pval&quot;</span><span class="p">,</span><span class="w">          </span>
<span class="w">                       </span><span class="s">&quot;AllStar_MELK_1&quot;</span><span class="w"> </span><span class="p">,</span><span class="s">&quot;Detection.Pval&quot;</span><span class="p">,</span><span class="w">           </span>
<span class="w">                       </span><span class="s">&quot;AllStar_MELK_2&quot;</span><span class="w"> </span><span class="p">,</span><span class="s">&quot;Detection.Pval&quot;</span><span class="w"> </span><span class="p">,</span><span class="w">          </span>
<span class="w">                       </span><span class="s">&quot;AllStar_MELK_3&quot;</span><span class="w"> </span><span class="p">,</span><span class="s">&quot;Detection.Pval&quot;</span><span class="w">  </span>
<span class="w">                       </span><span class="p">)</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">preview</span><span class="p">,</span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE32873/GSE32873_non_normalized_new.txt&quot;</span><span class="p">,</span>
<span class="w">            </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
<span class="n">GSE32873_non_norm</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.ilmn</span><span class="p">(</span><span class="n">files</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE32873/GSE32873_non_normalized_new.txt&quot;</span><span class="p">,</span>
<span class="w">                               </span><span class="n">expr</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;AllStar_&quot;</span><span class="p">,</span><span class="c1">#这里要根据数据实际的样子来改，有时候又变成了sample全</span>
<span class="w">                               </span><span class="n">other.columns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Detection&quot;</span><span class="p">,</span>
<span class="w">                               </span><span class="n">probeid</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ID_REF&quot;</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">GSE32873_non_norm</span><span class="o">$</span><span class="n">E</span><span class="p">)</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">GSE32873_non_norm</span><span class="o">$</span><span class="n">E</span><span class="p">)</span>
<span class="c1">## save data </span>
<span class="nf">save</span><span class="p">(</span><span class="n">GSE42242_non_norm</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE32873/GSE32873_non-normalized.Rdata&quot;</span><span class="p">)</span>
</code></pre></div>
总结：我们解压后的数据文件，可以打开看一下，如果数据量很大，每个GSM整百万行的那种，一般都是bead-level的数据，用beadarray::readIllumina
函数读取；对summary-level来说，如果是idat格式的，可以用 beadarray::readIdatFiles()来读取
。看到数据有四列，"ProbeID", "AVG_SIGNAL", "BEAD_STD", 'Avg_NBEADS', 'DETECTION'，需要做转换，就是改个名字+GSM的ID
，limma包读取一个列sample一列detection这种数据。</p>
<h4 id="3_1">3). 数据的标准化<a class="headerlink" href="#3_1" title="Permanent link">&para;</a></h4>
<p><strong>A.数据集GSE140882的标准化</strong>
前面下载数据和读入的代码：
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">beadarray</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">GEOquery</span><span class="p">)</span>
<span class="c1">##下载芯片原始数据</span>
<span class="nf">ifelse</span><span class="p">(</span><span class="o">!</span><span class="nf">dir.exists</span><span class="p">(</span><span class="s">&quot;geodata&quot;</span><span class="p">),</span><span class="nf">dir.create</span><span class="p">(</span><span class="s">&quot;geodata&quot;</span><span class="p">),</span><span class="nf">message</span><span class="p">(</span><span class="s">&quot;文件夹已存在&quot;</span><span class="p">))</span>
<span class="nf">getGEOSuppFiles</span><span class="p">(</span><span class="s">&quot;GSE140882&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">baseDir</span><span class="w"> </span><span class="o">=</span><span class="s">&quot;geodata&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">makeDirectory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span><span class="w"> </span>

<span class="n">path</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;geodata/GSE140882/GSE140882_RAW/&quot;</span>
<span class="n">GSE140882_BLdata</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readIllumina</span><span class="p">(</span><span class="n">dir</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">path</span><span class="p">)</span>
<span class="c1">## add annotation</span>
<span class="nf">annotation</span><span class="p">(</span><span class="n">GSE140882_BLdata</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w">  </span><span class="nf">suggestAnnotation</span><span class="p">(</span><span class="n">GSE140882_BLdata</span><span class="p">)</span>
<span class="nf">save</span><span class="p">(</span><span class="n">GSE140882_BLdata</span><span class="p">,</span><span class="w"> </span>
<span class="w">     </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE140882/GSE140882_beadlevel1.Rda&quot;</span><span class="p">)</span>
</code></pre></div>
加载前面处理的数据。提取targets仅仅是一部分信息，还需要自己整理。
<div class="highlight"><pre><span></span><code><span class="nf">load</span><span class="p">(</span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;geodata/GSE140882/GSE140882_beadlevel1.Rda&quot;</span><span class="p">)</span>
<span class="n">BLdata</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">GSE140882_BLdata</span>
<span class="nf">rm</span><span class="p">(</span><span class="n">GSE140882_BLdata</span><span class="p">)</span>
<span class="n">targets</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">BLdata</span><span class="o">@</span><span class="n">sectionData</span><span class="o">$</span><span class="n">Targets</span><span class="c1">#</span>
</code></pre></div></p>
<h3 id="4-soft">(4). 下载SOFT格式的数据<a class="headerlink" href="#4-soft" title="Permanent link">&para;</a></h3>
<p><div class="highlight"><pre><span></span><code><span class="n">gset</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">getGEO</span><span class="p">(</span><span class="n">GEO</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GSE7765&quot;</span><span class="p">,</span><span class="n">AnnotGPL</span><span class="o">=</span><span class="bp">F</span><span class="w"> </span><span class="p">,</span><span class="n">destdir</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;./geodata&quot;</span><span class="p">)</span>
<span class="n">exp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">exprs</span><span class="p">(</span><span class="n">gset</span><span class="p">[[</span><span class="m">1</span><span class="p">]])</span>
<span class="nf">head</span><span class="p">(</span><span class="n">exp</span><span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">])</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="o">&gt;</span><span class="w"> </span><span class="nf">head</span><span class="p">(</span><span class="n">exp</span><span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">])</span>
<span class="w">          </span><span class="n">GSM188013</span><span class="w"> </span><span class="n">GSM188014</span><span class="w"> </span><span class="n">GSM188016</span><span class="w"> </span><span class="n">GSM188018</span>
<span class="m">1007</span>_<span class="n">s_at</span><span class="w"> </span><span class="m">15630.200</span><span class="w"> </span><span class="m">17048.800</span><span class="w"> </span><span class="m">13667.500</span><span class="w"> </span><span class="m">15138.800</span>
<span class="m">1053</span>_<span class="n">at</span><span class="w">    </span><span class="m">3614.400</span><span class="w">  </span><span class="m">3563.220</span><span class="w">  </span><span class="m">2604.650</span><span class="w">  </span><span class="m">1945.710</span>
<span class="m">117</span>_<span class="n">at</span><span class="w">     </span><span class="m">1032.670</span><span class="w">  </span><span class="m">1164.150</span><span class="w">   </span><span class="m">510.692</span><span class="w">  </span><span class="m">5061.200</span>
<span class="m">121</span>_<span class="n">at</span><span class="w">     </span><span class="m">5917.800</span><span class="w">  </span><span class="m">6826.670</span><span class="w">  </span><span class="m">4562.440</span><span class="w">  </span><span class="m">5870.130</span>
<span class="m">1255</span>_<span class="n">g_at</span><span class="w">   </span><span class="m">224.525</span><span class="w">   </span><span class="m">395.025</span><span class="w">   </span><span class="m">207.087</span><span class="w">   </span><span class="m">164.835</span>
<span class="m">1294</span>_<span class="n">at</span><span class="w">     </span><span class="m">799.786</span><span class="w">   </span><span class="m">839.787</span><span class="w">   </span><span class="m">592.434</span><span class="w">   </span><span class="m">593.632</span>
</code></pre></div></p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  回到页面顶部
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2022 - 2025 DoubleHelix
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
    <a href="https://space.bilibili.com/434318322" target="_blank" rel="noopener" title="Bilibili | BioInfoCloud" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M488.6 104.1c16.7 18.1 24.4 39.7 23.3 65.7v202.4c-.4 26.4-9.2 48.1-26.5 65.1-17.2 17-39.1 25.9-65.5 26.7H92c-26.4-.8-48.2-9.8-65.3-27.2S.7 396.5 0 368.2V169.8c.8-26 9.7-47.6 26.7-65.7C43.8 87.8 65.5 78.8 92 78h29.4L96 52.2c-5.7-5.7-8.6-13-8.6-21.8S90.3 14.3 96 8.6 109 0 117.9 0s16.1 2.9 21.9 8.6L213.1 78h88l74.5-69.4C381.7 2.9 389.2 0 398 0s16.1 2.9 21.9 8.6c5.7 5.7 8.6 13 8.6 21.8s-2.9 16.1-8.6 21.8L394.6 78h29.3c26.4.8 48 9.8 64.7 26.1m-38.8 69.7c-.4-9.6-3.7-17.4-10.7-23.5-5.2-6.1-14-9.4-22.7-9.8H96c-9.6.4-17.4 3.7-23.6 9.8-6.1 6.1-9.4 13.9-9.8 23.5v194.4c0 9.2 3.3 17 9.8 23.5s14.4 9.8 23.6 9.8h320.4c9.2 0 17-3.3 23.3-9.8s9.7-14.3 10.1-23.5zm-264.3 42.7c6.3 6.3 9.7 14.1 10.1 23.2V273c-.4 9.2-3.7 16.9-9.8 23.2-6.2 6.3-14 9.5-23.6 9.5s-17.5-3.2-23.6-9.5-9.4-14-9.8-23.2v-33.3c.4-9.1 3.8-16.9 10.1-23.2s13.2-9.6 23.3-10c9.2.4 17 3.7 23.3 10m191.5 0c6.3 6.3 9.7 14.1 10.1 23.2V273c-.4 9.2-3.7 16.9-9.8 23.2s-14 9.5-23.6 9.5-17.4-3.2-23.6-9.5c-7-6.3-9.4-14-9.7-23.2v-33.3c.3-9.1 3.7-16.9 10-23.2s14.1-9.6 23.3-10c9.2.4 17 3.7 23.3 10"/></svg>
    </a>
  
    
    
    
    
    <a href="https://github.com/BioInfoCloud" target="_blank" rel="noopener" title="GitHub | BioInfoCloud" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../../..", "features": ["navigation.tabs", "navigation.top", "navigation.indexes", "navigation.expand", "search.suggest", "search.highlight"], "search": "../../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "\u5df2\u590d\u5236", "clipboard.copy": "\u590d\u5236", "search.result.more.one": "\u5728\u8be5\u9875\u4e0a\u8fd8\u6709 1 \u4e2a\u7b26\u5408\u6761\u4ef6\u7684\u7ed3\u679c", "search.result.more.other": "\u5728\u8be5\u9875\u4e0a\u8fd8\u6709 # \u4e2a\u7b26\u5408\u6761\u4ef6\u7684\u7ed3\u679c", "search.result.none": "\u6ca1\u6709\u627e\u5230\u7b26\u5408\u6761\u4ef6\u7684\u7ed3\u679c", "search.result.one": "\u627e\u5230 1 \u4e2a\u7b26\u5408\u6761\u4ef6\u7684\u7ed3\u679c", "search.result.other": "# \u4e2a\u7b26\u5408\u6761\u4ef6\u7684\u7ed3\u679c", "search.result.placeholder": "\u952e\u5165\u4ee5\u5f00\u59cb\u641c\u7d22", "search.result.term.missing": "\u7f3a\u5c11", "select.version": "\u9009\u62e9\u5f53\u524d\u7248\u672c"}, "version": null}</script>
    
    
      <script src="../../../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
        <script src="../../../mkdocs/javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://unpkg.com/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>